--- autotrace-0.31.1_orig/autotrace.m4	2008-04-01 20:53:59.000000000 +0000
+++ autotrace-0.31.1/autotrace.m4	2008-04-01 20:54:07.000000000 +0000
@@ -4,7 +4,7 @@
 dnl AM_PATH_AUTOTRACE([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
 dnl Test to see if libautotrace is installed, and define AUTOTRACE_CFLAGS, LIBS
 dnl
-AC_DEFUN(AM_PATH_AUTOTRACE,
+AC_DEFUN([AM_PATH_AUTOTRACE],
 [dnl
 dnl Get the cflags and libraries from the autotrace-config script
 dnl
@@ -157,4 +157,4 @@
   AC_SUBST(AUTOTRACE_CFLAGS)
   AC_SUBST(AUTOTRACE_LIBS)
   rm -f conf.autotracetest
-])
\ No newline at end of file
+])
