--- zlib-1.2.3_orig/zconf.h	2008-04-01 19:17:15.000000000 +0000
+++ zlib-1.2.3/zconf.h	2005-05-28 06:40:35.000000000 +0000
@@ -202,6 +202,9 @@
    /* If building or using zlib as a DLL, define ZLIB_DLL.
     * This is not mandatory, but it offers a little performance increase.
     */
+#  if defined(ZLIB_INTERNAL) && defined(DLL_EXPORT)
+#    define ZLIB_DLL
+#  endif
 #  ifdef ZLIB_DLL
 #    if defined(WIN32) && (!defined(__BORLANDC__) || (__BORLANDC__ >= 0x500))
 #      ifdef ZLIB_INTERNAL
