diff -ru -ru dcron-2.3.3_orig/Makefile dcron-2.3.3/Makefile
--- dcron-2.3.3_orig/Makefile	2006-09-21 15:28:14.000000000 +0200
+++ dcron-2.3.3/Makefile	2006-09-21 15:28:26.000000000 +0200
@@ -41,10 +41,10 @@
 	rm -f *.o dcron.tgz $(PROTOS)
 
 install: crond crontab
-	install -o root -g root -m 0755 -D crond $(DESTDIR)$(prefix)/sbin/crond
-	install -o root -g root -m 4755 -D crontab $(DESTDIR)$(prefix)/bin/crontab
-	install -o root -g root -m 0644 -D crontab.1 $(DESTDIR)$(prefix)/man/man1/crontab.1
-	install -o root -g root -m 0644 -D crond.8 $(DESTDIR)$(prefix)/man/man8/crond.8
+	install -m 0755 -D crond $(DESTDIR)$(prefix)/sbin/crond
+	install -m 4755 -D crontab $(DESTDIR)$(prefix)/bin/crontab
+	install -m 0644 -D crontab.1 $(DESTDIR)$(prefix)/man/man1/crontab.1
+	install -m 0644 -D crond.8 $(DESTDIR)$(prefix)/man/man8/crond.8
 
 tar: cleano
 	(cd ..; tar cvf - dcron | gzip -9 >dcron.tgz.tmp)
