diff -r -u -- popt-1.7_orig/Makefile.in popt-1.7/Makefile.in
--- popt-1.7_orig/Makefile.in	2007-05-07 16:37:22.000000000 +0200
+++ popt-1.7/Makefile.in	2007-05-07 16:38:11.000000000 +0200
@@ -73,7 +73,7 @@
 am_libpopt_la_OBJECTS = popt.lo findme.lo poptparse.lo poptconfig.lo \
 	popthelp.lo
 libpopt_la_OBJECTS = $(am_libpopt_la_OBJECTS)
-PROGRAMS = $(noinst_PROGRAMS)
+PROGRAMS = 
 am_test1_OBJECTS = test1.$(OBJEXT)
 test1_OBJECTS = $(am_test1_OBJECTS)
 am__DEPENDENCIES_1 = libpopt.la
