diff -ru httpd-2.2.10_orig/configure httpd-2.2.10/configure
--- httpd-2.2.10_orig/configure	2008-11-16 04:55:28.000000000 +0000
+++ httpd-2.2.10/configure	2008-11-16 04:59:55.000000000 +0000
@@ -21567,7 +21567,7 @@
 
 
 cat >>confdefs.h <<_ACEOF
-#define SERVER_CONFIG_FILE "${rel_sysconfdir}/${progname}.conf"
+#define SERVER_CONFIG_FILE "${rel_sysconfdir}/${progname%%-[0-9]*}.conf"
 _ACEOF
 
 
diff -ru httpd-2.2.10_orig/configure.in httpd-2.2.10/configure.in
--- httpd-2.2.10_orig/configure.in	2008-11-16 04:55:28.000000000 +0000
+++ httpd-2.2.10/configure.in	2008-11-16 04:59:10.000000000 +0000
@@ -642,7 +642,7 @@
 APR_EXPAND_VAR(ap_prefix, $prefix)
 AC_DEFINE_UNQUOTED(HTTPD_ROOT, "${ap_prefix}",
 	[Root directory of the Apache install area])
-AC_DEFINE_UNQUOTED(SERVER_CONFIG_FILE, "${rel_sysconfdir}/${progname}.conf",
+AC_DEFINE_UNQUOTED(SERVER_CONFIG_FILE, "${rel_sysconfdir}/${progname%%-[0-9]*}.conf",
 	[Location of the config file, relative to the Apache root directory])
 AC_DEFINE_UNQUOTED(AP_TYPES_CONFIG_FILE, "${rel_sysconfdir}/mime.types",
 	[Location of the MIME types config file, relative to the Apache root directory])
--- httpd-2.2.10_orig/Makefile.in	2008-11-16 04:55:29.000000000 +0000
+++ httpd-2.2.10/Makefile.in	2008-11-16 05:12:21.000000000 +0000
@@ -72,9 +72,6 @@
 	    	) > $(DESTDIR)$(sysconfdir)/original/$$i; \
 	    	chmod 0644 $(DESTDIR)$(sysconfdir)/original/$$i; \
 	    	file=$$i; \
-	    	if [ "$$i" = "httpd.conf" ]; then \
-	    		file=`echo $$i|sed s/.*.conf/$(PROGRAM_NAME).conf/`; \
-	    	fi; \
 	    	if test ! -f $(DESTDIR)$(sysconfdir)/$$file; then \
 	    		$(INSTALL_DATA) $(DESTDIR)$(sysconfdir)/original/$$i $(DESTDIR)$(sysconfdir)/$$file; \
 	    	fi; \
diff -ru httpd-2.2.10_orig/docs/conf/extra/httpd-ssl.conf.in httpd-2.2.10/docs/conf/extra/httpd-ssl.conf.in
--- httpd-2.2.10_orig/docs/conf/extra/httpd-ssl.conf.in	2008-11-16 05:42:48.000000000 +0000
+++ httpd-2.2.10/docs/conf/extra/httpd-ssl.conf.in	2008-11-16 05:51:13.000000000 +0000
@@ -96,16 +96,16 @@
 #   in mind that if you have both an RSA and a DSA certificate you
 #   can configure both in parallel (to also allow the use of DSA
 #   ciphers, etc.)
-SSLCertificateFile "@exp_sysconfdir@/server.crt"
-#SSLCertificateFile "@exp_sysconfdir@/server-dsa.crt"
+SSLCertificateFile "@exp_sysconfdir@/conf/ssl.crt/server.crt"
+#SSLCertificateFile "@exp_sysconfdir@/conf/ssl.crt/server-dsa.crt"
 
 #   Server Private Key:
 #   If the key is not combined with the certificate, use this
 #   directive to point at the key file.  Keep in mind that if
 #   you've both a RSA and a DSA private key you can configure
 #   both in parallel (to also allow the use of DSA ciphers, etc.)
-SSLCertificateKeyFile "@exp_sysconfdir@/server.key"
-#SSLCertificateKeyFile "@exp_sysconfdir@/server-dsa.key"
+SSLCertificateKeyFile "@exp_sysconfdir@/conf/ssl.key/server.key"
+#SSLCertificateKeyFile "@exp_sysconfdir@/conf/ssl.key/server-dsa.key"
 
 #   Server Certificate Chain:
 #   Point SSLCertificateChainFile at a file containing the
@@ -114,7 +114,7 @@
 #   the referenced file can be the same as SSLCertificateFile
 #   when the CA certificates are directly appended to the server
 #   certificate for convinience.
-#SSLCertificateChainFile "@exp_sysconfdir@/server-ca.crt"
+#SSLCertificateChainFile "@exp_sysconfdir@/conf/ssl.crt/server-ca.crt"
 
 #   Certificate Authority (CA):
 #   Set the CA certificate verification path where to find CA
@@ -123,8 +123,8 @@
 #   Note: Inside SSLCACertificatePath you need hash symlinks
 #         to point to the certificate files. Use the provided
 #         Makefile to update the hash symlinks after changes.
-#SSLCACertificatePath "@exp_sysconfdir@/ssl.crt"
-#SSLCACertificateFile "@exp_sysconfdir@/ssl.crt/ca-bundle.crt"
+#SSLCACertificatePath "@exp_sysconfdir@/conf/ssl.crt"
+#SSLCACertificateFile "@exp_sysconfdir@/conf/ssl.crt/ca-bundle.crt"
 
 #   Certificate Revocation Lists (CRL):
 #   Set the CA revocation path where to find CA CRLs for client
@@ -133,8 +133,8 @@
 #   Note: Inside SSLCARevocationPath you need hash symlinks
 #         to point to the certificate files. Use the provided
 #         Makefile to update the hash symlinks after changes.
-#SSLCARevocationPath "@exp_sysconfdir@/ssl.crl"
-#SSLCARevocationFile "@exp_sysconfdir@/ssl.crl/ca-bundle.crl"
+#SSLCARevocationPath "@exp_sysconfdir@/conf/ssl.crl"
+#SSLCARevocationFile "@exp_sysconfdir@/conf/ssl.crl/ca-bundle.crl"
 
 #   Client Authentication (Type):
 #   Client certificate verification type and depth.  Types are
