diff -ru readline-5.1_orig/shlib/Makefile.in readline-5.1/shlib/Makefile.in
--- readline-5.1_orig/shlib/Makefile.in	2006-01-24 17:21:28.000000000 +0100
+++ readline-5.1/shlib/Makefile.in	2006-01-24 17:22:03.000000000 +0100
@@ -164,7 +164,7 @@
 
 $(SHARED_READLINE):	$(SHARED_OBJ)
 	$(RM) $@
-	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_OBJ) $(SHLIB_LIBS)
+	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_OBJ) $(SHLIB_LIBS) -lterminfo
 
 $(SHARED_HISTORY):	$(SHARED_HISTOBJ) xmalloc.so
 	$(RM) $@
