args: -r -u -- izsound-0.3.0_orig izsound-0.3.0
diff -r -u -- izsound-0.3.0_orig/src/izsound/.deps/dspunit.Plo izsound-0.3.0/src/izsound/.deps/dspunit.Plo
diff -r -u -- izsound-0.3.0_orig/src/izsound/izsound/dspunit.h izsound-0.3.0/src/izsound/izsound/dspunit.h
--- izsound-0.3.0_orig/src/izsound/izsound/dspunit.h	2009-03-07 20:23:19.000000000 +0100
+++ izsound-0.3.0/src/izsound/izsound/dspunit.h	2009-03-07 20:25:08.000000000 +0100
@@ -98,7 +98,7 @@
   vector<SlotData*> m_outSlots;
 
   /** The binded slots tracker. */
-  bit_vector m_bindedSlots;
+  vector<bool> m_bindedSlots;
 
   /** The output dsp units. */
   vector<DspUnit*> m_outDspUnits;
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-bandfilter.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-bandfilter.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-blackhole.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-blackhole.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-crossfader.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-crossfader.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-datapicker.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-datapicker.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-delayextrastereo.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-delayextrastereo.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-demultiplexer.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-demultiplexer.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-flanger.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-flanger.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-fragmenter.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-fragmenter.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-libaooutput.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-libaooutput.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-maddecoder.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-maddecoder.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-oggfiledecoder.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-oggfiledecoder.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-ossoutput.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-ossoutput.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-pitch.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-pitch.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-silencer.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-silencer.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-volume.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-volume.Plo
diff -r -u -- izsound-0.3.0_orig/src/std-dsp/.deps/libstdizdsp_la-whitenoise.Plo izsound-0.3.0/src/std-dsp/.deps/libstdizdsp_la-whitenoise.Plo
