args: -r -u -- libvisual-0.4.0_orig libvisual-0.4.0
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_buffer.Plo libvisual-0.4.0/libvisual/.deps/lv_buffer.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_cache.Plo libvisual-0.4.0/libvisual/.deps/lv_cache.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_collection.Plo libvisual-0.4.0/libvisual/.deps/lv_collection.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_color.Plo libvisual-0.4.0/libvisual/.deps/lv_color.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_config.Plo libvisual-0.4.0/libvisual/.deps/lv_config.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_cpu.Plo libvisual-0.4.0/libvisual/.deps/lv_cpu.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_error.Plo libvisual-0.4.0/libvisual/.deps/lv_error.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_gl.Plo libvisual-0.4.0/libvisual/.deps/lv_gl.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_hashlist.Plo libvisual-0.4.0/libvisual/.deps/lv_hashlist.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_hashmap.Plo libvisual-0.4.0/libvisual/.deps/lv_hashmap.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_math.Plo libvisual-0.4.0/libvisual/.deps/lv_math.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_object.Plo libvisual-0.4.0/libvisual/.deps/lv_object.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_os.Plo libvisual-0.4.0/libvisual/.deps/lv_os.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_random.Plo libvisual-0.4.0/libvisual/.deps/lv_random.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_rectangle.Plo libvisual-0.4.0/libvisual/.deps/lv_rectangle.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_ringbuffer.Plo libvisual-0.4.0/libvisual/.deps/lv_ringbuffer.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_thread.Plo libvisual-0.4.0/libvisual/.deps/lv_thread.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_time.Plo libvisual-0.4.0/libvisual/.deps/lv_time.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_transform.Plo libvisual-0.4.0/libvisual/.deps/lv_transform.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_ui.Plo libvisual-0.4.0/libvisual/.deps/lv_ui.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/.deps/lv_utils.Plo libvisual-0.4.0/libvisual/.deps/lv_utils.Plo
diff -r -u -- libvisual-0.4.0_orig/libvisual/lv_cpu.c libvisual-0.4.0/libvisual/lv_cpu.c
--- libvisual-0.4.0_orig/libvisual/lv_cpu.c	2008-11-11 14:07:45.000000000 +0100
+++ libvisual-0.4.0/libvisual/lv_cpu.c	2008-11-11 14:10:14.000000000 +0100
@@ -76,7 +76,7 @@
 
 /* The sigill handlers */
 #if defined(VISUAL_ARCH_X86) //x86 (linux katmai handler check thing)
-#if defined(VISUAL_OS_LINUX) && defined(_POSIX_SOURCE) && defined(X86_FXSR_MAGIC)
+#if defined(VISUAL_OS_LINUX) && defined(_POSIX_SOURCE) /*&& defined(X86_FXSR_MAGIC)*/
 static void sigill_handler_sse( int signal, struct sigcontext sc )
 {
 	/* Both the "xorps %%xmm0,%%xmm0" and "divps %xmm0,%%xmm1"
