diff -ru net-tools-1.60_orig/Makefile net-tools-1.60/Makefile
--- net-tools-1.60_orig/Makefile	2006-02-13 13:56:19.000000000 +0000
+++ net-tools-1.60/Makefile	2006-02-13 13:56:57.000000000 +0000
@@ -114,7 +114,7 @@
 NET_LIB = $(NET_LIB_PATH)/lib$(NET_LIB_NAME).a
 
 CFLAGS	= $(COPTS) -I. -idirafter ./include/ -I$(NET_LIB_PATH)
-LDFLAGS	= $(LOPTS) -L$(NET_LIB_PATH)
+LDFLAGS	= $(LOPTS) -L$(NET_LIB_PATH) -Wl,-rpath,$(BASEDIR)/lib
 
 SUBDIRS	= man/ $(NET_LIB_PATH)/
 
