diff -r -u net-tools-1.60_orig/ifconfig.c net-tools-1.60/ifconfig.c
--- net-tools-1.60_orig/ifconfig.c	2007-09-27 02:46:40.000000000 +0200
+++ net-tools-1.60/ifconfig.c	2007-09-27 02:47:47.000000000 +0200
@@ -85,6 +85,8 @@
 #include "sockets.h"
 #include "util.h"
 
+extern int skfd;
+
 char *Release = RELEASE, *Version = "ifconfig 1.42 (2001-04-13)";
 
 int opt_a = 0;			/* show all interfaces          */
diff -r -u net-tools-1.60_orig/lib/interface.c net-tools-1.60/lib/interface.c
--- net-tools-1.60_orig/lib/interface.c	2007-09-27 02:46:40.000000000 +0200
+++ net-tools-1.60/lib/interface.c	2007-09-27 02:47:33.000000000 +0200
@@ -51,6 +51,8 @@
 #include "util.h"
 #include "intl.h"
 
+extern int skfd;
+
 #ifdef IFF_PORTSEL
 const char *if_port_text[][4] =
 {
diff -r -u net-tools-1.60_orig/netstat.c net-tools-1.60/netstat.c
--- net-tools-1.60_orig/netstat.c	2007-09-27 02:46:40.000000000 +0200
+++ net-tools-1.60/netstat.c	2007-09-27 02:48:05.000000000 +0200
@@ -122,6 +122,8 @@
 #define FEATURE_NETSTAT
 #include "lib/net-features.h"
 
+extern int skfd;
+
 char *Release = RELEASE, *Version = "netstat 1.42 (2001-04-15)", *Signature = "Fred Baumgarten, Alan Cox, Bernd Eckenfels, Phil Blundell, Tuan Hoang and others";
 
 
