diff -ru procps-3.2.5_orig/proc/module.mk procps-3.2.5/proc/module.mk
--- procps-3.2.5_orig/proc/module.mk	2006-02-03 16:31:17.000000000 +0000
+++ procps-3.2.5/proc/module.mk	2006-02-06 11:16:57.000000000 +0000
@@ -23,9 +23,9 @@
 NAME       :=  proc
 
 LIBVERSION := $(VERSION).$(SUBVERSION).$(MINORVERSION)
-ABIVERSION := 0
+ABIVERSION := $(VERSION)
 
-SOFILE     := lib$(NAME)-$(LIBVERSION).so
+SOFILE     := lib$(NAME).so.$(LIBVERSION)
 ifneq ($(ABIVERSION),0)
 SOLINK     := lib$(NAME).so
 SONAME     := lib$(NAME).so.$(ABIVERSION)
