diff -ru libsift-1.8_orig/src/Makefile libsift-1.8/src/Makefile
--- libsift-1.8_orig/src/Makefile	2008-07-14 18:02:24.000000000 +0200
+++ libsift-1.8/src/Makefile	2008-07-14 18:03:35.000000000 +0200
@@ -11,7 +11,7 @@
 LIBS_WIN=/r:gtk-sharp.dll /r:System.Drawing.dll /r:ICSharpCode.SharpZipLib.dll \
 	/r:glib-sharp.dll /r:gdk-sharp.dll
 MAKE_MONO=
-LIBS_MONO=/pkg:gtk-sharp /r:System.Drawing /r:ICSharpCode.SharpZipLib
+LIBS_MONO=/pkg:gtk-sharp-2.0 /r:System.Drawing /r:ICSharpCode.SharpZipLib
 
 ifeq ($(windows),yes)
 	CSC=$(CSC_WINDOWS)
@@ -29,7 +29,7 @@
 	BondBall.cs AreaFilter.cs
 	#CoverageFilter.cs FMM.cs
 
-all:	lib utils bin_install
+all:	lib 
 
 bin_install:
 	cp ICSharpCode.SharpZipLib.dll libsift.dll util/*.exe util/autopanog/*.exe \
