diff -r -u i686-pc-linux-gnu_orig/sleuthkit-2.09/src/auxtools/Makefile i686-pc-linux-gnu/sleuthkit-2.09/src/auxtools/Makefile
--- i686-pc-linux-gnu_orig/sleuthkit-2.09/src/auxtools/Makefile	2007-07-29 14:45:34.000000000 +0200
+++ i686-pc-linux-gnu/sleuthkit-2.09/src/auxtools/Makefile	2007-07-29 15:18:19.000000000 +0200
@@ -21,7 +21,7 @@
 manpages: 
 
 lib:	$(LIBOBJ)
-	$(AR) $(LIB) $?
+	$(AR) $(ARFLAGS) $(LIB) $?
 	$(RANLIB) $(LIB)
 
 $(BIN)/md5: md5.o md5c.o
diff -r -u i686-pc-linux-gnu_orig/sleuthkit-2.09/src/fstools/Makefile i686-pc-linux-gnu/sleuthkit-2.09/src/fstools/Makefile
--- i686-pc-linux-gnu_orig/sleuthkit-2.09/src/fstools/Makefile	2007-07-29 14:45:34.000000000 +0200
+++ i686-pc-linux-gnu/sleuthkit-2.09/src/fstools/Makefile	2007-07-29 15:18:19.000000000 +0200
@@ -32,7 +32,7 @@
 all:	lib $(PROGS)
 
 lib:	$(LIBOBJ)
-	$(AR) $(LIB) $?
+	$(AR) $(ARFLAGS) $(LIB) $?
 	$(RANLIB) $(LIB)
 
 $(BIN_DIR)/dls: dls.o $(LIB) 
diff -r -u i686-pc-linux-gnu_orig/sleuthkit-2.09/src/hashtools/Makefile i686-pc-linux-gnu/sleuthkit-2.09/src/hashtools/Makefile
--- i686-pc-linux-gnu_orig/sleuthkit-2.09/src/hashtools/Makefile	2007-07-29 14:45:34.000000000 +0200
+++ i686-pc-linux-gnu/sleuthkit-2.09/src/hashtools/Makefile	2007-07-29 15:18:19.000000000 +0200
@@ -16,7 +16,7 @@
 all:    lib $(PROGS)
 
 lib: 	$(LIBOBJ)
-	$(AR) $(LIB) $?
+	$(AR) $(ARFLAGS) $(LIB) $?
 	$(RANLIB) $(LIB)
 
 $(BIN)/hfind: hfind.o $(LIB)
diff -r -u i686-pc-linux-gnu_orig/sleuthkit-2.09/src/imgtools/Makefile i686-pc-linux-gnu/sleuthkit-2.09/src/imgtools/Makefile
--- i686-pc-linux-gnu_orig/sleuthkit-2.09/src/imgtools/Makefile	2007-07-29 14:45:34.000000000 +0200
+++ i686-pc-linux-gnu/sleuthkit-2.09/src/imgtools/Makefile	2007-07-29 15:18:19.000000000 +0200
@@ -19,7 +19,7 @@
 all:	lib $(PROGS)
 
 lib:	$(LIBOBJ)
-	$(AR) $(LIB) $?
+	$(AR) $(ARFLAGS) $(LIB) $?
 	$(RANLIB) $(LIB) 
 
 $(BIN_DIR)/img_stat: img_stat.o $(LIB)
diff -r -u i686-pc-linux-gnu_orig/sleuthkit-2.09/src/mmtools/Makefile i686-pc-linux-gnu/sleuthkit-2.09/src/mmtools/Makefile
--- i686-pc-linux-gnu_orig/sleuthkit-2.09/src/mmtools/Makefile	2007-07-29 14:45:34.000000000 +0200
+++ i686-pc-linux-gnu/sleuthkit-2.09/src/mmtools/Makefile	2007-07-29 15:18:19.000000000 +0200
@@ -19,7 +19,7 @@
 all:	lib $(PROGS)
 
 lib:	$(LIBOBJ)
-	$(AR) $(LIB) $?
+	$(AR) $(ARFLAGS) $(LIB) $?
 	$(RANLIB) $(LIB)
 
 $(BIN_DIR)/mmls: mmls.o $(LIB) 
diff -r -u i686-pc-linux-gnu_orig/sleuthkit-2.09/src/srchtools/Makefile i686-pc-linux-gnu/sleuthkit-2.09/src/srchtools/Makefile
--- i686-pc-linux-gnu_orig/sleuthkit-2.09/src/srchtools/Makefile	2007-07-29 14:45:34.000000000 +0200
+++ i686-pc-linux-gnu/sleuthkit-2.09/src/srchtools/Makefile	2007-07-29 15:18:19.000000000 +0200
@@ -17,7 +17,7 @@
 all:    lib $(PROGS)
 
 lib: $(LIBOBJ)
-	$(AR) $(LIB) $?
+	$(AR) $(ARFLAGS) $(LIB) $?
 	$(RANLIB) $(LIB)
 
 $(BIN)/srch_strings: srch_strings.o $(LIB)
