args: -r -u -N -- mod_cplusplus_orig mod_cplusplus
diff -r -u -N -- mod_cplusplus_orig/Makefile.am mod_cplusplus/Makefile.am
--- mod_cplusplus_orig/Makefile.am	2008-11-20 10:33:00.000000000 +0000
+++ mod_cplusplus/Makefile.am	2008-11-20 10:41:49.000000000 +0000
@@ -7,7 +7,5 @@
 install:
 	@echo ...installing module.......
 	cp src/.libs/libmod_cplusplus@SHLIB_EXT@ $(DESTDIR)@MODULES_DIR@
-	@echo ...checking config file......
-	if ! grep cplusplus $(DESTDIR)@CONF_DIR@/httpd.conf; then echo ...updating config file....;echo 'LoadModule cplusplus_module @MODULES_DIR@/libmod_cplusplus@SHLIB_EXT@' >> $(DESTDIR)@CONF_DIR@/httpd.conf ; else echo ***config file already contains modcplusplus***; fi
 	@echo ...install complete...
 
