--- binutils-2.19_orig/bfd/elf.c	2008-11-10 23:59:41.000000000 +0100
+++ binutils-2.19/bfd/elf.c	2008-11-11 00:12:39.000000000 +0100
@@ -6413,9 +6413,8 @@
 	  if (type == STT_OBJECT)
 	    sym.st_info = ELF_ST_INFO (STB_GLOBAL, STT_COMMON);
 	  else
-#else
-	    sym.st_info = ELF_ST_INFO (STB_GLOBAL, type);
 #endif
+	    sym.st_info = ELF_ST_INFO (STB_GLOBAL, type);
 	}
       else if (bfd_is_und_section (syms[idx]->section))
 	sym.st_info = ELF_ST_INFO (((flags & BSF_WEAK)
