diff -ru Linux-PAM-0.99.3.0_orig/modules/pam_rhosts/Makefile.in Linux-PAM-0.99.3.0/modules/pam_rhosts/Makefile.in
--- Linux-PAM-0.99.3.0_orig/modules/pam_rhosts/Makefile.in	2006-02-15 22:09:26.000000000 +0100
+++ Linux-PAM-0.99.3.0/modules/pam_rhosts/Makefile.in	2006-02-15 22:09:58.000000000 +0100
@@ -102,7 +102,7 @@
 CXXDEPMODE = @CXXDEPMODE@
 CXXFLAGS = @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
-DEFS = @DEFS@
+DEFS = @DEFS@ -DSYSCONFDIR=\"$(sysconfdir)\"
 DEPDIR = @DEPDIR@
 DOCDIR = @DOCDIR@
 ECHO = @ECHO@
