args: -r -u -- ccbuild-1.5.6_orig ccbuild-1.5.6
diff -r -u -- ccbuild-1.5.6_orig/src/.deps/ccbuild-ccbuild.Po ccbuild-1.5.6/src/.deps/ccbuild-ccbuild.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Arguments/.deps/ccbuild-Arguments.Po ccbuild-1.5.6/src/Arguments/.deps/ccbuild-Arguments.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Arguments/.deps/ccbuild-clear.Po ccbuild-1.5.6/src/Arguments/.deps/ccbuild-clear.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Arguments/.deps/ccbuild-destroy.Po ccbuild-1.5.6/src/Arguments/.deps/ccbuild-destroy.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Arguments/.deps/ccbuild-flagged.Po ccbuild-1.5.6/src/Arguments/.deps/ccbuild-flagged.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Arguments/.deps/ccbuild-getInstance.Po ccbuild-1.5.6/src/Arguments/.deps/ccbuild-getInstance.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Arguments/.deps/ccbuild-initialize.Po ccbuild-1.5.6/src/Arguments/.deps/ccbuild-initialize.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Arguments/.deps/ccbuild-outputOptions.Po ccbuild-1.5.6/src/Arguments/.deps/ccbuild-outputOptions.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Arguments/.deps/ccbuild-statics.Po ccbuild-1.5.6/src/Arguments/.deps/ccbuild-statics.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Arguments/.deps/ccbuild-value.Po ccbuild-1.5.6/src/Arguments/.deps/ccbuild-value.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Arguments/.deps/ccbuild-values.Po ccbuild-1.5.6/src/Arguments/.deps/ccbuild-values.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-addArgument.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-addArgument.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-addObject.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-addObject.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-ar.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-ar.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-cls.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-cls.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-compile.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-compile.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-compileCommand.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-compileCommand.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-countFirstLinkerArguments.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-countFirstLinkerArguments.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-lib.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-lib.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-libCommand.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-libCommand.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-link.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-link.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-linkCommand.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-linkCommand.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-operator_add.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-operator_add.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-precompile.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-precompile.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-precompileCommand.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-precompileCommand.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-splitInto.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-splitInto.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Compiler/.deps/ccbuild-statics.Po ccbuild-1.5.6/src/Compiler/.deps/ccbuild-statics.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-baseName.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-baseName.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-cleanPath.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-cleanPath.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-cwd.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-cwd.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-directoryName.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-directoryName.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-ensureDirectory.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-ensureDirectory.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-fileExists.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-fileExists.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-fileName.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-fileName.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-globDirectoriesInto.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-globDirectoriesInto.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-globFilesInto.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-globFilesInto.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-globInto.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-globInto.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-globSourceFilesInto.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-globSourceFilesInto.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-isReadable.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-isReadable.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-modTime.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-modTime.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-newer.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-newer.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-recursiveGlobDirectoriesInto.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-recursiveGlobDirectoriesInto.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-rmIfExists.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-rmIfExists.Po
diff -r -u -- ccbuild-1.5.6_orig/src/FileSystem/.deps/ccbuild-touch.Po ccbuild-1.5.6/src/FileSystem/.deps/ccbuild-touch.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Globals/.deps/ccbuild-Globals.Po ccbuild-1.5.6/src/Globals/.deps/ccbuild-Globals.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Globals/.deps/ccbuild-destroy.Po ccbuild-1.5.6/src/Globals/.deps/ccbuild-destroy.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Globals/.deps/ccbuild-getInstance.Po ccbuild-1.5.6/src/Globals/.deps/ccbuild-getInstance.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Globals/.deps/ccbuild-indexoperator.Po ccbuild-1.5.6/src/Globals/.deps/ccbuild-indexoperator.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Globals/.deps/ccbuild-statics.Po ccbuild-1.5.6/src/Globals/.deps/ccbuild-statics.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/.deps/ccbuild-MD5Info.Po ccbuild-1.5.6/src/MD5Info/.deps/ccbuild-MD5Info.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/.deps/ccbuild-contentHash.Po ccbuild-1.5.6/src/MD5Info/.deps/ccbuild-contentHash.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/.deps/ccbuild-destroy.Po ccbuild-1.5.6/src/MD5Info/.deps/ccbuild-destroy.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/.deps/ccbuild-getInstance.Po ccbuild-1.5.6/src/MD5Info/.deps/ccbuild-getInstance.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/.deps/ccbuild-hashFilenameFor.Po ccbuild-1.5.6/src/MD5Info/.deps/ccbuild-hashFilenameFor.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/.deps/ccbuild-load.Po ccbuild-1.5.6/src/MD5Info/.deps/ccbuild-load.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/.deps/ccbuild-old.Po ccbuild-1.5.6/src/MD5Info/.deps/ccbuild-old.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/.deps/ccbuild-save.Po ccbuild-1.5.6/src/MD5Info/.deps/ccbuild-save.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/.deps/ccbuild-statics.Po ccbuild-1.5.6/src/MD5Info/.deps/ccbuild-statics.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-MD5Hash.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-MD5Hash.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-clear.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-clear.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-extractor.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-extractor.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-from.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-from.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-fromHex.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-fromHex.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-hex.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-hex.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-inserter.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-inserter.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-opEqual.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-opEqual.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-opLess.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-opLess.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-opMinus.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-opMinus.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/.deps/ccbuild-random.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/.deps/ccbuild-random.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/GCrypt/.deps/ccbuild-GCrypt.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/GCrypt/.deps/ccbuild-GCrypt.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/GCrypt/.deps/ccbuild-finalize.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/GCrypt/.deps/ccbuild-finalize.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/GCrypt/.deps/ccbuild-from.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/GCrypt/.deps/ccbuild-from.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/GCrypt/.deps/ccbuild-update.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/GCrypt/.deps/ccbuild-update.Po
diff -r -u -- ccbuild-1.5.6_orig/src/MD5Info/MD5Hash/GCrypt/.deps/ccbuild-value.Po ccbuild-1.5.6/src/MD5Info/MD5Hash/GCrypt/.deps/ccbuild-value.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Options/.deps/ccbuild-statics.Po ccbuild-1.5.6/src/Options/.deps/ccbuild-statics.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Problem/.deps/ccbuild-Problem.Po ccbuild-1.5.6/src/Problem/.deps/ccbuild-Problem.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Resolver/.deps/ccbuild-Resolver.Po ccbuild-1.5.6/src/Resolver/.deps/ccbuild-Resolver.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Resolver/.deps/ccbuild-destroy.Po ccbuild-1.5.6/src/Resolver/.deps/ccbuild-destroy.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Resolver/.deps/ccbuild-expand.Po ccbuild-1.5.6/src/Resolver/.deps/ccbuild-expand.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Resolver/.deps/ccbuild-getInstance.Po ccbuild-1.5.6/src/Resolver/.deps/ccbuild-getInstance.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Resolver/.deps/ccbuild-loadIfExists.Po ccbuild-1.5.6/src/Resolver/.deps/ccbuild-loadIfExists.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Resolver/.deps/ccbuild-resolve.Po ccbuild-1.5.6/src/Resolver/.deps/ccbuild-resolve.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Resolver/.deps/ccbuild-resolveInto.Po ccbuild-1.5.6/src/Resolver/.deps/ccbuild-resolveInto.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Resolver/.deps/ccbuild-statics.Po ccbuild-1.5.6/src/Resolver/.deps/ccbuild-statics.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-Source.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-Source.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-basename.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-basename.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-build.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-build.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-buildBinTarget.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-buildBinTarget.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-buildHeader.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-buildHeader.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-buildObjectTarget.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-buildObjectTarget.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-changed.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-changed.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-dependencies.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-dependencies.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-directDeps.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-directDeps.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-directory.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-directory.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-genDeps.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-genDeps.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-hasSourceExtension.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-hasSourceExtension.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-isBinTarget.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-isBinTarget.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-isHeader.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-isHeader.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-isInternalHeader.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-isInternalHeader.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-isLibTarget.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-isLibTarget.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-isLocalHeader.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-isLocalHeader.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-isObjectTarget.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-isObjectTarget.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-outputFilename.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-outputFilename.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-producesOutput.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-producesOutput.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-reload.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-reload.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-scan.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-scan.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-stale.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-stale.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Source/.deps/ccbuild-upToDate.Po ccbuild-1.5.6/src/Source/.deps/ccbuild-upToDate.Po
diff -r -u -- ccbuild-1.5.6_orig/src/SourceScanner/.deps/ccbuild-SourceScanner.Po ccbuild-1.5.6/src/SourceScanner/.deps/ccbuild-SourceScanner.Po
diff -r -u -- ccbuild-1.5.6_orig/src/SourceScanner/.deps/ccbuild-hasDefine.Po ccbuild-1.5.6/src/SourceScanner/.deps/ccbuild-hasDefine.Po
diff -r -u -- ccbuild-1.5.6_orig/src/SourceScanner/.deps/ccbuild-hasMainFunction.Po ccbuild-1.5.6/src/SourceScanner/.deps/ccbuild-hasMainFunction.Po
diff -r -u -- ccbuild-1.5.6_orig/src/SourceScanner/.deps/ccbuild-includes.Po ccbuild-1.5.6/src/SourceScanner/.deps/ccbuild-includes.Po
diff -r -u -- ccbuild-1.5.6_orig/src/SourceScanner/.deps/ccbuild-storeGlobal.Po ccbuild-1.5.6/src/SourceScanner/.deps/ccbuild-storeGlobal.Po
diff -r -u -- ccbuild-1.5.6_orig/src/SourceScanner/.deps/ccbuild-storeIgnore.Po ccbuild-1.5.6/src/SourceScanner/.deps/ccbuild-storeIgnore.Po
diff -r -u -- ccbuild-1.5.6_orig/src/SourceScanner/.deps/ccbuild-storeLocal.Po ccbuild-1.5.6/src/SourceScanner/.deps/ccbuild-storeLocal.Po
diff -r -u -- ccbuild-1.5.6_orig/src/SourceScanner/.deps/ccbuild-yylex.Po ccbuild-1.5.6/src/SourceScanner/.deps/ccbuild-yylex.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Sources/.deps/ccbuild-Sources.Po ccbuild-1.5.6/src/Sources/.deps/ccbuild-Sources.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Sources/.deps/ccbuild-destroy.Po ccbuild-1.5.6/src/Sources/.deps/ccbuild-destroy.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Sources/.deps/ccbuild-erase.Po ccbuild-1.5.6/src/Sources/.deps/ccbuild-erase.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Sources/.deps/ccbuild-getInstance.Po ccbuild-1.5.6/src/Sources/.deps/ccbuild-getInstance.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Sources/.deps/ccbuild-indexoperator.Po ccbuild-1.5.6/src/Sources/.deps/ccbuild-indexoperator.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Sources/.deps/ccbuild-reloadStaleSources.Po ccbuild-1.5.6/src/Sources/.deps/ccbuild-reloadStaleSources.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Sources/.deps/ccbuild-statics.Po ccbuild-1.5.6/src/Sources/.deps/ccbuild-statics.Po
diff -r -u -- ccbuild-1.5.6_orig/src/Sources/indexoperator.cc ccbuild-1.5.6/src/Sources/indexoperator.cc
--- ccbuild-1.5.6_orig/src/Sources/indexoperator.cc	2009-04-09 22:31:37.000000000 +0200
+++ ccbuild-1.5.6/src/Sources/indexoperator.cc	2009-04-09 22:38:09.000000000 +0200
@@ -21,6 +21,8 @@
 
 #include "Sources.ih"
 
+#include <algorithm>
+
 namespace{
 class DerivedEqual
 {
@@ -58,7 +60,7 @@
 		return 0;
 	}
 
-	std::set<Source *>::iterator pos = find_if(d_sources.begin(), d_sources.end(), DerivedEqual(s));
+	std::set<Source *>::iterator pos = std::find_if(d_sources.begin(), d_sources.end(), DerivedEqual(s));
 
 	if(pos == d_sources.end())
 	{
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-aapFor.Po ccbuild-1.5.6/src/System/.deps/ccbuild-aapFor.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-aapForAll.Po ccbuild-1.5.6/src/System/.deps/ccbuild-aapForAll.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-addArguments.Po ccbuild-1.5.6/src/System/.deps/ccbuild-addArguments.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-build.Po ccbuild-1.5.6/src/System/.deps/ccbuild-build.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-buildAll.Po ccbuild-1.5.6/src/System/.deps/ccbuild-buildAll.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-changeTo.Po ccbuild-1.5.6/src/System/.deps/ccbuild-changeTo.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-check.Po ccbuild-1.5.6/src/System/.deps/ccbuild-check.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-clean.Po ccbuild-1.5.6/src/System/.deps/ccbuild-clean.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-collectTargets.Po ccbuild-1.5.6/src/System/.deps/ccbuild-collectTargets.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-depsFor.Po ccbuild-1.5.6/src/System/.deps/ccbuild-depsFor.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-destroy.Po ccbuild-1.5.6/src/System/.deps/ccbuild-destroy.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-distclean.Po ccbuild-1.5.6/src/System/.deps/ccbuild-distclean.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-dotgraphFor.Po ccbuild-1.5.6/src/System/.deps/ccbuild-dotgraphFor.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-dotgraphForAll.Po ccbuild-1.5.6/src/System/.deps/ccbuild-dotgraphForAll.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-exit.Po ccbuild-1.5.6/src/System/.deps/ccbuild-exit.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-icmake.Po ccbuild-1.5.6/src/System/.deps/ccbuild-icmake.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-inspect.Po ccbuild-1.5.6/src/System/.deps/ccbuild-inspect.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-lib.Po ccbuild-1.5.6/src/System/.deps/ccbuild-lib.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-localSourcesInto.Po ccbuild-1.5.6/src/System/.deps/ccbuild-localSourcesInto.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-makefileFor.Po ccbuild-1.5.6/src/System/.deps/ccbuild-makefileFor.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-makefileForAll.Po ccbuild-1.5.6/src/System/.deps/ccbuild-makefileForAll.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-md5.Po ccbuild-1.5.6/src/System/.deps/ccbuild-md5.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-parseArguments.Po ccbuild-1.5.6/src/System/.deps/ccbuild-parseArguments.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-projectName.Po ccbuild-1.5.6/src/System/.deps/ccbuild-projectName.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-resolveTest.Po ccbuild-1.5.6/src/System/.deps/ccbuild-resolveTest.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-sleep.Po ccbuild-1.5.6/src/System/.deps/ccbuild-sleep.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-statics.Po ccbuild-1.5.6/src/System/.deps/ccbuild-statics.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-system.Po ccbuild-1.5.6/src/System/.deps/ccbuild-system.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-tree.Po ccbuild-1.5.6/src/System/.deps/ccbuild-tree.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-trim.Po ccbuild-1.5.6/src/System/.deps/ccbuild-trim.Po
diff -r -u -- ccbuild-1.5.6_orig/src/System/.deps/ccbuild-trimmed.Po ccbuild-1.5.6/src/System/.deps/ccbuild-trimmed.Po
diff -r -u -- ccbuild-1.5.6_orig/src/ccbuild.cc ccbuild-1.5.6/src/ccbuild.cc
--- ccbuild-1.5.6_orig/src/ccbuild.cc	2009-04-09 22:31:37.000000000 +0200
+++ ccbuild-1.5.6/src/ccbuild.cc	2009-04-09 22:31:57.000000000 +0200
@@ -61,6 +61,7 @@
 #include <cstring>
 #include <vector>
 #include <cassert>
+#include <typeinfo>
 
 //#define DEBUGLEVEL 20
 #include "Misc/Preprocessor/Debug.hh"
