diff -r -u FreeLing-1.5_orig/doc/Makefile.am FreeLing-1.5/doc/Makefile.am
--- FreeLing-1.5_orig/doc/Makefile.am	2007-09-21 05:36:48.000000000 +0200
+++ FreeLing-1.5/doc/Makefile.am	2007-09-21 08:34:21.000000000 +0200
@@ -1,21 +1,21 @@
 
 pkgdata_DATA =
 
-diagdatadir = $(pkgdatadir)/doc/diagrams
+diagdatadir = $(docdir)/diagrams
 diagdata_DATA = diagrams/*
 
-refmandatadir = $(pkgdatadir)/doc/refman
+refmandatadir = $(docdir)/refman
 
-refmanlatexdatadir = $(pkgdatadir)/doc/refman/latex
+refmanlatexdatadir = $(docdir)/refman/latex
 refmanlatexdata_DATA = refman/latex/*.pdf
 
-refmanhtmldatadir = $(pkgdatadir)/doc/refman/html
+refmanhtmldatadir = $(docdir)/refman/html
 refmanhtmldata_DATA = refman/html/*
 
-usermandatadir = $(pkgdatadir)/doc/userman
+usermandatadir = $(docdir)/userman
 usermandata_DATA = userman/*.pdf userman/*.html
 
-usermanhtmldatadir = $(pkgdatadir)/doc/userman/html
+usermanhtmldatadir = $(docdir)/userman/html
 usermanhtmldata_DATA = userman/html/*
 
 
