diff -ru gdb-6.6_orig/gdb/main.c gdb-6.6/gdb/main.c
--- gdb-6.6_orig/gdb/main.c	2007-05-08 16:24:25.000000000 +0200
+++ gdb-6.6/gdb/main.c	2007-05-08 16:25:22.000000000 +0200
@@ -562,14 +562,6 @@
      installed.  Otherwize the help/version messages will be eaten by
      the interpreter's output handler.  */
 
-  if (print_version)
-    {
-      print_gdb_version (gdb_stdout);
-      wrap_here ("");
-      printf_filtered ("\n");
-      exit (0);
-    }
-
   if (print_help)
     {
       print_gdb_help (gdb_stdout);
@@ -577,23 +569,6 @@
       exit (0);
     }
 
-  /* FIXME: cagney/2003-02-03: The big hack (part 1 of 2) that lets
-     GDB retain the old MI1 interpreter startup behavior.  Output the
-     copyright message before the interpreter is installed.  That way
-     it isn't encapsulated in MI output.  */
-  if (!quiet && strcmp (interpreter_p, INTERP_MI1) == 0)
-    {
-      /* Print all the junk at the top, with trailing "..." if we are about
-         to read a symbol file (possibly slowly).  */
-      print_gdb_version (gdb_stdout);
-      if (symarg)
-	printf_filtered ("..");
-      wrap_here ("");
-      printf_filtered ("\n");
-      gdb_flush (gdb_stdout);	/* Force to screen during slow operations */
-    }
-
-
   /* Install the default UI.  All the interpreters should have had a
      look at things by now.  Initialize the default interpreter. */
 
@@ -612,22 +587,6 @@
       }
   }
 
-  /* FIXME: cagney/2003-02-03: The big hack (part 2 of 2) that lets
-     GDB retain the old MI1 interpreter startup behavior.  Output the
-     copyright message after the interpreter is installed when it is
-     any sane interpreter.  */
-  if (!quiet && !current_interp_named_p (INTERP_MI1))
-    {
-      /* Print all the junk at the top, with trailing "..." if we are about
-         to read a symbol file (possibly slowly).  */
-      print_gdb_version (gdb_stdout);
-      if (symarg)
-	printf_filtered ("..");
-      wrap_here ("");
-      printf_filtered ("\n");
-      gdb_flush (gdb_stdout);	/* Force to screen during slow operations */
-    }
-
   /* Set off error and warning messages with a blank line.  */
   error_pre_print = "\n";
   quit_pre_print = error_pre_print;
Only in gdb-6.6/gdb: main.c.rej
Only in gdb-6.6/gdb: main.c~
diff -ru gdb-6.6_orig/gdb/top.c gdb-6.6/gdb/top.c
--- gdb-6.6_orig/gdb/top.c	2007-05-08 16:24:25.000000000 +0200
+++ gdb-6.6/gdb/top.c	2007-05-08 16:24:30.000000000 +0200
@@ -67,7 +67,7 @@
 /* Default command line prompt.  This is overriden in some configs. */
 
 #ifndef DEFAULT_PROMPT
-#define DEFAULT_PROMPT	"(gdb) "
+#define DEFAULT_PROMPT	"gdb>"
 #endif
 
 /* Initialization file name for gdb.  This is overridden in some configs.  */
