--- libxml2-2.6.31_orig/configure.in	2008-07-12 22:59:19.000000000 +0200
+++ libxml2-2.6.31/configure.in	2008-07-12 22:59:56.000000000 +0200
@@ -44,9 +44,9 @@
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_CPP
-AC_PATH_PROG(RM, rm, /bin/rm)
-AC_PATH_PROG(MV, mv, /bin/mv)
-AC_PATH_PROG(TAR, tar, /bin/tar)
+dnl AC_PATH_PROG(RM, rm, /bin/rm)
+dnl AC_PATH_PROG(MV, mv, /bin/mv)
+dnl AC_PATH_PROG(TAR, tar, /bin/tar)
 AC_PATH_PROG(PERL, perl, /usr/bin/perl)
 AC_PATH_PROG(WGET, wget, /usr/bin/wget)
 AC_PATH_PROG(XMLLINT, xmllint, /usr/bin/xmllint)
