diff -ru xdebug-2.0.3_orig/xdebug_code_coverage.c xdebug-2.0.3/xdebug_code_coverage.c
--- xdebug-2.0.3_orig/xdebug_code_coverage.c	2008-12-16 05:52:29.000000000 +0100
+++ xdebug-2.0.3/xdebug_code_coverage.c	2008-12-16 05:58:27.000000000 +0100
@@ -22,7 +22,7 @@
 #include "xdebug_var.h"
 #include "xdebug_code_coverage.h"
 
-extern ZEND_DECLARE_MODULE_GLOBALS(xdebug);
+extern zend_xdebug_globals xdebug_globals;
 
 void xdebug_coverage_line_dtor(void *data)
 {
diff -ru xdebug-2.0.3_orig/xdebug_superglobals.c xdebug-2.0.3/xdebug_superglobals.c
--- xdebug-2.0.3_orig/xdebug_superglobals.c	2008-12-16 05:52:29.000000000 +0100
+++ xdebug-2.0.3/xdebug_superglobals.c	2008-12-16 05:58:43.000000000 +0100
@@ -22,7 +22,7 @@
 #include "xdebug_superglobals.h"
 #include "SAPI.h"
 
-extern ZEND_DECLARE_MODULE_GLOBALS(xdebug);
+extern zend_xdebug_globals xdebug_globals;
 
 void xdebug_superglobals_dump_dtor(void *user, void *ptr)
 {
