--- libtool-2.2.4_orig/libltdl/config/ltmain.sh	2008-06-26 04:49:39.000000000 +0200
+++ libtool-2.2.4/libltdl/config/ltmain.sh	2008-06-26 04:51:05.000000000 +0200
@@ -2042,7 +2042,7 @@
 	  test -n "$relink_command" && srcname="$realname"T
 
 	  # Install the shared library and build the symlinks.
-	  func_show_eval "$install_prog $dir/$srcname $destdir/$realname" \
+	  func_show_eval "$install_prog -m 644 $dir/$srcname $destdir/$realname" \
 	      'exit $?'
 	  tstripme="$stripme"
 	  case $host_os in
