--- libtool-2.2.4_orig/libltdl/config/ltmain.sh	2008-06-26 04:55:25.000000000 +0200
+++ libtool-2.2.4/libltdl/config/ltmain.sh	2008-06-26 05:12:31.000000000 +0200
@@ -588,8 +588,8 @@
 nonopt=
 execute_dlfiles=
 preserve_args=
-lo2o="s/\\.lo\$/.${objext}/"
-o2lo="s/\\.${objext}\$/.lo/"
+lo2o="s/\\.lo\$/.${objext:-o}/"
+o2lo="s/\\.${objext:-o}\$/.lo/"
 extracted_archives=
 extracted_serial=0
 
@@ -1287,7 +1287,7 @@
     # Calculate the filename of the output object if compiler does
     # not support -o with -c
     if test "$compiler_c_o" = no; then
-      output_obj=`$ECHO "X$srcfile" | $Xsed -e 's%^.*/%%' -e 's%\.[^.]*$%%'`.${objext}
+      output_obj=`$ECHO "X$srcfile" | $Xsed -e 's%^.*/%%' -e 's%\.[^.]*$%%'`.${objext:-o}
       lockfile="$output_obj.lock"
     else
       output_obj=
@@ -1599,7 +1599,7 @@
 If OUTPUT-FILE ends in \`.a' or \`.lib', then a standard library is created
 using \`ar' and \`ranlib', or on Windows using \`lib'.
 
-If OUTPUT-FILE ends in \`.lo' or \`.${objext}', then a reloadable object file
+If OUTPUT-FILE ends in \`.lo' or \`.${objext:-o}', then a reloadable object file
 is created, otherwise an executable program is created."
         ;;
 
@@ -1974,7 +1974,7 @@
 
       # Do each installation.
       case $file in
-      *.$libext)
+      *.${libext:-a})
 	# Do the static libraries later.
 	staticlibs="$staticlibs $file"
 	;;
@@ -2103,7 +2103,7 @@
 	  func_lo2o "$destfile"
 	  staticdest=$func_lo2o_result
 	  ;;
-	*.$objext)
+	*.${objext:-o})
 	  staticdest="$destfile"
 	  destfile=
 	  ;;
@@ -2510,7 +2510,7 @@
 	func_show_eval '$RM "$output_objdir/$my_dlsyms" "$nlist" "${nlist}S" "${nlist}T"'
 
 	# Transform the symbol file into the correct name.
-	symfileobj="$output_objdir/${my_outputname}S.$objext"
+	symfileobj="$output_objdir/${my_outputname}S.${objext:-o}"
 	case $host in
 	*cygwin* | *mingw* )
 	  if test -f "$output_objdir/$my_outputname.def"; then
@@ -2683,7 +2683,7 @@
         func_extract_an_archive "$my_xdir" "$my_xabs"
 	;;
       esac
-      my_oldobjs="$my_oldobjs "`find $my_xdir -name \*.$objext -print -o -name \*.lo -print | $NL2SP`
+      my_oldobjs="$my_oldobjs "`find $my_xdir -name \*.${objext:-o} -print -o -name \*.lo -print | $NL2SP`
     done
 
     func_extract_archives_result="$my_oldobjs"
@@ -4253,7 +4253,7 @@
 	arg="$func_quote_for_eval_result"
 	;;
 
-      *.$objext)
+      *.${objext:-o})
 	# A standard object.
 	objs="$objs $arg"
 	;;
@@ -4341,7 +4341,7 @@
 	fi
 	;;
 
-      *.$libext)
+      *.${libext:-a})
 	# An archive.
 	deplibs="$deplibs $arg"
 	old_deplibs="$old_deplibs $arg"
@@ -4415,8 +4415,8 @@
     "")
       func_fatal_help "you must specify an output file"
       ;;
-    *.$libext) linkmode=oldlib ;;
-    *.lo | *.$objext) linkmode=obj ;;
+    *.${libext:-a}) linkmode=oldlib ;;
+    *.lo | *.${objext:-o}) linkmode=obj ;;
     *.la) linkmode=lib ;;
     *) linkmode=prog ;; # Anything else should be a program.
     esac
@@ -4683,7 +4683,7 @@
 	  continue
 	  ;;
 	*.la) lib="$deplib" ;;
-	*.$libext)
+	*.${libext:-a})
 	  if test "$pass" = conv; then
 	    deplibs="$deplib $deplibs"
 	    continue
@@ -4715,7 +4715,7 @@
 		$ECHO "*** I have the capability to make that library automatically link in when"
 		$ECHO "*** you link to this library.  But I can only do this if you have a"
 		$ECHO "*** shared version of the library, which you do not appear to have"
-		$ECHO "*** because the file extensions .$libext of this argument makes me believe"
+		$ECHO "*** because the file extensions .${libext:-a} of this argument makes me believe"
 		$ECHO "*** that it is just a static archive that I should not use here."
 	      else
 		$ECHO
@@ -4737,8 +4737,8 @@
 	    continue
 	    ;;
 	  esac # linkmode
-	  ;; # *.$libext
-	*.lo | *.$objext)
+	  ;; # *.${libext:-a}
+	*.lo | *.${objext:-o})
 	  if test "$pass" = conv; then
 	    deplibs="$deplib $deplibs"
 	  elif test "$linkmode" = prog; then
@@ -5623,7 +5623,7 @@
 	  # Some compilers have problems with a `.al' extension so
 	  # convenience libraries should have the same extension an
 	  # archive normally would.
-	  oldlibs="$output_objdir/$libname.$libext $oldlibs"
+	  oldlibs="$output_objdir/$libname.${libext:-a} $oldlibs"
 	  build_libtool_libs=convenience
 	  build_old_libs=yes
 	fi
@@ -5878,7 +5878,7 @@
 	tempremovelist=`$ECHO "$output_objdir/*"`
 	for p in $tempremovelist; do
 	  case $p in
-	    *.$objext)
+	    *.${objext:-o})
 	       ;;
 	    $output_objdir/$outputname | $output_objdir/$libname.* | $output_objdir/${libname}${release}.*)
 	       if test "X$precious_files_regex" != "X"; then
@@ -5898,10 +5898,10 @@
 
       # Now set the variables for building old libraries.
       if test "$build_old_libs" = yes && test "$build_libtool_libs" != convenience ; then
-	oldlibs="$oldlibs $output_objdir/$libname.$libext"
+	oldlibs="$oldlibs $output_objdir/$libname.${libext:-a}"
 
 	# Transform .lo files to .o files.
-	oldobjs="$objs "`$ECHO "X$libobjs" | $SP2NL | $Xsed -e '/\.'${libext}'$/d' -e "$lo2o" | $NL2SP`
+	oldobjs="$objs "`$ECHO "X$libobjs" | $SP2NL | $Xsed -e '/\.'${libext:-a}'$/d' -e "$lo2o" | $NL2SP`
       fi
 
       # Eliminate all temporary directories.
@@ -6279,7 +6279,7 @@
 	      $ECHO "*** \`nm' from GNU binutils and a full rebuild may help."
 	    fi
 	    if test "$build_old_libs" = no; then
-	      oldlibs="$output_objdir/$libname.$libext"
+	      oldlibs="$output_objdir/$libname.${libext:-a}"
 	      build_libtool_libs=module
 	      build_old_libs=yes
 	    else
@@ -6297,7 +6297,7 @@
 	      $ECHO "*** it was explicitly requested with -no-undefined,"
 	      $ECHO "*** libtool will only create a static version of it."
 	      if test "$build_old_libs" = no; then
-		oldlibs="$output_objdir/$libname.$libext"
+		oldlibs="$output_objdir/$libname.${libext:-a}"
 		build_libtool_libs=module
 		build_old_libs=yes
 	      else
@@ -6643,7 +6643,7 @@
 	  else
 	    if test -n "$save_libobjs"; then
 	      func_verbose "creating reloadable object files..."
-	      output=$output_objdir/$output_la-${k}.$objext
+	      output=$output_objdir/$output_la-${k}.${objext:-o}
 	      eval test_cmds=\"$reload_cmds\"
 	      func_len " $test_cmds"
 	      len0=$func_len_result
@@ -6669,10 +6669,10 @@
 		    # the last one created.
 		    eval concat_cmds=\"\$concat_cmds~$reload_cmds $objlist $last_robj~\$RM $last_robj\"
 		  fi
-		  last_robj=$output_objdir/$output_la-${k}.$objext
+		  last_robj=$output_objdir/$output_la-${k}.${objext:-o}
 		  func_arith $k + 1
 		  k=$func_arith_result
-		  output=$output_objdir/$output_la-${k}.$objext
+		  output=$output_objdir/$output_la-${k}.${objext:-o}
 		  objlist=$obj
 		  func_len " $last_robj"
 		  func_arith $len0 + $func_len_result
@@ -6918,7 +6918,7 @@
       fi
 
       # Create the old-style object.
-      reload_objs="$objs$old_deplibs "`$ECHO "X$libobjs" | $SP2NL | $Xsed -e '/\.'${libext}$'/d' -e '/\.lib$/d' -e "$lo2o" | $NL2SP`" $reload_conv_objs" ### testsuite: skip nested quoting test
+      reload_objs="$objs$old_deplibs "`$ECHO "X$libobjs" | $SP2NL | $Xsed -e '/\.'${libext:-a}$'/d' -e '/\.lib$/d' -e "$lo2o" | $NL2SP`" $reload_conv_objs" ### testsuite: skip nested quoting test
 
       output="$obj"
       func_execute_cmds "$reload_cmds" 'exit $?'
@@ -7166,8 +7166,8 @@
 	func_show_eval "$link_command" 'exit_status=$?'
 
 	# Delete the generated files.
-	if test -f "$output_objdir/${outputname}S.${objext}"; then
-	  func_show_eval '$RM "$output_objdir/${outputname}S.${objext}"'
+	if test -f "$output_objdir/${outputname}S.${objext:-o}"; then
+	  func_show_eval '$RM "$output_objdir/${outputname}S.${objext:-o}"'
 	fi
 
 	exit $exit_status
@@ -7488,7 +7488,7 @@
     case $output in
     *.la)
       old_library=
-      test "$build_old_libs" = yes && old_library="$libname.$libext"
+      test "$build_old_libs" = yes && old_library="$libname.${libext:-a}"
       func_verbose "creating $output"
 
       # Preserve any variables that may affect compiler behavior
@@ -7802,7 +7802,7 @@
 
 	    # note $name still contains .exe if it was in $file originally
 	    # as does the version of $file that was added into $rmfiles
-	    rmfiles="$rmfiles $objdir/$name $objdir/${name}S.${objext}"
+	    rmfiles="$rmfiles $objdir/$name $objdir/${name}S.${objext:-o}"
 	    if test "$fast_install" = yes && test -n "$relink_command"; then
 	      rmfiles="$rmfiles $objdir/lt-$name"
 	    fi
