diff -ru cpplibs-0.2b-20031008_orig/prapi/Histogram.cc cpplibs-0.2b-20031008/prapi/Histogram.cc
--- cpplibs-0.2b-20031008_orig/prapi/Histogram.cc	2008-07-14 00:56:16.000000000 +0200
+++ cpplibs-0.2b-20031008/prapi/Histogram.cc	2008-07-14 01:11:59.000000000 +0200
@@ -20,7 +20,7 @@
 {
 	List<int> IntegerHistogram::getFeatureVector(const Matrix<int>& mat) throw (FeatureExtractionException&)
 	{
-		int len = (mat.max()+1) >? _iLength;
+		int len = (mat.max()+1) > _iLength ? _iLength : mat.max()+1;
 		List<int> result(len);
 		result.setLength(len);
 		result = 0;
