diff -ru -ru net-snmp-5.4_orig/net-snmp-config.in net-snmp-5.4/net-snmp-config.in
--- net-snmp-5.4_orig/net-snmp-config.in	2007-01-18 15:08:15.000000000 +0100
+++ net-snmp-5.4/net-snmp-config.in	2007-01-18 15:10:12.000000000 +0100
@@ -88,10 +88,10 @@
       @MODULE_LIST@
       ;;
     --default-mibs|--mibs|--MIBS)
-      echo @DEFAULT_MIBS@
+      echo @NETSNMP_DEFAULT_MIBS@
       ;;
     --default-mibdirs|--mibdirs|--MIBDIRS)
-      echo @DEFAULT_MIBDIRS@
+      echo @NETSNMP_DEFAULT_MIBDIRS@
       ;;
     --snmpconfpath|--SNMPCONFPATH)
       echo "@SNMPCONFPATH@@ENV_SEPARATOR@@SNMPSHAREPATH@@ENV_SEPARATOR@@SNMPLIBPATH@@ENV_SEPARATOR@$HOME/.snmp@ENV_SEPARATOR@@PERSISTENT_DIRECTORY@"
