diff -ru log4cxx-0.9.7_orig/Makefile.am log4cxx-0.9.7/Makefile.am
--- log4cxx-0.9.7_orig/Makefile.am	2007-09-17 02:38:35.000000000 +0200
+++ log4cxx-0.9.7/Makefile.am	2007-09-17 02:54:06.000000000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = docs src msvc include tests performance examples
+SUBDIRS = docs src msvc include performance examples
 EXTRA_DIST = autogen.sh configure include license.apl
 dist-hook:
 	rm -rf\
diff -ru log4cxx-0.9.7_orig/include/log4cxx/xml/domconfigurator.h log4cxx-0.9.7/include/log4cxx/xml/domconfigurator.h
--- log4cxx-0.9.7_orig/include/log4cxx/xml/domconfigurator.h	2007-09-17 02:38:35.000000000 +0200
+++ log4cxx-0.9.7/include/log4cxx/xml/domconfigurator.h	2007-09-17 02:39:01.000000000 +0200
@@ -208,10 +208,10 @@
 				spi::LoggerRepositoryPtr& repository);
 
 		protected:
-			String DOMConfigurator::subst(const String& value);
+			String subst(const String& value);
 
 		protected:
-			void * appenderBag;
+			void *appenderBag;
 
 			helpers::Properties props;
 			spi::LoggerRepositoryPtr repository;
