diff -ru libwmf-0.2.8.4_orig/src/Makefile.in libwmf-0.2.8.4/src/Makefile.in
--- libwmf-0.2.8.4_orig/src/Makefile.in	2006-02-06 10:20:37.000000000 +0000
+++ libwmf-0.2.8.4/src/Makefile.in	2006-02-06 10:21:07.000000000 +0000
@@ -276,7 +276,7 @@
 libwmflite_la_LDFLAGS = $(SYS_LIBM) \
 	-no-undefined \
 	-version-info $(LT_LITE_CURRENT):$(LT_LITE_REVISION):$(LT_LITE_AGE) \
-	-release $(LT_RELEASE) -export-dynamic
+	-export-dynamic
 
 libwmf_la_SOURCES = font.c stream.c wmf.c xml.c
 @LIBWMF_OPT_TRIO_FALSE@LIBTRIO = 
@@ -287,7 +287,7 @@
 libwmf_la_LDFLAGS = \
 	-no-undefined \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
-	-release $(LT_RELEASE) -export-dynamic
+	-export-dynamic
 
 loaderdir = $(libdir)/$(GDK_PIXBUF_DIR)
 loader_LTLIBRARIES = $(GDK_PIXBUF_PLUGIN)
