diff -ru cmake-2.6.0_orig/Bootstrap.cmk/cmake_bootstrap.log cmake-2.6.0/Bootstrap.cmk/cmake_bootstrap.log
Only in cmake-2.6.0/: CMakeCPackOptions.cmake
diff -ru cmake-2.6.0_orig/CMakeCache.txt cmake-2.6.0/CMakeCache.txt
Only in cmake-2.6.0/CMakeFiles: CMakeCCompiler.cmake
Only in cmake-2.6.0/CMakeFiles: CMakeCXXCompiler.cmake
Only in cmake-2.6.0/CMakeFiles: CMakeDirectoryInformation.cmake
diff -ru cmake-2.6.0_orig/CMakeFiles/CMakeError.log cmake-2.6.0/CMakeFiles/CMakeError.log
Only in cmake-2.6.0/CMakeFiles: CMakeFiles
diff -ru cmake-2.6.0_orig/CMakeFiles/CMakeOutput.log cmake-2.6.0/CMakeFiles/CMakeOutput.log
diff -ru cmake-2.6.0_orig/CMakeFiles/CMakeSystem.cmake cmake-2.6.0/CMakeFiles/CMakeSystem.cmake
Only in cmake-2.6.0_orig/CMakeFiles: CompilerIdC
Only in cmake-2.6.0_orig/CMakeFiles: CompilerIdCXX
Only in cmake-2.6.0/CMakeFiles: Continuous.dir
Only in cmake-2.6.0/CMakeFiles: ContinuousBuild.dir
Only in cmake-2.6.0/CMakeFiles: ContinuousConfigure.dir
Only in cmake-2.6.0/CMakeFiles: ContinuousCoverage.dir
Only in cmake-2.6.0/CMakeFiles: ContinuousMemCheck.dir
Only in cmake-2.6.0/CMakeFiles: ContinuousStart.dir
Only in cmake-2.6.0/CMakeFiles: ContinuousSubmit.dir
Only in cmake-2.6.0/CMakeFiles: ContinuousTest.dir
Only in cmake-2.6.0/CMakeFiles: ContinuousUpdate.dir
Only in cmake-2.6.0/CMakeFiles: Experimental.dir
Only in cmake-2.6.0/CMakeFiles: ExperimentalBuild.dir
Only in cmake-2.6.0/CMakeFiles: ExperimentalConfigure.dir
Only in cmake-2.6.0/CMakeFiles: ExperimentalCoverage.dir
Only in cmake-2.6.0/CMakeFiles: ExperimentalMemCheck.dir
Only in cmake-2.6.0/CMakeFiles: ExperimentalStart.dir
Only in cmake-2.6.0/CMakeFiles: ExperimentalSubmit.dir
Only in cmake-2.6.0/CMakeFiles: ExperimentalTest.dir
Only in cmake-2.6.0/CMakeFiles: ExperimentalUpdate.dir
Only in cmake-2.6.0/CMakeFiles: Makefile.cmake
Only in cmake-2.6.0/CMakeFiles: Makefile2
Only in cmake-2.6.0/CMakeFiles: Nightly.dir
Only in cmake-2.6.0/CMakeFiles: NightlyBuild.dir
Only in cmake-2.6.0/CMakeFiles: NightlyConfigure.dir
Only in cmake-2.6.0/CMakeFiles: NightlyCoverage.dir
Only in cmake-2.6.0/CMakeFiles: NightlyMemCheck.dir
Only in cmake-2.6.0/CMakeFiles: NightlyMemoryCheck.dir
Only in cmake-2.6.0/CMakeFiles: NightlyStart.dir
Only in cmake-2.6.0/CMakeFiles: NightlySubmit.dir
Only in cmake-2.6.0/CMakeFiles: NightlyTest.dir
Only in cmake-2.6.0/CMakeFiles: NightlyUpdate.dir
Only in cmake-2.6.0/CMakeFiles: progress.make
Only in cmake-2.6.0/CMakeFiles: uninstall.dir
Only in cmake-2.6.0_orig: CMakeGraphVizOptions.cmake
Only in cmake-2.6.0/: CPackConfig.cmake
Only in cmake-2.6.0/: CPackSourceConfig.cmake
Only in cmake-2.6.0_orig: CTestConfig.cmake
Only in cmake-2.6.0/: CTestCustom.cmake
Only in cmake-2.6.0/: CTestCustom.ctest
Only in cmake-2.6.0/: CTestScript.cmake
Only in cmake-2.6.0_orig: DartConfig.cmake
Only in cmake-2.6.0/: DartConfiguration.tcl
Only in cmake-2.6.0/: DartLocal.conf
Only in cmake-2.6.0/: DartTestfile.txt
Only in cmake-2.6.0/Docs: Copyright.txt
Only in cmake-2.6.0/Docs: ccmake.1
Only in cmake-2.6.0/Docs: ccmake.html
Only in cmake-2.6.0/Docs: ccmake.txt
Only in cmake-2.6.0/Docs: cmake-commands.html
Only in cmake-2.6.0/Docs: cmake-commands.txt
Only in cmake-2.6.0/Docs: cmake-compatcommands.html
Only in cmake-2.6.0/Docs: cmake-compatcommands.txt
Only in cmake-2.6.0/Docs: cmake-modules.html
Only in cmake-2.6.0/Docs: cmake-modules.txt
Only in cmake-2.6.0/Docs: cmake-properties.html
Only in cmake-2.6.0/Docs: cmake-properties.txt
Only in cmake-2.6.0/Docs: cmake-variables.html
Only in cmake-2.6.0/Docs: cmake-variables.txt
Only in cmake-2.6.0/Docs: cmake.1
Only in cmake-2.6.0/Docs: cmake.html
Only in cmake-2.6.0/Docs: cmake.txt
Only in cmake-2.6.0/Docs: cmakecommands.1
Only in cmake-2.6.0/Docs: cmakecompat.1
Only in cmake-2.6.0/Docs: cmakemodules.1
Only in cmake-2.6.0/Docs: cmakeprops.1
Only in cmake-2.6.0/Docs: cmakevars.1
Only in cmake-2.6.0/Docs: cpack.1
Only in cmake-2.6.0/Docs: cpack.html
Only in cmake-2.6.0/Docs: cpack.txt
Only in cmake-2.6.0/Docs: ctest.1
Only in cmake-2.6.0/Docs: ctest.html
Only in cmake-2.6.0/Docs: ctest.txt
Only in cmake-2.6.0/: Makefile
Only in cmake-2.6.0/Modules: .NoDartCoverage
Only in cmake-2.6.0/Modules: CMakeFiles
Only in cmake-2.6.0/Modules: DartTestfile.txt
Only in cmake-2.6.0/Modules: Makefile
Only in cmake-2.6.0_orig/Modules/Platform: AIX-VisualAge-Fortran.cmake
Only in cmake-2.6.0_orig/Modules/Platform: AIX.cmake
Only in cmake-2.6.0_orig/Modules/Platform: BSDOS.cmake
Only in cmake-2.6.0_orig/Modules/Platform: BeOS.cmake
Only in cmake-2.6.0_orig/Modules/Platform: BlueGeneL.cmake
Only in cmake-2.6.0/Modules/Platform: CMakeFiles
Only in cmake-2.6.0_orig/Modules/Platform: CYGWIN-g77.cmake
Only in cmake-2.6.0_orig/Modules/Platform: CYGWIN.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Catamount.cmake
Only in cmake-2.6.0/Modules/Platform: DartTestfile.txt
Only in cmake-2.6.0_orig/Modules/Platform: Darwin-icc.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Darwin-icpc.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Darwin-xlc.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Darwin.cmake
Only in cmake-2.6.0_orig/Modules/Platform: DragonFly.cmake
Only in cmake-2.6.0_orig/Modules/Platform: FreeBSD.cmake
Only in cmake-2.6.0_orig/Modules/Platform: GNU.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Generic-ADSP-ASM.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Generic-ADSP-C.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Generic-ADSP-CXX.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Generic-ADSP-Common.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Generic-SDCC-C.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Generic.cmake
Only in cmake-2.6.0_orig/Modules/Platform: HP-UX.cmake
Only in cmake-2.6.0_orig/Modules/Platform: IRIX.cmake
Only in cmake-2.6.0_orig/Modules/Platform: IRIX64.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-GNU-Fortran.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-Intel-C.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-Intel-CXX.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-Intel-Fortran.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-PGI-C.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-PGI-CXX.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-SunPro-C.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-SunPro-CXX.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-SunPro-Fortran.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-VisualAge-C.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-VisualAge-Fortran.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-como.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-icpc.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux-ifort.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Linux.cmake
Only in cmake-2.6.0_orig/Modules/Platform: MP-RAS.cmake
Only in cmake-2.6.0/Modules/Platform: Makefile
Only in cmake-2.6.0_orig/Modules/Platform: NetBSD.cmake
Only in cmake-2.6.0_orig/Modules/Platform: OSF1.cmake
Only in cmake-2.6.0_orig/Modules/Platform: OpenBSD.cmake
Only in cmake-2.6.0_orig/Modules/Platform: QNX.cmake
Only in cmake-2.6.0_orig/Modules/Platform: RISCos.cmake
Only in cmake-2.6.0_orig/Modules/Platform: SCO_SV.cmake
Only in cmake-2.6.0_orig/Modules/Platform: SINIX.cmake
Only in cmake-2.6.0_orig/Modules/Platform: SunOS-SunPro-Fortran.cmake
Only in cmake-2.6.0_orig/Modules/Platform: SunOS.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Tru64.cmake
Only in cmake-2.6.0_orig/Modules/Platform: ULTRIX.cmake
Only in cmake-2.6.0_orig/Modules/Platform: UNIX_SV.cmake
Only in cmake-2.6.0_orig/Modules/Platform: UnixPaths.cmake
Only in cmake-2.6.0_orig/Modules/Platform: UnixWare.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Windows-bcc32.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Windows-cl.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Windows-df.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Windows-g++.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Windows-g77.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Windows-gcc.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Windows-icl.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Windows-ifort.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Windows-wcl386.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Windows.cmake
Only in cmake-2.6.0_orig/Modules/Platform: WindowsPaths.cmake
Only in cmake-2.6.0_orig/Modules/Platform: Xenix.cmake
Only in cmake-2.6.0_orig/Modules/Platform: cl.cmake
Only in cmake-2.6.0/Modules/Platform: cmake_install.cmake
Only in cmake-2.6.0_orig/Modules/Platform: eCos.cmake
Only in cmake-2.6.0_orig/Modules/Platform: g77.cmake
Only in cmake-2.6.0_orig/Modules/Platform: gas.cmake
Only in cmake-2.6.0_orig/Modules/Platform: gcc.cmake
Only in cmake-2.6.0_orig/Modules/Platform: kFreeBSD.cmake
Only in cmake-2.6.0_orig/Modules/Platform: syllable.cmake
Only in cmake-2.6.0_orig/Modules/Platform: xlf.cmake
Only in cmake-2.6.0/Modules: cmake_install.cmake
Only in cmake-2.6.0/Source: CMakeFiles
diff -ru cmake-2.6.0_orig/Source/CMakeLists.txt cmake-2.6.0/Source/CMakeLists.txt
--- cmake-2.6.0_orig/Source/CMakeLists.txt	2008-06-26 11:03:03.000000000 +0200
+++ cmake-2.6.0/Source/CMakeLists.txt	2008-06-26 12:41:15.000000000 +0200
@@ -242,13 +242,13 @@
 ENDIF(APPLE)
 
 
-IF (WIN32)
+#IF (WIN32)
   SET(SRCS ${SRCS}
     cmCallVisualStudioMacro.cxx
     cmCallVisualStudioMacro.h
     )
 
-  IF(NOT UNIX)
+#  IF(NOT UNIX)
     SET(SRCS ${SRCS}
       cmGlobalBorlandMakefileGenerator.cxx
       cmGlobalBorlandMakefileGenerator.h
@@ -279,11 +279,16 @@
       cmLocalVisualStudio7Generator.h
       cmLocalVisualStudioGenerator.cxx
       cmLocalVisualStudioGenerator.h
-      cmWin32ProcessExecution.cxx
-      cmWin32ProcessExecution.h
       )
-  ENDIF(NOT UNIX)
-ENDIF (WIN32)
+      
+#   IF(WIN32 AND NOT UNIX AND NOT LINUX)
+#     SET(SRCS ${SRCS}
+#         cmWin32ProcessExecution.cxx
+#         cmWin32ProcessExecution.h)
+#   ENDIF(WIN32 AND NOT UNIX AND NOT LINUX)
+
+#  ENDIF(NOT UNIX)
+#ENDIF (WIN32)
 
 # create a library used by the command line and the GUI
 ADD_LIBRARY(CMakeLib ${SRCS})
Only in cmake-2.6.0/Source/CPack: cmCPackConfigure.h
Only in cmake-2.6.0/Source/CursesDialog/form: CMakeFiles
Only in cmake-2.6.0/Source/CursesDialog/form: DartTestfile.txt
Only in cmake-2.6.0/Source/CursesDialog/form: Makefile
Only in cmake-2.6.0/Source/CursesDialog/form: cmFormConfigure.h
Only in cmake-2.6.0/Source/CursesDialog/form: cmake_install.cmake
Only in cmake-2.6.0/Source/CursesDialog/form: libcmForm.a
Only in cmake-2.6.0/Source: DartTestfile.txt
Only in cmake-2.6.0/Source: Makefile
Only in cmake-2.6.0_orig/Source/QtDialog: CMakeIngestOSXBundleLibraries.cmake
Only in cmake-2.6.0/Source: cmConfigure.h
diff -ru cmake-2.6.0_orig/Source/cmGlobalVisualStudio71Generator.cxx cmake-2.6.0/Source/cmGlobalVisualStudio71Generator.cxx
--- cmake-2.6.0_orig/Source/cmGlobalVisualStudio71Generator.cxx	2008-06-26 11:03:05.000000000 +0200
+++ cmake-2.6.0/Source/cmGlobalVisualStudio71Generator.cxx	2008-06-26 12:06:30.000000000 +0200
@@ -14,7 +14,9 @@
      PURPOSE.  See the above copyright notices for more information.
 
 =========================================================================*/
+#if defined(_WIN32)
 #include "windows.h" // this must be first to define GetCurrentDirectory
+#endif
 #include "cmGlobalVisualStudio71Generator.h"
 #include "cmLocalVisualStudio7Generator.h"
 #include "cmMakefile.h"
diff -ru cmake-2.6.0_orig/Source/cmGlobalVisualStudio7Generator.cxx cmake-2.6.0/Source/cmGlobalVisualStudio7Generator.cxx
--- cmake-2.6.0_orig/Source/cmGlobalVisualStudio7Generator.cxx	2008-06-26 11:03:03.000000000 +0200
+++ cmake-2.6.0/Source/cmGlobalVisualStudio7Generator.cxx	2008-06-26 12:07:15.000000000 +0200
@@ -14,7 +14,9 @@
      PURPOSE.  See the above copyright notices for more information.
 
 =========================================================================*/
+#if defined(_WIN32)
 #include "windows.h" // this must be first to define GetCurrentDirectory
+#endif
 #include "cmGlobalVisualStudio7Generator.h"
 #include "cmGeneratedFileStream.h"
 #include "cmLocalVisualStudio7Generator.h"
@@ -652,6 +654,7 @@
     {
     return;
     }
+#if defined(_WIN32)
   std::string ret;
   UUID uid;
   unsigned char *uidstr;
@@ -661,8 +664,9 @@
   RpcStringFree(&uidstr);
   ret = cmSystemTools::UpperCase(ret);
   this->CMakeInstance->AddCacheEntry(guidStoreName.c_str(), 
-                                     ret.c_str(), "Stored GUID", 
+                                     ret.c_str(), "Stored GUID",
                                      cmCacheManager::INTERNAL);
+#endif
 }
 
 std::vector<std::string> *cmGlobalVisualStudio7Generator::GetConfigurations()
diff -ru cmake-2.6.0_orig/Source/cmGlobalVisualStudio8Generator.cxx cmake-2.6.0/Source/cmGlobalVisualStudio8Generator.cxx
--- cmake-2.6.0_orig/Source/cmGlobalVisualStudio8Generator.cxx	2008-06-26 11:03:03.000000000 +0200
+++ cmake-2.6.0/Source/cmGlobalVisualStudio8Generator.cxx	2008-06-26 12:07:34.000000000 +0200
@@ -14,7 +14,9 @@
      PURPOSE.  See the above copyright notices for more information.
 
 =========================================================================*/
+#if defined(_WIN32)
 #include "windows.h" // this must be first to define GetCurrentDirectory
+#endif
 #include "cmGlobalVisualStudio8Generator.h"
 #include "cmLocalVisualStudio7Generator.h"
 #include "cmMakefile.h"
diff -ru cmake-2.6.0_orig/Source/cmGlobalVisualStudio8Win64Generator.cxx cmake-2.6.0/Source/cmGlobalVisualStudio8Win64Generator.cxx
--- cmake-2.6.0_orig/Source/cmGlobalVisualStudio8Win64Generator.cxx	2008-06-26 11:03:03.000000000 +0200
+++ cmake-2.6.0/Source/cmGlobalVisualStudio8Win64Generator.cxx	2008-06-26 12:08:10.000000000 +0200
@@ -14,7 +14,9 @@
      PURPOSE.  See the above copyright notices for more information.
 
 =========================================================================*/
+#ifdef _WIN32
 #include "windows.h" // this must be first to define GetCurrentDirectory
+#endif
 #include "cmGlobalVisualStudio8Win64Generator.h"
 #include "cmLocalVisualStudio7Generator.h"
 #include "cmMakefile.h"
diff -ru cmake-2.6.0_orig/Source/cmGlobalVisualStudio9Generator.cxx cmake-2.6.0/Source/cmGlobalVisualStudio9Generator.cxx
--- cmake-2.6.0_orig/Source/cmGlobalVisualStudio9Generator.cxx	2008-06-26 11:03:03.000000000 +0200
+++ cmake-2.6.0/Source/cmGlobalVisualStudio9Generator.cxx	2008-06-26 12:07:54.000000000 +0200
@@ -14,7 +14,9 @@
      PURPOSE.  See the above copyright notices for more information.
 
 =========================================================================*/
+#if defined(_WIN32)
 #include "windows.h" // this must be first to define GetCurrentDirectory
+#endif
 #include "cmGlobalVisualStudio9Generator.h"
 #include "cmLocalVisualStudio7Generator.h"
 #include "cmMakefile.h"
diff -ru cmake-2.6.0_orig/Source/cmGlobalVisualStudioGenerator.cxx cmake-2.6.0/Source/cmGlobalVisualStudioGenerator.cxx
--- cmake-2.6.0_orig/Source/cmGlobalVisualStudioGenerator.cxx	2008-06-26 11:03:05.000000000 +0200
+++ cmake-2.6.0/Source/cmGlobalVisualStudioGenerator.cxx	2008-06-26 12:10:14.000000000 +0200
@@ -401,13 +401,16 @@
 }
 
 //----------------------------------------------------------------------------
+#if defined(_WIN32)
 #include <windows.h>
+#endif
 
 //----------------------------------------------------------------------------
 bool IsVisualStudioMacrosFileRegistered(const std::string& macrosFile,
   const std::string& regKeyBase,
   std::string& nextAvailableSubKeyName)
 {
+#if defined(_WIN32)
   bool macrosRegistered = false;
 
   std::string s1;
@@ -570,6 +573,9 @@
     }
 
   return macrosRegistered;
+#else
+  return false;
+#endif
 }
 
 //----------------------------------------------------------------------------
@@ -578,6 +584,7 @@
   const std::string& macrosFile,
   const std::string& regKeyBase)
 {
+#if defined(_WIN32)
   std::string keyname = regKeyBase + "\\OtherProjects7";
   HKEY hkey = NULL;
   LONG result = RegOpenKeyEx(HKEY_CURRENT_USER, keyname.c_str(), 0,
@@ -639,6 +646,7 @@
     std::cout << "error opening key: " << keyname << std::endl;
     std::cout << std::endl;
     }
+#endif
 }
 
 //----------------------------------------------------------------------------
diff -ru cmake-2.6.0_orig/Source/cmLocalVisualStudio7Generator.cxx cmake-2.6.0/Source/cmLocalVisualStudio7Generator.cxx
--- cmake-2.6.0_orig/Source/cmLocalVisualStudio7Generator.cxx	2008-06-26 11:03:03.000000000 +0200
+++ cmake-2.6.0/Source/cmLocalVisualStudio7Generator.cxx	2008-06-26 12:11:02.000000000 +0200
@@ -2320,9 +2320,12 @@
 }
 
 //----------------------------------------------------------------------------
+#if defined(_WIN32)
 #include <windows.h>
+#endif
 static bool cmLVS6G_IsFAT(const char* dir)
 {
+#if defined(_WIN32)
   if(dir[0] && dir[1] == ':')
     {
     char volRoot[4] = "_:/";
@@ -2334,5 +2337,6 @@
       return true;
       }
     }
+#endif
   return false;
 }
diff -ru cmake-2.6.0_orig/Source/cmLocalVisualStudioGenerator.cxx cmake-2.6.0/Source/cmLocalVisualStudioGenerator.cxx
--- cmake-2.6.0_orig/Source/cmLocalVisualStudioGenerator.cxx	2008-06-26 11:03:03.000000000 +0200
+++ cmake-2.6.0/Source/cmLocalVisualStudioGenerator.cxx	2008-06-26 12:11:20.000000000 +0200
@@ -19,7 +19,9 @@
 #include "cmMakefile.h"
 #include "cmSourceFile.h"
 #include "cmSystemTools.h"
+#if defined(_WIN32)
 #include "windows.h"
+#endif
 
 //----------------------------------------------------------------------------
 cmLocalVisualStudioGenerator::cmLocalVisualStudioGenerator()
diff -ru cmake-2.6.0_orig/Source/cmWin32ProcessExecution.h cmake-2.6.0/Source/cmWin32ProcessExecution.h
--- cmake-2.6.0_orig/Source/cmWin32ProcessExecution.h	2008-06-26 11:03:02.000000000 +0200
+++ cmake-2.6.0/Source/cmWin32ProcessExecution.h	2008-06-26 12:42:09.000000000 +0200
@@ -18,6 +18,7 @@
 #define cmWin32ProcessExecution_h
 
 #include "cmStandardIncludes.h"
+#if defined(_WIN32)
 #include "windows.h"
 
 class cmMakefile;
@@ -170,5 +171,6 @@
   bool HideWindows;
 };
 
+#endif
 
 #endif
diff -ru cmake-2.6.0_orig/Source/cmake.cxx cmake-2.6.0/Source/cmake.cxx
--- cmake-2.6.0_orig/Source/cmake.cxx	2008-06-26 11:03:03.000000000 +0200
+++ cmake-2.6.0/Source/cmake.cxx	2008-06-26 12:45:57.000000000 +0200
@@ -58,7 +58,7 @@
 #endif
 
 // include the generator
-#if defined(_WIN32) && !defined(__CYGWIN__)
+#if 1 //defined(_WIN32) && !defined(__CYGWIN__)
 #  if !defined(CMAKE_BOOT_MINGW)
 #    include "cmGlobalVisualStudio6Generator.h"
 #    include "cmGlobalVisualStudio7Generator.h"
@@ -2386,7 +2386,7 @@
 
 void cmake::AddDefaultGenerators()
 {
-#if defined(_WIN32) && !defined(__CYGWIN__)
+#if 1 //defined(_WIN32) && !defined(__CYGWIN__)
 # if !defined(CMAKE_BOOT_MINGW)
   this->Generators[cmGlobalVisualStudio6Generator::GetActualName()] =
     &cmGlobalVisualStudio6Generator::New;
Only in cmake-2.6.0/Source: cmake_install.cmake
Only in cmake-2.6.0/Source: cmsys
Only in cmake-2.6.0/Source/kwsys: CMakeFiles
Only in cmake-2.6.0_orig/Source/kwsys: CTestConfig.cmake
Only in cmake-2.6.0_orig/Source/kwsys: CheckCXXSourceRuns.cmake
Only in cmake-2.6.0/Source/kwsys: DartTestfile.txt
Only in cmake-2.6.0/Source/kwsys: Makefile
Only in cmake-2.6.0/Source/kwsys: cmake_install.cmake
Only in cmake-2.6.0/Source/kwsys: cmsysTestsC.c
Only in cmake-2.6.0/Source/kwsys: cmsysTestsCxx.cxx
Only in cmake-2.6.0_orig/Source/kwsys: kwsysDateStamp.cmake
Only in cmake-2.6.0/Source/kwsys: libcmsys.a
Only in cmake-2.6.0/Source/kwsys: libcmsysTestDynload.so
Only in cmake-2.6.0/Source/kwsys: libcmsys_c.a
Only in cmake-2.6.0/Source/kwsys: testSystemTools.h
Only in cmake-2.6.0/Source: libCMakeLib.a
Only in cmake-2.6.0/Source: libCPackLib.a
Only in cmake-2.6.0/Source: libCTestLib.a
Only in cmake-2.6.0/Templates: CMakeFiles
Only in cmake-2.6.0_orig/Templates: CMakeVisualStudio6Configurations.cmake
Only in cmake-2.6.0/Templates: DartTestfile.txt
Only in cmake-2.6.0/Templates: Makefile
Only in cmake-2.6.0/Templates: cmake_install.cmake
Only in cmake-2.6.0/: Testing
Only in cmake-2.6.0/Tests: CMakeFiles
Only in cmake-2.6.0/Tests/CMakeTests: CMakeFiles
Only in cmake-2.6.0/Tests/CMakeTests: DartTestfile.txt
Only in cmake-2.6.0_orig/Tests/CMakeTests: DummyToolchain.cmake
Only in cmake-2.6.0/Tests/CMakeTests: FindBaseTest.cmake
Only in cmake-2.6.0/Tests/CMakeTests: IncludeTest.cmake
Only in cmake-2.6.0/Tests/CMakeTests: ListTest.cmake
Only in cmake-2.6.0/Tests/CMakeTests: Makefile
Only in cmake-2.6.0/Tests/CMakeTests: ToolchainTest.cmake
Only in cmake-2.6.0/Tests/CMakeTests: VariableWatchTest.cmake
Only in cmake-2.6.0/Tests/CMakeTests: cmake_install.cmake
Only in cmake-2.6.0_orig/Tests/CommandLineTest: PreLoad.cmake
Only in cmake-2.6.0_orig/Tests/Complex/Executable: Included.cmake
Only in cmake-2.6.0_orig/Tests/Complex/Library: test_preprocess.cmake
Only in cmake-2.6.0_orig/Tests/Complex: VarTests.cmake
Only in cmake-2.6.0_orig/Tests/ComplexOneConfig/Executable: Included.cmake
Only in cmake-2.6.0_orig/Tests/ComplexOneConfig/Library: test_preprocess.cmake
Only in cmake-2.6.0_orig/Tests/ComplexOneConfig: VarTests.cmake
Only in cmake-2.6.0_orig/Tests/ComplexRelativePaths/Executable: Included.cmake
Only in cmake-2.6.0_orig/Tests/ComplexRelativePaths/Library: test_preprocess.cmake
Only in cmake-2.6.0_orig/Tests/ComplexRelativePaths: VarTests.cmake
Only in cmake-2.6.0_orig/Tests/CustomCommand: check_mark.cmake
Only in cmake-2.6.0/Tests: DartTestfile.txt
Only in cmake-2.6.0_orig/Tests/FindPackageTest: FindVersionTestA.cmake
Only in cmake-2.6.0_orig/Tests/FindPackageTest: FindVersionTestB.cmake
Only in cmake-2.6.0_orig/Tests/FindPackageTest: FindVersionTestC.cmake
Only in cmake-2.6.0_orig/Tests/FunctionTest: Util.cmake
Only in cmake-2.6.0/Tests: Makefile
Only in cmake-2.6.0_orig/Tests/SimpleInstall: InstallScript1.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstall: InstallScript2.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstall: InstallScript3.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstall: InstallScript4.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstall: PackageScript.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstall: PostInstall.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstall: PreInstall.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstallS2: InstallScript1.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstallS2: InstallScript2.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstallS2: InstallScript3.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstallS2: InstallScript4.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstallS2: PackageScript.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstallS2: PostInstall.cmake
Only in cmake-2.6.0_orig/Tests/SimpleInstallS2: PreInstall.cmake
Only in cmake-2.6.0_orig/Tests/Testing: DartConfig.cmake
Only in cmake-2.6.0_orig/Tests/Tutorial/Step7: CTestConfig.cmake
Only in cmake-2.6.0_orig/Tests/Tutorial/Step7: build1.cmake
Only in cmake-2.6.0_orig/Tests/Tutorial/Step7: build2.cmake
Only in cmake-2.6.0/Tests: cmake_install.cmake
Only in cmake-2.6.0/Utilities: CMakeFiles
Only in cmake-2.6.0/Utilities: DartTestfile.txt
Only in cmake-2.6.0/Utilities/Doxygen: CMakeFiles
Only in cmake-2.6.0/Utilities/Doxygen: DartTestfile.txt
Only in cmake-2.6.0/Utilities/Doxygen: Makefile
Only in cmake-2.6.0/Utilities/Doxygen: cmake_install.cmake
Only in cmake-2.6.0/Utilities/KWStyle: CMakeFiles
Only in cmake-2.6.0/Utilities/KWStyle: DartTestfile.txt
Only in cmake-2.6.0/Utilities/KWStyle: Makefile
Only in cmake-2.6.0/Utilities/KWStyle: cmake_install.cmake
Only in cmake-2.6.0/Utilities: Makefile
Only in cmake-2.6.0/Utilities: cmThirdParty.h
Only in cmake-2.6.0/Utilities: cmake_install.cmake
Only in cmake-2.6.0/Utilities/cmcompress: CMakeFiles
Only in cmake-2.6.0/Utilities/cmcompress: DartTestfile.txt
Only in cmake-2.6.0/Utilities/cmcompress: Makefile
Only in cmake-2.6.0/Utilities/cmcompress: cmake_install.cmake
Only in cmake-2.6.0/Utilities/cmcompress: libcmcompress.a
Only in cmake-2.6.0/Utilities/cmcurl: CMakeFiles
Only in cmake-2.6.0/Utilities/cmcurl: DartTestfile.txt
Only in cmake-2.6.0/Utilities/cmcurl: Makefile
Only in cmake-2.6.0/Utilities/cmcurl: cmake_install.cmake
Only in cmake-2.6.0/Utilities/cmcurl: config.h
Only in cmake-2.6.0/Utilities/cmcurl: libcmcurl.a
Only in cmake-2.6.0/Utilities/cmexpat: CMakeFiles
Only in cmake-2.6.0/Utilities/cmexpat: DartTestfile.txt
Only in cmake-2.6.0/Utilities/cmexpat: Makefile
Only in cmake-2.6.0/Utilities/cmexpat: cmake_install.cmake
Only in cmake-2.6.0/Utilities/cmexpat: expatConfig.h
Only in cmake-2.6.0/Utilities/cmexpat: expatDllConfig.h
Only in cmake-2.6.0/Utilities/cmexpat: libcmexpat.a
Only in cmake-2.6.0/Utilities/cmtar: CMakeFiles
Only in cmake-2.6.0/Utilities/cmtar: DartTestfile.txt
Only in cmake-2.6.0/Utilities/cmtar: Makefile
Only in cmake-2.6.0/Utilities/cmtar: cmake_install.cmake
Only in cmake-2.6.0/Utilities/cmtar: libcmtar.a
Only in cmake-2.6.0/Utilities/cmtar: libtar
Only in cmake-2.6.0/Utilities/cmtar: libtarint
Only in cmake-2.6.0/Utilities/cmtar/listhash: libtar_hash.c
Only in cmake-2.6.0/Utilities/cmtar/listhash: libtar_list.c
Only in cmake-2.6.0/Utilities/cmxmlrpc: CMakeFiles
Only in cmake-2.6.0/Utilities/cmxmlrpc: DartTestfile.txt
Only in cmake-2.6.0/Utilities/cmxmlrpc: Makefile
Only in cmake-2.6.0/Utilities/cmxmlrpc: cmake_install.cmake
Only in cmake-2.6.0/Utilities/cmxmlrpc: libcmXMLRPC.a
Only in cmake-2.6.0/Utilities/cmxmlrpc: xmlrpc_amconfig.h
Only in cmake-2.6.0/Utilities/cmxmlrpc: xmlrpc_config.h
Only in cmake-2.6.0/Utilities/cmzlib: CMakeFiles
Only in cmake-2.6.0/Utilities/cmzlib: DartTestfile.txt
Only in cmake-2.6.0/Utilities/cmzlib: Makefile
Only in cmake-2.6.0/Utilities/cmzlib: cmake_install.cmake
Only in cmake-2.6.0/Utilities/cmzlib: libcmzlib.a
Only in cmake-2.6.0/Utilities/cmzlib: zlibDllConfig.h
Only in cmake-2.6.0/: bin
Only in cmake-2.6.0/: cmake_install.cmake
Only in cmake-2.6.0/: cmake_uninstall.cmake
Only in cmake-2.6.0/: la
