--- bundle/ngx_lua-0.10.15/src/ngx_http_lua_util.c	2019-11-04 09:41:48.275203259 +0100
+++ bundle/ngx_lua-0.10.15/src/ngx_http_lua_util.c.orig	2019-11-04 09:41:41.667137445 +0100
@@ -808,11 +808,11 @@
                 "if phase == 'init_worker' or phase == 'init' then\n"
                     "return\n"
                 "end\n"
-               /* "ngx_log(ngx_WARN, 'writing a global lua variable "
+                "ngx_log(ngx_WARN, 'writing a global lua variable "
                          "(\\'', tostring(key), '\\') which may lead to "
                          "race conditions between concurrent requests, so "
                          "prefer the use of \\'local\\' variables', "
-                         "traceback('', 2))\n"*/
+                         "traceback('', 2))\n"
             "end\n"
             "setmetatable(_G, { __newindex = newindex })\n"
             ;
