diff -ru -ru netpbm-10.31_orig/converter/other/Makefile netpbm-10.31/converter/other/Makefile
--- netpbm-10.31_orig/converter/other/Makefile	2006-09-25 15:30:28.000000000 +0200
+++ netpbm-10.31/converter/other/Makefile	2006-09-25 15:31:45.000000000 +0200
@@ -7,7 +7,7 @@
 
 include $(BUILDDIR)/Makefile.config
 
-SUBDIRS = jbig pnmtopalm jpeg2000 cameratopam
+SUBDIRS = jbig pnmtopalm cameratopam
 ifneq ($(BUILD_FIASCO), N)
   SUBDIRS += fiasco
 endif
diff -ru -ru netpbm-10.31_orig/converter/other/cameratopam/identify.c netpbm-10.31/converter/other/cameratopam/identify.c
--- netpbm-10.31_orig/converter/other/cameratopam/identify.c	2006-09-25 15:30:28.000000000 +0200
+++ netpbm-10.31/converter/other/cameratopam/identify.c	2006-09-25 15:32:12.000000000 +0200
@@ -25,8 +25,8 @@
 
 
 /* This does the same as the function of the same name in the GNU C library */
-static const char *memmem (const char *haystack, size_t haystacklen,
-                     const char *needle, size_t needlelen)
+static const char *my_memmem (const char *haystack, size_t haystacklen,
+                              const char *needle, size_t needlelen)
 {
   const char *c;
   for (c = haystack; c <= haystack + haystacklen - needlelen; c++)
@@ -316,7 +316,7 @@
   fread (head, 1, 32, ifp);
   fseek (ifp, 0, SEEK_END);
   fsize = ftell(ifp);
-  if ((c = (char*)memmem (head, 32, "MMMMRawT", 8))) {
+  if ((c = (char*)my_memmem (head, 32, "MMMMRawT", 8))) {
     strcpy (make, "Phase One");
     data_offset = c - head;
     fseek (ifp, data_offset + 8, SEEK_SET);
