--- db-4.5.20/dist/Makefile.in	2006-09-13 18:22:38.000000000 +0200
+++ db-4.5.20_patched/dist/Makefile.in	2007-08-31 19:32:32.000000000 +0200
@@ -62,11 +62,11 @@
 
 libdb_base=	libdb
 libdb=		$(libdb_base).a
-libdb_version=	$(libdb_base)-$(LIBVERSION).a
-libso=		$(libdb_base)-$(LIBVERSION)@SOSUFFIX@
-libso_target=	$(libdb_base)-$(LIBVERSION).la
+libdb_version=	$(libdb_base).a
+libso=		$(libdb_base)@SOSUFFIX@.$(LIBVERSION)
+libso_target=	$(libdb_base).la
 libso_default=	$(libdb_base)@SOSUFFIX@
-libso_major=	$(libdb_base)-$(LIBMAJOR)@SOSUFFIX@
+libso_major=	$(libdb_base)@SOSUFFIX@.$(LIBMAJOR)
 
 ##################################################
 # C++ API.
@@ -81,11 +81,11 @@
 
 libcxx_base=	libdb_cxx
 libcxx=		$(libcxx_base).a
-libcxx_version=	$(libcxx_base)-$(LIBVERSION).a
-libxso=		$(libcxx_base)-$(LIBVERSION)@SOSUFFIX@
-libxso_target=	$(libcxx_base)-$(LIBVERSION).la
+libcxx_version=	$(libcxx_base).a
+libxso=		$(libcxx_base)@SOSUFFIX@.$(LIBVERSION)
+libxso_target=	$(libcxx_base).la
 libxso_default=	$(libcxx_base)@SOSUFFIX@
-libxso_major=	$(libcxx_base)-$(LIBMAJOR)@SOSUFFIX@
+libxso_major=	$(libcxx_base)@SOSUFFIX@.$(LIBMAJOR)
 
 ##################################################
 # Java API.
@@ -108,12 +108,12 @@
 libj_jarfile=	db.jar
 libj_exjarfile=	dbexamples.jar
 libjso_base=	libdb_java
-libjso=		$(libjso_base)-$(LIBVERSION)@JMODSUFFIX@
-libjso_static=	$(libjso_base)-$(LIBVERSION).a
-libjso_target=	$(libjso_base)-$(LIBVERSION).la
+libjso=		$(libjso_base)@JMODSUFFIX@.$(LIBVERSION)
+libjso_static=	$(libjso_base).a
+libjso_target=	$(libjso_base).la
 libjso_default=	$(libjso_base)@JMODSUFFIX@
-libjso_major=	$(libjso_base)-$(LIBMAJOR)@JMODSUFFIX@
-libjso_g=	$(libjso_base)-$(LIBVERSION)_g@JMODSUFFIX@
+libjso_major=	$(libjso_base)@JMODSUFFIX@.$(LIBMAJOR)
+libjso_g=	$(libjso_base)_g@JMODSUFFIX@.$(LIBVERSION)
 
 ##################################################
 # TCL API.
@@ -123,11 +123,11 @@
 TCL_INCLUDE_SPEC=	@TCL_INCLUDE_SPEC@
 LIBTSO_LIBS=	@LIBTSO_LIBS@ @LIBSO_LIBS@
 libtso_base=	libdb_tcl
-libtso=		$(libtso_base)-$(LIBVERSION)@MODSUFFIX@
-libtso_static=	$(libtso_base)-$(LIBVERSION).a
-libtso_target=	$(libtso_base)-$(LIBVERSION).la
+libtso=		$(libtso_base)@MODSUFFIX@.$(LIBVERSION)
+libtso_static=	$(libtso_base).a
+libtso_target=	$(libtso_base).la
 libtso_default=	$(libtso_base)@MODSUFFIX@
-libtso_major=	$(libtso_base)-$(LIBMAJOR)@MODSUFFIX@
+libtso_major=	$(libtso_base)@MODSUFFIX@.$(LIBMAJOR)
 
 ##################################################
 # db_dump185 UTILITY
