--- lzma-4.32.5_orig/src/liblzmadec/Makefile.am	2008-03-29 08:24:05.000000000 +0000
+++ lzma-4.32.5/src/liblzmadec/Makefile.am	2008-03-29 08:24:28.000000000 +0000
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES = liblzmadec.la
 liblzmadec_la_SOURCES = private.h main.c buffer.c io.c
-liblzmadec_la_LDFLAGS = -version-info 0:0:0
+liblzmadec_la_LDFLAGS = -rpath $(libdir) -no-undefined -version-info 0:0:0
 
 lzmadecincludedir = $(includedir)
 lzmadecinclude_HEADERS = lzmadec.h
