diff -r -u e2retrieve_orig/Makefile e2retrieve/Makefile
--- e2retrieve_orig/Makefile	2007-07-29 11:10:02.000000000 +0200
+++ e2retrieve/Makefile	2007-07-29 11:12:49.000000000 +0200
@@ -60,8 +60,8 @@
 devel: $(PROJECT_NAME) $(IHM_NAME)
 
 $(PROJECT_NAME): $(OBJS_COMMON) $(OBJS_TXT)
-#	$(CC) -Wl,-static -o $@ $(OBJS_COMMON) $(OBJS_TXT)
-	$(CC) -o $@ $(OBJS_COMMON) $(OBJS_TXT)
+	$(CC) -static -o $@ $(OBJS_COMMON) $(OBJS_TXT)
+#	$(CC) -o $@ $(OBJS_COMMON) $(OBJS_TXT)
 ifneq ($(MAKECMDGOALS),devel)
 	$(STRIP) $@
 endif
