diff -ru python-support-0.7.5_orig/debian/control python-support-0.7.5/debian/control
--- python-support-0.7.5_orig/debian/control	2007-11-22 20:14:11.000000000 +0100
+++ python-support-0.7.5/debian/control	2007-11-22 20:14:23.000000000 +0100
@@ -5,8 +5,8 @@
 Build-Depends: debhelper (>= 4.1), cdbs
 Build-Depends-Indep: python (>= 2.3)
 Standards-Version: 3.7.2
-
 Package: python-support
+Version: 0.7.5
 Architecture: all
 Depends: python (>= 2.3)
 Conflicts: debhelper (<< 5.0.38)
