diff -ru nnfw-1.1.5-src_orig/CMakeCache.txt nnfw-1.1.5-src/CMakeCache.txt
diff -ru nnfw-1.1.5-src_orig/CMakeFiles/Makefile.cmake nnfw-1.1.5-src/CMakeFiles/Makefile.cmake
Only in nnfw-1.1.5-src_orig/CMakeFiles: Progress
diff -ru nnfw-1.1.5-src_orig/CMakeFiles/cnnfw.dir/CXX.includecache nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/CXX.includecache
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: backpropagationalgo.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: biasedcluster.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: cluster.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: copylinker.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: ddecluster.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: dotlinker.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: fakecluster.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: ionnfw.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: learningalgorithm.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: libcompetitivefunctions.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: liboutputfunctions.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: libperiodicfunctions.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: libradialfunctions.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: linker.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: matrixlinker.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: messages.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: neuralnet.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: nnfwfactory.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: normlinker.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: outputfunction.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: propertized.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: random.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: realmat.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: realvec.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: simplecluster.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: sparsematrixlinker.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: types.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: updatable.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: utils.o
Only in nnfw-1.1.5-src/CMakeFiles/cnnfw.dir/src: xmlnnfw.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir: CXX.includecache
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir: depend.internal
diff -ru nnfw-1.1.5-src_orig/CMakeFiles/nnfw.dir/depend.make nnfw-1.1.5-src/CMakeFiles/nnfw.dir/depend.make
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: backpropagationalgo.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: biasedcluster.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: cluster.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: copylinker.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: ddecluster.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: dotlinker.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: fakecluster.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: ionnfw.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: learningalgorithm.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: libcompetitivefunctions.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: liboutputfunctions.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: libperiodicfunctions.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: libradialfunctions.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: linker.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: matrixlinker.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: messages.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: neuralnet.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: nnfwfactory.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: normlinker.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: outputfunction.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: propertized.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: random.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: realmat.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: realvec.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: simplecluster.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: sparsematrixlinker.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: types.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: updatable.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: utils.o
Only in nnfw-1.1.5-src/CMakeFiles/nnfw.dir/src: xmlnnfw.o
diff -ru nnfw-1.1.5-src_orig/include/backpropagationalgo.h nnfw-1.1.5-src/include/backpropagationalgo.h
--- nnfw-1.1.5-src_orig/include/backpropagationalgo.h	2009-02-25 17:43:23.000000000 +0100
+++ nnfw-1.1.5-src/include/backpropagationalgo.h	2009-02-25 17:46:52.000000000 +0100
@@ -116,7 +116,7 @@
 	 *  teaching_input ) method. If your network has more than one output layer you have to call
 	 *  setTeachingInput() for all the output clusters before calling getError() for any of the clusters.
 	 */
-	const RealVec& getError( Cluster* );
+	RealVec getError( Cluster* );
 	//@}
 
 private:
diff -ru nnfw-1.1.5-src_orig/include/cluster.h nnfw-1.1.5-src/include/cluster.h
--- nnfw-1.1.5-src_orig/include/cluster.h	2009-02-25 17:43:22.000000000 +0100
+++ nnfw-1.1.5-src/include/cluster.h	2009-02-25 17:45:40.000000000 +0100
@@ -247,7 +247,7 @@
 
     /*! Get the Output function
      */
-    OutputFunction* const getFunction() const {
+    OutputFunction* getFunction() const {
         return updater;
     };
 
diff -ru nnfw-1.1.5-src_orig/include/memutils.h nnfw-1.1.5-src/include/memutils.h
--- nnfw-1.1.5-src_orig/include/memutils.h	2009-02-25 17:43:22.000000000 +0100
+++ nnfw-1.1.5-src/include/memutils.h	2009-02-25 17:44:50.000000000 +0100
@@ -23,6 +23,8 @@
 /*! \file
  */
 
+#include <cstring>
+
 #include "types.h"
 
 namespace nnfw {
Only in nnfw-1.1.5-src: libcnnfw.a
Only in nnfw-1.1.5-src: libnnfw.a
diff -ru nnfw-1.1.5-src_orig/src/backpropagationalgo.cpp nnfw-1.1.5-src/src/backpropagationalgo.cpp
--- nnfw-1.1.5-src_orig/src/backpropagationalgo.cpp	2009-02-25 17:43:23.000000000 +0100
+++ nnfw-1.1.5-src/src/backpropagationalgo.cpp	2009-02-25 17:46:17.000000000 +0100
@@ -66,7 +66,7 @@
 	return;
 }
 
-const RealVec& BackPropagationAlgo::getError( Cluster* cl ) {
+RealVec BackPropagationAlgo::getError( Cluster* cl ) {
 	if ( mapIndex.count( cl ) == 0 ) {
 		nWarning() << "Cluster not present in BackPropagationAlgo";
 		return RealVec();
