diff -ruN vamp-aubio-plugins-0.3.2_orig/plugins/Notes.cpp vamp-aubio-plugins-0.3.2/plugins/Notes.cpp
--- vamp-aubio-plugins-0.3.2_orig/plugins/Notes.cpp	2009-03-11 16:45:31.000000000 +0100
+++ vamp-aubio-plugins-0.3.2/plugins/Notes.cpp	2009-03-11 16:50:36.000000000 +0100
@@ -14,7 +14,8 @@
 
 */
 
-#include <math.h>
+#include <cmath>
+#include <algorithm>
 #include "Notes.h"
 
 using std::string;
@@ -395,7 +396,7 @@
     if (median < 45.0) return;
 
     float freq = median;
-    int midiPitch = (int)FLOOR(aubio_freqtomidi(freq) + 0.5);
+    int midiPitch = (int)std::floor(aubio_freqtomidi(freq) + 0.5);
     
     if (m_avoidLeaps) {
         if (m_prevPitch >= 0) {
Binary files vamp-aubio-plugins-0.3.2_orig/plugins/Notes.o and vamp-aubio-plugins-0.3.2/plugins/Notes.o differ
Binary files vamp-aubio-plugins-0.3.2_orig/plugins/Onset.o and vamp-aubio-plugins-0.3.2/plugins/Onset.o differ
Binary files vamp-aubio-plugins-0.3.2_orig/plugins/Pitch.o and vamp-aubio-plugins-0.3.2/plugins/Pitch.o differ
Binary files vamp-aubio-plugins-0.3.2_orig/plugins/Tempo.o and vamp-aubio-plugins-0.3.2/plugins/Tempo.o differ
Binary files vamp-aubio-plugins-0.3.2_orig/vamp-aubio.so and vamp-aubio-plugins-0.3.2/vamp-aubio.so differ
