--- libexnode-0.5_orig/Makefile.am	2009-02-21 01:24:53.000000000 +0100
+++ libexnode-0.5/Makefile.am	2009-02-21 01:25:24.000000000 +0100
@@ -3,8 +3,8 @@
 EXTRA_DIST = include COPYRIGHT
 
 install-data-local:
-	$(mkinstalldirs) $(prefix)/include/libexnode
-	$(mkinstalldirs) $(prefix)/lib
+	$(mkinstalldirs) $(DESTDIR)$(prefix)/include/libexnode
+	$(mkinstalldirs) $(DESTDIR)$(prefix)/lib
 
 #lib_LIBRARIES = libexnode.a
 
