diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/abort_prec_max.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/abort_prec_max.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/add1sp.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/add1sp.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/atan2.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/atan2.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/cache.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/cache.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/check.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/check.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/clears.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/clears.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/cmp_d.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/cmp_d.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/cmp_ld.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/cmp_ld.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/const_catalan.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/const_catalan.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/constant.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/constant.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/cot.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/cot.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/coth.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/coth.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/csc.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/csc.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/csch.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/csch.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/eint.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/eint.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/erf.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/erf.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/erfc.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/erfc.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/fits_intmax.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/fits_intmax.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/fits_uintmax.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/fits_uintmax.Plo
Only in mpfr-2.3.1_orig/build/i486-linux-diet/.deps: fits_uintmax.Tpo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/free_cache.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/free_cache.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/get_d64.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/get_d64.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/get_f.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/get_f.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/get_patches.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/get_patches.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/get_si.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/get_si.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/get_sj.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/get_sj.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/get_ui.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/get_ui.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/get_uj.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/get_uj.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/get_z.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/get_z.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/inits.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/inits.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/inits2.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/inits2.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/int_ceil_log2.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/int_ceil_log2.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/isqrt.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/isqrt.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/iszero.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/iszero.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/jn.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/jn.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/lngamma.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/lngamma.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/logging.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/logging.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/mp_clz_tab.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/mp_clz_tab.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/mpfr-gmp.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/mpfr-gmp.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/mpn_exp.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/mpn_exp.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/mulders.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/mulders.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/pow_z.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/pow_z.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/remquo.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/remquo.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/root.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/root.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/round_near_x.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/round_near_x.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/round_p.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/round_p.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/sec.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/sec.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/sech.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/sech.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/set_d64.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/set_d64.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/set_si_2exp.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/set_si_2exp.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/set_sj.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/set_sj.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/set_ui_2exp.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/set_ui_2exp.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/set_uj.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/set_uj.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/sgn.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/sgn.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/si_op.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/si_op.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/sqr.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/sqr.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/stack_interface.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/stack_interface.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/strtofr.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/strtofr.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/sub1sp.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/sub1sp.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/subnormal.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/subnormal.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/sum.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/sum.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/version.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/version.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/yn.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/yn.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/zeta.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/zeta.Plo
diff -ru mpfr-2.3.1_orig/build/i486-linux-diet/.deps/zeta_ui.Plo mpfr-2.3.1/build/i486-linux-diet/.deps/zeta_ui.Plo
Only in mpfr-2.3.1/build/i486-linux-diet: .libs
Only in mpfr-2.3.1/build/i486-linux-diet: abort_prec_max.lo
Only in mpfr-2.3.1/build/i486-linux-diet: abort_prec_max.o
Only in mpfr-2.3.1/build/i486-linux-diet: add1sp.lo
Only in mpfr-2.3.1/build/i486-linux-diet: add1sp.o
Only in mpfr-2.3.1/build/i486-linux-diet: atan2.lo
Only in mpfr-2.3.1/build/i486-linux-diet: atan2.o
Only in mpfr-2.3.1/build/i486-linux-diet: cache.lo
Only in mpfr-2.3.1/build/i486-linux-diet: cache.o
Only in mpfr-2.3.1/build/i486-linux-diet: check.lo
Only in mpfr-2.3.1/build/i486-linux-diet: check.o
Only in mpfr-2.3.1/build/i486-linux-diet: clears.lo
Only in mpfr-2.3.1/build/i486-linux-diet: clears.o
Only in mpfr-2.3.1/build/i486-linux-diet: cmp_d.lo
Only in mpfr-2.3.1/build/i486-linux-diet: cmp_d.o
Only in mpfr-2.3.1/build/i486-linux-diet: cmp_ld.lo
Only in mpfr-2.3.1/build/i486-linux-diet: cmp_ld.o
Only in mpfr-2.3.1/build/i486-linux-diet: const_catalan.lo
Only in mpfr-2.3.1/build/i486-linux-diet: const_catalan.o
Only in mpfr-2.3.1/build/i486-linux-diet: constant.lo
Only in mpfr-2.3.1/build/i486-linux-diet: constant.o
Only in mpfr-2.3.1/build/i486-linux-diet: cot.lo
Only in mpfr-2.3.1/build/i486-linux-diet: cot.o
Only in mpfr-2.3.1/build/i486-linux-diet: coth.lo
Only in mpfr-2.3.1/build/i486-linux-diet: coth.o
Only in mpfr-2.3.1/build/i486-linux-diet: csc.lo
Only in mpfr-2.3.1/build/i486-linux-diet: csc.o
Only in mpfr-2.3.1/build/i486-linux-diet: csch.lo
Only in mpfr-2.3.1/build/i486-linux-diet: csch.o
Only in mpfr-2.3.1/build/i486-linux-diet: eint.lo
Only in mpfr-2.3.1/build/i486-linux-diet: eint.o
Only in mpfr-2.3.1/build/i486-linux-diet: erf.lo
Only in mpfr-2.3.1/build/i486-linux-diet: erf.o
Only in mpfr-2.3.1/build/i486-linux-diet: erfc.lo
Only in mpfr-2.3.1/build/i486-linux-diet: erfc.o
Only in mpfr-2.3.1/build/i486-linux-diet: fits_intmax.lo
Only in mpfr-2.3.1/build/i486-linux-diet: fits_intmax.o
Only in mpfr-2.3.1/build/i486-linux-diet: fits_uintmax.lo
Only in mpfr-2.3.1/build/i486-linux-diet: fits_uintmax.o
Only in mpfr-2.3.1/build/i486-linux-diet: free_cache.lo
Only in mpfr-2.3.1/build/i486-linux-diet: free_cache.o
Only in mpfr-2.3.1/build/i486-linux-diet: get_d64.lo
Only in mpfr-2.3.1/build/i486-linux-diet: get_d64.o
Only in mpfr-2.3.1/build/i486-linux-diet: get_f.lo
Only in mpfr-2.3.1/build/i486-linux-diet: get_f.o
Only in mpfr-2.3.1/build/i486-linux-diet: get_patches.lo
Only in mpfr-2.3.1/build/i486-linux-diet: get_patches.o
Only in mpfr-2.3.1/build/i486-linux-diet: get_si.lo
Only in mpfr-2.3.1/build/i486-linux-diet: get_si.o
Only in mpfr-2.3.1/build/i486-linux-diet: get_sj.lo
Only in mpfr-2.3.1/build/i486-linux-diet: get_sj.o
Only in mpfr-2.3.1/build/i486-linux-diet: get_ui.lo
Only in mpfr-2.3.1/build/i486-linux-diet: get_ui.o
Only in mpfr-2.3.1/build/i486-linux-diet: get_uj.lo
Only in mpfr-2.3.1/build/i486-linux-diet: get_uj.o
Only in mpfr-2.3.1/build/i486-linux-diet: get_z.lo
Only in mpfr-2.3.1/build/i486-linux-diet: get_z.o
Only in mpfr-2.3.1/build/i486-linux-diet: inits.lo
Only in mpfr-2.3.1/build/i486-linux-diet: inits.o
Only in mpfr-2.3.1/build/i486-linux-diet: inits2.lo
Only in mpfr-2.3.1/build/i486-linux-diet: inits2.o
Only in mpfr-2.3.1/build/i486-linux-diet: int_ceil_log2.lo
Only in mpfr-2.3.1/build/i486-linux-diet: int_ceil_log2.o
Only in mpfr-2.3.1/build/i486-linux-diet: isqrt.lo
Only in mpfr-2.3.1/build/i486-linux-diet: isqrt.o
Only in mpfr-2.3.1/build/i486-linux-diet: iszero.lo
Only in mpfr-2.3.1/build/i486-linux-diet: iszero.o
Only in mpfr-2.3.1/build/i486-linux-diet: jn.lo
Only in mpfr-2.3.1/build/i486-linux-diet: jn.o
Only in mpfr-2.3.1/build/i486-linux-diet: libmpfr.la
Only in mpfr-2.3.1/build/i486-linux-diet: lngamma.lo
Only in mpfr-2.3.1/build/i486-linux-diet: lngamma.o
Only in mpfr-2.3.1/build/i486-linux-diet: logging.lo
Only in mpfr-2.3.1/build/i486-linux-diet: logging.o
Only in mpfr-2.3.1/build/i486-linux-diet: mp_clz_tab.lo
Only in mpfr-2.3.1/build/i486-linux-diet: mp_clz_tab.o
Only in mpfr-2.3.1/build/i486-linux-diet: mpfr-gmp.lo
Only in mpfr-2.3.1/build/i486-linux-diet: mpfr-gmp.o
Only in mpfr-2.3.1/build/i486-linux-diet: mpn_exp.lo
Only in mpfr-2.3.1/build/i486-linux-diet: mpn_exp.o
Only in mpfr-2.3.1/build/i486-linux-diet: mulders.lo
Only in mpfr-2.3.1/build/i486-linux-diet: mulders.o
Only in mpfr-2.3.1/build/i486-linux-diet: pow_z.lo
Only in mpfr-2.3.1/build/i486-linux-diet: pow_z.o
Only in mpfr-2.3.1/build/i486-linux-diet: remquo.lo
Only in mpfr-2.3.1/build/i486-linux-diet: remquo.o
Only in mpfr-2.3.1/build/i486-linux-diet: root.lo
Only in mpfr-2.3.1/build/i486-linux-diet: root.o
Only in mpfr-2.3.1/build/i486-linux-diet: round_near_x.lo
Only in mpfr-2.3.1/build/i486-linux-diet: round_near_x.o
Only in mpfr-2.3.1/build/i486-linux-diet: round_p.lo
Only in mpfr-2.3.1/build/i486-linux-diet: round_p.o
Only in mpfr-2.3.1/build/i486-linux-diet: sec.lo
Only in mpfr-2.3.1/build/i486-linux-diet: sec.o
Only in mpfr-2.3.1/build/i486-linux-diet: sech.lo
Only in mpfr-2.3.1/build/i486-linux-diet: sech.o
Only in mpfr-2.3.1/build/i486-linux-diet: set_d64.lo
Only in mpfr-2.3.1/build/i486-linux-diet: set_d64.o
Only in mpfr-2.3.1/build/i486-linux-diet: set_si_2exp.lo
Only in mpfr-2.3.1/build/i486-linux-diet: set_si_2exp.o
Only in mpfr-2.3.1/build/i486-linux-diet: set_sj.lo
Only in mpfr-2.3.1/build/i486-linux-diet: set_sj.o
Only in mpfr-2.3.1/build/i486-linux-diet: set_ui_2exp.lo
Only in mpfr-2.3.1/build/i486-linux-diet: set_ui_2exp.o
Only in mpfr-2.3.1/build/i486-linux-diet: set_uj.lo
Only in mpfr-2.3.1/build/i486-linux-diet: set_uj.o
Only in mpfr-2.3.1/build/i486-linux-diet: sgn.lo
Only in mpfr-2.3.1/build/i486-linux-diet: sgn.o
Only in mpfr-2.3.1/build/i486-linux-diet: si_op.lo
Only in mpfr-2.3.1/build/i486-linux-diet: si_op.o
Only in mpfr-2.3.1/build/i486-linux-diet: sqr.lo
Only in mpfr-2.3.1/build/i486-linux-diet: sqr.o
Only in mpfr-2.3.1/build/i486-linux-diet: stack_interface.lo
Only in mpfr-2.3.1/build/i486-linux-diet: stack_interface.o
Only in mpfr-2.3.1/build/i486-linux-diet: strtofr.lo
Only in mpfr-2.3.1/build/i486-linux-diet: strtofr.o
Only in mpfr-2.3.1/build/i486-linux-diet: sub1sp.lo
Only in mpfr-2.3.1/build/i486-linux-diet: sub1sp.o
Only in mpfr-2.3.1/build/i486-linux-diet: subnormal.lo
Only in mpfr-2.3.1/build/i486-linux-diet: subnormal.o
Only in mpfr-2.3.1/build/i486-linux-diet: sum.lo
Only in mpfr-2.3.1/build/i486-linux-diet: sum.o
Only in mpfr-2.3.1/build/i486-linux-diet: version.lo
Only in mpfr-2.3.1/build/i486-linux-diet: version.o
Only in mpfr-2.3.1/build/i486-linux-diet: yn.lo
Only in mpfr-2.3.1/build/i486-linux-diet: yn.o
Only in mpfr-2.3.1/build/i486-linux-diet: zeta.lo
Only in mpfr-2.3.1/build/i486-linux-diet: zeta.o
Only in mpfr-2.3.1/build/i486-linux-diet: zeta_ui.lo
Only in mpfr-2.3.1/build/i486-linux-diet: zeta_ui.o
diff -ru mpfr-2.3.1_orig/fits_intmax.c mpfr-2.3.1/fits_intmax.c
--- mpfr-2.3.1_orig/fits_intmax.c	2008-05-30 12:29:33.000000000 +0200
+++ mpfr-2.3.1/fits_intmax.c	2008-05-30 12:33:00.000000000 +0200
@@ -40,6 +40,14 @@
 
 #include "mpfr-impl.h"
 
+#ifndef INTMAX_MAX
+# define INTMAX_MAX (__INT64_C(9223372036854775807))
+#endif
+
+#ifndef INTMAX_MIN
+# define INTMAX_MIN (-__INT64_C(9223372036854775807)-1)
+#endif
+
 #ifdef _MPFR_H_HAVE_INTMAX_T
 
 /* We can't use fits_s.h <= mpfr_cmp_ui */
diff -ru mpfr-2.3.1_orig/fits_uintmax.c mpfr-2.3.1/fits_uintmax.c
--- mpfr-2.3.1_orig/fits_uintmax.c	2008-05-30 12:29:33.000000000 +0200
+++ mpfr-2.3.1/fits_uintmax.c	2008-05-30 12:31:37.000000000 +0200
@@ -40,6 +40,10 @@
 
 #include "mpfr-impl.h"
 
+#ifndef UINTMAX_MAX
+#define UINTMAX_MAX (__UINT64_C(18446744073709551615))
+#endif
+
 #ifdef _MPFR_H_HAVE_INTMAX_T
 
 /* We can't use fits_u.h <= mpfr_cmp_ui */
diff -ru mpfr-2.3.1_orig/get_sj.c mpfr-2.3.1/get_sj.c
--- mpfr-2.3.1_orig/get_sj.c	2008-05-30 12:29:33.000000000 +0200
+++ mpfr-2.3.1/get_sj.c	2008-05-30 12:34:02.000000000 +0200
@@ -40,6 +40,18 @@
 
 #include "mpfr-impl.h"
 
+#ifndef INTMAX_MIN
+# define INTMAX_MIN (-__INT64_C(9223372036854775807)-1)
+#endif
+
+#ifndef INTMAX_MAX
+# define INTMAX_MAX (__INT64_C(9223372036854775807))
+#endif
+
+#ifndef UINTMAX_MAX
+# define UINTMAX_MAX (__UINT64_C(18446744073709551615))
+#endif
+
 #ifdef _MPFR_H_HAVE_INTMAX_T
 
 intmax_t
diff -ru mpfr-2.3.1_orig/get_uj.c mpfr-2.3.1/get_uj.c
--- mpfr-2.3.1_orig/get_uj.c	2008-05-30 12:29:33.000000000 +0200
+++ mpfr-2.3.1/get_uj.c	2008-05-30 12:34:24.000000000 +0200
@@ -40,6 +40,10 @@
 
 #include "mpfr-impl.h"
 
+#ifndef UINTMAX_MAX
+# define UINTMAX_MAX (__UINT64_C(18446744073709551615))
+#endif
+
 #ifdef _MPFR_H_HAVE_INTMAX_T
 
 uintmax_t
