diff -r -u fann-2.1.0_orig/src/Makefile.in fann-2.1.0/src/Makefile.in
--- fann-2.1.0_orig/src/Makefile.in	2007-05-31 16:45:07.685043200 +0200
+++ fann-2.1.0/src/Makefile.in	2007-05-31 16:45:31.679545600 +0200
@@ -198,7 +198,7 @@
 top_srcdir = @top_srcdir@
 SUBDIRS = include
 lib_LTLIBRARIES = libfloatfann.la libdoublefann.la libfixedfann.la libfann.la
-AM_LDFLAGS = -version-info 2:1:0
+AM_LDFLAGS = -version-info 2:1:0 -no-undefined
 AM_CFLAGS = -D_REENTRANT
 libfloatfann_la_SOURCES = floatfann.c
 libdoublefann_la_SOURCES = doublefann.c
