diff -r -u netkit-rpc-0.18-pre1_orig/rpcgen/rpc_cout.c netkit-rpc-0.18-pre1/rpcgen/rpc_cout.c
--- netkit-rpc-0.18-pre1_orig/rpcgen/rpc_cout.c	2007-11-20 13:51:29.000000000 -0600
+++ netkit-rpc-0.18-pre1/rpcgen/rpc_cout.c	2007-11-20 13:52:43.000000000 -0600
@@ -102,6 +102,7 @@
 		emit_typedef(def);
 		break;
 	default:
+    break;
 	  /* can't happen */
 	}
 	print_trailer();
@@ -666,6 +667,7 @@
     f_print(fout,"\t\t   }\n\t\t };\n");
 
   default:
+    break;
     /* ?... do nothing I guess */
   }
 }
diff -r -u netkit-rpc-0.18-pre1_orig/rpcgen/rpc_hout.c netkit-rpc-0.18-pre1/rpcgen/rpc_hout.c
--- netkit-rpc-0.18-pre1_orig/rpcgen/rpc_hout.c	2007-11-20 13:51:29.000000000 -0600
+++ netkit-rpc-0.18-pre1/rpcgen/rpc_hout.c	2007-11-20 13:53:09.000000000 -0600
@@ -107,6 +107,7 @@
 		pprogramdef(def);
 		break;
 	  default:
+      break;
 	    /* ?... shouldn't happen I guess */
 	}
 }
