diff -r -u curl-7.15.1_orig/configure curl-7.15.1/configure
--- curl-7.15.1_orig/configure	2007-05-29 12:48:55.137521600 +0200
+++ curl-7.15.1/configure	2007-05-29 13:01:40.127523200 +0200
@@ -28980,15 +28980,15 @@
     { echo "$as_me:$LINENO: checking if gmtime_r exists and works" >&5
 echo $ECHO_N "checking if gmtime_r exists and works... $ECHO_C" >&6; }
     if test "$cross_compiling" = yes; then
-
+ :
 else
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+   :; }
 fi
-]])
+#]])
 
   fi
 else
diff -r -u curl-7.15.1_orig/lib/ca-bundle.h curl-7.15.1/lib/ca-bundle.h
--- curl-7.15.1_orig/lib/ca-bundle.h	2007-05-29 12:48:56.829955200 +0200
+++ curl-7.15.1/lib/ca-bundle.h	2007-05-29 13:05:03.349742400 +0200
@@ -1 +1,2 @@
-/* ca bundle path set in here*/
+/* This file is generated automatically */
+#define CURL_CA_BUNDLE "/usr/share/curl/curl-ca-bundle.crt"
