--- configure.orig	2008-08-18 01:54:29.000000000 +0200
+++ configure	2008-08-18 01:54:33.000000000 +0200
@@ -1701,7 +1701,7 @@
 fi
 
 
-PKGS="aterm toolbuslib error-support c-library tide-support pt-support ptable-support asf-support sdf-support config-support relation-stores sglr asc-support pgen sdf"
+PKGS="toolbuslib error-support c-library tide-support pt-support ptable-support asf-support sdf-support config-support relation-stores sglr asc-support pgen sdf"
 
 
                     ac_config_files="$ac_config_files Makefile pre-checks/Makefile"
