args: -r -u -- cognitivevision-0.9_orig cognitivevision-0.9
diff -r -u -- cognitivevision-0.9_orig/CMakeCache.txt cognitivevision-0.9/CMakeCache.txt
diff -r -u -- cognitivevision-0.9_orig/CMakeFiles/Makefile.cmake cognitivevision-0.9/CMakeFiles/Makefile.cmake
diff -r -u -- cognitivevision-0.9_orig/CMakeLists.txt cognitivevision-0.9/CMakeLists.txt
--- cognitivevision-0.9_orig/CMakeLists.txt	2009-02-28 23:07:47.000000000 +0100
+++ cognitivevision-0.9/CMakeLists.txt	2009-02-28 23:09:30.000000000 +0100
@@ -17,7 +17,7 @@
 # Add version to the compiler flags
 ADD_DEFINITIONS(-DPROJECT_VERSION=\\"${PROJECT_VERSION}\\")
 # Treats all warnings as errors and full optimization
-ADD_DEFINITIONS(-Wall -Werror -O3)
+ADD_DEFINITIONS(-Wall -O2)
 
 # Define mind libs
 SET(MIND_INC_DIR
diff -r -u -- cognitivevision-0.9_orig/apps/MindView/src/CMakeFiles/MindView.dir/depend.make cognitivevision-0.9/apps/MindView/src/CMakeFiles/MindView.dir/depend.make
diff -r -u -- cognitivevision-0.9_orig/apps/MindView/src/CMakeFiles/MindView.dir/flags.make cognitivevision-0.9/apps/MindView/src/CMakeFiles/MindView.dir/flags.make
diff -r -u -- cognitivevision-0.9_orig/libs/Blobs/CMakeFiles/Blobs.dir/flags.make cognitivevision-0.9/libs/Blobs/CMakeFiles/Blobs.dir/flags.make
diff -r -u -- cognitivevision-0.9_orig/libs/URBI/jpeg-6b/CMakeFiles/jpeg_6b.dir/flags.make cognitivevision-0.9/libs/URBI/jpeg-6b/CMakeFiles/jpeg_6b.dir/flags.make
diff -r -u -- cognitivevision-0.9_orig/libs/URBI/src/liburbi/CMakeFiles/urbi-cpp.dir/flags.make cognitivevision-0.9/libs/URBI/src/liburbi/CMakeFiles/urbi-cpp.dir/flags.make
diff -r -u -- cognitivevision-0.9_orig/libs/cppunit/src/CMakeFiles/cppunit.dir/depend.make cognitivevision-0.9/libs/cppunit/src/CMakeFiles/cppunit.dir/depend.make
diff -r -u -- cognitivevision-0.9_orig/libs/cppunit/src/CMakeFiles/cppunit.dir/flags.make cognitivevision-0.9/libs/cppunit/src/CMakeFiles/cppunit.dir/flags.make
diff -r -u -- cognitivevision-0.9_orig/libs/maxflow/CMakeFiles/maxflow.dir/flags.make cognitivevision-0.9/libs/maxflow/CMakeFiles/maxflow.dir/flags.make
diff -r -u -- cognitivevision-0.9_orig/mind/src/CMakeFiles/mind.dir/depend.make cognitivevision-0.9/mind/src/CMakeFiles/mind.dir/depend.make
diff -r -u -- cognitivevision-0.9_orig/mind/src/CMakeFiles/mind.dir/flags.make cognitivevision-0.9/mind/src/CMakeFiles/mind.dir/flags.make
diff -r -u -- cognitivevision-0.9_orig/tools/CMakeFiles/blackcalibtool.dir/depend.make cognitivevision-0.9/tools/CMakeFiles/blackcalibtool.dir/depend.make
diff -r -u -- cognitivevision-0.9_orig/tools/CMakeFiles/blackcalibtool.dir/flags.make cognitivevision-0.9/tools/CMakeFiles/blackcalibtool.dir/flags.make
diff -r -u -- cognitivevision-0.9_orig/tools/CMakeFiles/calibtool.dir/depend.make cognitivevision-0.9/tools/CMakeFiles/calibtool.dir/depend.make
diff -r -u -- cognitivevision-0.9_orig/tools/CMakeFiles/calibtool.dir/flags.make cognitivevision-0.9/tools/CMakeFiles/calibtool.dir/flags.make
diff -r -u -- cognitivevision-0.9_orig/tools/CMakeFiles/capturetool.dir/depend.make cognitivevision-0.9/tools/CMakeFiles/capturetool.dir/depend.make
diff -r -u -- cognitivevision-0.9_orig/tools/CMakeFiles/capturetool.dir/flags.make cognitivevision-0.9/tools/CMakeFiles/capturetool.dir/flags.make
diff -r -u -- cognitivevision-0.9_orig/tools/CMakeFiles/motiontool.dir/depend.make cognitivevision-0.9/tools/CMakeFiles/motiontool.dir/depend.make
diff -r -u -- cognitivevision-0.9_orig/tools/CMakeFiles/motiontool.dir/flags.make cognitivevision-0.9/tools/CMakeFiles/motiontool.dir/flags.make
