Only in BitchX/: .config.h
Only in BitchX/: BitchX.spec
diff -ru BitchX_orig/Makefile BitchX/Makefile
Only in BitchX/bx-conf: Makefile
Only in BitchX/: config.log
Only in BitchX/: config.status
Only in BitchX/dll: Makefile
Only in BitchX/dll/abot: Makefile
Only in BitchX/dll/acro: Makefile
Only in BitchX/dll/aim: Makefile
Only in BitchX/dll/aim/toc: Makefile
Only in BitchX/dll/amp: Makefile
Only in BitchX/dll/arcfour: Makefile
Only in BitchX/dll/autocycle: Makefile
Only in BitchX/dll/blowfish: Makefile
Only in BitchX/dll/cavlink: Makefile
Only in BitchX/dll/cdrom: Makefile
Only in BitchX/dll/encrypt: Makefile
Only in BitchX/dll/europa: Makefile
Only in BitchX/dll/fserv: Makefile
Only in BitchX/dll/hint: Makefile
Only in BitchX/dll/identd: Makefile
Only in BitchX/dll/nap: Makefile
Only in BitchX/dll/nicklist: Makefile
Only in BitchX/dll/pkga: Makefile
Only in BitchX/dll/possum: Makefile
Only in BitchX/dll/qbx: Makefile
Only in BitchX/dll/qmail: Makefile
Only in BitchX/dll/scan: Makefile
Only in BitchX/dll/wavplay: Makefile
Only in BitchX/dll/xmms: Makefile
Only in BitchX/doc: BitchX.bat
diff -ru BitchX_orig/doc/Makefile BitchX/doc/Makefile
Only in BitchX/: gtkBitchX.spec
diff -ru BitchX_orig/include/ctcp.h BitchX/include/ctcp.h
--- BitchX_orig/include/ctcp.h	2009-11-06 00:39:21.000000000 +0100
+++ BitchX/include/ctcp.h	2009-11-06 00:43:52.000000000 +0100
@@ -56,7 +56,7 @@
 extern CtcpEntryDll *dll_ctcp;
 
 
-extern		char	*ctcp_type[];
+//extern		char	*ctcp_type[];
 extern		int	sed;
 extern		int	in_ctcp_flag;
 
Only in BitchX/include: defs.h
diff -ru BitchX_orig/include/struct.h BitchX/include/struct.h
--- BitchX_orig/include/struct.h	2009-11-06 00:39:21.000000000 +0100
+++ BitchX/include/struct.h	2009-11-06 00:45:32.000000000 +0100
@@ -1064,7 +1064,7 @@
 	int	delete;
 }	TimerList;
 
-extern TimerList *PendingTimers;
+//extern TimerList *PendingTimers;
 typedef struct nicktab_stru
 {
 	struct nicktab_stru *next;
Only in BitchX/: la
Only in BitchX/source: BitchX
Only in BitchX/source: Makefile
diff -ru BitchX_orig/source/Makefile.in BitchX/source/Makefile.in
--- BitchX_orig/source/Makefile.in	2009-11-06 00:39:21.000000000 +0100
+++ BitchX/source/Makefile.in	2009-11-06 00:53:30.000000000 +0100
@@ -37,7 +37,7 @@
 IRCLIB = @IRCLIB@
 
 CC = @CC@
-DEFS = @INCLUDES@
+DEFS = -D_GNU_SOURCE @INCLUDES@
 LIBS = @LIBS@
 
 # Tcl library.
Only in BitchX/source: alias.o
Only in BitchX/source: alist.o
Only in BitchX/source: array.o
Only in BitchX/source: art.o
Only in BitchX/source: banlist.o
Only in BitchX/source: bcompat.o
Only in BitchX/source: bircsig.o
Only in BitchX/source: bot_link.o
Only in BitchX/source: cdcc.o
Only in BitchX/source: cdns.o
Only in BitchX/source: chelp.o
Only in BitchX/source: commands.o
Only in BitchX/source: commands2.o
Only in BitchX/source: compat.o
Only in BitchX/source: cset.o
Only in BitchX/source: ctcp.o
Only in BitchX/source: dcc.o
Only in BitchX/source: debug.o
Only in BitchX/source: encrypt.o
Only in BitchX/source: exec.o
Only in BitchX/source: files.o
Only in BitchX/source: flood.o
Only in BitchX/source: fset.o
Only in BitchX/source: functions.o
Only in BitchX/source: funny.o
Only in BitchX/source: glob.o
Only in BitchX/source: hash.o
Only in BitchX/source: hebrew.o
Only in BitchX/source: help.o
Only in BitchX/source: history.o
Only in BitchX/source: hook.o
Only in BitchX/source: if.o
Only in BitchX/source: ignore.o
Only in BitchX/source: input.o
Only in BitchX/source: irc.o
Only in BitchX/source: ircaux.o
Only in BitchX/source: ircsig.o
Only in BitchX/source: keys.o
Only in BitchX/source: lastlog.o
Only in BitchX/source: list.o
Only in BitchX/source: log.o
Only in BitchX/source: mail.o
Only in BitchX/source: misc.o
Only in BitchX/source: modules.o
Only in BitchX/source: names.o
Only in BitchX/source: network.o
Only in BitchX/source: newio.o
Only in BitchX/source: notice.o
Only in BitchX/source: notify.o
Only in BitchX/source: numbers.o
Only in BitchX/source: output.o
Only in BitchX/source: parse.o
Only in BitchX/source: queue.o
Only in BitchX/source: readlog.o
Only in BitchX/source: reg.o
Only in BitchX/source: scr-bx
Only in BitchX/source: scr-bx.o
Only in BitchX/source: screen.o
Only in BitchX/source: server.o
Only in BitchX/source: stack.o
Only in BitchX/source: status.o
Only in BitchX/source: sterm.c
Only in BitchX/source: sterm.o
Only in BitchX/source: struct.o
Only in BitchX/source: tcl_public.o
Only in BitchX/source: term.o
Only in BitchX/source: timer.o
Only in BitchX/source: translat.o
Only in BitchX/source: user.o
Only in BitchX/source: userlist.o
Only in BitchX/source: vars.o
Only in BitchX/source: who.o
Only in BitchX/source: whowas.o
Only in BitchX/source: window.o
Only in BitchX/source: wnetwork.c
Only in BitchX/source: wnetwork.o
Only in BitchX/source: words.o
Only in BitchX/source: wserv
Only in BitchX/source: wserv.o
Only in BitchX/source: wterm.c
Only in BitchX/source: wterm.o
Only in BitchX/: stamp-h
