diff -ru xml2var_orig/Makefile xml2var/Makefile
--- xml2var_orig/Makefile	2008-04-05 09:10:42.000000000 +0000
+++ xml2var/Makefile	2008-04-05 09:11:15.000000000 +0000
@@ -6,7 +6,6 @@
 all:	xml2var 
 xml2var: xml2var.o 
 	$(CC) $(CFLAGS) -oxml2var xml2var.o $(LDFLAGS) $(LIBS) 
-	strip xml2var
 install:
 	cp -av xml2var /etc/rc.d/bin
 clean: 
diff -ru xml2var_orig/xml2var.c xml2var/xml2var.c
--- xml2var_orig/xml2var.c	2008-04-05 09:10:42.000000000 +0000
+++ xml2var/xml2var.c	2008-04-05 09:10:57.000000000 +0000
@@ -93,8 +93,8 @@
 void disclaimer()
 {
 printf("$Id: xml2var.c,v 1.5 2003/08/13 14:22:28 tino Exp $\n");
-printf("Copyright (C) 2001 Sherryl Manalo <sm@antitachyon.com>
-                   and Martin Willner <mw@antitachyon.com>\n");
+printf("Copyright (C) 2001 Sherryl Manalo <sm@antitachyon.com>\n"
+       "               and Martin Willner <mw@antitachyon.com>\n");
 printf("\nusage: xml2var [xmlfile]\n");
 exit(0);
 }
