diff -ru w32api-3.11_orig/lib/directx/Makefile.in w32api-3.11/lib/directx/Makefile.in
--- w32api-3.11_orig/lib/directx/Makefile.in	2008-12-04 06:34:04.000000000 +0100
+++ w32api-3.11/lib/directx/Makefile.in	2008-12-04 06:34:23.000000000 +0100
@@ -78,7 +78,7 @@
 EXTRA_INCLUDES = -I$(srcdir)/../../mingw/include
 endif
 
-INCLUDES = -I$(srcdir)/../include -I$(srcdir)/../include/directx $(EXTRA_INCLUDES)
+INCLUDES = -I$(srcdir)/../../include -I$(srcdir)/../../include/directx $(EXTRA_INCLUDES)
 
 CFLAGS = @CFLAGS@
 ALL_CFLAGS = $(CFLAGS) $(INCLUDES)
