diff -ru splitpipe-0.4_orig/display.cc splitpipe-0.4/display.cc
--- splitpipe-0.4_orig/display.cc	2009-03-09 14:54:24.000000000 +0100
+++ splitpipe-0.4/display.cc	2009-03-09 15:14:56.000000000 +0100
@@ -19,7 +19,8 @@
 #include <ncurses.h>
 #include <string>
 #include <stdexcept>
-#include <time.h>
+#include <ctime>
+#include <cstring>
 
 using namespace std;
 
Only in splitpipe-0.4: display.o
Only in splitpipe-0.4: joinpipe
diff -ru splitpipe-0.4_orig/joinpipe.cc splitpipe-0.4/joinpipe.cc
--- splitpipe-0.4_orig/joinpipe.cc	2009-03-09 14:54:24.000000000 +0100
+++ splitpipe-0.4/joinpipe.cc	2009-03-09 15:07:41.000000000 +0100
@@ -29,6 +29,7 @@
 #include <netinet/in.h>
 #include <stdexcept>
 #include <vector>
+#include <cstdlib>
 #include "misc.hh"
 #include "md5.hh"
 
Only in splitpipe-0.4: joinpipe.o
diff -ru splitpipe-0.4_orig/main.cc splitpipe-0.4/main.cc
--- splitpipe-0.4_orig/main.cc	2009-03-09 14:54:25.000000000 +0100
+++ splitpipe-0.4/main.cc	2009-03-09 15:14:44.000000000 +0100
@@ -1,4 +1,5 @@
 #include <libgen.h>
+#include <cstring>
 #include "misc.hh"
 
 int main(int argc, char** argv)
Only in splitpipe-0.4: main.o
diff -ru splitpipe-0.4_orig/md5.hh splitpipe-0.4/md5.hh
--- splitpipe-0.4_orig/md5.hh	2009-03-09 14:54:24.000000000 +0100
+++ splitpipe-0.4/md5.hh	2009-03-09 15:07:28.000000000 +0100
@@ -3,6 +3,7 @@
 
 #include <string>
 #include <stdint.h>
+#include <cstring>
 
 namespace {
   struct md5_context
Only in splitpipe-0.4: md5.o
diff -ru splitpipe-0.4_orig/misc.cc splitpipe-0.4/misc.cc
--- splitpipe-0.4_orig/misc.cc	2009-03-09 14:54:25.000000000 +0100
+++ splitpipe-0.4/misc.cc	2009-03-09 15:15:07.000000000 +0100
@@ -7,6 +7,7 @@
 #include <stdint.h>
 #include <cerrno>
 #include <cstdio>
+#include <cstring>
 
 using namespace std;
 
Only in splitpipe-0.4: misc.o
Only in splitpipe-0.4: splitpipe
Only in splitpipe-0.4: splitpipe.o
