--- fltk-1.3.x-r6530_orig/src/fl_font_x.cxx	2008-12-04 21:24:54.000000000 +0100
+++ fltk-1.3.x-r6530/src/fl_font_x.cxx	2008-12-04 21:38:03.000000000 +0100
@@ -299,10 +299,10 @@
 
 void fl_text_extents(const char *c, int n, int &dx, int &dy, int &W, int &H) {
 #warning fl_text_extents is only a test stub in Xlib build at present
-  w = 0; h = 0;
-  fl_measure(c, &w, &h, 0);
+  W = 0; W = 0;
+  fl_measure(c, W, H, 0);
   dx = 0;
-  dy = fl_descent() - h;
+  dy = fl_descent() - H;
 } // fl_text_extents
 
 
