args: -r -u -N -- xbelSync.0.8.a7_orig xbelSync.0.8.a7
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-BookmarksPanel.Po xbelSync.0.8.a7/.deps/xbelSync-BookmarksPanel.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-Browser.Po xbelSync.0.8.a7/.deps/xbelSync-Browser.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-BrowserPathSelector.Po xbelSync.0.8.a7/.deps/xbelSync-BrowserPathSelector.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-Configuration.Po xbelSync.0.8.a7/.deps/xbelSync-Configuration.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-Downloader.Po xbelSync.0.8.a7/.deps/xbelSync-Downloader.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-ElementsTreeView.Po xbelSync.0.8.a7/.deps/xbelSync-ElementsTreeView.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-Exception.Po xbelSync.0.8.a7/.deps/xbelSync-Exception.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-FolderSelection.Po xbelSync.0.8.a7/.deps/xbelSync-FolderSelection.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-HtmlParser.Po xbelSync.0.8.a7/.deps/xbelSync-HtmlParser.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-Logger.Po xbelSync.0.8.a7/.deps/xbelSync-Logger.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-Netscape.Po xbelSync.0.8.a7/.deps/xbelSync-Netscape.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-Network.Po xbelSync.0.8.a7/.deps/xbelSync-Network.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-Synchronizer.Po xbelSync.0.8.a7/.deps/xbelSync-Synchronizer.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-Thread.Po xbelSync.0.8.a7/.deps/xbelSync-Thread.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-WindowBackups.Po xbelSync.0.8.a7/.deps/xbelSync-WindowBackups.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-WindowBrowserType.Po xbelSync.0.8.a7/.deps/xbelSync-WindowBrowserType.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-WindowBrowsers.Po xbelSync.0.8.a7/.deps/xbelSync-WindowBrowsers.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-WindowDuplicates.Po xbelSync.0.8.a7/.deps/xbelSync-WindowDuplicates.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-WindowEdit.Po xbelSync.0.8.a7/.deps/xbelSync-WindowEdit.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-WindowFind.Po xbelSync.0.8.a7/.deps/xbelSync-WindowFind.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-WindowMain.Po xbelSync.0.8.a7/.deps/xbelSync-WindowMain.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-WindowNetwork.Po xbelSync.0.8.a7/.deps/xbelSync-WindowNetwork.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-WindowOutOfDate.Po xbelSync.0.8.a7/.deps/xbelSync-WindowOutOfDate.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-WindowSort.Po xbelSync.0.8.a7/.deps/xbelSync-WindowSort.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-XBEL.Po xbelSync.0.8.a7/.deps/xbelSync-XBEL.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-XPM.Po xbelSync.0.8.a7/.deps/xbelSync-XPM.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-ZlibWrapper.Po xbelSync.0.8.a7/.deps/xbelSync-ZlibWrapper.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-g_stream.Po xbelSync.0.8.a7/.deps/xbelSync-g_stream.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-types.Po xbelSync.0.8.a7/.deps/xbelSync-types.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/.deps/xbelSync-xbelSync.Po xbelSync.0.8.a7/.deps/xbelSync-xbelSync.Po
diff -r -u -N -- xbelSync.0.8.a7_orig/Makefile xbelSync.0.8.a7/Makefile
diff -r -u -N -- xbelSync.0.8.a7_orig/Makefile.in xbelSync.0.8.a7/Makefile.in
diff -r -u -N -- xbelSync.0.8.a7_orig/aclocal.m4 xbelSync.0.8.a7/aclocal.m4
diff -r -u -N -- xbelSync.0.8.a7_orig/autogen.sh xbelSync.0.8.a7/autogen.sh
--- xbelSync.0.8.a7_orig/autogen.sh	1970-01-01 01:00:00.000000000 +0100
+++ xbelSync.0.8.a7/autogen.sh	2009-04-25 12:00:58.000000000 +0200
@@ -0,0 +1,10 @@
+#!/bin/sh
+#
+# 20090425
+
+
+aclocal --force -I m4
+autoheader --force
+automake --force --copy --foreign --add-missing --foreign --add-missing
+aclocal --force -I m4
+autoconf --force
diff -r -u -N -- xbelSync.0.8.a7_orig/autom4te.cache/output.0 xbelSync.0.8.a7/autom4te.cache/output.0
diff -r -u -N -- xbelSync.0.8.a7_orig/autom4te.cache/output.1 xbelSync.0.8.a7/autom4te.cache/output.1
diff -r -u -N -- xbelSync.0.8.a7_orig/autom4te.cache/requests xbelSync.0.8.a7/autom4te.cache/requests
diff -r -u -N -- xbelSync.0.8.a7_orig/autom4te.cache/traces.0 xbelSync.0.8.a7/autom4te.cache/traces.0
diff -r -u -N -- xbelSync.0.8.a7_orig/autom4te.cache/traces.1 xbelSync.0.8.a7/autom4te.cache/traces.1
diff -r -u -N -- xbelSync.0.8.a7_orig/aux/depcomp xbelSync.0.8.a7/aux/depcomp
diff -r -u -N -- xbelSync.0.8.a7_orig/aux/install-sh xbelSync.0.8.a7/aux/install-sh
diff -r -u -N -- xbelSync.0.8.a7_orig/config.log xbelSync.0.8.a7/config.log
diff -r -u -N -- xbelSync.0.8.a7_orig/config.status xbelSync.0.8.a7/config.status
diff -r -u -N -- xbelSync.0.8.a7_orig/configure xbelSync.0.8.a7/configure
diff -r -u -N -- xbelSync.0.8.a7_orig/configure.ac xbelSync.0.8.a7/configure.ac
--- xbelSync.0.8.a7_orig/configure.ac	2009-04-25 11:57:30.000000000 +0200
+++ xbelSync.0.8.a7/configure.ac	2009-04-25 12:00:32.000000000 +0200
@@ -11,6 +11,10 @@
 AC_SUBST(GTKMM_CFLAGS)
 AC_SUBST(GTKMM_LIBS)
 
+PKG_CHECK_MODULES(GIOMM, giomm-2.4 >= 2.4.0)
+AC_SUBST(GIOMM_CFLAGS)
+AC_SUBST(GIOMM_LIBS)
+
 PKG_CHECK_MODULES(CURL, libcurl >= 7.12.0)
 AC_SUBST(CURL_CFLAGS)
 AC_SUBST(CURL_LIBS)
diff -r -u -N -- xbelSync.0.8.a7_orig/include/Browsers/Network.h xbelSync.0.8.a7/include/Browsers/Network.h
--- xbelSync.0.8.a7_orig/include/Browsers/Network.h	2009-04-25 11:57:30.000000000 +0200
+++ xbelSync.0.8.a7/include/Browsers/Network.h	2009-04-25 12:01:32.000000000 +0200
@@ -7,7 +7,7 @@
 #include <string>
 #include <utility>
 #include <vector>
-
+#include <memory>
 
 namespace xbelSync
 {
diff -r -u -N -- xbelSync.0.8.a7_orig/include/GUI/WindowBrowserType.h xbelSync.0.8.a7/include/GUI/WindowBrowserType.h
--- xbelSync.0.8.a7_orig/include/GUI/WindowBrowserType.h	2009-04-25 11:57:30.000000000 +0200
+++ xbelSync.0.8.a7/include/GUI/WindowBrowserType.h	2009-04-25 12:05:09.000000000 +0200
@@ -6,6 +6,7 @@
 
 #include <gtkmm.h>
 #include <string>
+#include <memory>
 #include "Browsers/Browser.h"
 #include "GUI/FolderSelection.h"
 
diff -r -u -N -- xbelSync.0.8.a7_orig/include/Synchronizer.h xbelSync.0.8.a7/include/Synchronizer.h
--- xbelSync.0.8.a7_orig/include/Synchronizer.h	2009-04-25 11:57:30.000000000 +0200
+++ xbelSync.0.8.a7/include/Synchronizer.h	2009-04-25 12:03:38.000000000 +0200
@@ -4,7 +4,7 @@
 #include "enums.h"
 #include "SynchronizerObserver.h"
 #include <vector>
-
+#include <string>
 
 namespace xbelSync
 {
diff -r -u -N -- xbelSync.0.8.a7_orig/include/config.h xbelSync.0.8.a7/include/config.h
diff -r -u -N -- xbelSync.0.8.a7_orig/include/config.h.in~ xbelSync.0.8.a7/include/config.h.in~
diff -r -u -N -- xbelSync.0.8.a7_orig/src/Browsers/Netscape.cpp xbelSync.0.8.a7/src/Browsers/Netscape.cpp
--- xbelSync.0.8.a7_orig/src/Browsers/Netscape.cpp	2009-04-25 11:57:30.000000000 +0200
+++ xbelSync.0.8.a7/src/Browsers/Netscape.cpp	2009-04-25 12:02:39.000000000 +0200
@@ -10,6 +10,7 @@
 // TODO: move Window method in separate file
 #include <gtkmm.h>
 #include <glibmm.h>
+#include <cstring>
 
 const char *DOCTYPE = "<!DOCTYPE NETSCAPE-Bookmark-file-1>";
 
diff -r -u -N -- xbelSync.0.8.a7_orig/src/GUI/ElementsTreeView.cpp xbelSync.0.8.a7/src/GUI/ElementsTreeView.cpp
--- xbelSync.0.8.a7_orig/src/GUI/ElementsTreeView.cpp	2009-04-25 11:57:30.000000000 +0200
+++ xbelSync.0.8.a7/src/GUI/ElementsTreeView.cpp	2009-04-25 12:04:25.000000000 +0200
@@ -3,6 +3,7 @@
 #include "types.h"
 #include <gtkmm/stock.h>
 #include <gdkmm/pixbufloader.h>
+#include <cstring>
 
 using namespace Gtk;
 using namespace std;
diff -r -u -N -- xbelSync.0.8.a7_orig/src/GUI/WindowSort.cpp xbelSync.0.8.a7/src/GUI/WindowSort.cpp
--- xbelSync.0.8.a7_orig/src/GUI/WindowSort.cpp	2009-04-25 11:57:30.000000000 +0200
+++ xbelSync.0.8.a7/src/GUI/WindowSort.cpp	2009-04-25 12:06:25.000000000 +0200
@@ -1,4 +1,5 @@
 #include <string>
+#include <memory>
 #include "GUI/WindowSort.h"
 
 using namespace std;
diff -r -u -N -- xbelSync.0.8.a7_orig/src/Utils/Downloader.cpp xbelSync.0.8.a7/src/Utils/Downloader.cpp
--- xbelSync.0.8.a7_orig/src/Utils/Downloader.cpp	2009-04-25 11:57:30.000000000 +0200
+++ xbelSync.0.8.a7/src/Utils/Downloader.cpp	2009-04-25 12:02:00.000000000 +0200
@@ -2,6 +2,7 @@
 #include "Utils/Exception.h"
 #include <glibmm/main.h>
 #include <curl/multi.h>
+#include <cstring>
 
 using namespace std;
 
