--- xmldiff-0.6.8_orig/setup.py	2008-04-04 03:48:08.000000000 +0000
+++ xmldiff-0.6.8/setup.py	2008-04-04 03:49:03.000000000 +0000
@@ -16,11 +16,7 @@
 # You should have received a copy of the GNU General Public License along with
 # this program; if not, write to the Free Software Foundation, Inc.,
 # 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
-""" Generic Setup script, takes package info from __pkginfo__.py file """
-
-__revision__ = '$Id: setup.py,v 1.20 2005-01-12 14:21:47 syt Exp $'
-
-from __future__ import nested_scopes
+from __future__  import nested_scopes
 import os
 import sys
 import shutil
