diff -ru acl-2.2.47_orig/include/buildmacros acl-2.2.47/include/buildmacros
--- acl-2.2.47_orig/include/buildmacros	2008-12-05 03:50:32.000000000 +0100
+++ acl-2.2.47/include/buildmacros	2008-12-05 04:24:25.000000000 +0100
@@ -27,7 +27,7 @@
 	  $(LFILES:.l=.o) \
 	  $(YFILES:%.y=%.tab.o)
 
-INSTALL	= $(TOPDIR)/install-sh -o $(PKG_USER) -g $(PKG_GROUP)
+INSTALL	= $(TOPDIR)/install-sh 
 
 SHELL = /bin/sh
 IMAGES_DIR = $(TOPDIR)/all-images
