diff -ru utest-0.20_orig/Makefile.conf utest-0.20/Makefile.conf
--- utest-0.20_orig/Makefile.conf	2008-07-14 17:07:29.000000000 +0200
+++ utest-0.20/Makefile.conf	2008-07-14 17:07:39.000000000 +0200
@@ -12,22 +12,22 @@
 
 ### pod
 
-A:=$(warning "finding pod2html")
-cfg_pod2html:=$(shell spod2html --version 2>>config/log)
-ifeq ("$(cfg_pod2html)","")
-	cfg_pod2html:=$(shell pod2html --help 2>>config/log)
-	ifeq ("$(cfg_pod2html)","")
-		A:=$(warning "NO POD2HTML or SPOD2HTML found, cannot generate documentation")
-		cfg_pod2html:=echo NO POD2HTML 
-		cfg_pod2htmlout:=
-	else
+#A:=$(warning "finding pod2html")
+#cfg_pod2html:=$(shell spod2html --version 2>>config/log)
+#ifeq ("$(cfg_pod2html)","")
+#	cfg_pod2html:=$(shell pod2html --help 2>>config/log)
+#	ifeq ("$(cfg_pod2html)","")
+#		A:=$(warning "NO POD2HTML or SPOD2HTML found, cannot generate documentation")
+#		cfg_pod2html:=echo NO POD2HTML
+#		cfg_pod2htmlout:=
+#	else
 		cfg_pod2html:=pod2html
 		cfg_pod2htmlout:=--outfile
-	endif
-else
-	cfg_pod2html:=spod2html --pod
-	cfg_pod2htmlout:=
-endif
+#	endif
+#else
+#	cfg_pod2html:=spod2html --pod
+#	cfg_pod2htmlout:=
+#endif
 
 
 ### Document uploading
