--- wt-2.2.1_orig/src/Wt/Ext/CMakeLists.txt	2008-11-06 01:23:47.000000000 +0100
+++ wt-2.2.1/src/Wt/Ext/CMakeLists.txt	2008-11-06 01:52:12.000000000 +0100
@@ -1,2 +1,32 @@
-INSTALL_FILES(/include/Wt/Ext "...+[^Ch~]$")
+INSTALL(FILES AbstractButton
+              AbstractToggleButton
+              Button
+              Calendar
+              CheckBox
+              ComboBox
+              Component
+              Container
+              DataStore
+              DateField
+              Dialog
+              FormField
+              LineEdit
+              Makefile
+              Menu
+              MenuItem
+              MessageBox
+              NumberField
+              PagingToolBar
+              Panel
+              ProgressDialog
+              RadioButton
+              Splitter
+              SplitterHandle
+              TabWidget
+              TableView
+              TextEdit
+              ToolBar
+              ToolTipConfig
+              Widget
+              DESTINATION include/Wt/Ext)
 INSTALL_FILES(/include/Wt/Ext ExtDllDefs.h)
