Only in udhcp-0.9.8: clientpacket.o
diff -ru udhcp-0.9.8_orig/dhcpc.c udhcp-0.9.8/dhcpc.c
--- udhcp-0.9.8_orig/dhcpc.c	2007-11-24 05:22:55.000000000 +0100
+++ udhcp-0.9.8/dhcpc.c	2007-11-24 05:23:49.000000000 +0100
@@ -132,6 +132,7 @@
 		state = INIT_SELECTING;
 		break;
 	case INIT_SELECTING:
+    break;
 	}
 
 	/* start things over */
Only in udhcp-0.9.8: dhcpc.o
Only in udhcp-0.9.8: dumpleases
Only in udhcp-0.9.8: dumpleases.o
Only in udhcp-0.9.8: script.o
Only in udhcp-0.9.8: udhcpc
Binary files udhcp-0.9.8_orig/udhcpd and udhcp-0.9.8/udhcpd differ
