diff -r -u pantheios-1.0.1-beta25_orig/src/core/api.cpp pantheios-1.0.1-beta25/src/core/api.cpp
--- pantheios-1.0.1-beta25_orig/src/core/api.cpp	2007-04-19 19:12:14.000000000 +0200
+++ pantheios-1.0.1-beta25/src/core/api.cpp	2007-04-19 19:13:03.000000000 +0200
@@ -219,7 +219,7 @@
     platformstl::thread_mutex*                  s_pmxBEI;
 
     // Memory useds for the mutexes
-    union
+    union s_mx_mem
     {
         double          d;
         long double     ld;
