args: -r -u -- gephex-0.4.4_orig gephex-0.4.4
diff -r -u -- gephex-0.4.4_orig/base/src/TestFramework/.deps/CompositeTest.Po gephex-0.4.4/base/src/TestFramework/.deps/CompositeTest.Po
diff -r -u -- gephex-0.4.4_orig/base/src/TestFramework/.deps/LeafTest.Po gephex-0.4.4/base/src/TestFramework/.deps/LeafTest.Po
diff -r -u -- gephex-0.4.4_orig/base/src/TestFramework/.deps/Tester.Po gephex-0.4.4/base/src/TestFramework/.deps/Tester.Po
diff -r -u -- gephex-0.4.4_orig/base/src/TestFramework/.deps/easytest.Po gephex-0.4.4/base/src/TestFramework/.deps/easytest.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/convertbin.Po gephex-0.4.4/base/src/net/.deps/convertbin.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/domainclientsocket.Po gephex-0.4.4/base/src/net/.deps/domainclientsocket.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/domainserversocket.Po gephex-0.4.4/base/src/net/.deps/domainserversocket.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/fdsocket.Po gephex-0.4.4/base/src/net/.deps/fdsocket.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/isocket.Po gephex-0.4.4/base/src/net/.deps/isocket.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/netexceptions.Po gephex-0.4.4/base/src/net/.deps/netexceptions.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/protocol.Po gephex-0.4.4/base/src/net/.deps/protocol.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/socketadaptor.Po gephex-0.4.4/base/src/net/.deps/socketadaptor.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/socketutils.Po gephex-0.4.4/base/src/net/.deps/socketutils.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/sotimeout.Po gephex-0.4.4/base/src/net/.deps/sotimeout.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/tagutils.Po gephex-0.4.4/base/src/net/.deps/tagutils.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/tcpclientsocket.Po gephex-0.4.4/base/src/net/.deps/tcpclientsocket.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/.deps/tcpserversocket.Po gephex-0.4.4/base/src/net/.deps/tcpserversocket.Po
diff -r -u -- gephex-0.4.4_orig/base/src/net/fdsocket.cpp gephex-0.4.4/base/src/net/fdsocket.cpp
--- gephex-0.4.4_orig/base/src/net/fdsocket.cpp	2008-11-21 20:26:18.000000000 +0100
+++ gephex-0.4.4/base/src/net/fdsocket.cpp	2008-11-21 21:22:53.000000000 +0100
@@ -39,6 +39,8 @@
 #include <errno.h>
 #endif
 
+#include <cstring>
+
 #include "netexceptions.h"
 #include "socketutils.h"
 
diff -r -u -- gephex-0.4.4_orig/base/src/net/protocol.cpp gephex-0.4.4/base/src/net/protocol.cpp
--- gephex-0.4.4_orig/base/src/net/protocol.cpp	2008-11-21 20:26:18.000000000 +0100
+++ gephex-0.4.4/base/src/net/protocol.cpp	2008-11-21 21:22:17.000000000 +0100
@@ -26,6 +26,7 @@
 
 #include <cassert>
 #include <cstdio>
+#include <cstring>
 
 #if defined(HAVE_CONFIG_H)
 #include "config.h"
diff -r -u -- gephex-0.4.4_orig/base/src/net/tagutils.cpp gephex-0.4.4/base/src/net/tagutils.cpp
--- gephex-0.4.4_orig/base/src/net/tagutils.cpp	2008-11-21 20:26:18.000000000 +0100
+++ gephex-0.4.4/base/src/net/tagutils.cpp	2008-11-21 21:22:35.000000000 +0100
@@ -25,6 +25,7 @@
 #include "convertbin.h"
 #include "utils/buffer.h"
 
+#include <cstring>
 
 namespace net {
 
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/controlvaluereceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/controlvaluereceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/controlvaluesendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/controlvaluesendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/enginecontrolreceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/enginecontrolreceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/enginecontrolsendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/enginecontrolsendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/errorreceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/errorreceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/errorsendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/errorsendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/graphdatareceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/graphdatareceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/graphdatasendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/graphdatasendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/graphnamereceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/graphnamereceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/graphnamesendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/graphnamesendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/modelcontrolreceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/modelcontrolreceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/modelcontrolsendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/modelcontrolsendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/modelstatusreceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/modelstatusreceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/modelstatussendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/modelstatussendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/moduleclassinforeceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/moduleclassinforeceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/moduleclassinfosendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/moduleclassinfosendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/moduleconstructiondumbreceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/moduleconstructiondumbreceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/moduleconstructiondumbsendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/moduleconstructiondumbsendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/moduledatareceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/moduledatareceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/moduledatasendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/moduledatasendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/modulestatisticsreceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/modulestatisticsreceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/modulestatisticssendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/modulestatisticssendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/renderercontrolreceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/renderercontrolreceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/renderercontrolsendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/renderercontrolsendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/rendererstatusreceivernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/rendererstatusreceivernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/netinterfaces/.deps/rendererstatussendernet.Po gephex-0.4.4/base/src/netinterfaces/.deps/rendererstatussendernet.Po
diff -r -u -- gephex-0.4.4_orig/base/src/thread/.deps/imutex.Po gephex-0.4.4/base/src/thread/.deps/imutex.Po
diff -r -u -- gephex-0.4.4_orig/base/src/thread/.deps/ithread.Po gephex-0.4.4/base/src/thread/.deps/ithread.Po
diff -r -u -- gephex-0.4.4_orig/base/src/thread/.deps/threadposix.Po gephex-0.4.4/base/src/thread/.deps/threadposix.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/buffer.Po gephex-0.4.4/base/src/utils/.deps/buffer.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/bufferstream.Po gephex-0.4.4/base/src/utils/.deps/bufferstream.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/bufferutils.Po gephex-0.4.4/base/src/utils/.deps/bufferutils.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/circularbuffer.Po gephex-0.4.4/base/src/utils/.deps/circularbuffer.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/configfile.Po gephex-0.4.4/base/src/utils/.deps/configfile.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/configmanager.Po gephex-0.4.4/base/src/utils/.deps/configmanager.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/filesystem.Po gephex-0.4.4/base/src/utils/.deps/filesystem.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/itokenizer.Po gephex-0.4.4/base/src/utils/.deps/itokenizer.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/sharedlibrary.Po gephex-0.4.4/base/src/utils/.deps/sharedlibrary.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/spawn.Po gephex-0.4.4/base/src/utils/.deps/spawn.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/streamlogger.Po gephex-0.4.4/base/src/utils/.deps/streamlogger.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/streamtokenizer.Po gephex-0.4.4/base/src/utils/.deps/streamtokenizer.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/string_.Po gephex-0.4.4/base/src/utils/.deps/string_.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/stringtokenizer.Po gephex-0.4.4/base/src/utils/.deps/stringtokenizer.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/structreader.Po gephex-0.4.4/base/src/utils/.deps/structreader.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/structscanner.Po gephex-0.4.4/base/src/utils/.deps/structscanner.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/timing.Po gephex-0.4.4/base/src/utils/.deps/timing.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/.deps/xmlutils.Po gephex-0.4.4/base/src/utils/.deps/xmlutils.Po
diff -r -u -- gephex-0.4.4_orig/base/src/utils/circularbuffer.cpp gephex-0.4.4/base/src/utils/circularbuffer.cpp
--- gephex-0.4.4_orig/base/src/utils/circularbuffer.cpp	2008-11-21 20:26:18.000000000 +0100
+++ gephex-0.4.4/base/src/utils/circularbuffer.cpp	2008-11-21 21:21:01.000000000 +0100
@@ -24,6 +24,7 @@
 
 #include <stdexcept>
 #include <cassert>
+#include <cstring>
 
 #include "buffer.h"
 
diff -r -u -- gephex-0.4.4_orig/base/src/utils/spawn.cpp gephex-0.4.4/base/src/utils/spawn.cpp
--- gephex-0.4.4_orig/base/src/utils/spawn.cpp	2008-11-21 20:26:18.000000000 +0100
+++ gephex-0.4.4/base/src/utils/spawn.cpp	2008-11-21 21:21:56.000000000 +0100
@@ -23,6 +23,8 @@
 #include "spawn.h"
 
 #include <stdexcept>
+#include <cstdlib>
+#include <cstring>
 
 #if defined(HAVE_CONFIG_H)
 #include "config.h"
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectv1dmodule/.deps/effectv1dmodule.Plo gephex-0.4.4/effectvmodules/src/effectv1dmodule/.deps/effectv1dmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectv1dmodule/.deps/effectv1dmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectv1dmodule/.deps/effectv1dmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvagingmodule/.deps/effectvagingmodule.Plo gephex-0.4.4/effectvmodules/src/effectvagingmodule/.deps/effectvagingmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvagingmodule/.deps/effectvagingmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvagingmodule/.deps/effectvagingmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvbaltanmodule/.deps/effectvbaltanmodule.Plo gephex-0.4.4/effectvmodules/src/effectvbaltanmodule/.deps/effectvbaltanmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvbaltanmodule/.deps/effectvbaltanmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvbaltanmodule/.deps/effectvbaltanmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvblurzoommodule/.deps/effectvblurzoommodule.Plo gephex-0.4.4/effectvmodules/src/effectvblurzoommodule/.deps/effectvblurzoommodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvblurzoommodule/.deps/effectvblurzoommodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvblurzoommodule/.deps/effectvblurzoommodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvburnmodule/.deps/effectvburnmodule.Plo gephex-0.4.4/effectvmodules/src/effectvburnmodule/.deps/effectvburnmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvburnmodule/.deps/effectvburnmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvburnmodule/.deps/effectvburnmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvchameleonmodule/.deps/effectvchameleonmodule.Plo gephex-0.4.4/effectvmodules/src/effectvchameleonmodule/.deps/effectvchameleonmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvchameleonmodule/.deps/effectvchameleonmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvchameleonmodule/.deps/effectvchameleonmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvcyclemodule/.deps/effectvcyclemodule.Plo gephex-0.4.4/effectvmodules/src/effectvcyclemodule/.deps/effectvcyclemodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvcyclemodule/.deps/effectvcyclemodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvcyclemodule/.deps/effectvcyclemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvdicemodule/.deps/effectvdicemodule.Plo gephex-0.4.4/effectvmodules/src/effectvdicemodule/.deps/effectvdicemodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvdicemodule/.deps/effectvdicemodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvdicemodule/.deps/effectvdicemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvdiffmodule/.deps/effectvdiffmodule.Plo gephex-0.4.4/effectvmodules/src/effectvdiffmodule/.deps/effectvdiffmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvdiffmodule/.deps/effectvdiffmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvdiffmodule/.deps/effectvdiffmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvdizzymodule/.deps/effectvdizzymodule.Plo gephex-0.4.4/effectvmodules/src/effectvdizzymodule/.deps/effectvdizzymodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvdizzymodule/.deps/effectvdizzymodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvdizzymodule/.deps/effectvdizzymodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvdotmodule/.deps/effectvdotmodule.Plo gephex-0.4.4/effectvmodules/src/effectvdotmodule/.deps/effectvdotmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvdotmodule/.deps/effectvdotmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvdotmodule/.deps/effectvdotmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvedgemodule/.deps/effectvedgemodule.Plo gephex-0.4.4/effectvmodules/src/effectvedgemodule/.deps/effectvedgemodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvedgemodule/.deps/effectvedgemodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvedgemodule/.deps/effectvedgemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvfiremodule/.deps/effectvfiremodule.Plo gephex-0.4.4/effectvmodules/src/effectvfiremodule/.deps/effectvfiremodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvfiremodule/.deps/effectvfiremodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvfiremodule/.deps/effectvfiremodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvholomodule/.deps/effectvholomodule.Plo gephex-0.4.4/effectvmodules/src/effectvholomodule/.deps/effectvholomodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvholomodule/.deps/effectvholomodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvholomodule/.deps/effectvholomodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvlensmodule/.deps/effectvlensmodule.Plo gephex-0.4.4/effectvmodules/src/effectvlensmodule/.deps/effectvlensmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvlensmodule/.deps/effectvlensmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvlensmodule/.deps/effectvlensmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvlifemodule/.deps/effectvlifemodule.Plo gephex-0.4.4/effectvmodules/src/effectvlifemodule/.deps/effectvlifemodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvlifemodule/.deps/effectvlifemodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvlifemodule/.deps/effectvlifemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvmatrixmodule/.deps/effectvmatrixmodule.Plo gephex-0.4.4/effectvmodules/src/effectvmatrixmodule/.deps/effectvmatrixmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvmatrixmodule/.deps/effectvmatrixmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvmatrixmodule/.deps/effectvmatrixmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvmosaicmodule/.deps/effectvmosaicmodule.Plo gephex-0.4.4/effectvmodules/src/effectvmosaicmodule/.deps/effectvmosaicmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvmosaicmodule/.deps/effectvmosaicmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvmosaicmodule/.deps/effectvmosaicmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvnervousmodule/.deps/effectvnervousmodule.Plo gephex-0.4.4/effectvmodules/src/effectvnervousmodule/.deps/effectvnervousmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvnervousmodule/.deps/effectvnervousmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvnervousmodule/.deps/effectvnervousmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvnoisemodule/.deps/effectvnoisemodule.Plo gephex-0.4.4/effectvmodules/src/effectvnoisemodule/.deps/effectvnoisemodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvnoisemodule/.deps/effectvnoisemodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvnoisemodule/.deps/effectvnoisemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvpredatormodule/.deps/effectvpredatormodule.Plo gephex-0.4.4/effectvmodules/src/effectvpredatormodule/.deps/effectvpredatormodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvpredatormodule/.deps/effectvpredatormodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvpredatormodule/.deps/effectvpredatormodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvpupmodule/.deps/effectvpupmodule.Plo gephex-0.4.4/effectvmodules/src/effectvpupmodule/.deps/effectvpupmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvpupmodule/.deps/effectvpupmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvpupmodule/.deps/effectvpupmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvpuzzlemodule/.deps/effectvpuzzlemodule.Plo gephex-0.4.4/effectvmodules/src/effectvpuzzlemodule/.deps/effectvpuzzlemodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvpuzzlemodule/.deps/effectvpuzzlemodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvpuzzlemodule/.deps/effectvpuzzlemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvquarkmodule/.deps/effectvquarkmodule.Plo gephex-0.4.4/effectvmodules/src/effectvquarkmodule/.deps/effectvquarkmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvquarkmodule/.deps/effectvquarkmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvquarkmodule/.deps/effectvquarkmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvrdsmodule/.deps/effectvrdsmodule.Plo gephex-0.4.4/effectvmodules/src/effectvrdsmodule/.deps/effectvrdsmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvrdsmodule/.deps/effectvrdsmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvrdsmodule/.deps/effectvrdsmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvrevmodule/.deps/effectvrevmodule.Plo gephex-0.4.4/effectvmodules/src/effectvrevmodule/.deps/effectvrevmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvrevmodule/.deps/effectvrevmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvrevmodule/.deps/effectvrevmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvripplemodule/.deps/effectvripplemodule.Plo gephex-0.4.4/effectvmodules/src/effectvripplemodule/.deps/effectvripplemodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvripplemodule/.deps/effectvripplemodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvripplemodule/.deps/effectvripplemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvrndmmodule/.deps/effectvrndmmodule.Plo gephex-0.4.4/effectvmodules/src/effectvrndmmodule/.deps/effectvrndmmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvrndmmodule/.deps/effectvrndmmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvrndmmodule/.deps/effectvrndmmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvscrollmodule/.deps/effectvscrollmodule.Plo gephex-0.4.4/effectvmodules/src/effectvscrollmodule/.deps/effectvscrollmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvscrollmodule/.deps/effectvscrollmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvscrollmodule/.deps/effectvscrollmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvshagadelicmodule/.deps/effectvshagadelicmodule.Plo gephex-0.4.4/effectvmodules/src/effectvshagadelicmodule/.deps/effectvshagadelicmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvshagadelicmodule/.deps/effectvshagadelicmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvshagadelicmodule/.deps/effectvshagadelicmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvsimuramodule/.deps/effectvsimuramodule.Plo gephex-0.4.4/effectvmodules/src/effectvsimuramodule/.deps/effectvsimuramodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvsimuramodule/.deps/effectvsimuramodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvsimuramodule/.deps/effectvsimuramodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvsparkmodule/.deps/effectvsparkmodule.Plo gephex-0.4.4/effectvmodules/src/effectvsparkmodule/.deps/effectvsparkmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvsparkmodule/.deps/effectvsparkmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvsparkmodule/.deps/effectvsparkmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvspiralmodule/.deps/effectvspiralmodule.Plo gephex-0.4.4/effectvmodules/src/effectvspiralmodule/.deps/effectvspiralmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvspiralmodule/.deps/effectvspiralmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvspiralmodule/.deps/effectvspiralmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvstreakmodule/.deps/effectvstreakmodule.Plo gephex-0.4.4/effectvmodules/src/effectvstreakmodule/.deps/effectvstreakmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvstreakmodule/.deps/effectvstreakmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvstreakmodule/.deps/effectvstreakmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvtransformmodule/.deps/effectvtransformmodule.Plo gephex-0.4.4/effectvmodules/src/effectvtransformmodule/.deps/effectvtransformmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvtransformmodule/.deps/effectvtransformmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvtransformmodule/.deps/effectvtransformmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvwarholmodule/.deps/effectvwarholmodule.Plo gephex-0.4.4/effectvmodules/src/effectvwarholmodule/.deps/effectvwarholmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvwarholmodule/.deps/effectvwarholmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvwarholmodule/.deps/effectvwarholmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvwarpmodule/.deps/effectvwarpmodule.Plo gephex-0.4.4/effectvmodules/src/effectvwarpmodule/.deps/effectvwarpmodule.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/effectvwarpmodule/.deps/effectvwarpmodule_auto.Plo gephex-0.4.4/effectvmodules/src/effectvwarpmodule/.deps/effectvwarpmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/effectvmodules/src/libeffectv/.deps/libeffectv.Po gephex-0.4.4/effectvmodules/src/libeffectv/.deps/libeffectv.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/c_input.Po gephex-0.4.4/engine/src/dllloader/.deps/c_input.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/c_output.Po gephex-0.4.4/engine/src/dllloader/.deps/c_output.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/c_outputplug.Po gephex-0.4.4/engine/src/dllloader/.deps/c_outputplug.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/c_type.Po gephex-0.4.4/engine/src/dllloader/.deps/c_type.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/c_typeclass.Po gephex-0.4.4/engine/src/dllloader/.deps/c_typeclass.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/dllloader.Po gephex-0.4.4/engine/src/dllloader/.deps/dllloader.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/frei0r_module.Po gephex-0.4.4/engine/src/dllloader/.deps/frei0r_module.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/frei0r_module_class.Po gephex-0.4.4/engine/src/dllloader/.deps/frei0r_module_class.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/frei0r_module_plugin.Po gephex-0.4.4/engine/src/dllloader/.deps/frei0r_module_plugin.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/gephex_module.Po gephex-0.4.4/engine/src/dllloader/.deps/gephex_module.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/gephex_module_class.Po gephex-0.4.4/engine/src/dllloader/.deps/gephex_module_class.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/gephex_module_plugin.Po gephex-0.4.4/engine/src/dllloader/.deps/gephex_module_plugin.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/gephex_type_plugin.Po gephex-0.4.4/engine/src/dllloader/.deps/gephex_type_plugin.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/moduleclassspec.Po gephex-0.4.4/engine/src/dllloader/.deps/moduleclassspec.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/moduleinfoutils.Po gephex-0.4.4/engine/src/dllloader/.deps/moduleinfoutils.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/dllloader/.deps/nameresolver.Po gephex-0.4.4/engine/src/dllloader/.deps/nameresolver.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/engine/.deps/bufferedsender.Po gephex-0.4.4/engine/src/engine/.deps/bufferedsender.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/engine/.deps/controller.Po gephex-0.4.4/engine/src/engine/.deps/controller.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/engine/.deps/main.Po gephex-0.4.4/engine/src/engine/.deps/main.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/engine/.deps/netlogger.Po gephex-0.4.4/engine/src/engine/.deps/netlogger.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/engine/.deps/scheduler.Po gephex-0.4.4/engine/src/engine/.deps/scheduler.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/model/.deps/controlvalueset.Po gephex-0.4.4/engine/src/model/.deps/controlvalueset.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/model/.deps/graph.Po gephex-0.4.4/engine/src/model/.deps/graph.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/model/.deps/graphconnection.Po gephex-0.4.4/engine/src/model/.deps/graphconnection.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/model/.deps/graphfilesystem.Po gephex-0.4.4/engine/src/model/.deps/graphfilesystem.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/model/.deps/graphnode.Po gephex-0.4.4/engine/src/model/.deps/graphnode.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/model/.deps/graphparser.Po gephex-0.4.4/engine/src/model/.deps/graphparser.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/model/.deps/graphserial.Po gephex-0.4.4/engine/src/model/.deps/graphserial.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/model/.deps/model.Po gephex-0.4.4/engine/src/model/.deps/model.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/renderer/.deps/modulefactory.Po gephex-0.4.4/engine/src/renderer/.deps/modulefactory.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/renderer/.deps/renderer.Po gephex-0.4.4/engine/src/renderer/.deps/renderer.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/renderer/.deps/runtimesystem.Po gephex-0.4.4/engine/src/renderer/.deps/runtimesystem.Po
diff -r -u -- gephex-0.4.4_orig/engine/src/renderer/.deps/typefactory.Po gephex-0.4.4/engine/src/renderer/.deps/typefactory.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/audioaddmodule/.deps/audioaddmodule.Plo gephex-0.4.4/modules/src/audioaddmodule/.deps/audioaddmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audioaddmodule/.deps/audioaddmodule_auto.Plo gephex-0.4.4/modules/src/audioaddmodule/.deps/audioaddmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiobdmodule/.deps/audiobdmodule.Plo gephex-0.4.4/modules/src/audiobdmodule/.deps/audiobdmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiobdmodule/.deps/audiobdmodule_auto.Plo gephex-0.4.4/modules/src/audiobdmodule/.deps/audiobdmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audioenergymodule/.deps/audioenergymodule.Plo gephex-0.4.4/modules/src/audioenergymodule/.deps/audioenergymodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audioenergymodule/.deps/audioenergymodule_auto.Plo gephex-0.4.4/modules/src/audioenergymodule/.deps/audioenergymodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audioinmodule/.deps/a_cvt.Plo gephex-0.4.4/modules/src/audioinmodule/.deps/a_cvt.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audioinmodule/.deps/alsadriver.Plo gephex-0.4.4/modules/src/audioinmodule/.deps/alsadriver.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audioinmodule/.deps/audioinmodule.Plo gephex-0.4.4/modules/src/audioinmodule/.deps/audioinmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audioinmodule/.deps/audioinmodule_auto.Plo gephex-0.4.4/modules/src/audioinmodule/.deps/audioinmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audioinmodule/.deps/ossdriver.Plo gephex-0.4.4/modules/src/audioinmodule/.deps/ossdriver.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiooutmodule/.deps/a_cvt.Plo gephex-0.4.4/modules/src/audiooutmodule/.deps/a_cvt.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiooutmodule/.deps/alsaoutdriver.Plo gephex-0.4.4/modules/src/audiooutmodule/.deps/alsaoutdriver.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiooutmodule/.deps/audiooutmodule.Plo gephex-0.4.4/modules/src/audiooutmodule/.deps/audiooutmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiooutmodule/.deps/audiooutmodule_auto.Plo gephex-0.4.4/modules/src/audiooutmodule/.deps/audiooutmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiooutmodule/.deps/ossoutdriver.Plo gephex-0.4.4/modules/src/audiooutmodule/.deps/ossoutdriver.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiospecmodule/.deps/audiospecmodule.Plo gephex-0.4.4/modules/src/audiospecmodule/.deps/audiospecmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiospecmodule/.deps/audiospecmodule_auto.Plo gephex-0.4.4/modules/src/audiospecmodule/.deps/audiospecmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiosynthmodule/.deps/audiosynthmodule.Plo gephex-0.4.4/modules/src/audiosynthmodule/.deps/audiosynthmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiosynthmodule/.deps/audiosynthmodule_auto.Plo gephex-0.4.4/modules/src/audiosynthmodule/.deps/audiosynthmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiosynthmodule/.deps/synth.Plo gephex-0.4.4/modules/src/audiosynthmodule/.deps/synth.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiovismodule/.deps/audiovismodule.Plo gephex-0.4.4/modules/src/audiovismodule/.deps/audiovismodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiovismodule/.deps/audiovismodule_auto.Plo gephex-0.4.4/modules/src/audiovismodule/.deps/audiovismodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiowavegenmodule/.deps/audiowavegenmodule.Plo gephex-0.4.4/modules/src/audiowavegenmodule/.deps/audiowavegenmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/audiowavegenmodule/.deps/audiowavegenmodule_auto.Plo gephex-0.4.4/modules/src/audiowavegenmodule/.deps/audiowavegenmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/blurmodule/.deps/blurmodule.Plo gephex-0.4.4/modules/src/blurmodule/.deps/blurmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/blurmodule/.deps/blurmodule_auto.Plo gephex-0.4.4/modules/src/blurmodule/.deps/blurmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/bwmodule/.deps/bwmodule.Plo gephex-0.4.4/modules/src/bwmodule/.deps/bwmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/bwmodule/.deps/bwmodule_auto.Plo gephex-0.4.4/modules/src/bwmodule/.deps/bwmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/capturemodule/.deps/capturemodule.Plo gephex-0.4.4/modules/src/capturemodule/.deps/capturemodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/capturemodule/.deps/capturemodule_auto.Plo gephex-0.4.4/modules/src/capturemodule/.deps/capturemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/capturemodule/.deps/v4l1.Plo gephex-0.4.4/modules/src/capturemodule/.deps/v4l1.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/capturemodule/.deps/v4l2capturedriver.Plo gephex-0.4.4/modules/src/capturemodule/.deps/v4l2capturedriver.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/capturemodule/.deps/v4lcapturedriver.Plo gephex-0.4.4/modules/src/capturemodule/.deps/v4lcapturedriver.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/capturemodule/v4l2capturedriver.cpp gephex-0.4.4/modules/src/capturemodule/v4l2capturedriver.cpp
--- gephex-0.4.4_orig/modules/src/capturemodule/v4l2capturedriver.cpp	2008-11-21 20:26:30.000000000 +0100
+++ gephex-0.4.4/modules/src/capturemodule/v4l2capturedriver.cpp	2008-11-21 21:28:45.000000000 +0100
@@ -31,6 +31,7 @@
 #include <algorithm>
 #include <stdint.h>
 #include <iostream>
+#include <cstring>
 
 #include <cassert>
 
diff -r -u -- gephex-0.4.4_orig/modules/src/capturemodule/v4lcapturedriver.cpp gephex-0.4.4/modules/src/capturemodule/v4lcapturedriver.cpp
--- gephex-0.4.4_orig/modules/src/capturemodule/v4lcapturedriver.cpp	2008-11-21 20:26:30.000000000 +0100
+++ gephex-0.4.4/modules/src/capturemodule/v4lcapturedriver.cpp	2008-11-21 21:28:14.000000000 +0100
@@ -30,6 +30,7 @@
 #include "stdint.h"
 
 #include <cassert>
+#include <cstring>
 
 // for stat
 #include <sys/types.h>
diff -r -u -- gephex-0.4.4_orig/modules/src/centermodule/.deps/centermodule.Plo gephex-0.4.4/modules/src/centermodule/.deps/centermodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/centermodule/.deps/centermodule_auto.Plo gephex-0.4.4/modules/src/centermodule/.deps/centermodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/chromakeymodule/.deps/chromakeymodule-chromakeymodule.Po gephex-0.4.4/modules/src/chromakeymodule/.deps/chromakeymodule-chromakeymodule.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/chromakeymodule/.deps/chromakeymodule-chromakeymodule_auto.Po gephex-0.4.4/modules/src/chromakeymodule/.deps/chromakeymodule-chromakeymodule_auto.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/color2numbermodule/.deps/color2numbermodule.Plo gephex-0.4.4/modules/src/color2numbermodule/.deps/color2numbermodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/color2numbermodule/.deps/color2numbermodule_auto.Plo gephex-0.4.4/modules/src/color2numbermodule/.deps/color2numbermodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/colorizemodule/.deps/colorizemodule.Plo gephex-0.4.4/modules/src/colorizemodule/.deps/colorizemodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/colorizemodule/.deps/colorizemodule_auto.Plo gephex-0.4.4/modules/src/colorizemodule/.deps/colorizemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/diffmodule/.deps/diffmodule.Plo gephex-0.4.4/modules/src/diffmodule/.deps/diffmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/diffmodule/.deps/diffmodule_auto.Plo gephex-0.4.4/modules/src/diffmodule/.deps/diffmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/doepfermodule/.deps/doepfermodule.Plo gephex-0.4.4/modules/src/doepfermodule/.deps/doepfermodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/doepfermodule/.deps/doepfermodule_auto.Plo gephex-0.4.4/modules/src/doepfermodule/.deps/doepfermodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/dplmodule/.deps/dplmodule.Plo gephex-0.4.4/modules/src/dplmodule/.deps/dplmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/dplmodule/.deps/dplmodule_auto.Plo gephex-0.4.4/modules/src/dplmodule/.deps/dplmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/ffmpegoutmodule/.deps/ffmpegoutmodule.Plo gephex-0.4.4/modules/src/ffmpegoutmodule/.deps/ffmpegoutmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/ffmpegoutmodule/.deps/ffmpegoutmodule_auto.Plo gephex-0.4.4/modules/src/ffmpegoutmodule/.deps/ffmpegoutmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/ffmpegoutmodule/.deps/ffmpegwriter.Plo gephex-0.4.4/modules/src/ffmpegoutmodule/.deps/ffmpegwriter.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/ffmpegoutmodule/ffmpegwriter.cpp gephex-0.4.4/modules/src/ffmpegoutmodule/ffmpegwriter.cpp
--- gephex-0.4.4_orig/modules/src/ffmpegoutmodule/ffmpegwriter.cpp	2008-11-21 20:26:30.000000000 +0100
+++ gephex-0.4.4/modules/src/ffmpegoutmodule/ffmpegwriter.cpp	2008-11-21 21:30:06.000000000 +0100
@@ -22,6 +22,8 @@
 #include <iostream>
 #include <string>
 #include <stdexcept>
+#include <cstdlib>
+#include <cstring>
 
 
 FFMpegWriter::FFMpegWriter(const std::string &filename, int xres, int yres, 
diff -r -u -- gephex-0.4.4_orig/modules/src/flashmodule/.deps/flashmodule.Plo gephex-0.4.4/modules/src/flashmodule/.deps/flashmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/flashmodule/.deps/flashmodule_auto.Plo gephex-0.4.4/modules/src/flashmodule/.deps/flashmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/flipflopmodule/.deps/flipflopmodule.Plo gephex-0.4.4/modules/src/flipflopmodule/.deps/flipflopmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/flipflopmodule/.deps/flipflopmodule_auto.Plo gephex-0.4.4/modules/src/flipflopmodule/.deps/flipflopmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/fpsmodule/.deps/fpsmodule.Plo gephex-0.4.4/modules/src/fpsmodule/.deps/fpsmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/fpsmodule/.deps/fpsmodule_auto.Plo gephex-0.4.4/modules/src/fpsmodule/.deps/fpsmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/frbinmodule/.deps/frbinmodule-bmpdriver.Po gephex-0.4.4/modules/src/frbinmodule/.deps/frbinmodule-bmpdriver.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/frbinmodule/.deps/frbinmodule-ffmpegdriver.Po gephex-0.4.4/modules/src/frbinmodule/.deps/frbinmodule-ffmpegdriver.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/frbinmodule/.deps/frbinmodule-framecache.Po gephex-0.4.4/modules/src/frbinmodule/.deps/frbinmodule-framecache.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/frbinmodule/.deps/frbinmodule-frbinmodule.Po gephex-0.4.4/modules/src/frbinmodule/.deps/frbinmodule-frbinmodule.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/frbinmodule/.deps/frbinmodule-frbinmodule_auto.Po gephex-0.4.4/modules/src/frbinmodule/.deps/frbinmodule-frbinmodule_auto.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/frbinmodule/.deps/frbinmodule-sdlimagedriver.Po gephex-0.4.4/modules/src/frbinmodule/.deps/frbinmodule-sdlimagedriver.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/frbinmodule/bmpdriver.cpp gephex-0.4.4/modules/src/frbinmodule/bmpdriver.cpp
--- gephex-0.4.4_orig/modules/src/frbinmodule/bmpdriver.cpp	2008-11-21 20:26:28.000000000 +0100
+++ gephex-0.4.4/modules/src/frbinmodule/bmpdriver.cpp	2008-11-21 21:24:44.000000000 +0100
@@ -27,6 +27,7 @@
 #include <memory>
 #include <cstdlib>
 #include <cstdio>
+#include <cstring>
 
 //----------------------------------------------------------------------
 
diff -r -u -- gephex-0.4.4_orig/modules/src/frbinmodule/ffmpegdriver.cpp gephex-0.4.4/modules/src/frbinmodule/ffmpegdriver.cpp
--- gephex-0.4.4_orig/modules/src/frbinmodule/ffmpegdriver.cpp	2008-11-21 20:26:28.000000000 +0100
+++ gephex-0.4.4/modules/src/frbinmodule/ffmpegdriver.cpp	2008-11-21 21:26:09.000000000 +0100
@@ -30,6 +30,7 @@
 #include <iostream>
 #include <cassert>
 #include <cmath>
+#include <cstring>
 
 #if defined _MSC_VER
 #define EMULATE_INTTYPES
diff -r -u -- gephex-0.4.4_orig/modules/src/frboutmodule/.deps/frboutmodule.Plo gephex-0.4.4/modules/src/frboutmodule/.deps/frboutmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/frboutmodule/.deps/frboutmodule_auto.Plo gephex-0.4.4/modules/src/frboutmodule/.deps/frboutmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/frboutmodule/.deps/sdloutput.Plo gephex-0.4.4/modules/src/frboutmodule/.deps/sdloutput.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/frboutmodule/.deps/slist.Plo gephex-0.4.4/modules/src/frboutmodule/.deps/slist.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/frboutmodule/.deps/vloopbackoutput.Plo gephex-0.4.4/modules/src/frboutmodule/.deps/vloopbackoutput.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/frboutmodule/.deps/x11output.Plo gephex-0.4.4/modules/src/frboutmodule/.deps/x11output.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/frboutmodule/.deps/x11stuff.Plo gephex-0.4.4/modules/src/frboutmodule/.deps/x11stuff.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/funparsemodule/.deps/builtin_functions.Plo gephex-0.4.4/modules/src/funparsemodule/.deps/builtin_functions.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/funparsemodule/.deps/env.Plo gephex-0.4.4/modules/src/funparsemodule/.deps/env.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/funparsemodule/.deps/expression.Plo gephex-0.4.4/modules/src/funparsemodule/.deps/expression.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/funparsemodule/.deps/expresso.Plo gephex-0.4.4/modules/src/funparsemodule/.deps/expresso.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/funparsemodule/.deps/funparsemodule.Plo gephex-0.4.4/modules/src/funparsemodule/.deps/funparsemodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/funparsemodule/.deps/funparsemodule_auto.Plo gephex-0.4.4/modules/src/funparsemodule/.deps/funparsemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/funparsemodule/.deps/parser.tab.Plo gephex-0.4.4/modules/src/funparsemodule/.deps/parser.tab.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/funparsemodule/.deps/scanner.Plo gephex-0.4.4/modules/src/funparsemodule/.deps/scanner.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/funparsemodule/.deps/slist.Plo gephex-0.4.4/modules/src/funparsemodule/.deps/slist.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/funparsemodule/.deps/string_copy.Plo gephex-0.4.4/modules/src/funparsemodule/.deps/string_copy.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/gradnormmodule/.deps/gradnormmodule-gradnormmodule_auto.Po gephex-0.4.4/modules/src/gradnormmodule/.deps/gradnormmodule-gradnormmodule_auto.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/gradnormmodule/.deps/gradnormmodule.Po gephex-0.4.4/modules/src/gradnormmodule/.deps/gradnormmodule.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/ifsgenmodule/.deps/ifsgenmodule.Plo gephex-0.4.4/modules/src/ifsgenmodule/.deps/ifsgenmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/ifsgenmodule/.deps/ifsgenmodule_auto.Plo gephex-0.4.4/modules/src/ifsgenmodule/.deps/ifsgenmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/ifsmodule/.deps/ifsmodule.Plo gephex-0.4.4/modules/src/ifsmodule/.deps/ifsmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/ifsmodule/.deps/ifsmodule_auto.Plo gephex-0.4.4/modules/src/ifsmodule/.deps/ifsmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/ifsxfadermodule/.deps/ifsxfadermodule.Plo gephex-0.4.4/modules/src/ifsxfadermodule/.deps/ifsxfadermodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/ifsxfadermodule/.deps/ifsxfadermodule_auto.Plo gephex-0.4.4/modules/src/ifsxfadermodule/.deps/ifsxfadermodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/imagebuffermodule/.deps/imagebuffermodule.Plo gephex-0.4.4/modules/src/imagebuffermodule/.deps/imagebuffermodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/imagebuffermodule/.deps/imagebuffermodule_auto.Plo gephex-0.4.4/modules/src/imagebuffermodule/.deps/imagebuffermodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/imgarithmodule/.deps/imgarithmodule-imgarithmodule.Po gephex-0.4.4/modules/src/imgarithmodule/.deps/imgarithmodule-imgarithmodule.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/imgarithmodule/.deps/imgarithmodule-imgarithmodule_auto.Po gephex-0.4.4/modules/src/imgarithmodule/.deps/imgarithmodule-imgarithmodule_auto.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/isingnoizemodule/.deps/isingnoizemodule.Plo gephex-0.4.4/modules/src/isingnoizemodule/.deps/isingnoizemodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/isingnoizemodule/.deps/isingnoizemodule_auto.Plo gephex-0.4.4/modules/src/isingnoizemodule/.deps/isingnoizemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/joystickmodule/.deps/joystickmodule.Plo gephex-0.4.4/modules/src/joystickmodule/.deps/joystickmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/joystickmodule/.deps/joystickmodule_auto.Plo gephex-0.4.4/modules/src/joystickmodule/.deps/joystickmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/jumpmodule/.deps/jumpmodule.Plo gephex-0.4.4/modules/src/jumpmodule/.deps/jumpmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/jumpmodule/.deps/jumpmodule_auto.Plo gephex-0.4.4/modules/src/jumpmodule/.deps/jumpmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/libcolorconv/.deps/libcolorconv.Po gephex-0.4.4/modules/src/libcolorconv/.deps/libcolorconv.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/libgrid/.deps/libgrid.Po gephex-0.4.4/modules/src/libgrid/.deps/libgrid.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/libmidi/.deps/libmidi.Po gephex-0.4.4/modules/src/libmidi/.deps/libmidi.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/libmidi/.deps/midiutil.Po gephex-0.4.4/modules/src/libmidi/.deps/midiutil.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/libscale/.deps/libscale.Po gephex-0.4.4/modules/src/libscale/.deps/libscale.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/midiccdecmodule/.deps/midiccdecmodule.Plo gephex-0.4.4/modules/src/midiccdecmodule/.deps/midiccdecmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midiccdecmodule/.deps/midiccdecmodule_auto.Plo gephex-0.4.4/modules/src/midiccdecmodule/.deps/midiccdecmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midiinmodule/.deps/alsamidiindriver.Plo gephex-0.4.4/modules/src/midiinmodule/.deps/alsamidiindriver.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midiinmodule/.deps/alsaseqmidiindriver.Plo gephex-0.4.4/modules/src/midiinmodule/.deps/alsaseqmidiindriver.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midiinmodule/.deps/midiinmodule.Plo gephex-0.4.4/modules/src/midiinmodule/.deps/midiinmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midiinmodule/.deps/midiinmodule_auto.Plo gephex-0.4.4/modules/src/midiinmodule/.deps/midiinmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midiinmodule/.deps/ossmidiindriver.Plo gephex-0.4.4/modules/src/midiinmodule/.deps/ossmidiindriver.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midikeyrangedecmodule/.deps/midikeyrangedecmodule.Plo gephex-0.4.4/modules/src/midikeyrangedecmodule/.deps/midikeyrangedecmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midikeyrangedecmodule/.deps/midikeyrangedecmodule_auto.Plo gephex-0.4.4/modules/src/midikeyrangedecmodule/.deps/midikeyrangedecmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midinotedecmodule/.deps/midinotedecmodule.Plo gephex-0.4.4/modules/src/midinotedecmodule/.deps/midinotedecmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midinotedecmodule/.deps/midinotedecmodule_auto.Plo gephex-0.4.4/modules/src/midinotedecmodule/.deps/midinotedecmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midipitchwdecmodule/.deps/midipitchwdecmodule.Plo gephex-0.4.4/modules/src/midipitchwdecmodule/.deps/midipitchwdecmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midipitchwdecmodule/.deps/midipitchwdecmodule_auto.Plo gephex-0.4.4/modules/src/midipitchwdecmodule/.deps/midipitchwdecmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midiprgchdecmodule/.deps/midiprgchdecmodule.Plo gephex-0.4.4/modules/src/midiprgchdecmodule/.deps/midiprgchdecmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/midiprgchdecmodule/.deps/midiprgchdecmodule_auto.Plo gephex-0.4.4/modules/src/midiprgchdecmodule/.deps/midiprgchdecmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/mirrormodule/.deps/mirrormodule.Plo gephex-0.4.4/modules/src/mirrormodule/.deps/mirrormodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/mirrormodule/.deps/mirrormodule_auto.Plo gephex-0.4.4/modules/src/mirrormodule/.deps/mirrormodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/netcontrolmodule/.deps/netcontrolmodule.Plo gephex-0.4.4/modules/src/netcontrolmodule/.deps/netcontrolmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/netcontrolmodule/.deps/netcontrolmodule_auto.Plo gephex-0.4.4/modules/src/netcontrolmodule/.deps/netcontrolmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/noisemodule/.deps/noisemodule.Plo gephex-0.4.4/modules/src/noisemodule/.deps/noisemodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/noisemodule/.deps/noisemodule_auto.Plo gephex-0.4.4/modules/src/noisemodule/.deps/noisemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numarithmodule/.deps/numarithmodule.Plo gephex-0.4.4/modules/src/numarithmodule/.deps/numarithmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numarithmodule/.deps/numarithmodule_auto.Plo gephex-0.4.4/modules/src/numarithmodule/.deps/numarithmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/number2colormodule/.deps/number2colormodule.Plo gephex-0.4.4/modules/src/number2colormodule/.deps/number2colormodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/number2colormodule/.deps/number2colormodule_auto.Plo gephex-0.4.4/modules/src/number2colormodule/.deps/number2colormodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numbermodule/.deps/numbermodule.Plo gephex-0.4.4/modules/src/numbermodule/.deps/numbermodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numbermodule/.deps/numbermodule_auto.Plo gephex-0.4.4/modules/src/numbermodule/.deps/numbermodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numenvmodule/.deps/numenvmodule.Plo gephex-0.4.4/modules/src/numenvmodule/.deps/numenvmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numenvmodule/.deps/numenvmodule_auto.Plo gephex-0.4.4/modules/src/numenvmodule/.deps/numenvmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numlfomodule/.deps/numlfomodule.Plo gephex-0.4.4/modules/src/numlfomodule/.deps/numlfomodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numlfomodule/.deps/numlfomodule_auto.Plo gephex-0.4.4/modules/src/numlfomodule/.deps/numlfomodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numoscmodule/.deps/numoscmodule.Plo gephex-0.4.4/modules/src/numoscmodule/.deps/numoscmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numoscmodule/.deps/numoscmodule_auto.Plo gephex-0.4.4/modules/src/numoscmodule/.deps/numoscmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numposmodule/.deps/numposmodule.Plo gephex-0.4.4/modules/src/numposmodule/.deps/numposmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numposmodule/.deps/numposmodule_auto.Plo gephex-0.4.4/modules/src/numposmodule/.deps/numposmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numxfadermodule/.deps/numxfadermodule.Plo gephex-0.4.4/modules/src/numxfadermodule/.deps/numxfadermodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/numxfadermodule/.deps/numxfadermodule_auto.Plo gephex-0.4.4/modules/src/numxfadermodule/.deps/numxfadermodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/oscinmodule/.deps/oscinmodule.Plo gephex-0.4.4/modules/src/oscinmodule/.deps/oscinmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/oscinmodule/.deps/oscinmodule_auto.Plo gephex-0.4.4/modules/src/oscinmodule/.deps/oscinmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/oscmergemodule/.deps/oscmergemodule.Plo gephex-0.4.4/modules/src/oscmergemodule/.deps/oscmergemodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/oscmergemodule/.deps/oscmergemodule_auto.Plo gephex-0.4.4/modules/src/oscmergemodule/.deps/oscmergemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/oscnummodule/.deps/oscnummodule.Plo gephex-0.4.4/modules/src/oscnummodule/.deps/oscnummodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/oscnummodule/.deps/oscnummodule_auto.Plo gephex-0.4.4/modules/src/oscnummodule/.deps/oscnummodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/oscoutmodule/.deps/oscoutmodule.Plo gephex-0.4.4/modules/src/oscoutmodule/.deps/oscoutmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/oscoutmodule/.deps/oscoutmodule_auto.Plo gephex-0.4.4/modules/src/oscoutmodule/.deps/oscoutmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/overlaymodule/.deps/overlaymodule.Plo gephex-0.4.4/modules/src/overlaymodule/.deps/overlaymodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/overlaymodule/.deps/overlaymodule_auto.Plo gephex-0.4.4/modules/src/overlaymodule/.deps/overlaymodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/pixelizemodule/.deps/pixelizemodule.Plo gephex-0.4.4/modules/src/pixelizemodule/.deps/pixelizemodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/pixelizemodule/.deps/pixelizemodule_auto.Plo gephex-0.4.4/modules/src/pixelizemodule/.deps/pixelizemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/planemodule/.deps/planemodule.Plo gephex-0.4.4/modules/src/planemodule/.deps/planemodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/planemodule/.deps/planemodule_auto.Plo gephex-0.4.4/modules/src/planemodule/.deps/planemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/plasmamodule/.deps/plasmamodule.Plo gephex-0.4.4/modules/src/plasmamodule/.deps/plasmamodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/plasmamodule/.deps/plasmamodule_auto.Plo gephex-0.4.4/modules/src/plasmamodule/.deps/plasmamodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/pongmodule/.deps/pongmodule.Plo gephex-0.4.4/modules/src/pongmodule/.deps/pongmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/pongmodule/.deps/pongmodule_auto.Plo gephex-0.4.4/modules/src/pongmodule/.deps/pongmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/posnummodule/.deps/posnummodule.Plo gephex-0.4.4/modules/src/posnummodule/.deps/posnummodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/posnummodule/.deps/posnummodule_auto.Plo gephex-0.4.4/modules/src/posnummodule/.deps/posnummodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/rangeconvertmodule/.deps/rangeconvertmodule.Plo gephex-0.4.4/modules/src/rangeconvertmodule/.deps/rangeconvertmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/rangeconvertmodule/.deps/rangeconvertmodule_auto.Plo gephex-0.4.4/modules/src/rangeconvertmodule/.deps/rangeconvertmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/resultmodule/.deps/resultmodule.Plo gephex-0.4.4/modules/src/resultmodule/.deps/resultmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/resultmodule/.deps/resultmodule_auto.Plo gephex-0.4.4/modules/src/resultmodule/.deps/resultmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/rotozoommodule/.deps/rotozoommodule.Plo gephex-0.4.4/modules/src/rotozoommodule/.deps/rotozoommodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/rotozoommodule/.deps/rotozoommodule_auto.Plo gephex-0.4.4/modules/src/rotozoommodule/.deps/rotozoommodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/sdlfontmodule/.deps/sdlfontmodule.Plo gephex-0.4.4/modules/src/sdlfontmodule/.deps/sdlfontmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/sdlfontmodule/.deps/sdlfontmodule_auto.Plo gephex-0.4.4/modules/src/sdlfontmodule/.deps/sdlfontmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/signalgenmodule/.deps/signalgenmodule.Plo gephex-0.4.4/modules/src/signalgenmodule/.deps/signalgenmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/signalgenmodule/.deps/signalgenmodule_auto.Plo gephex-0.4.4/modules/src/signalgenmodule/.deps/signalgenmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/signalgenmodule/.deps/signals.Plo gephex-0.4.4/modules/src/signalgenmodule/.deps/signals.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/signalplotmodule/.deps/signalplotmodule.Plo gephex-0.4.4/modules/src/signalplotmodule/.deps/signalplotmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/signalplotmodule/.deps/signalplotmodule_auto.Plo gephex-0.4.4/modules/src/signalplotmodule/.deps/signalplotmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/staticcolormodule/.deps/staticcolormodule-staticcolormodule.Po gephex-0.4.4/modules/src/staticcolormodule/.deps/staticcolormodule-staticcolormodule.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/staticcolormodule/.deps/staticcolormodule-staticcolormodule_auto.Po gephex-0.4.4/modules/src/staticcolormodule/.deps/staticcolormodule-staticcolormodule_auto.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/stringtokenizermodule/.deps/stringtokenizermodule.Plo gephex-0.4.4/modules/src/stringtokenizermodule/.deps/stringtokenizermodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/stringtokenizermodule/.deps/stringtokenizermodule_auto.Plo gephex-0.4.4/modules/src/stringtokenizermodule/.deps/stringtokenizermodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/timesourcemodule/.deps/timesourcemodule.Plo gephex-0.4.4/modules/src/timesourcemodule/.deps/timesourcemodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/timesourcemodule/.deps/timesourcemodule_auto.Plo gephex-0.4.4/modules/src/timesourcemodule/.deps/timesourcemodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/tunnelmodule/.deps/tunnelmodule-tunnelmodule.Po gephex-0.4.4/modules/src/tunnelmodule/.deps/tunnelmodule-tunnelmodule.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/tunnelmodule/.deps/tunnelmodule-tunnelmodule_auto.Po gephex-0.4.4/modules/src/tunnelmodule/.deps/tunnelmodule-tunnelmodule_auto.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/twolayermodule/.deps/twolayermodule.Plo gephex-0.4.4/modules/src/twolayermodule/.deps/twolayermodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/twolayermodule/.deps/twolayermodule_auto.Plo gephex-0.4.4/modules/src/twolayermodule/.deps/twolayermodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/xfademodule/.deps/xfademodule-xfademodule.Po gephex-0.4.4/modules/src/xfademodule/.deps/xfademodule-xfademodule.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/xfademodule/.deps/xfademodule-xfademodule_auto.Po gephex-0.4.4/modules/src/xfademodule/.deps/xfademodule-xfademodule_auto.Po
diff -r -u -- gephex-0.4.4_orig/modules/src/y4minmodule/.deps/y4minmodule.Plo gephex-0.4.4/modules/src/y4minmodule/.deps/y4minmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/y4minmodule/.deps/y4minmodule_auto.Plo gephex-0.4.4/modules/src/y4minmodule/.deps/y4minmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/yuv4mpeg2module/.deps/y4moutputmodule.Plo gephex-0.4.4/modules/src/yuv4mpeg2module/.deps/y4moutputmodule.Plo
diff -r -u -- gephex-0.4.4_orig/modules/src/yuv4mpeg2module/.deps/y4moutputmodule_auto.Plo gephex-0.4.4/modules/src/yuv4mpeg2module/.deps/y4moutputmodule_auto.Plo
diff -r -u -- gephex-0.4.4_orig/qtgui/src/guimodel/.deps/controlmodel.Po gephex-0.4.4/qtgui/src/guimodel/.deps/controlmodel.Po
diff -r -u -- gephex-0.4.4_orig/qtgui/src/guimodel/.deps/controlvaluedispatcher.Po gephex-0.4.4/qtgui/src/guimodel/.deps/controlvaluedispatcher.Po
diff -r -u -- gephex-0.4.4_orig/qtgui/src/guimodel/.deps/enginewrapper.Po gephex-0.4.4/qtgui/src/guimodel/.deps/enginewrapper.Po
diff -r -u -- gephex-0.4.4_orig/qtgui/src/guimodel/.deps/graphmodel.Po gephex-0.4.4/qtgui/src/guimodel/.deps/graphmodel.Po
diff -r -u -- gephex-0.4.4_orig/qtgui/src/guimodel/.deps/moduleclassmodel.Po gephex-0.4.4/qtgui/src/guimodel/.deps/moduleclassmodel.Po
diff -r -u -- gephex-0.4.4_orig/qtgui/src/guimodel/.deps/moduledatadispatcher.Po gephex-0.4.4/qtgui/src/guimodel/.deps/moduledatadispatcher.Po
diff -r -u -- gephex-0.4.4_orig/qtgui/src/guimodel/.deps/moduleinfo.Po gephex-0.4.4/qtgui/src/guimodel/.deps/moduleinfo.Po
diff -r -u -- gephex-0.4.4_orig/qtgui/src/guimodel/.deps/point.Po gephex-0.4.4/qtgui/src/guimodel/.deps/point.Po
diff -r -u -- gephex-0.4.4_orig/qtgui/src/guimodel/.deps/xpm.Po gephex-0.4.4/qtgui/src/guimodel/.deps/xpm.Po
diff -r -u -- gephex-0.4.4_orig/qtgui/src/guimodel/moduleclassmodel.cpp gephex-0.4.4/qtgui/src/guimodel/moduleclassmodel.cpp
--- gephex-0.4.4_orig/qtgui/src/guimodel/moduleclassmodel.cpp	2008-11-21 20:26:32.000000000 +0100
+++ gephex-0.4.4/qtgui/src/guimodel/moduleclassmodel.cpp	2008-11-21 21:31:55.000000000 +0100
@@ -24,6 +24,7 @@
 
 #include <sstream>
 #include <cassert>
+#include <cstring>
 
 //TODO
 #include <iostream>
diff -r -u -- gephex-0.4.4_orig/types/src/audiotype/.deps/audiotype.Plo gephex-0.4.4/types/src/audiotype/.deps/audiotype.Plo
diff -r -u -- gephex-0.4.4_orig/types/src/envtype/.deps/envtype.Plo gephex-0.4.4/types/src/envtype/.deps/envtype.Plo
diff -r -u -- gephex-0.4.4_orig/types/src/framebuffertype/.deps/framebuffertype.Plo gephex-0.4.4/types/src/framebuffertype/.deps/framebuffertype.Plo
diff -r -u -- gephex-0.4.4_orig/types/src/ifstype/.deps/ifstype.Plo gephex-0.4.4/types/src/ifstype/.deps/ifstype.Plo
diff -r -u -- gephex-0.4.4_orig/types/src/lfotype/.deps/lfotype.Plo gephex-0.4.4/types/src/lfotype/.deps/lfotype.Plo
diff -r -u -- gephex-0.4.4_orig/types/src/miditype/.deps/miditype.Plo gephex-0.4.4/types/src/miditype/.deps/miditype.Plo
diff -r -u -- gephex-0.4.4_orig/types/src/numbertype/.deps/numbertype.Plo gephex-0.4.4/types/src/numbertype/.deps/numbertype.Plo
diff -r -u -- gephex-0.4.4_orig/types/src/osctype/.deps/osctype.Plo gephex-0.4.4/types/src/osctype/.deps/osctype.Plo
diff -r -u -- gephex-0.4.4_orig/types/src/positiontype/.deps/positiontype.Plo gephex-0.4.4/types/src/positiontype/.deps/positiontype.Plo
diff -r -u -- gephex-0.4.4_orig/types/src/rgbtype/.deps/rgbtype.Plo gephex-0.4.4/types/src/rgbtype/.deps/rgbtype.Plo
diff -r -u -- gephex-0.4.4_orig/types/src/stringtype/.deps/stringtype.Plo gephex-0.4.4/types/src/stringtype/.deps/stringtype.Plo
diff -r -u -- gephex-0.4.4_orig/util/src/cpuinfo/.deps/cpuinfo.Po gephex-0.4.4/util/src/cpuinfo/.deps/cpuinfo.Po
diff -r -u -- gephex-0.4.4_orig/util/src/joystick_midi_net/.deps/joystick_midi_net.Po gephex-0.4.4/util/src/joystick_midi_net/.deps/joystick_midi_net.Po
diff -r -u -- gephex-0.4.4_orig/util/src/libgeo/.deps/libgeo.Po gephex-0.4.4/util/src/libgeo/.deps/libgeo.Po
diff -r -u -- gephex-0.4.4_orig/util/src/libjoystick/.deps/libjoystick.Plo gephex-0.4.4/util/src/libjoystick/.deps/libjoystick.Plo
diff -r -u -- gephex-0.4.4_orig/util/src/libjoystick/.deps/linuxjoystickdriver.Plo gephex-0.4.4/util/src/libjoystick/.deps/linuxjoystickdriver.Plo
diff -r -u -- gephex-0.4.4_orig/util/src/libjoystick/.deps/sdljoystickdriver.Plo gephex-0.4.4/util/src/libjoystick/.deps/sdljoystickdriver.Plo
diff -r -u -- gephex-0.4.4_orig/util/src/misc/.deps/libmisc.Po gephex-0.4.4/util/src/misc/.deps/libmisc.Po
