diff -ru apache_1.3.36_orig/src/main/http_log.c apache_1.3.36/src/main/http_log.c
--- apache_1.3.36_orig/src/main/http_log.c	2006-02-24 11:53:33.000000000 +0100
+++ apache_1.3.36/src/main/http_log.c	2006-02-24 11:53:55.000000000 +0100
@@ -309,11 +309,11 @@
 	logf = NULL;
     }
 
-    if (logf) {
+/*    if (logf) {
 	len = ap_snprintf(errstr, sizeof(errstr), "[%s] ", ap_get_time());
-    } else {
+    } else {*/
 	len = 0;
-    }
+/*    }*/
 
     len += ap_snprintf(errstr + len, sizeof(errstr) - len,
 	    "[%s] ", priorities[level & APLOG_LEVELMASK].t_name);
