--- jpeg-6b_orig/makefile.cfg	2008-03-28 21:13:36.000000000 +0000
+++ jpeg-6b/makefile.cfg	2008-03-28 21:20:53.000000000 +0000
@@ -169,7 +169,7 @@
 # with libtool:
 libjpeg.la: @A2K_DEPS@ $(LIBOBJECTS)
 	$(LIBTOOL) --mode=link $(CC) -o libjpeg.la $(LIBOBJECTS) \
-		-rpath $(libdir) -version-info $(JPEG_LIB_VERSION)
+		-no-undefined -rpath $(libdir) -version-info $(JPEG_LIB_VERSION)
 
 # sample programs:
 
