diff -r -u -- glibmm-2.18.1_orig/glib/glibmm/timeval.cc glibmm-2.18.1/glib/glibmm/timeval.cc
--- glibmm-2.18.1_orig/glib/glibmm/timeval.cc	2008-11-05 09:18:56.000000000 +0100
+++ glibmm-2.18.1/glib/glibmm/timeval.cc	2008-11-05 09:19:30.000000000 +0100
@@ -20,6 +20,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#include <glib/gtypes.h>
 #include <glib/gmain.h>
 #include <glib/gmessages.h>
 #include <glib/gtimer.h>
