diff -r -u -- gcc-4.1.2_orig/libstdc++-v3/configure gcc-4.1.2/libstdc++-v3/configure
--- gcc-4.1.2_orig/libstdc++-v3/configure	2007-05-21 21:01:28.000000000 +0200
+++ gcc-4.1.2/libstdc++-v3/configure	2007-05-21 21:01:45.000000000 +0200
@@ -87133,7 +87133,7 @@
   *)
     { { echo "$as_me:$LINENO: error: No support for this host/target combination." >&5
 echo "$as_me: error: No support for this host/target combination." >&2;}
-   { (exit 1); exit 1; }; }
+   :; }
    ;;
 esac
 
