diff -r -u TUIO_CPP_orig/TuioClient.cpp TUIO_CPP/TuioClient.cpp
--- TUIO_CPP_orig/TuioClient.cpp	2007-07-31 21:06:40.000000000 +0200
+++ TUIO_CPP/TuioClient.cpp	2007-07-31 21:06:54.000000000 +0200
@@ -29,7 +29,7 @@
 */
 
 #include "TuioClient.h"
-
+#include <string.h>
 
 #ifndef WIN32
 static void* ThreadFunc( void* obj )
diff -r -u TUIO_CPP_orig/TuioDump.cpp TUIO_CPP/TuioDump.cpp
--- TUIO_CPP_orig/TuioDump.cpp	2007-07-31 21:06:40.000000000 +0200
+++ TUIO_CPP/TuioDump.cpp	2007-07-31 21:07:41.000000000 +0200
@@ -29,7 +29,8 @@
 */
 
 #include "TuioDump.h"
-		
+#include <string.h>
+
 void TuioDump::addTuioObj(unsigned int s_id, unsigned  int f_id) {
 	std::cout << "added object " << f_id << " (" << s_id << ")" << std::endl;
 }
diff -r -u TUIO_CPP_orig/oscpack/osc/OscPrintReceivedElements.cpp TUIO_CPP/oscpack/osc/OscPrintReceivedElements.cpp
--- TUIO_CPP_orig/oscpack/osc/OscPrintReceivedElements.cpp	2007-07-31 21:06:40.000000000 +0200
+++ TUIO_CPP/oscpack/osc/OscPrintReceivedElements.cpp	2007-07-31 21:07:23.000000000 +0200
@@ -32,7 +32,7 @@
 #include <iostream>
 #include <iomanip>
 #include <ctime>
-
+#include <string.h>
 
 namespace osc{
 
