diff -ru CableSwig_orig/Cable/Generators/CMakeFiles/CableGenerators.dir/depend.make CableSwig/Cable/Generators/CMakeFiles/CableGenerators.dir/depend.make
diff -ru CableSwig_orig/Cable/Generators/cableSystemTools.cxx CableSwig/Cable/Generators/cableSystemTools.cxx
--- CableSwig_orig/Cable/Generators/cableSystemTools.cxx	2008-11-09 19:42:09.000000000 +0100
+++ CableSwig/Cable/Generators/cableSystemTools.cxx	2008-11-09 20:13:14.000000000 +0100
@@ -17,6 +17,7 @@
 #include "cableSystemTools.h"
 
 #include <vector>
+#include <cstring>
 
 #include <ctype.h>
 #include <sys/stat.h>
diff -ru CableSwig_orig/Cable/Parsers/cableXMLParser.cxx CableSwig/Cable/Parsers/cableXMLParser.cxx
--- CableSwig_orig/Cable/Parsers/cableXMLParser.cxx	2008-11-09 19:42:09.000000000 +0100
+++ CableSwig/Cable/Parsers/cableXMLParser.cxx	2008-11-09 20:12:53.000000000 +0100
@@ -18,6 +18,8 @@
 
 #include "../Expat/expat.h"
 
+#include <cstring>
+
 namespace cable
 {
 
diff -ru CableSwig_orig/Executables/CMakeFiles/cableidx.dir/depend.make CableSwig/Executables/CMakeFiles/cableidx.dir/depend.make
diff -ru CableSwig_orig/Executables/CMakeFiles/cswig.dir/depend.make CableSwig/Executables/CMakeFiles/cswig.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/backend.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/backend.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/cpp.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/cpp.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genattr.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genattr.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genattrtab.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genattrtab.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/gencheck.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/gencheck.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/gencodes.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/gencodes.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genconditions.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genconditions.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genconfig.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genconfig.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genconstants.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genconstants.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genemit.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genemit.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genextract.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genextract.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genflags.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genflags.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/gengenrtl.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/gengenrtl.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/gengtype.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/gengtype.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genopinit.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genopinit.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genoutput.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genoutput.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genpeep.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genpeep.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genpreds.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genpreds.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/CMakeFiles/genrecog.dir/depend.make CableSwig/GCC/gcc/CMakeFiles/genrecog.dir/depend.make
diff -ru CableSwig_orig/GCC/gcc/cp/CMakeFiles/gccxml_cc1plus.dir/depend.make CableSwig/GCC/gcc/cp/CMakeFiles/gccxml_cc1plus.dir/depend.make
diff -ru CableSwig_orig/GCC/libiberty/CMakeFiles/iberty.dir/depend.make CableSwig/GCC/libiberty/CMakeFiles/iberty.dir/depend.make
diff -ru CableSwig_orig/GCC_XML/GXFront/CMakeFiles/gccxml.dir/depend.make CableSwig/GCC_XML/GXFront/CMakeFiles/gccxml.dir/depend.make
diff -ru CableSwig_orig/GCC_XML/GXFront/gxConfiguration.cxx CableSwig/GCC_XML/GXFront/gxConfiguration.cxx
--- CableSwig_orig/GCC_XML/GXFront/gxConfiguration.cxx	2008-11-09 19:42:11.000000000 +0100
+++ CableSwig/GCC_XML/GXFront/gxConfiguration.cxx	2008-11-09 20:26:48.000000000 +0100
@@ -16,6 +16,7 @@
 =========================================================================*/
 #include "gxConfiguration.h"
 
+#include <cstring>
 #include <gxsys/ios/sstream>
 
 //----------------------------------------------------------------------------
diff -ru CableSwig_orig/GCC_XML/KWSys/CMakeFiles/gxsys.dir/depend.make CableSwig/GCC_XML/KWSys/CMakeFiles/gxsys.dir/depend.make
diff -ru CableSwig_orig/SWIG/Source/CMakeFiles/swig.dir/depend.make CableSwig/SWIG/Source/CMakeFiles/swig.dir/depend.make
diff -ru CableSwig_orig/SWIG/Source/CMakeFiles/swigLib.dir/depend.make CableSwig/SWIG/Source/CMakeFiles/swigLib.dir/depend.make
