diff -r -u litesql-0.3.2_orig/include/litesql/persistent.hpp litesql-0.3.2/include/litesql/persistent.hpp
--- litesql-0.3.2_orig/include/litesql/persistent.hpp	2007-10-25 17:55:40.000000000 +0200
+++ litesql-0.3.2/include/litesql/persistent.hpp	2007-10-25 18:02:48.000000000 +0200
@@ -32,10 +32,10 @@
 protected:
     typedef map<string, vector<pair<FieldType, string> > > Updates;
     /** executed when object is created into database */    
-    string Persistent::insert(Record& tables, 
-                              Records& fieldRecs,
-                              Records& values,
-                              const string& sequence);
+    string insert(Record& tables, 
+                  Records& fieldRecs,
+                  Records& values,
+                  const string& sequence);
     void update(Updates& updates); 
     template <class T>
     void updateField(litesql::Persistent::Updates& updates, 
