diff -ru vstgui_orig/vstcontrols.cpp vstgui/vstcontrols.cpp
--- vstgui_orig/vstcontrols.cpp	2008-11-27 16:46:00.000000000 +0100
+++ vstgui/vstcontrols.cpp	2008-11-27 16:46:15.000000000 +0100
@@ -41,6 +41,8 @@
 // Date: 2007/01/21
 //-----------------------------------------------------------------------------
 
+#define DBG(...)
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
diff -ru vstgui_orig/vstgui.cpp vstgui/vstgui.cpp
--- vstgui_orig/vstgui.cpp	2008-11-27 16:46:00.000000000 +0100
+++ vstgui/vstgui.cpp	2008-11-27 16:46:31.000000000 +0100
@@ -43,7 +43,8 @@
 #include <cstdio>
 #include <cstdlib>
 #include <cmath>
-#include <string.h>
+#include <iostream>
+#include <cstring>
 
 #include "vstgui.h"
 #include "audioeffectx.h"
