diff -r -u gcc-3.4.6_orig/gcc/Makefile.in gcc-3.4.6/gcc/Makefile.in
--- gcc-3.4.6_orig/gcc/Makefile.in	2007-08-05 01:00:23.000000000 +0000
+++ gcc-3.4.6/gcc/Makefile.in	2007-08-05 01:21:13.000000000 +0000
@@ -689,7 +689,7 @@
 INSN_ATTR_H = insn-attr.h $(srcdir)/insn-addr.h $(srcdir)/varray.h
 C_COMMON_H = c-common.h $(SPLAY_TREE_H) $(CPPLIB_H)
 C_TREE_H = c-tree.h $(C_COMMON_H)
-SYSTEM_H = system.h hwint.h $(srcdir)/../include/libiberty.h
+SYSTEM_H = system.h hwint.h
 PREDICT_H = predict.h predict.def
 CPPLIB_H = cpplib.h line-map.h
 PRETTY_PRINT_H = pretty-print.h input.h $(OBSTACK_H)
@@ -714,15 +714,15 @@
 
 # Build and host support libraries.  FORBUILD is either
 # .. or ../$(build_alias) depending on whether host != build.
-LIBIBERTY = ../libiberty/libiberty.a
-BUILD_LIBIBERTY = @FORBUILD@/libiberty/libiberty.a
+LIBIBERTY = -liberty
+BUILD_LIBIBERTY = -liberty
 
 # Dependencies on the intl and portability libraries.
-LIBDEPS= $(LIBIBERTY) $(LIBINTL_DEP) $(LIBICONV_DEP)
+LIBDEPS= $(LIBIBERTY_DEP) $(LIBINTL_DEP) $(LIBICONV_DEP)
 
 # Likewise, for use in the tools that must run on this machine
 # even if we are cross-building GCC.
-BUILD_LIBDEPS= $(BUILD_LIBIBERTY)
+BUILD_LIBDEPS= $(BUILD_LIBIBERTY_DEP)
 
 # How to link with both our special library facilities
 # and the system's installed libraries.
@@ -2531,7 +2531,7 @@
 	$(FIXINCSRCDIR)/server.h $(FIXINCSRCDIR)/inclhack.def specs.ready
 	(MAKE="$(MAKE)"; srcdir=`cd $(srcdir)/fixinc && ${PWD_COMMAND}` ; \
 	CC="$(CC_FOR_BUILD)"; CFLAGS="$(BUILD_CFLAGS)"; LDFLAGS="$(BUILD_LDFLAGS)"; \
-	WARN_CFLAGS="$(WARN_CFLAGS)"; LIBERTY=`${PWD_COMMAND}`/"$(BUILD_LIBIBERTY)"; \
+	WARN_CFLAGS="$(WARN_CFLAGS)"; LIBERTY="$(BUILD_LIBIBERTY)"; \
 	export MAKE srcdir CC CFLAGS LDFLAGS WARN_CFLAGS LIBERTY; \
 	cd ./fixinc && \
 	$(SHELL) $${srcdir}/mkfixinc.sh $(build) $(target))
