diff -ru -wru mod_ap2_snmp_1.04/mod_ap2_snmp.c mod_ap2_snmp_1.04_orig/mod_ap2_snmp.c
--- mod_ap2_snmp_1.04/mod_ap2_snmp.c	2007-02-15 15:53:27.000000000 +0100
+++ mod_ap2_snmp_1.04_orig/mod_ap2_snmp.c	2007-02-15 15:52:20.000000000 +0100
@@ -451,14 +451,37 @@
 	*/
 	switch(r->status)
 	{
-	    case 400: snmp_set("APACHE2-MIB::httpError400.0",'=',"1"); break;
-	    case 403: snmp_set("APACHE2-MIB::httpError403.0",'=',"1"); break;
-	    case 404: snmp_set("APACHE2-MIB::httpError404.0",'=',"1"); break;
-	    case 405: snmp_set("APACHE2-MIB::httpError405.0",'=',"1"); break;
-	    case 500: snmp_set("APACHE2-MIB::httpError500.0",'=',"1"); break;
-	    case 501: snmp_set("APACHE2-MIB::httpError501.0",'=',"1"); break;
-	    case 505: snmp_set("APACHE2-MIB::httpError505.0",'=',"1"); break;
-	    default: break;
+	    case 400:
+        	snmp_set("APACHE2-MIB::httpError400.0",'=',"1");
+        	break;
+
+	    case 403:
+        	snmp_set("APACHE2-MIB::httpError403.0",'=',"1");
+        	break;
+
+	    case 404:
+        	snmp_set("APACHE2-MIB::httpError404.0",'=',"1");
+        	break;
+
+	    case 405:
+        	snmp_set("APACHE2-MIB::httpError405.0",'=',"1");
+        	break;
+
+	    case 500:
+        	snmp_set("APACHE2-MIB::httpError500.0",'=',"1");
+        	break;
+
+	    case 501:
+        	snmp_set("APACHE2-MIB::httpError501.0",'=',"1");
+        	break;
+
+	    case 505:
+        	snmp_set("APACHE2-MIB::httpError505.0",'=',"1");
+        	break;
+
+	    default:
+	    	break;
+
 	}
 	MOD_SNMP_DEBUG("AP2_SNMP: hook - log_transaction (end)");
 }
