args: -r -u -- pm-medialibrary-0.2.4_orig pm-medialibrary-0.2.4
diff -r -u -- pm-medialibrary-0.2.4_orig/src/.deps/Database.Po pm-medialibrary-0.2.4/src/.deps/Database.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/.deps/Frontend.Po pm-medialibrary-0.2.4/src/.deps/Frontend.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/.deps/Library.Po pm-medialibrary-0.2.4/src/.deps/Library.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/.deps/Log.Po pm-medialibrary-0.2.4/src/.deps/Log.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/.deps/main.Po pm-medialibrary-0.2.4/src/.deps/main.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/Database.cpp pm-medialibrary-0.2.4/src/Database.cpp
--- pm-medialibrary-0.2.4_orig/src/Database.cpp	2009-02-28 12:34:44.000000000 +0100
+++ pm-medialibrary-0.2.4/src/Database.cpp	2009-02-28 12:39:07.000000000 +0100
@@ -26,8 +26,9 @@
  *****************************************************************************/
 
 #include <iostream>
+#include <cstring>
 #include "Database.h"
-#include "Datadict.h"
+#include "DataDict.h"
 #include "Error.h"
 
 CDatabase::CDatabase() {
diff -r -u -- pm-medialibrary-0.2.4_orig/src/Frontend.cpp pm-medialibrary-0.2.4/src/Frontend.cpp
--- pm-medialibrary-0.2.4_orig/src/Frontend.cpp	2009-02-28 12:34:44.000000000 +0100
+++ pm-medialibrary-0.2.4/src/Frontend.cpp	2009-02-28 12:40:11.000000000 +0100
@@ -28,11 +28,13 @@
 #include "config.h"
 #include "Frontend.h"
 #include "Database.h"
-#include "Datadict.h"
+#include "DataDict.h"
 
 #include "Error.h"
 #include "Log.h"
 
+#include <cstring>
+
 #if (!defined(HAVE_STRCASECMP)) && (defined(HAVE_STRICMP))
 #define strcasecmp stricmp
 #endif
diff -r -u -- pm-medialibrary-0.2.4_orig/src/datadict/.deps/DataDict.Po pm-medialibrary-0.2.4/src/datadict/.deps/DataDict.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/datadict/.deps/DataDictCols.Po pm-medialibrary-0.2.4/src/datadict/.deps/DataDictCols.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/datadict/.deps/DataDictTabs.Po pm-medialibrary-0.2.4/src/datadict/.deps/DataDictTabs.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/dcraw/.deps/libdcraw_a-dcraw.Po pm-medialibrary-0.2.4/src/dcraw/.deps/libdcraw_a-dcraw.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/error/.deps/Error.Po pm-medialibrary-0.2.4/src/error/.deps/Error.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/error/Error.cpp pm-medialibrary-0.2.4/src/error/Error.cpp
--- pm-medialibrary-0.2.4_orig/src/error/Error.cpp	2009-02-28 12:34:44.000000000 +0100
+++ pm-medialibrary-0.2.4/src/error/Error.cpp	2009-02-28 12:35:04.000000000 +0100
@@ -26,6 +26,7 @@
  *****************************************************************************/
 
 #include <string>
+#include <cstring>
 #include "Error.h"
 
 CError::CError(const char* szFile, const int iLine, const char* szFunc, char* szTxt)
diff -r -u -- pm-medialibrary-0.2.4_orig/src/md5/.deps/md5.Po pm-medialibrary-0.2.4/src/md5/.deps/md5.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/server/.deps/Asset.Po pm-medialibrary-0.2.4/src/server/.deps/Asset.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/server/.deps/Folder.Po pm-medialibrary-0.2.4/src/server/.deps/Folder.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/server/.deps/FolderAsset.Po pm-medialibrary-0.2.4/src/server/.deps/FolderAsset.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/server/.deps/Info.Po pm-medialibrary-0.2.4/src/server/.deps/Info.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/server/.deps/Part.Po pm-medialibrary-0.2.4/src/server/.deps/Part.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/server/.deps/Project.Po pm-medialibrary-0.2.4/src/server/.deps/Project.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/server/.deps/Server.Po pm-medialibrary-0.2.4/src/server/.deps/Server.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/server/.deps/Storage.Po pm-medialibrary-0.2.4/src/server/.deps/Storage.Po
diff -r -u -- pm-medialibrary-0.2.4_orig/src/server/.deps/Version.Po pm-medialibrary-0.2.4/src/server/.deps/Version.Po
