diff -ru actualiza_orig/Makefile.scam actualiza/Makefile.scam
--- actualiza_orig/Makefile.scam	2008-07-17 10:37:36.000000000 +0200
+++ actualiza/Makefile.scam	2008-07-17 10:37:43.000000000 +0200
@@ -2,7 +2,7 @@
 VERSION=0.8.0
 EXTRA=cpp qt kde sharedlib backreference expand
 
-CXXFLAGS=-Wall -Werror -DTUXUM_ACTUALIZA
+CXXFLAGS=-Wall -DTUXUM_ACTUALIZA
 CXXFLAGS=-DPREFIX=\"${prefix}/\" -DVERSION=\"${VERSION}\" -g -O3
 LDFLAGS=-lkhtml
 
diff -ru actualiza_orig/PackageListToolTip.h actualiza/PackageListToolTip.h
--- actualiza_orig/PackageListToolTip.h	2008-07-17 10:37:36.000000000 +0200
+++ actualiza/PackageListToolTip.h	2008-07-17 10:39:24.000000000 +0200
@@ -35,7 +35,7 @@
 class PackageListViewToolTip : public QToolTip{
 	public:
 		PackageListViewToolTip(PackageListView *_packageListView,
-							   QWidget* parent, char* ):QToolTip(parent)	{	
+							   QWidget* parent, const char* ):QToolTip(parent)	{	
 			packageListView=_packageListView;
 		}
 		virtual ~PackageListViewToolTip(){}
