diff -ru iconvircproxy-1.6.3_orig/Makefile.sets.in iconvircproxy-1.6.3/Makefile.sets.in
--- iconvircproxy-1.6.3_orig/Makefile.sets.in	2008-07-04 10:31:39.000000000 +0200
+++ iconvircproxy-1.6.3/Makefile.sets.in	2008-07-04 10:32:58.000000000 +0200
@@ -1,7 +1,7 @@
 CPPDEBUG=-g -pipe
 
 # Options for compiling, generally.
-CPPFLAGS=-DVERSION=\"$(VERSION)\" $(CPPDEBUG)
+CPPFLAGS=-DVERSION=\"$(VERSION)\" -DSYSCONFDIR=\"$(SYSCONFDIR)\" $(CPPDEBUG)
 #         -fmessage-length=0
 
 # C specific warnings
diff -ru iconvircproxy-1.6.3_orig/config.cc iconvircproxy-1.6.3/config.cc
--- iconvircproxy-1.6.3_orig/config.cc	2008-07-04 10:31:39.000000000 +0200
+++ iconvircproxy-1.6.3/config.cc	2008-07-04 10:33:32.000000000 +0200
@@ -1,7 +1,7 @@
 #include "config.hh"
 #include "confparser.hh"
 
-static const char cfgfilename[] = "ircproxy.cfg";
+static const char cfgfilename[] = SYSCONFDIR"/ircproxy.cfg";
 
 static ConfParser config;
 static class ConfigLoader
