diff -r -u -- DB_DataObject-1.8.5_orig/DataObject/createTables.php DB_DataObject-1.8.5/DataObject/createTables.php
--- DB_DataObject-1.8.5_orig/DataObject/createTables.php	2007-05-15 14:23:56.000000000 +0200
+++ DB_DataObject-1.8.5/DataObject/createTables.php	2007-05-15 14:24:09.000000000 +0200
@@ -1,5 +1,8 @@
-#!/usr/bin/php -q
+#!@bindir@/php5 -q
 <?php
+  
+ini_set('include_path', '@libdir@/php5:@libdir@/php5/DB:@libdir@/php5/PEAR');
+
 // +----------------------------------------------------------------------+
 // | PHP Version 4                                                        |
 // +----------------------------------------------------------------------+
diff -r -u -- DB_DataObject-1.8.5_orig/DataObject/createTables.php DB_DataObject-1.8.5/DataObject/createTables.php
--- DB_DataObject-1.8.5_orig/DataObject/createTables.php	2007-05-15 14:27:55.000000000 +0200
+++ DB_DataObject-1.8.5/DataObject/createTables.php	2007-05-15 14:28:31.000000000 +0200
@@ -36,7 +36,7 @@
 }
 
 if (!@$_SERVER['argv'][1]) {
-    PEAR::raiseError("\nERROR: createTable.php usage:\n\nC:\php\pear\DB\DataObjects\createTable.php example.ini\n\n", null, PEAR_ERROR_DIE);
+    PEAR::raiseError("\nERROR: createTable.php usage:\n\n".$_SERVER['argv'][0]." example.ini\n\n", null, PEAR_ERROR_DIE);
     exit;
 }
 
