diff -ru jslibs-0.9-r1881_orig/src/host/Makefile jslibs-0.9-r1881/src/host/Makefile
--- jslibs-0.9-r1881_orig/src/host/Makefile	2009-02-06 09:29:36.000000000 +0100
+++ jslibs-0.9-r1881/src/host/Makefile	2009-02-06 09:32:33.000000000 +0100
@@ -1,5 +1,6 @@
 TARGET     = host.a
 SRC        = $(wildcard *.cpp)
+DEFINES    = -DDLL_DIR=\"/usr/lib/js\"
 INCLUDES   = `pkg-config js --cflags`
 STATICLIBS =
 SHAREDLIBS =
diff -ru jslibs-0.9-r1881_orig/src/host/host.cpp jslibs-0.9-r1881/src/host/host.cpp
--- jslibs-0.9-r1881_orig/src/host/host.cpp	2009-02-06 09:29:36.000000000 +0100
+++ jslibs-0.9-r1881/src/host/host.cpp	2009-02-06 09:34:42.000000000 +0100
@@ -207,7 +207,11 @@
 	const char *fileName;
 	J_CHK( JsvalToString(cx, &argv[0], &fileName) );
 	char libFileName[PATH_MAX];
-	strcpy( libFileName, fileName );
+  char *libName;
+	strcpy( libFileName, DLL_DIR );
+  strcat( libFileName, "/");
+  libName = libFileName + strlen(libFileName);
+  strcat( libFileName, fileName );
 	strcat( libFileName, DLL_EXT );
 // MAC OSX: 	'@executable_path' ??
 /*
@@ -246,7 +250,7 @@
 	J_S_ASSERT( libFileName != NULL && *libFileName != '\0', "Invalid module filename." );
 	JLLibraryHandler module;
 	module = JLDynamicLibraryOpen(libFileName);
-	J_S_ASSERT_1( JLDynamicLibraryOk(module), "Unable to load the 'module %s'.", libFileName );
+	J_S_ASSERT_1( JLDynamicLibraryOk(module), "Unable to load the 'module %s'.", libName );
 
 	ModuleInitFunction moduleInit;
 	moduleInit = (ModuleInitFunction)JLDynamicLibrarySymbol(module, NAME_MODULE_INIT);
