diff -ru librsvg-2.22.3_orig/rsvg-paint-server.h librsvg-2.22.3/rsvg-paint-server.h
--- librsvg-2.22.3_orig/rsvg-paint-server.h	2009-02-06 08:59:07.000000000 +0100
+++ librsvg-2.22.3/rsvg-paint-server.h	2009-02-06 08:59:55.000000000 +0100
@@ -116,6 +116,13 @@
     guint32 rgb;
 };
 
+enum _RsvgPaintServerType {
+    RSVG_PAINT_SERVER_RAD_GRAD,
+    RSVG_PAINT_SERVER_LIN_GRAD,
+    RSVG_PAINT_SERVER_SOLID,
+    RSVG_PAINT_SERVER_PATTERN
+};
+
 typedef struct _RsvgSolidColour RsvgPaintServerColour;
 typedef enum _RsvgPaintServerType RsvgPaintServerType;
 typedef union _RsvgPaintServerCore RsvgPaintServerCore;
@@ -127,13 +134,6 @@
     RsvgPattern *pattern;
 };
 
-enum _RsvgPaintServerType {
-    RSVG_PAINT_SERVER_RAD_GRAD,
-    RSVG_PAINT_SERVER_LIN_GRAD,
-    RSVG_PAINT_SERVER_SOLID,
-    RSVG_PAINT_SERVER_PATTERN
-};
-
 struct _RsvgPaintServer {
     int refcnt;
     RsvgPaintServerType type;
