diff -ru opencv-1.0.0_orig/otherlibs/highgui/cvcap_ffmpeg.cpp opencv-1.0.0/otherlibs/highgui/cvcap_ffmpeg.cpp
--- opencv-1.0.0_orig/otherlibs/highgui/cvcap_ffmpeg.cpp	2009-02-07 19:05:08.000000000 +0100
+++ opencv-1.0.0/otherlibs/highgui/cvcap_ffmpeg.cpp	2009-02-07 19:06:01.000000000 +0100
@@ -781,7 +781,7 @@
 
 	if (!(mywriter->fmt->flags & AVFMT_NOFILE)) {
 		/* close the output file */
-		url_fclose(&mywriter->oc->pb);
+		url_fclose(mywriter->oc->pb);
 	}
 
 	/* free the stream */
