--- libtool-1.5.24_orig/ltmain.in	2007-09-22 06:47:14.000000000 +0200
+++ libtool-1.5.24/ltmain.in	2007-09-22 10:43:02.000000000 +0200
@@ -6028,8 +6028,8 @@
 	  test -n "$relink_command" && srcname="$realname"T
 
 	  # Install the shared library and build the symlinks.
-	  $show "$install_prog $dir/$srcname $destdir/$realname"
-	  $run eval "$install_prog $dir/$srcname $destdir/$realname" || exit $?
+	  $show "$install_prog -m 644 $dir/$srcname $destdir/$realname"
+	  $run eval "$install_prog -m 644 $dir/$srcname $destdir/$realname" || exit $?
 	  if test -n "$stripme" && test -n "$striplib"; then
 	    $show "$striplib $destdir/$realname"
 	    $run eval "$striplib $destdir/$realname" || exit $?
