diff -ru apachetop-0.12.6_orig/src/resolver.h apachetop-0.12.6/src/resolver.h
--- apachetop-0.12.6_orig/src/resolver.h	2008-07-17 15:36:15.000000000 +0200
+++ apachetop-0.12.6/src/resolver.h	2008-07-17 15:36:25.000000000 +0200
@@ -10,8 +10,8 @@
 class Resolver
 {
 	public:
-	Resolver::Resolver(void);
-	Resolver::~Resolver(void);
+	Resolver(void);
+	~Resolver(void);
 	int add_request(char *request, enum resolver_action act);
 
 
