diff -r -u SndObj-2.6.4_orig/src/AdSyn.cpp SndObj-2.6.4/src/AdSyn.cpp
--- SndObj-2.6.4_orig/src/AdSyn.cpp	2007-07-25 00:21:58.000000000 +0200
+++ SndObj-2.6.4/src/AdSyn.cpp	2007-07-25 00:24:48.000000000 +0200
@@ -3,6 +3,7 @@
 // and licensing information
 
 #include "AdSyn.h"
+#include <string.h>
 
 AdSyn::AdSyn(){
 
Only in SndObj-2.6.4/src: AdSyn.os
Only in SndObj-2.6.4/src: Convol.os
Only in SndObj-2.6.4/src: EnvTable.os
diff -r -u SndObj-2.6.4_orig/src/FFT.cpp SndObj-2.6.4/src/FFT.cpp
--- SndObj-2.6.4_orig/src/FFT.cpp	2007-07-25 00:21:58.000000000 +0200
+++ SndObj-2.6.4/src/FFT.cpp	2007-07-25 00:23:02.000000000 +0200
@@ -8,6 +8,7 @@
 //           Victor Lazzarini, 2003
 /////////////////////////////////////////////////
 #include "FFT.h"
+#include <string.h>
 
 FFT::FFT(){
 
Only in SndObj-2.6.4/src: FFT.os
Only in SndObj-2.6.4/src: FastOsc.os
Only in SndObj-2.6.4/src: HammingTable.os
Only in SndObj-2.6.4/src: HarmTable.os
Only in SndObj-2.6.4/src: HiPass.os
diff -r -u SndObj-2.6.4_orig/src/IFAdd.cpp SndObj-2.6.4/src/IFAdd.cpp
--- SndObj-2.6.4_orig/src/IFAdd.cpp	2007-07-25 00:21:58.000000000 +0200
+++ SndObj-2.6.4/src/IFAdd.cpp	2007-07-25 00:25:13.000000000 +0200
@@ -3,6 +3,7 @@
 // and licensing information
 
 #include "IFAdd.h"
+#include <string.h>
 
 IFAdd::IFAdd(){
 }
Only in SndObj-2.6.4/src: IFAdd.os
diff -r -u SndObj-2.6.4_orig/src/IFFT.cpp SndObj-2.6.4/src/IFFT.cpp
--- SndObj-2.6.4_orig/src/IFFT.cpp	2007-07-25 00:21:58.000000000 +0200
+++ SndObj-2.6.4/src/IFFT.cpp	2007-07-25 00:23:16.000000000 +0200
@@ -8,7 +8,7 @@
 //           Victor Lazzarini, 2003
 /////////////////////////////////////////////////
 #include "IFFT.h"
-
+#include <string.h>
 
 IFFT::IFFT(){
 
Only in SndObj-2.6.4/src: IFFT.os
Only in SndObj-2.6.4/src: IFGram.os
Only in SndObj-2.6.4/src: ImpulseTable.os
Only in SndObj-2.6.4/src: LoPassTable.os
Only in SndObj-2.6.4/src: LowPass.os
Only in SndObj-2.6.4/src: NoteTable.os
Only in SndObj-2.6.4/src: Osc.os
Only in SndObj-2.6.4/src: Osci.os
diff -r -u SndObj-2.6.4_orig/src/PVA.cpp SndObj-2.6.4/src/PVA.cpp
--- SndObj-2.6.4_orig/src/PVA.cpp	2007-07-25 00:21:58.000000000 +0200
+++ SndObj-2.6.4/src/PVA.cpp	2007-07-25 00:23:31.000000000 +0200
@@ -8,6 +8,7 @@
 //           Victor Lazzarini, 2003
 /////////////////////////////////////////////////
 #include "PVA.h"
+#include <string.h>
 
 PVA::PVA(){
   m_rotcount = 0;
Only in SndObj-2.6.4/src: PVA.os
Only in SndObj-2.6.4/src: PVBlur.os
Only in SndObj-2.6.4/src: PVEnvTable.os
Only in SndObj-2.6.4/src: PVFilter.os
Only in SndObj-2.6.4/src: PVMask.os
Only in SndObj-2.6.4/src: PVMix.os
Only in SndObj-2.6.4/src: PVMorph.os
diff -r -u SndObj-2.6.4_orig/src/PVRead.cpp SndObj-2.6.4/src/PVRead.cpp
--- SndObj-2.6.4_orig/src/PVRead.cpp	2007-07-25 00:21:58.000000000 +0200
+++ SndObj-2.6.4/src/PVRead.cpp	2007-07-25 00:24:03.000000000 +0200
@@ -3,6 +3,7 @@
 // and licensing information
 
 #include "PVRead.h"
+#include <string.h>
 
 PVRead::PVRead(){
 	
Only in SndObj-2.6.4/src: PVRead.os
diff -r -u SndObj-2.6.4_orig/src/PVS.cpp SndObj-2.6.4/src/PVS.cpp
--- SndObj-2.6.4_orig/src/PVS.cpp	2007-07-25 00:21:58.000000000 +0200
+++ SndObj-2.6.4/src/PVS.cpp	2007-07-25 00:23:48.000000000 +0200
@@ -9,7 +9,7 @@
 //
 /////////////////////////////////////////////////
 #include "PVS.h"
-
+#include <string.h>
 
 PVS::PVS(){
   m_rotcount = m_vecsize;
Only in SndObj-2.6.4/src: PVS.os
Only in SndObj-2.6.4/src: PVTable.os
Only in SndObj-2.6.4/src: PVTransp.os
Only in SndObj-2.6.4/src: PlnTable.os
diff -r -u SndObj-2.6.4_orig/src/ReSyn.cpp SndObj-2.6.4/src/ReSyn.cpp
--- SndObj-2.6.4_orig/src/ReSyn.cpp	2007-07-25 00:21:58.000000000 +0200
+++ SndObj-2.6.4/src/ReSyn.cpp	2007-07-25 00:25:01.000000000 +0200
@@ -4,6 +4,7 @@
 
 #include "ReSyn.h"
 #include "IFGram.h"
+#include <string.h>
 
 ReSyn::ReSyn(){
   AddMsg("pitch", 31);
Only in SndObj-2.6.4/src: ReSyn.os
Only in SndObj-2.6.4/src: SinAnal.os
diff -r -u SndObj-2.6.4_orig/src/SinSyn.cpp SndObj-2.6.4/src/SinSyn.cpp
--- SndObj-2.6.4_orig/src/SinSyn.cpp	2007-07-25 00:21:58.000000000 +0200
+++ SndObj-2.6.4/src/SinSyn.cpp	2007-07-25 00:24:34.000000000 +0200
@@ -3,6 +3,7 @@
 // and licensing information
 
 #include "SinSyn.h"
+#include <string.h>
 
 SinSyn::SinSyn(){
 
Only in SndObj-2.6.4/src: SinSyn.os
Only in SndObj-2.6.4/src: SndASIO.os
Only in SndObj-2.6.4/src: SndAiff.os
Only in SndObj-2.6.4/src: SndBuffer.os
Only in SndObj-2.6.4/src: SndCoreAudio.os
Only in SndObj-2.6.4/src: SndFIO.os
Only in SndObj-2.6.4/src: SndIO.os
Only in SndObj-2.6.4/src: SndJackIO.os
Only in SndObj-2.6.4/src: SndMidi.os
Only in SndObj-2.6.4/src: SndMidiIn.os
Only in SndObj-2.6.4/src: SndPVOCEX.os
Only in SndObj-2.6.4/src: SndRTIO.os
Only in SndObj-2.6.4/src: SndRTThread.os
diff -r -u SndObj-2.6.4_orig/src/SndRead.cpp SndObj-2.6.4/src/SndRead.cpp
--- SndObj-2.6.4_orig/src/SndRead.cpp	2007-07-25 00:21:58.000000000 +0200
+++ SndObj-2.6.4/src/SndRead.cpp	2007-07-25 00:25:25.000000000 +0200
@@ -5,6 +5,7 @@
 #include "SndRead.h"
 #include "SndWave.h"
 #include "SndAiff.h"
+#include <string.h>
 
 SndRead::SndRead(){
 	
Only in SndObj-2.6.4/src: SndRead.os
Only in SndObj-2.6.4/src: SndSinIO.os
Only in SndObj-2.6.4/src: SndTable.os
Only in SndObj-2.6.4/src: SndThread.os
Only in SndObj-2.6.4/src: SndWave.os
Only in SndObj-2.6.4/src: SndWaveX.os
Only in SndObj-2.6.4/src: SpecCart.os
Only in SndObj-2.6.4/src: SpecCombine.os
Only in SndObj-2.6.4/src: SpecEnvTable.os
Only in SndObj-2.6.4/src: SpecIn.os
Only in SndObj-2.6.4/src: SpecInterp.os
Only in SndObj-2.6.4/src: SpecMult.os
Only in SndObj-2.6.4/src: SpecPolar.os
Only in SndObj-2.6.4/src: SpecSplit.os
Only in SndObj-2.6.4/src: SpecThresh.os
Only in SndObj-2.6.4/src: SpecVoc.os
Only in SndObj-2.6.4/src: TpTz.os
Only in SndObj-2.6.4/src: TrisegTable.os
Only in SndObj-2.6.4/src: UsrDefTable.os
Only in SndObj-2.6.4/src: UsrHarmTable.os
Only in SndObj-2.6.4/src/examples: Ladspaplug.os
Only in SndObj-2.6.4/src/examples: ladspa_example.os
Only in SndObj-2.6.4/src/examples: morph_tilde.os
Only in SndObj-2.6.4/src/rfftw: config.os
Only in SndObj-2.6.4/src/rfftw: executor.os
Only in SndObj-2.6.4/src/rfftw: fcr_1.os
Only in SndObj-2.6.4/src/rfftw: fcr_10.os
Only in SndObj-2.6.4/src/rfftw: fcr_11.os
Only in SndObj-2.6.4/src/rfftw: fcr_12.os
Only in SndObj-2.6.4/src/rfftw: fcr_128.os
Only in SndObj-2.6.4/src/rfftw: fcr_13.os
Only in SndObj-2.6.4/src/rfftw: fcr_14.os
Only in SndObj-2.6.4/src/rfftw: fcr_15.os
Only in SndObj-2.6.4/src/rfftw: fcr_16.os
Only in SndObj-2.6.4/src/rfftw: fcr_2.os
Only in SndObj-2.6.4/src/rfftw: fcr_3.os
Only in SndObj-2.6.4/src/rfftw: fcr_32.os
Only in SndObj-2.6.4/src/rfftw: fcr_4.os
Only in SndObj-2.6.4/src/rfftw: fcr_5.os
Only in SndObj-2.6.4/src/rfftw: fcr_6.os
Only in SndObj-2.6.4/src/rfftw: fcr_64.os
Only in SndObj-2.6.4/src/rfftw: fcr_7.os
Only in SndObj-2.6.4/src/rfftw: fcr_8.os
Only in SndObj-2.6.4/src/rfftw: fcr_9.os
Only in SndObj-2.6.4/src/rfftw: fftwnd.os
Only in SndObj-2.6.4/src/rfftw: fhb_10.os
Only in SndObj-2.6.4/src/rfftw: fhb_16.os
Only in SndObj-2.6.4/src/rfftw: fhb_2.os
Only in SndObj-2.6.4/src/rfftw: fhb_3.os
Only in SndObj-2.6.4/src/rfftw: fhb_32.os
Only in SndObj-2.6.4/src/rfftw: fhb_4.os
Only in SndObj-2.6.4/src/rfftw: fhb_5.os
Only in SndObj-2.6.4/src/rfftw: fhb_6.os
Only in SndObj-2.6.4/src/rfftw: fhb_7.os
Only in SndObj-2.6.4/src/rfftw: fhb_8.os
Only in SndObj-2.6.4/src/rfftw: fhb_9.os
Only in SndObj-2.6.4/src/rfftw: fhf_10.os
Only in SndObj-2.6.4/src/rfftw: fhf_16.os
Only in SndObj-2.6.4/src/rfftw: fhf_2.os
Only in SndObj-2.6.4/src/rfftw: fhf_3.os
Only in SndObj-2.6.4/src/rfftw: fhf_32.os
Only in SndObj-2.6.4/src/rfftw: fhf_4.os
Only in SndObj-2.6.4/src/rfftw: fhf_5.os
Only in SndObj-2.6.4/src/rfftw: fhf_6.os
Only in SndObj-2.6.4/src/rfftw: fhf_7.os
Only in SndObj-2.6.4/src/rfftw: fhf_8.os
Only in SndObj-2.6.4/src/rfftw: fhf_9.os
Only in SndObj-2.6.4/src/rfftw: fn_1.os
Only in SndObj-2.6.4/src/rfftw: fn_10.os
Only in SndObj-2.6.4/src/rfftw: fn_11.os
Only in SndObj-2.6.4/src/rfftw: fn_12.os
Only in SndObj-2.6.4/src/rfftw: fn_13.os
Only in SndObj-2.6.4/src/rfftw: fn_14.os
Only in SndObj-2.6.4/src/rfftw: fn_15.os
Only in SndObj-2.6.4/src/rfftw: fn_16.os
Only in SndObj-2.6.4/src/rfftw: fn_2.os
Only in SndObj-2.6.4/src/rfftw: fn_3.os
Only in SndObj-2.6.4/src/rfftw: fn_32.os
Only in SndObj-2.6.4/src/rfftw: fn_4.os
Only in SndObj-2.6.4/src/rfftw: fn_5.os
Only in SndObj-2.6.4/src/rfftw: fn_6.os
Only in SndObj-2.6.4/src/rfftw: fn_64.os
Only in SndObj-2.6.4/src/rfftw: fn_7.os
Only in SndObj-2.6.4/src/rfftw: fn_8.os
Only in SndObj-2.6.4/src/rfftw: fn_9.os
Only in SndObj-2.6.4/src/rfftw: fni_1.os
Only in SndObj-2.6.4/src/rfftw: fni_10.os
Only in SndObj-2.6.4/src/rfftw: fni_11.os
Only in SndObj-2.6.4/src/rfftw: fni_12.os
Only in SndObj-2.6.4/src/rfftw: fni_13.os
Only in SndObj-2.6.4/src/rfftw: fni_14.os
Only in SndObj-2.6.4/src/rfftw: fni_15.os
Only in SndObj-2.6.4/src/rfftw: fni_16.os
Only in SndObj-2.6.4/src/rfftw: fni_2.os
Only in SndObj-2.6.4/src/rfftw: fni_3.os
Only in SndObj-2.6.4/src/rfftw: fni_32.os
Only in SndObj-2.6.4/src/rfftw: fni_4.os
Only in SndObj-2.6.4/src/rfftw: fni_5.os
Only in SndObj-2.6.4/src/rfftw: fni_6.os
Only in SndObj-2.6.4/src/rfftw: fni_64.os
Only in SndObj-2.6.4/src/rfftw: fni_7.os
Only in SndObj-2.6.4/src/rfftw: fni_8.os
Only in SndObj-2.6.4/src/rfftw: fni_9.os
Only in SndObj-2.6.4/src/rfftw: frc_1.os
Only in SndObj-2.6.4/src/rfftw: frc_10.os
Only in SndObj-2.6.4/src/rfftw: frc_11.os
Only in SndObj-2.6.4/src/rfftw: frc_12.os
Only in SndObj-2.6.4/src/rfftw: frc_128.os
Only in SndObj-2.6.4/src/rfftw: frc_13.os
Only in SndObj-2.6.4/src/rfftw: frc_14.os
Only in SndObj-2.6.4/src/rfftw: frc_15.os
Only in SndObj-2.6.4/src/rfftw: frc_16.os
Only in SndObj-2.6.4/src/rfftw: frc_2.os
Only in SndObj-2.6.4/src/rfftw: frc_3.os
Only in SndObj-2.6.4/src/rfftw: frc_32.os
Only in SndObj-2.6.4/src/rfftw: frc_4.os
Only in SndObj-2.6.4/src/rfftw: frc_5.os
Only in SndObj-2.6.4/src/rfftw: frc_6.os
Only in SndObj-2.6.4/src/rfftw: frc_64.os
Only in SndObj-2.6.4/src/rfftw: frc_7.os
Only in SndObj-2.6.4/src/rfftw: frc_8.os
Only in SndObj-2.6.4/src/rfftw: frc_9.os
Only in SndObj-2.6.4/src/rfftw: ftw_10.os
Only in SndObj-2.6.4/src/rfftw: ftw_16.os
Only in SndObj-2.6.4/src/rfftw: ftw_2.os
Only in SndObj-2.6.4/src/rfftw: ftw_3.os
Only in SndObj-2.6.4/src/rfftw: ftw_32.os
Only in SndObj-2.6.4/src/rfftw: ftw_4.os
Only in SndObj-2.6.4/src/rfftw: ftw_5.os
Only in SndObj-2.6.4/src/rfftw: ftw_6.os
Only in SndObj-2.6.4/src/rfftw: ftw_64.os
Only in SndObj-2.6.4/src/rfftw: ftw_7.os
Only in SndObj-2.6.4/src/rfftw: ftw_8.os
Only in SndObj-2.6.4/src/rfftw: ftw_9.os
Only in SndObj-2.6.4/src/rfftw: ftwi_10.os
Only in SndObj-2.6.4/src/rfftw: ftwi_16.os
Only in SndObj-2.6.4/src/rfftw: ftwi_2.os
Only in SndObj-2.6.4/src/rfftw: ftwi_3.os
Only in SndObj-2.6.4/src/rfftw: ftwi_32.os
Only in SndObj-2.6.4/src/rfftw: ftwi_4.os
Only in SndObj-2.6.4/src/rfftw: ftwi_5.os
Only in SndObj-2.6.4/src/rfftw: ftwi_6.os
Only in SndObj-2.6.4/src/rfftw: ftwi_64.os
Only in SndObj-2.6.4/src/rfftw: ftwi_7.os
Only in SndObj-2.6.4/src/rfftw: ftwi_8.os
Only in SndObj-2.6.4/src/rfftw: ftwi_9.os
Only in SndObj-2.6.4/src/rfftw: generic.os
Only in SndObj-2.6.4/src/rfftw: malloc.os
Only in SndObj-2.6.4/src/rfftw: planner.os
Only in SndObj-2.6.4/src/rfftw: putils.os
Only in SndObj-2.6.4/src/rfftw: rader.os
Only in SndObj-2.6.4/src/rfftw: rconfig.os
Only in SndObj-2.6.4/src/rfftw: rexec.os
Only in SndObj-2.6.4/src/rfftw: rexec2.os
Only in SndObj-2.6.4/src/rfftw: rfftwf77.os
Only in SndObj-2.6.4/src/rfftw: rfftwnd.os
Only in SndObj-2.6.4/src/rfftw: rgeneric.os
Only in SndObj-2.6.4/src/rfftw: rplanner.os
Only in SndObj-2.6.4/src/rfftw: timer.os
Only in SndObj-2.6.4/src/rfftw: twiddle.os
Only in SndObj-2.6.4/src/rfftw: wisdom.os
Only in SndObj-2.6.4/src/rfftw: wisdomio.os
