diff -ru -ru goffice-0.2.1_orig/plugins/plot_radar/gog-radar.c goffice-0.2.1/plugins/plot_radar/gog-radar.c
--- goffice-0.2.1_orig/plugins/plot_radar/gog-radar.c	2006-10-18 15:59:13.000000000 +0200
+++ goffice-0.2.1/plugins/plot_radar/gog-radar.c	2006-10-18 16:03:17.000000000 +0200
@@ -387,7 +387,7 @@
 typedef GogPlotViewClass	GogRTViewClass;
 
 static double
-fmin (double a, double b)
+__fmin (double a, double b)
 {
 	return (a < b) ? a : b;
 }
@@ -518,7 +518,7 @@
 			      GogObject const *cur_selection,
 			      GogObject **obj, char **name)
 {
-	double radius = fmin (view->allocation.h, view->allocation.w)/2.0;
+	double radius = __fmin (view->allocation.h, view->allocation.w)/2.0;
 
 	x -= view->allocation.x + view->allocation.w/2.;
 	y -= view->allocation.y + view->allocation.h/2.;
