diff -r -u automake-1.10_orig/lib/config.sub automake-1.10/lib/config.sub
--- automake-1.10_orig/lib/config.sub	2006-10-15 19:02:52.000000000 +0200
+++ automake-1.10/lib/config.sub	2007-08-04 09:17:08.000000000 +0200
@@ -683,6 +683,10 @@
 		basic_machine=i386-pc
 		os=-mingw32
 		;;
+	msys)
+		basic_machine=i386-pc
+		os=-msys
+		;;
 	miniframe)
 		basic_machine=m68000-convergent
 		;;
@@ -1211,7 +1215,7 @@
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* \
 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
-	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
+	      | -mingw32* | -msys* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
 	      | -openstep* | -oskit* | -conix* | -pw32* | -nonstopux* \
