diff -ru gtk+extra-1.0.0_orig/gtkextra.m4 gtk+extra-1.0.0/gtkextra.m4
--- gtk+extra-1.0.0_orig/gtkextra.m4	2008-08-04 23:25:38.000000000 +0200
+++ gtk+extra-1.0.0/gtkextra.m4	2008-08-04 23:25:51.000000000 +0200
@@ -5,7 +5,7 @@
 dnl AM_PATH_GTK_EXTRA([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND [, MODULES]]]])
 dnl Test for GTK_EXTRA, and define GTK_EXTRA_CFLAGS and GTK_EXTRA_LIBS
 dnl
-AC_DEFUN(AM_PATH_GTK_EXTRA,
+AC_DEFUN([AM_PATH_GTK_EXTRA],
 [dnl 
 dnl Get the cflags and libraries from the gtkextra-config script
 dnl
