--- gcc-3.4.6_orig/gcc/config.gcc	2005-07-19 23:08:48.000000000 +0200
+++ gcc-3.4.6/gcc/config.gcc	2007-09-05 00:39:08.000000000 +0200
@@ -1162,8 +1162,8 @@
 	tm_file="${tm_file} i386/sysv4.h i386/unix.h i386/vxworks.h"
 	tmake_file="${tmake_file} i386/t-vxworks"
 	;;
-i[34567]86-*-pe | i[34567]86-*-cygwin*)
+i[34567]86-*-pe |  i[34567]86-*-cygwin* |  i[34567]86-*-msys*)
 	tm_file="${tm_file} i386/unix.h i386/bsd.h i386/gas.h dbxcoff.h i386/cygming.h i386/cygwin.h"
 	target_gtfiles="\$(srcdir)/config/i386/winnt.c"
 	xm_file=i386/xm-cygwin.h
 	tmake_file="i386/t-cygwin i386/t-cygming"
