diff -ru -ru gcc-3.4.6_orig/gcc/config/host-linux.c gcc-3.4.6/gcc/config/host-linux.c
--- gcc-3.4.6_orig/gcc/config/host-linux.c	2007-04-04 08:08:35.000000000 +0000
+++ gcc-3.4.6/gcc/config/host-linux.c	2007-04-04 08:09:05.000000000 +0000
@@ -26,6 +26,9 @@
 #include "hosthooks.h"
 #include "hosthooks-def.h"
 
+#ifndef SSIZE_MAX
+#define SSIZE_MAX 2147483647
+#endif //ndef SSIZE_MAX
 
 /* Linux has a feature called exec-shield-randomize that perturbs the
    address of non-fixed mapped segments by a (relatively) small amount.
