--- lzo-2.02_orig/configure.ac	2008-06-09 23:08:47.000000000 +0200
+++ lzo-2.02/configure.ac	2008-06-09 23:07:44.000000000 +0200
@@ -52,9 +52,9 @@
 
 AC_PROG_CC
 AC_PROG_CPP
-mfx_PROG_CPPFLAGS
+dnl mfx_PROG_CPPFLAGS
 AC_C_CONST
-mfx_LZO_CHECK_ENDIAN
+dnl mfx_LZO_CHECK_ENDIAN
 AC_SYS_LARGEFILE
 
 AM_INIT_AUTOMAKE
@@ -113,15 +113,17 @@
 # // Checks for header files
 # ************************************************************************/
 
-mfx_ACC_CHECK_HEADERS
+dnl mfx_ACC_CHECK_HEADERS
+
 dnl AC_CHECK_HEADERS([sys/param.h sys/resource.h sys/times.h])
-if test "X$ac_cv_header_limits_h" != Xyes; then
-    AC_MSG_ERROR([<limits.h> header not found])
-fi
-mfx_CHECK_HEADER_SANE_LIMITS_H
-if test "X$mfx_cv_header_sane_limits_h" != Xyes; then
-    AC_MSG_ERROR([your <limits.h> header is broken - for details see config.log])
-fi
+dnl if test "X$ac_cv_header_limits_h" != Xyes; then
+dnl     AC_MSG_ERROR([<limits.h> header not found])
+dnl fi
+dnl mfx_CHECK_HEADER_SANE_LIMITS_H
+dnl
+dnl if test "X$mfx_cv_header_sane_limits_h" != Xyes; then
+dnl     AC_MSG_ERROR([your <limits.h> header is broken - for details see config.log])
+dnl fi
 
 
 # /***********************************************************************
@@ -133,23 +135,23 @@
 AC_TYPE_SIZE_T
 AC_TYPE_SIGNAL
 
-mfx_ACC_CHECK_SIZEOF
-mfx_CHECK_SIZEOF
+dnl mfx_ACC_CHECK_SIZEOF
+dnl mfx_CHECK_SIZEOF
 
 
 # /***********************************************************************
 # // Checks for library functions
 # ************************************************************************/
 
-mfx_ACC_CHECK_FUNCS
-mfx_CHECK_LIB_WINMM
+dnl mfx_ACC_CHECK_FUNCS
+dnl mfx_CHECK_LIB_WINMM
 
 
 # /***********************************************************************
 # // Write output files
 # ************************************************************************/
 
-mfx_MINIACC_ACCCHK(["-I$srcdir"],["src/miniacc.h"])
+dnl mfx_MINIACC_ACCCHK(["-I$srcdir"],["src/miniacc.h"])
 
 if test -r .Conf.settings2; then
     . ./.Conf.settings2
