diff -ru imap-utils-svn52_orig/setup.py imap-utils-svn52/setup.py
--- imap-utils-svn52_orig/setup.py	2009-02-09 20:36:53.000000000 +0100
+++ imap-utils-svn52/setup.py	2009-02-09 20:44:56.000000000 +0100
@@ -4,7 +4,7 @@
 '''
 Setup script for installation of IMAP utils using distutils
 '''
-__author__ = 'Michal Čihař'
+__author__ = 'Michal'
 __email__ = 'michal@cihar.com'
 __license__ = '''
 Copyright (c) 2003 - 2007 Michal Čihař
@@ -30,11 +30,11 @@
     version = IMAPUtils.__version__,
     description = "IMAP utils",
     long_description = "Set of simple utilities for managing IMAP emails.",
-    author = u"Michal Čihař",
+    author = "Michal",
     author_email = "michal@cihar.com",
-    maintainer = u"Michal Čihař",
+    maintainer = "Michal",
     maintainer_email = "michal@cihar.com",
-    platforms = ['Linux','Mac OSX','Windows XP/2000/NT','Windows 95/98/ME'],
+    latforms = ['Linux','Mac OSX','Windows XP/2000/NT','Windows 95/98/ME'],
     keywords = ['IMAP', 'mail', 'email'],
     url = "http://cihar.com/software/imap-utils/",
     download_url = "http://cihar.com/software/imap-utils/",
@@ -51,5 +51,5 @@
         'Topic :: Utilities',
     ],
     packages = ['IMAPUtils'],
-    scripts = ['imap-purge', 'imap-folders', 'imap-stats'],
+    scripts = ['imap-purge', 'imap-folders', 'imap-stats', 'fetch-messages', 'process-messages', 'store-message'],
     )
