diff -u ez-ipupdate-3.0.11b7_orig/example.conf ez-ipupdate-3.0.11b7/example.conf
--- ez-ipupdate-3.0.11b7_orig/example.conf	2008-11-15 19:28:09.000000000 +0100
+++ ez-ipupdate-3.0.11b7/example.conf	2008-11-15 19:30:47.000000000 +0100
@@ -1,23 +1,16 @@
-#!/usr/local/bin/ez-ipupdate -c
+#!/usr/bin/ez-ipupdate -c
 #
 # example config file for ez-ipupdate
-#
 # this file is actually executable!
 #
 
-service-type=<type of service>
-user=myuserid:mypassword
-interface=eth1
-#host=mydomain.whatever.com
-
-# other options:
+service-type=dyndns
+script=/usr/bin/get_ext_ip
 #address=<ip address>
-#cache-file=/etc/ez-ipupdate.cache.eth1
+#cache-file=/var/cache/ez-ipupdate/eth0
-#daemon
+daemon
 #debug
-#foreground
-#host=<host>
-#interface=<interface>
+foreground
 #mx=<mail exchanger>
 #retrys=<number of trys>
 #run-as-user=<user>
