--- PyQt-x11-gpl-4.3.1/configure.py	2007-09-28 16:07:42.000000000 +0000
+++ PyQt-x11-gpl-4.3.1-hacked/configure.py	2007-10-30 16:55:41.000000000 +0000
@@ -1328,25 +1328,6 @@
     print "Type 'no' to decline the terms of the license."
     print
 
-    while 1:
-        try:
-            resp = raw_input("Do you accept the terms of the license? ")
-        except KeyboardInterrupt:
-            raise SystemExit
-        except:
-            resp = ""
-
-        resp = string.lower(string.strip(resp))
-
-        if resp == "yes":
-            break
-
-        if resp == "no":
-            sys.exit(0)
-
-        if resp == "l":
-            os.system("more LICENSE")
-
     # If there should be a license file then check it is where it should be.
     if lfile:
         if os.access(os.path.join("sip", lfile), os.F_OK):
