diff -ru miniloop-0.0_orig/Engine.cpp miniloop-0.0/Engine.cpp
--- miniloop-0.0_orig/Engine.cpp	2009-03-27 05:37:13.000000000 +0000
+++ miniloop-0.0/Engine.cpp	2009-03-27 05:37:33.000000000 +0000
@@ -28,6 +28,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <sndfile.h>
+#include <cstring>
 #include "Config.hpp"
 #include "Engine.hpp"
 using namespace miniloop;
Only in miniloop-0.0/: Engine.o
Only in miniloop-0.0/: FileBrowserWindow.o
Only in miniloop-0.0/: LoopWindow.o
Only in miniloop-0.0/: main.o
Only in miniloop-0.0/: miniloop
