diff -ru qmailalizer_orig/config.h qmailalizer/config.h
--- qmailalizer_orig/config.h	2008-07-18 02:55:29.000000000 +0200
+++ qmailalizer/config.h	2008-07-18 02:56:56.000000000 +0200
@@ -7,3 +7,4 @@
 /* Version number of package */
 #define VERSION "0.1"
 
+
diff -ru qmailalizer_orig/src/config.h qmailalizer/src/config.h
--- qmailalizer_orig/src/config.h	2008-07-18 02:55:29.000000000 +0200
+++ qmailalizer/src/config.h	2008-07-18 02:57:03.000000000 +0200
@@ -30,4 +30,4 @@
     LDomains *ldWurzel;
 };
 
-#endif  
\ No newline at end of file
+#endif  
diff -ru qmailalizer_orig/src/data.h qmailalizer/src/data.h
--- qmailalizer_orig/src/data.h	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/data.h	2008-07-18 02:57:03.000000000 +0200
@@ -68,4 +68,4 @@
   unsigned long monthbytes[12];
 };
 
-#endif
\ No newline at end of file
+#endif
diff -ru qmailalizer_orig/src/graphs.h qmailalizer/src/graphs.h
--- qmailalizer_orig/src/graphs.h	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/graphs.h	2008-07-18 02:57:03.000000000 +0200
@@ -28,4 +28,4 @@
     char *domain_view (Domains *domain);
 };
 
-#endif
\ No newline at end of file
+#endif
diff -ru qmailalizer_orig/src/message.cc qmailalizer/src/message.cc
--- qmailalizer_orig/src/message.cc	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/message.cc	2008-07-18 02:57:03.000000000 +0200
@@ -328,4 +328,4 @@
     }
   }
   return adeliverie;
-}
\ No newline at end of file
+}
diff -ru qmailalizer_orig/src/message.h qmailalizer/src/message.h
--- qmailalizer_orig/src/message.h	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/message.h	2008-07-18 02:57:03.000000000 +0200
@@ -55,4 +55,4 @@
     Message *newmessage(unsigned long msnr,struct tm *datum);
 };
 
-#endif
\ No newline at end of file
+#endif
diff -ru qmailalizer_orig/src/parse.cc qmailalizer/src/parse.cc
--- qmailalizer_orig/src/parse.cc	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/parse.cc	2008-07-18 02:57:03.000000000 +0200
@@ -348,4 +348,4 @@
   saveresults ();
   xmloutput->SaveResult (config->xmlfile);
   return 0;
-}
\ No newline at end of file
+}
diff -ru qmailalizer_orig/src/parse.h qmailalizer/src/parse.h
--- qmailalizer_orig/src/parse.h	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/parse.h	2008-07-18 02:57:03.000000000 +0200
@@ -36,4 +36,4 @@
     int save();
 };
 
-#endif
\ No newline at end of file
+#endif
diff -ru qmailalizer_orig/src/results.h qmailalizer/src/results.h
--- qmailalizer_orig/src/results.h	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/results.h	2008-07-18 02:57:03.000000000 +0200
@@ -56,4 +56,4 @@
     void ChangeUsage (Domains *domain,unsigned long usage[12],unsigned long bytes[12]);
 };
 
-#endif
\ No newline at end of file
+#endif
diff -ru qmailalizer_orig/src/stuff.cc qmailalizer/src/stuff.cc
--- qmailalizer_orig/src/stuff.cc	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/stuff.cc	2008-07-18 02:57:03.000000000 +0200
@@ -201,4 +201,4 @@
   return digits;
 }
 
-#endif						    
\ No newline at end of file
+#endif						    
diff -ru qmailalizer_orig/src/stuff.h qmailalizer/src/stuff.h
--- qmailalizer_orig/src/stuff.h	2008-07-18 02:55:29.000000000 +0200
+++ qmailalizer/src/stuff.h	2008-07-18 02:57:03.000000000 +0200
@@ -12,4 +12,4 @@
 unsigned char *ltoa(unsigned long value,unsigned char *digits, int base);
 unsigned char *dtoa(double value,unsigned char *digits, int base);
 int strpos(char *s,char c);
-char *strlower(char *s);
\ No newline at end of file
+char *strlower(char *s);
diff -ru qmailalizer_orig/src/tai64.h qmailalizer/src/tai64.h
--- qmailalizer_orig/src/tai64.h	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/tai64.h	2008-07-18 02:57:03.000000000 +0200
@@ -74,4 +74,4 @@
       out(".",1); out(num,fmt_uint0(num,nanosecs,9));  */
     }
   return t;
-}
\ No newline at end of file
+}
diff -ru qmailalizer_orig/src/xmlinput.cc qmailalizer/src/xmlinput.cc
--- qmailalizer_orig/src/xmlinput.cc	2008-07-18 02:55:29.000000000 +0200
+++ qmailalizer/src/xmlinput.cc	2008-07-18 02:57:03.000000000 +0200
@@ -256,4 +256,4 @@
   }
   free (datum);
   free (local);
-}
\ No newline at end of file
+}
diff -ru qmailalizer_orig/src/xmlinput.h qmailalizer/src/xmlinput.h
--- qmailalizer_orig/src/xmlinput.h	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/xmlinput.h	2008-07-18 02:57:15.000000000 +0200
@@ -1,6 +1,5 @@
 #ifndef XMLINPUT_H
 #define XMLINPUT_H
-#define _XOPEN_SOURCE
 #include "config.h"
 #include "results.h"
 #include "message.h"
@@ -35,4 +34,4 @@
     
     int LoadResults ();
 };
-#endif
\ No newline at end of file
+#endif
diff -ru qmailalizer_orig/src/xmloutput.h qmailalizer/src/xmloutput.h
--- qmailalizer_orig/src/xmloutput.h	2008-07-18 02:55:30.000000000 +0200
+++ qmailalizer/src/xmloutput.h	2008-07-18 02:57:03.000000000 +0200
@@ -31,4 +31,4 @@
     void SaveResult (char *FileName);
 };
 
-#endif
\ No newline at end of file
+#endif
