diff -r -u -- php-5.2.2_orig/configure php-5.2.2/configure
--- php-5.2.2_orig/configure	2007-05-22 14:20:47.000000000 +0200
+++ php-5.2.2/configure	2007-05-22 14:20:47.000000000 +0200
@@ -20447,14 +20447,14 @@
   c|c_r|pthread*) ;;
   *) 
     if test "$ext_shared" = "yes"; then
-        OPENSSL_SHARED_LIBADD="$OPENSSL_SHARED_LIBADD -lgssapi_krb5" 
+        OPENSSL_SHARED_LIBADD="$OPENSSL_SHARED_LIBADD -lkrb5" 
     else
       
   
   case gssapi_krb5 in
   c|c_r|pthread*) ;;
   *) 
-      LIBS="$LIBS -lgssapi_krb5" 
+      LIBS="$LIBS -lkrb5" 
    ;;
   esac
 
@@ -20493,14 +20493,14 @@
   c|c_r|pthread*) ;;
   *) 
     if test "$ext_shared" = "yes"; then
-        OPENSSL_SHARED_LIBADD="$OPENSSL_SHARED_LIBADD -lk5crypto" 
+        OPENSSL_SHARED_LIBADD="$OPENSSL_SHARED_LIBADD" 
     else
       
   
   case k5crypto in
   c|c_r|pthread*) ;;
   *) 
-      LIBS="$LIBS -lk5crypto" 
+      LIBS="$LIBS" 
    ;;
   esac
 
@@ -46946,14 +46946,14 @@
   c|c_r|pthread*) ;;
   *) 
     if test "$ext_shared" = "yes"; then
-        IMAP_SHARED_LIBADD="$IMAP_SHARED_LIBADD -lgssapi_krb5" 
+        IMAP_SHARED_LIBADD="$IMAP_SHARED_LIBADD -lkrb5" 
     else
       
   
   case gssapi_krb5 in
   c|c_r|pthread*) ;;
   *) 
-      LIBS="$LIBS -lgssapi_krb5" 
+      LIBS="$LIBS -lkrb5" 
    ;;
   esac
 
@@ -46992,14 +46992,14 @@
   c|c_r|pthread*) ;;
   *) 
     if test "$ext_shared" = "yes"; then
-        IMAP_SHARED_LIBADD="$IMAP_SHARED_LIBADD -lk5crypto" 
+        IMAP_SHARED_LIBADD="$IMAP_SHARED_LIBADD" 
     else
       
   
   case k5crypto in
   c|c_r|pthread*) ;;
   *) 
-      LIBS="$LIBS -lk5crypto" 
+      LIBS="$LIBS" 
    ;;
   esac
 
Only in php-5.2.2: configure~
