diff -ru amygdala-0.4.0_orig/amygdala/.deps/axon.Plo amygdala-0.4.0/amygdala/.deps/axon.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/axonnode.Plo amygdala-0.4.0/amygdala/.deps/axonnode.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/basicspikeoutput.Plo amygdala-0.4.0/amygdala/.deps/basicspikeoutput.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/dendrite.Plo amygdala-0.4.0/amygdala/.deps/dendrite.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/euler.Plo amygdala-0.4.0/amygdala/.deps/euler.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/factory.Plo amygdala-0.4.0/amygdala/.deps/factory.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/feedforwardlayer.Plo amygdala-0.4.0/amygdala/.deps/feedforwardlayer.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/flatlayer.Plo amygdala-0.4.0/amygdala/.deps/flatlayer.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/genome.Plo amygdala-0.4.0/amygdala/.deps/genome.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/lsmtopology.Plo amygdala-0.4.0/amygdala/.deps/lsmtopology.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/nconnector.Plo amygdala-0.4.0/amygdala/.deps/nconnector.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/netloader.Plo amygdala-0.4.0/amygdala/.deps/netloader.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/neurongroup.Plo amygdala-0.4.0/amygdala/.deps/neurongroup.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/neuronproperties.Plo amygdala-0.4.0/amygdala/.deps/neuronproperties.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/outputmanager.Plo amygdala-0.4.0/amygdala/.deps/outputmanager.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/pdeltatrainer.Plo amygdala-0.4.0/amygdala/.deps/pdeltatrainer.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/physicalproperties.Plo amygdala-0.4.0/amygdala/.deps/physicalproperties.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/properties.Plo amygdala-0.4.0/amygdala/.deps/properties.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/scheduledspikeinput.Plo amygdala-0.4.0/amygdala/.deps/scheduledspikeinput.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/simplenet.Plo amygdala-0.4.0/amygdala/.deps/simplenet.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/spikeinput.Plo amygdala-0.4.0/amygdala/.deps/spikeinput.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/spikeoutput.Plo amygdala-0.4.0/amygdala/.deps/spikeoutput.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/spikingneuronproperties.Plo amygdala-0.4.0/amygdala/.deps/spikingneuronproperties.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/statictrainer.Plo amygdala-0.4.0/amygdala/.deps/statictrainer.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/statisticsoutput.Plo amygdala-0.4.0/amygdala/.deps/statisticsoutput.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/synapse.Plo amygdala-0.4.0/amygdala/.deps/synapse.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/synapseproperties.Plo amygdala-0.4.0/amygdala/.deps/synapseproperties.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/trainer.Plo amygdala-0.4.0/amygdala/.deps/trainer.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/utilities.Plo amygdala-0.4.0/amygdala/.deps/utilities.Plo
diff -ru amygdala-0.4.0_orig/amygdala/.deps/visualstub.Plo amygdala-0.4.0/amygdala/.deps/visualstub.Plo
Only in amygdala-0.4.0/amygdala/.libs: axon.o
Only in amygdala-0.4.0/amygdala/.libs: axonnode.o
Only in amygdala-0.4.0/amygdala/.libs: basicspikeoutput.o
Only in amygdala-0.4.0/amygdala/.libs: dendrite.o
Only in amygdala-0.4.0/amygdala/.libs: euler.o
Only in amygdala-0.4.0/amygdala/.libs: factory.o
Only in amygdala-0.4.0/amygdala/.libs: feedforwardlayer.o
Only in amygdala-0.4.0/amygdala/.libs: flatlayer.o
Only in amygdala-0.4.0/amygdala/.libs: genome.o
Only in amygdala-0.4.0/amygdala/.libs: libamygdala.a
Only in amygdala-0.4.0/amygdala/.libs: libamygdala.la
Only in amygdala-0.4.0/amygdala/.libs: libamygdala.lai
Only in amygdala-0.4.0/amygdala/.libs: libamygdala.so
Only in amygdala-0.4.0/amygdala/.libs: libamygdala.so.0
Only in amygdala-0.4.0/amygdala/.libs: libamygdala.so.0.4.0
Only in amygdala-0.4.0/amygdala/.libs: lsmtopology.o
Only in amygdala-0.4.0/amygdala/.libs: nconnector.o
Only in amygdala-0.4.0/amygdala/.libs: netloader.o
Only in amygdala-0.4.0/amygdala/.libs: neurongroup.o
Only in amygdala-0.4.0/amygdala/.libs: neuronproperties.o
Only in amygdala-0.4.0/amygdala/.libs: outputmanager.o
Only in amygdala-0.4.0/amygdala/.libs: pdeltatrainer.o
Only in amygdala-0.4.0/amygdala/.libs: physicalproperties.o
Only in amygdala-0.4.0/amygdala/.libs: properties.o
Only in amygdala-0.4.0/amygdala/.libs: scheduledspikeinput.o
Only in amygdala-0.4.0/amygdala/.libs: simplenet.o
Only in amygdala-0.4.0/amygdala/.libs: spikeinput.o
Only in amygdala-0.4.0/amygdala/.libs: spikeoutput.o
Only in amygdala-0.4.0/amygdala/.libs: spikingneuronproperties.o
Only in amygdala-0.4.0/amygdala/.libs: statictrainer.o
Only in amygdala-0.4.0/amygdala/.libs: statisticsoutput.o
Only in amygdala-0.4.0/amygdala/.libs: synapse.o
Only in amygdala-0.4.0/amygdala/.libs: synapseproperties.o
Only in amygdala-0.4.0/amygdala/.libs: trainer.o
Only in amygdala-0.4.0/amygdala/.libs: utilities.o
Only in amygdala-0.4.0/amygdala/.libs: visualstub.o
diff -ru amygdala-0.4.0_orig/amygdala/alphaneuron.cpp amygdala-0.4.0/amygdala/alphaneuron.cpp
--- amygdala-0.4.0_orig/amygdala/alphaneuron.cpp	2009-02-26 02:15:04.000000000 +0100
+++ amygdala-0.4.0/amygdala/alphaneuron.cpp	2009-02-26 02:03:03.000000000 +0100
@@ -15,6 +15,7 @@
  ***************************************************************************/
 
 #include <cmath>
+#include <cstdlib>
 
 #include "alphaneuron.h"
 #include "euler.h"
diff -ru amygdala-0.4.0_orig/amygdala/alphaneuron.h amygdala-0.4.0/amygdala/alphaneuron.h
--- amygdala-0.4.0_orig/amygdala/alphaneuron.h	2009-02-26 02:15:04.000000000 +0100
+++ amygdala-0.4.0/amygdala/alphaneuron.h	2009-02-26 02:03:03.000000000 +0100
@@ -105,7 +105,7 @@
     
     virtual void SpikeCleanup();
 
-/*    /** Determine the maximum value of a weight for this neuron and
+/*   * Determine the maximum value of a weight for this neuron and
      * set maxScaledWeight to this value. This will be used
      * as the multiplication factor to convert to and from the
      * normalized weight values that are used in the public
diff -ru amygdala-0.4.0_orig/amygdala/amygdalaclass.cpp amygdala-0.4.0/amygdala/amygdalaclass.cpp
--- amygdala-0.4.0_orig/amygdala/amygdalaclass.cpp	2009-02-26 02:15:04.000000000 +0100
+++ amygdala-0.4.0/amygdala/amygdalaclass.cpp	2009-02-26 02:03:03.000000000 +0100
@@ -17,6 +17,8 @@
 
 #include "amygdalaclass.h"
 
+#include <cstdlib>
+
 using namespace Amygdala;
 
 AmygdalaClass::~AmygdalaClass()
Only in amygdala-0.4.0/amygdala: axon.lo
Only in amygdala-0.4.0/amygdala: axon.o
Only in amygdala-0.4.0/amygdala: axonnode.lo
Only in amygdala-0.4.0/amygdala: axonnode.o
Only in amygdala-0.4.0/amygdala: basicspikeoutput.lo
Only in amygdala-0.4.0/amygdala: basicspikeoutput.o
Only in amygdala-0.4.0/amygdala: dendrite.lo
Only in amygdala-0.4.0/amygdala: dendrite.o
Only in amygdala-0.4.0/amygdala: euler.lo
Only in amygdala-0.4.0/amygdala: euler.o
Only in amygdala-0.4.0/amygdala: factory.lo
Only in amygdala-0.4.0/amygdala: factory.o
diff -ru amygdala-0.4.0_orig/amygdala/feedforwardlayer.cpp amygdala-0.4.0/amygdala/feedforwardlayer.cpp
--- amygdala-0.4.0_orig/amygdala/feedforwardlayer.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/feedforwardlayer.cpp	2009-02-26 02:06:15.000000000 +0100
@@ -16,6 +16,7 @@
 
 #include <stdexcept>
 #include <string>
+#include <cstdlib>
  
 #include "feedforwardlayer.h"
 #include "synapseproperties.h"
diff -ru amygdala-0.4.0_orig/amygdala/feedforwardlayer.h amygdala-0.4.0/amygdala/feedforwardlayer.h
--- amygdala-0.4.0_orig/amygdala/feedforwardlayer.h	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/feedforwardlayer.h	2009-02-26 02:06:29.000000000 +0100
@@ -54,7 +54,7 @@
                         SynapseProperties & sProps,
                         const float percentConnect = 100);
 
-/*    /** Form random connections to another FeedForwardLayer with weights distributed according to a random
+/*   ** Form random connections to another FeedForwardLayer with weights distributed according to a random
      * Gaussian distribution.  Note that this will form connections with StaticSynapses only.
      * @param outLayer The output FeedForwardLayer
      * @param percentConnect The percentage of neurons in this layer that should form connections to outLayer
Only in amygdala-0.4.0/amygdala: feedforwardlayer.lo
Only in amygdala-0.4.0/amygdala: feedforwardlayer.o
Only in amygdala-0.4.0/amygdala: flatlayer.lo
Only in amygdala-0.4.0/amygdala: flatlayer.o
diff -ru amygdala-0.4.0_orig/amygdala/genome.cpp amygdala-0.4.0/amygdala/genome.cpp
--- amygdala-0.4.0_orig/amygdala/genome.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/genome.cpp	2009-02-26 02:09:03.000000000 +0100
@@ -35,6 +35,8 @@
 #include <stdio.h>
 #include <iostream>
 #include <stdexcept>
+#include <cstring>
+
 using namespace Amygdala;
 using namespace std;
 
Only in amygdala-0.4.0/amygdala: genome.lo
Only in amygdala-0.4.0/amygdala: genome.o
Only in amygdala-0.4.0/amygdala: libamygdala.la
Only in amygdala-0.4.0/amygdala: lsmtopology.lo
Only in amygdala-0.4.0/amygdala: lsmtopology.o
Only in amygdala-0.4.0/amygdala: nconnector.lo
Only in amygdala-0.4.0/amygdala: nconnector.o
diff -ru amygdala-0.4.0_orig/amygdala/netloader.cpp amygdala-0.4.0/amygdala/netloader.cpp
--- amygdala-0.4.0_orig/amygdala/netloader.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/netloader.cpp	2009-02-26 02:08:36.000000000 +0100
@@ -32,6 +32,7 @@
 
 #include <stdexcept>
 #include <iostream>
+#include <cstring>
 
 #include <libxml/parser.h>
 #include <libxml/parserInternals.h>
diff -ru amygdala-0.4.0_orig/amygdala/netloader.h amygdala-0.4.0/amygdala/netloader.h
--- amygdala-0.4.0_orig/amygdala/netloader.h	2009-02-26 02:15:04.000000000 +0100
+++ amygdala-0.4.0/amygdala/netloader.h	2009-02-26 02:03:03.000000000 +0100
@@ -31,7 +31,7 @@
 namespace Amygdala {
 
 class Neuron;
-class Factory;
+//class Factory;
 class NeuronProperties;
 class SynapseProperties;
 class NConnector;
Only in amygdala-0.4.0/amygdala: netloader.lo
Only in amygdala-0.4.0/amygdala: netloader.o
diff -ru amygdala-0.4.0_orig/amygdala/network.cpp amygdala-0.4.0/amygdala/network.cpp
--- amygdala-0.4.0_orig/amygdala/network.cpp	2009-02-26 02:15:04.000000000 +0100
+++ amygdala-0.4.0/amygdala/network.cpp	2009-02-26 02:03:03.000000000 +0100
@@ -35,6 +35,7 @@
 #include <time.h>
 #include <pthread.h>
 #include <unistd.h>
+#include <cstring>
 
 #include <stdexcept>
 
Only in amygdala-0.4.0/amygdala: neurongroup.lo
Only in amygdala-0.4.0/amygdala: neurongroup.o
Only in amygdala-0.4.0/amygdala: neuronproperties.lo
Only in amygdala-0.4.0/amygdala: neuronproperties.o
Only in amygdala-0.4.0/amygdala: outputmanager.lo
Only in amygdala-0.4.0/amygdala: outputmanager.o
diff -ru amygdala-0.4.0_orig/amygdala/pdeltatrainer.cpp amygdala-0.4.0/amygdala/pdeltatrainer.cpp
--- amygdala-0.4.0_orig/amygdala/pdeltatrainer.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/pdeltatrainer.cpp	2009-02-26 02:09:51.000000000 +0100
@@ -16,6 +16,7 @@
  ***************************************************************************/
 
 #include <map>
+#include <cstdlib>
 #include "pdeltatrainer.h"
 #include "utilities.h"
 #include "synapse.h"
Only in amygdala-0.4.0/amygdala: pdeltatrainer.lo
Only in amygdala-0.4.0/amygdala: pdeltatrainer.o
diff -ru amygdala-0.4.0_orig/amygdala/physicalproperties.cpp amygdala-0.4.0/amygdala/physicalproperties.cpp
--- amygdala-0.4.0_orig/amygdala/physicalproperties.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/physicalproperties.cpp	2009-02-26 02:06:56.000000000 +0100
@@ -19,6 +19,8 @@
 #include <string>
 #include <map>
 #include "utilities.h"
+#include <cstring>
+#include <cstdlib>
 
 using namespace std;
 using namespace Amygdala;
Only in amygdala-0.4.0/amygdala: physicalproperties.lo
Only in amygdala-0.4.0/amygdala: physicalproperties.o
Only in amygdala-0.4.0/amygdala: properties.lo
Only in amygdala-0.4.0/amygdala: properties.o
diff -ru amygdala-0.4.0_orig/amygdala/scheduledspikeinput.cpp amygdala-0.4.0/amygdala/scheduledspikeinput.cpp
--- amygdala-0.4.0_orig/amygdala/scheduledspikeinput.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/scheduledspikeinput.cpp	2009-02-26 02:07:57.000000000 +0100
@@ -21,6 +21,7 @@
 #include "inputneuron.h"
 #include "utilities.h"
 #include "logging.h"
+#include <cstring>
 
 // includes for XML
 #include <libxml/parser.h>
Only in amygdala-0.4.0/amygdala: scheduledspikeinput.lo
Only in amygdala-0.4.0/amygdala: scheduledspikeinput.o
Only in amygdala-0.4.0/amygdala: simplenet.lo
Only in amygdala-0.4.0/amygdala: simplenet.o
Only in amygdala-0.4.0/amygdala: spikeinput.lo
Only in amygdala-0.4.0/amygdala: spikeinput.o
Only in amygdala-0.4.0/amygdala: spikeoutput.lo
Only in amygdala-0.4.0/amygdala: spikeoutput.o
diff -ru amygdala-0.4.0_orig/amygdala/spikingneuronproperties.cpp amygdala-0.4.0/amygdala/spikingneuronproperties.cpp
--- amygdala-0.4.0_orig/amygdala/spikingneuronproperties.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/spikingneuronproperties.cpp	2009-02-26 02:07:17.000000000 +0100
@@ -1,6 +1,8 @@
 #include "spikingneuronproperties.h"
 #include "utilities.h"
 
+#include <cstdlib>
+
 using namespace Amygdala;
 using namespace Utilities;
 
Only in amygdala-0.4.0/amygdala: spikingneuronproperties.lo
Only in amygdala-0.4.0/amygdala: spikingneuronproperties.o
diff -ru amygdala-0.4.0_orig/amygdala/statictrainer.cpp amygdala-0.4.0/amygdala/statictrainer.cpp
--- amygdala-0.4.0_orig/amygdala/statictrainer.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/statictrainer.cpp	2009-02-26 02:09:32.000000000 +0100
@@ -19,6 +19,7 @@
 #include "synapse.h"
 #include "utilities.h"
 #include "network.h"
+#include <cstdlib>
 
 using namespace Amygdala;
 using namespace Utilities;
Only in amygdala-0.4.0/amygdala: statictrainer.lo
Only in amygdala-0.4.0/amygdala: statictrainer.o
Only in amygdala-0.4.0/amygdala: statisticsoutput.lo
Only in amygdala-0.4.0/amygdala: statisticsoutput.o
diff -ru amygdala-0.4.0_orig/amygdala/synapse.cpp amygdala-0.4.0/amygdala/synapse.cpp
--- amygdala-0.4.0_orig/amygdala/synapse.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/synapse.cpp	2009-02-26 02:05:51.000000000 +0100
@@ -28,8 +28,8 @@
 using namespace Amygdala;
 using namespace std;
 
-StaticSynapse::NeuronConnector StaticSynapse::Connect;
-DynamicSynapse::NeuronConnector DynamicSynapse::Connect;
+StaticSynapse::NeuronConnectorType StaticSynapse::Connect;
+DynamicSynapse::NeuronConnectorType DynamicSynapse::Connect;
 
 //-------------- Methods of class Synapse ---------------------------------//
 Synapse::Synapse(SynapseProperties& synProps):
diff -ru amygdala-0.4.0_orig/amygdala/synapse.h amygdala-0.4.0/amygdala/synapse.h
--- amygdala-0.4.0_orig/amygdala/synapse.h	2009-02-26 02:15:04.000000000 +0100
+++ amygdala-0.4.0/amygdala/synapse.h	2009-02-26 02:03:03.000000000 +0100
@@ -108,9 +108,9 @@
     friend class NeuronConnector;
 public:
     typedef StaticSynapseProperties PropertyType;
-    typedef NeuronConnector<StaticSynapse, PropertyType> NeuronConnector;
+    typedef NeuronConnector<StaticSynapse, PropertyType> NeuronConnectorType;
 
-    static NeuronConnector Connect;
+    static NeuronConnectorType Connect;
     
     virtual ~StaticSynapse() {};
 
@@ -158,9 +158,9 @@
     friend class NeuronConnector;
 public:
     typedef DynamicSynapseProperties PropertyType;
-    typedef NeuronConnector<DynamicSynapse, PropertyType> NeuronConnector;
+    typedef NeuronConnector<DynamicSynapse, PropertyType> NeuronConnectorType;
 
-    static NeuronConnector Connect;
+    static NeuronConnectorType Connect;
     
     virtual ~DynamicSynapse() {};
 
Only in amygdala-0.4.0/amygdala: synapse.lo
Only in amygdala-0.4.0_orig/amygdala: synapse.loT
Only in amygdala-0.4.0/amygdala: synapse.o
diff -ru amygdala-0.4.0_orig/amygdala/synapseproperties.cpp amygdala-0.4.0/amygdala/synapseproperties.cpp
--- amygdala-0.4.0_orig/amygdala/synapseproperties.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/amygdala/synapseproperties.cpp	2009-02-26 02:07:32.000000000 +0100
@@ -19,6 +19,8 @@
 #include "logging.h"
 #include "utilities.h"
 
+#include <cstdlib>
+
 using namespace Amygdala;
 using namespace Utilities;
 
Only in amygdala-0.4.0/amygdala: synapseproperties.lo
Only in amygdala-0.4.0/amygdala: synapseproperties.o
Only in amygdala-0.4.0/amygdala: trainer.lo
Only in amygdala-0.4.0/amygdala: trainer.o
Only in amygdala-0.4.0/amygdala: utilities.lo
Only in amygdala-0.4.0/amygdala: utilities.o
Only in amygdala-0.4.0/amygdala: visualstub.lo
Only in amygdala-0.4.0/amygdala: visualstub.o
diff -ru amygdala-0.4.0_orig/geneserver/.deps/chromosome.Po amygdala-0.4.0/geneserver/.deps/chromosome.Po
diff -ru amygdala-0.4.0_orig/geneserver/.deps/gene.Po amygdala-0.4.0/geneserver/.deps/gene.Po
diff -ru amygdala-0.4.0_orig/geneserver/.deps/genome.Po amygdala-0.4.0/geneserver/.deps/genome.Po
diff -ru amygdala-0.4.0_orig/geneserver/.deps/gfile.Po amygdala-0.4.0/geneserver/.deps/gfile.Po
diff -ru amygdala-0.4.0_orig/geneserver/.deps/main.Po amygdala-0.4.0/geneserver/.deps/main.Po
Only in amygdala-0.4.0/geneserver: .libs
diff -ru amygdala-0.4.0_orig/geneserver/chromosome.cpp amygdala-0.4.0/geneserver/chromosome.cpp
--- amygdala-0.4.0_orig/geneserver/chromosome.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/geneserver/chromosome.cpp	2009-02-26 02:10:28.000000000 +0100
@@ -19,7 +19,7 @@
 #include "genome.h"
 #include "gene.h"
 #include "amygdala/logging.h"
-
+#include <cstdlib>
 
 Chromosome::Chromosome(){
 }
Only in amygdala-0.4.0/geneserver: chromosome.o
Only in amygdala-0.4.0/geneserver: gene.o
Only in amygdala-0.4.0/geneserver: geneserver
diff -ru amygdala-0.4.0_orig/geneserver/genome.cpp amygdala-0.4.0/geneserver/genome.cpp
--- amygdala-0.4.0_orig/geneserver/genome.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/geneserver/genome.cpp	2009-02-26 02:10:16.000000000 +0100
@@ -27,7 +27,8 @@
 #include <sstream>
 #include <iostream>
 #include <string>
-#include <stdio.h>
+#include <cstdio>
+#include <cstdlib>
 
 using namespace std;
 
Only in amygdala-0.4.0/geneserver: genome.o
Only in amygdala-0.4.0/geneserver: gfile.o
diff -ru amygdala-0.4.0_orig/geneserver/main.cpp amygdala-0.4.0/geneserver/main.cpp
--- amygdala-0.4.0_orig/geneserver/main.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/geneserver/main.cpp	2009-02-26 02:10:42.000000000 +0100
@@ -32,6 +32,7 @@
 #include <fstream>
 #include <string>
 #include <stdexcept>
+#include <cstdlib>
 
 #include "genome.h"
 #include "gfile.h"
Only in amygdala-0.4.0/geneserver: main.o
diff -ru amygdala-0.4.0_orig/samples/lsm/.deps/lsmmouse.Po amygdala-0.4.0/samples/lsm/.deps/lsmmouse.Po
diff -ru amygdala-0.4.0_orig/samples/lsm/.deps/musspikeinput.Po amygdala-0.4.0/samples/lsm/.deps/musspikeinput.Po
Only in amygdala-0.4.0/samples/lsm: .libs
Only in amygdala-0.4.0/samples/lsm: lsmmouse
diff -ru amygdala-0.4.0_orig/samples/lsm/lsmmouse.cpp amygdala-0.4.0/samples/lsm/lsmmouse.cpp
--- amygdala-0.4.0_orig/samples/lsm/lsmmouse.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/samples/lsm/lsmmouse.cpp	2009-02-26 02:13:47.000000000 +0100
@@ -2,6 +2,7 @@
 #include <fstream>
 #include <cstdlib>
 #include <stdexcept>
+#include <cstring>
 
 #include <amygdala/network.h>
 #include <amygdala/inputneuron.h>
Only in amygdala-0.4.0/samples/lsm: lsmmouse.o
Only in amygdala-0.4.0/samples/lsm: musspikeinput.o
diff -ru amygdala-0.4.0_orig/samples/lsmcount/.deps/lsmcount.Po amygdala-0.4.0/samples/lsmcount/.deps/lsmcount.Po
diff -ru amygdala-0.4.0_orig/samples/lsmcount/.deps/lsmspikeinput.Po amygdala-0.4.0/samples/lsmcount/.deps/lsmspikeinput.Po
Only in amygdala-0.4.0/samples/lsmcount: .libs
Only in amygdala-0.4.0/samples/lsmcount: lsmcount
Only in amygdala-0.4.0/samples/lsmcount: lsmcount.o
Only in amygdala-0.4.0/samples/lsmcount: lsmspikeinput.o
diff -ru amygdala-0.4.0_orig/samples/mouse/.deps/main.Po amygdala-0.4.0/samples/mouse/.deps/main.Po
diff -ru amygdala-0.4.0_orig/samples/mouse/.deps/mouse.Po amygdala-0.4.0/samples/mouse/.deps/mouse.Po
diff -ru amygdala-0.4.0_orig/samples/mouse/.deps/musspikeinput.Po amygdala-0.4.0/samples/mouse/.deps/musspikeinput.Po
Only in amygdala-0.4.0/samples/mouse: .libs
diff -ru amygdala-0.4.0_orig/samples/mouse/main.cpp amygdala-0.4.0/samples/mouse/main.cpp
--- amygdala-0.4.0_orig/samples/mouse/main.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/samples/mouse/main.cpp	2009-02-26 02:12:06.000000000 +0100
@@ -24,6 +24,8 @@
 #include "defaultdata.h"
 #include <string>
 #include <iostream>
+#include <cstdlib>
+#include <cstring>
 
 using namespace std;
 using namespace Amygdala;
Only in amygdala-0.4.0/samples/mouse: main.o
diff -ru amygdala-0.4.0_orig/samples/mouse/mouse.cpp amygdala-0.4.0/samples/mouse/mouse.cpp
--- amygdala-0.4.0_orig/samples/mouse/mouse.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/samples/mouse/mouse.cpp	2009-02-26 02:11:26.000000000 +0100
@@ -18,7 +18,8 @@
 
 #include <iostream>
 #include <fstream>
-#include <stdlib.h>
+#include <cstdlib>
+#include <cstring>
 #include <vector>
 #include <string>
 #include "mouse.h"
@@ -153,7 +154,7 @@
     AlphaNeuron::Factory makeAlphaNeuron;
 
     // Synapses also typedef NeuronConnector templates.
-    StaticSynapse::NeuronConnector nc;
+    StaticSynapse::NeuronConnectorType nc;
     StaticSynapseProperties ssp(areaAWeight, 1000);
 
     int x = -8;
Only in amygdala-0.4.0/samples/mouse: mouse.o
Only in amygdala-0.4.0/samples/mouse: musspikeinput.o
Only in amygdala-0.4.0/samples/mouse: runmouse
diff -ru amygdala-0.4.0_orig/samples/pdelta/.deps/main.Po amygdala-0.4.0/samples/pdelta/.deps/main.Po
Only in amygdala-0.4.0/samples/pdelta: .libs
diff -ru amygdala-0.4.0_orig/samples/pdelta/main.cpp amygdala-0.4.0/samples/pdelta/main.cpp
--- amygdala-0.4.0_orig/samples/pdelta/main.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/samples/pdelta/main.cpp	2009-02-26 02:14:33.000000000 +0100
@@ -1,5 +1,6 @@
 #include <iostream>
 #include <string>
+#include <cstdlib>
 #include <amygdala/network.h>
 #include <amygdala/flatlayer.h>
 #include <amygdala/fastneuron.h>
Only in amygdala-0.4.0/samples/pdelta: main.o
Only in amygdala-0.4.0/samples/pdelta: pdclassify
diff -ru amygdala-0.4.0_orig/samples/pole2d/.deps/pole2d.Po amygdala-0.4.0/samples/pole2d/.deps/pole2d.Po
diff -ru amygdala-0.4.0_orig/samples/pole2d/.deps/poleinput.Po amygdala-0.4.0/samples/pole2d/.deps/poleinput.Po
diff -ru amygdala-0.4.0_orig/samples/pole2d/.deps/poleoutput.Po amygdala-0.4.0/samples/pole2d/.deps/poleoutput.Po
diff -ru amygdala-0.4.0_orig/samples/pole2d/.deps/polephysics.Po amygdala-0.4.0/samples/pole2d/.deps/polephysics.Po
Only in amygdala-0.4.0/samples/pole2d: .libs
Only in amygdala-0.4.0/samples/pole2d: pole2d
diff -ru amygdala-0.4.0_orig/samples/pole2d/pole2d.cpp amygdala-0.4.0/samples/pole2d/pole2d.cpp
--- amygdala-0.4.0_orig/samples/pole2d/pole2d.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/samples/pole2d/pole2d.cpp	2009-02-26 02:14:04.000000000 +0100
@@ -26,6 +26,7 @@
 #include <iostream>
 #include <cstdlib>
 #include <stdexcept>
+#include <cstring>
 
 #include "amygdala/simplenet.h"
 #include "amygdala/basicneuron.h"
Only in amygdala-0.4.0/samples/pole2d: pole2d.o
diff -ru amygdala-0.4.0_orig/samples/pole2d/poleinput.cpp amygdala-0.4.0/samples/pole2d/poleinput.cpp
--- amygdala-0.4.0_orig/samples/pole2d/poleinput.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/samples/pole2d/poleinput.cpp	2009-02-26 02:14:20.000000000 +0100
@@ -23,6 +23,7 @@
 #include "poleinput.h"
 #include <iostream>
 #include <cmath>
+#include <cstring>
 
 using namespace Amygdala;
 using namespace std;
Only in amygdala-0.4.0/samples/pole2d: poleinput.o
Only in amygdala-0.4.0/samples/pole2d: poleoutput.o
Only in amygdala-0.4.0/samples/pole2d: polephysics.o
diff -ru amygdala-0.4.0_orig/samples/spikeloop/.deps/main.Po amygdala-0.4.0/samples/spikeloop/.deps/main.Po
diff -ru amygdala-0.4.0_orig/samples/spikeloop/.deps/spikeloop.Po amygdala-0.4.0/samples/spikeloop/.deps/spikeloop.Po
Only in amygdala-0.4.0/samples/spikeloop: .libs
Only in amygdala-0.4.0/samples/spikeloop: main.o
Only in amygdala-0.4.0/samples/spikeloop: spikeloop
diff -ru amygdala-0.4.0_orig/samples/spikeloop/spikeloop.cpp amygdala-0.4.0/samples/spikeloop/spikeloop.cpp
--- amygdala-0.4.0_orig/samples/spikeloop/spikeloop.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/samples/spikeloop/spikeloop.cpp	2009-02-26 02:14:50.000000000 +0100
@@ -112,7 +112,7 @@
         NeuronCoordinates(tmpLayer, x, y);
     }
 
-    StaticSynapse::NeuronConnector neuronConnector;
+    StaticSynapse::NeuronConnectorType neuronConnector;
     StaticSynapseProperties sProps(0.5, 1000);
 
     for (uint i=0; i<layers.size() - 1; i++) {
Only in amygdala-0.4.0/samples/spikeloop: spikeloop.o
diff -ru amygdala-0.4.0_orig/samples/testgeneserver/.deps/handler.Po amygdala-0.4.0/samples/testgeneserver/.deps/handler.Po
diff -ru amygdala-0.4.0_orig/samples/testgeneserver/.deps/main.Po amygdala-0.4.0/samples/testgeneserver/.deps/main.Po
diff -ru amygdala-0.4.0_orig/samples/testgeneserver/.deps/tester.Po amygdala-0.4.0/samples/testgeneserver/.deps/tester.Po
Only in amygdala-0.4.0/samples/testgeneserver: .libs
Only in amygdala-0.4.0/samples/testgeneserver: handler.o
diff -ru amygdala-0.4.0_orig/samples/testgeneserver/main.cpp amygdala-0.4.0/samples/testgeneserver/main.cpp
--- amygdala-0.4.0_orig/samples/testgeneserver/main.cpp	2009-02-26 02:05:16.000000000 +0100
+++ amygdala-0.4.0/samples/testgeneserver/main.cpp	2009-02-26 02:10:54.000000000 +0100
@@ -18,6 +18,7 @@
 #include <stdexcept>
 #include <iostream>
 #include <string>
+#include <cstdlib>
 
 #include <tester.h>
 
Only in amygdala-0.4.0/samples/testgeneserver: main.o
Only in amygdala-0.4.0/samples/testgeneserver: tester.o
Only in amygdala-0.4.0/samples/testgeneserver: testgeneserver
diff -ru amygdala-0.4.0_orig/test/.deps/main.Po amygdala-0.4.0/test/.deps/main.Po
Only in amygdala-0.4.0/test: .libs
Only in amygdala-0.4.0/test: apitest
Only in amygdala-0.4.0/test: main.o
