diff -r -u dietlibc-0.30_orig/Makefile dietlibc-0.30/Makefile
--- dietlibc-0.30_orig/Makefile	2007-08-04 08:38:35.000000000 +0000
+++ dietlibc-0.30/Makefile	2007-08-04 08:40:29.000000000 +0000
@@ -84,7 +84,8 @@
 # ARCH=$(MYARCH)
 
 OBJDIR=bin-$(ARCH)
-ILIBDIR=$(LIBDIR)-$(ARCH)
+ILIBDIR=$(LIBDIR)
+#-$(ARCH)
 
 HOME=$(shell pwd)
 
diff -r -u dietlibc-0.30_orig/Makefile dietlibc-0.30/Makefile
--- dietlibc-0.30_orig/Makefile	2007-08-04 09:02:40.000000000 +0000
+++ dietlibc-0.30/Makefile	2007-08-04 09:03:08.000000000 +0000
@@ -83,9 +83,8 @@
 
 # ARCH=$(MYARCH)
 
-OBJDIR=bin-$(ARCH)
+OBJDIR=bin
 ILIBDIR=$(LIBDIR)
-#-$(ARCH)
 
 HOME=$(shell pwd)
 
@@ -297,10 +296,10 @@
 $(SYSCALLOBJ): syscalls.h
 
 $(OBJDIR)/elftrunc: $(OBJDIR)/diet contrib/elftrunc.c
-	bin-$(MYARCH)/diet $(CROSS)$(CC) $(CFLAGS) -o $@ contrib/elftrunc.c
+	bin/diet $(CROSS)$(CC) $(CFLAGS) -o $@ contrib/elftrunc.c
 
 $(OBJDIR)/dnsd: $(OBJDIR)/diet contrib/dnsd.c
-	bin-$(MYARCH)/diet $(CROSS)$(CC) $(CFLAGS) -o $@ contrib/dnsd.c
+	bin/diet $(CROSS)$(CC) $(CFLAGS) -o $@ contrib/dnsd.c
 
 VERSION=dietlibc-$(shell head -n 1 CHANGES|sed 's/://')
 CURNAME=$(notdir $(shell pwd))
Only in dietlibc-0.30: bin
diff -r -u dietlibc-0.30_orig/diet.c dietlibc-0.30/diet.c
--- dietlibc-0.30_orig/diet.c	2007-08-04 09:02:40.000000000 +0000
+++ dietlibc-0.30/diet.c	2007-08-04 09:06:42.000000000 +0000
@@ -95,12 +95,12 @@
   platform=alloca(strlen(diethome)+100);
   strcpy(platform,diethome);
 #ifdef INSTALLVERSION
-  strcat(platform,"/lib-");
+  strcat(platform,"/lib");
 #else
 #ifndef __DYN_LIB
-  strcat(platform,"/bin-");
+  strcat(platform,"/bin");
 #else
-  strcat(platform,"/pic-");
+  strcat(platform,"/pic");
 #endif
 #endif
   strcpy(dashL,"-L");
@@ -139,8 +139,9 @@
       if (tmp3<tmp2) tmp2=tmp3;
       if (tmp2-cc>90) error("platform name too long!\n");
       shortplatform=platform+len;
-      memmove(shortplatform,argv[1],(size_t)(tmp2-cc));
-      platform[tmp2-cc+len]=0;
+      *shortplatform='\0';
+//      memmove(shortplatform,argv[1],(size_t)(tmp2-cc));
+//      platform[tmp2-cc+len]=0;
       if (shortplatform[0]=='i' && shortplatform[2]=='8' && shortplatform[3]=='6') shortplatform[1]='3';
     } else {
 #ifdef __sparc__
@@ -189,7 +190,7 @@
 #endif
       {
 	char *tmp=platform+strlen(platform);
-	strcpy(tmp,shortplatform);
+	//strcpy(tmp,shortplatform);
 	shortplatform=tmp;
       }
     }
