diff -ru dietlibc-0.31_orig/Makefile dietlibc-0.31/Makefile
--- dietlibc-0.31_orig/Makefile	2008-05-26 12:56:07.000000000 +0200
+++ dietlibc-0.31/Makefile	2008-05-26 13:12:51.000000000 +0200
@@ -261,7 +261,9 @@
 DYN_LIBC_PIC = $(LIBOBJ) $(LIBSTDIOOBJ) $(LIBUGLYOBJ) \
 $(LIBCRUFTOBJ) $(LIBCRYPTOBJ) $(LIBSHELLOBJ) $(LIBREGEXOBJ)
 
-DYN_LIBC_OBJ = $(PICODIR)/dyn_syscalls.o $(PICODIR)/errlist.o \
+DYN_LIBC_OBJ = \
+  $(PICODIR)/dyn_syscalls.o \
+  $(PICODIR)/errlist.o \
 	$(patsubst $(OBJDIR)/%.o,$(PICODIR)/%.o,$(DYN_LIBC_PIC))
 
 DYN_PTHREAD_OBJS = $(patsubst $(OBJDIR)/%.o,$(PICODIR)/%.o,$(LIBPTHREAD_OBJS))
Only in dietlibc-0.31_orig/bin: __dtostr.o
Only in dietlibc-0.31_orig/bin: __end_parse.o
Only in dietlibc-0.31_orig/bin: __fcntl64.o
Only in dietlibc-0.31_orig/bin: __finite.o
Only in dietlibc-0.31_orig/bin: __fprepare_parse.o
Only in dietlibc-0.31_orig/bin: __fstat64.o
Only in dietlibc-0.31_orig/bin: __fstatfs64.o
Only in dietlibc-0.31_orig/bin: __ftruncate64.o
Only in dietlibc-0.31_orig/bin: __getcwd.o
Only in dietlibc-0.31_orig/bin: __getpagesize.o
Only in dietlibc-0.31_orig/bin: __guard.o
Only in dietlibc-0.31_orig/bin: __half.o
Only in dietlibc-0.31_orig/bin: __isinf.o
Only in dietlibc-0.31_orig/bin: __isnan.o
Only in dietlibc-0.31_orig/bin: __lltostr.o
Only in dietlibc-0.31_orig/bin: __longjmp.o
Only in dietlibc-0.31_orig/bin: __lstat64.o
Only in dietlibc-0.31_orig/bin: __ltostr.o
Only in dietlibc-0.31_orig/bin: __mcount.o
Only in dietlibc-0.31_orig/bin: __nice.o
Only in dietlibc-0.31_orig/bin: __parse.o
Only in dietlibc-0.31_orig/bin: __parse_1.o
Only in dietlibc-0.31_orig/bin: __parse_nws.o
Only in dietlibc-0.31_orig/bin: __parse_ws.o
Only in dietlibc-0.31_orig/bin: __pread.o
Only in dietlibc-0.31_orig/bin: __prepare_parse.o
Only in dietlibc-0.31_orig/bin: __ptrace.o
Only in dietlibc-0.31_orig/bin: __pwrite.o
Only in dietlibc-0.31_orig/bin: __reboot.o
Only in dietlibc-0.31_orig/bin: __restore_rt.o
Only in dietlibc-0.31_orig/bin: __sendfile64.o
Only in dietlibc-0.31_orig/bin: __signalfd.o
Only in dietlibc-0.31_orig/bin: __stat64.o
Only in dietlibc-0.31_orig/bin: __stat64_cvt.o
Only in dietlibc-0.31_orig/bin: __statfs64.o
Only in dietlibc-0.31_orig/bin: __stime.o
Only in dietlibc-0.31_orig/bin: __sysctl.o
Only in dietlibc-0.31_orig/bin: __ten.o
Only in dietlibc-0.31_orig/bin: __testandset.o
Only in dietlibc-0.31_orig/bin: __truncate64.o
Only in dietlibc-0.31_orig/bin: __utmp_block_signals.o
Only in dietlibc-0.31_orig/bin: __utmp_io.o
Only in dietlibc-0.31_orig/bin: __v_printf.o
Only in dietlibc-0.31_orig/bin: __v_scanf.o
Only in dietlibc-0.31_orig/bin: _brk.o
Only in dietlibc-0.31_orig/bin: _llseek.o
Only in dietlibc-0.31_orig/bin: abort.o
Only in dietlibc-0.31_orig/bin: abs.o
Only in dietlibc-0.31_orig/bin: accept.o
Only in dietlibc-0.31_orig/bin: access.o
Only in dietlibc-0.31_orig/bin: acos.o
Only in dietlibc-0.31_orig/bin: add_key.o
Only in dietlibc-0.31_orig/bin: addmntent.o
Only in dietlibc-0.31_orig/bin: adjtime.o
Only in dietlibc-0.31_orig/bin: adjtimex.o
Only in dietlibc-0.31_orig/bin: alarm.o
Only in dietlibc-0.31_orig/bin: alloc.o
Only in dietlibc-0.31_orig/bin: alphasort.o
Only in dietlibc-0.31_orig/bin: alphasort64.o
Only in dietlibc-0.31_orig/bin: asctime.o
Only in dietlibc-0.31_orig/bin: asctime_r.o
Only in dietlibc-0.31_orig/bin: asin.o
Only in dietlibc-0.31_orig/bin: asprintf.o
Only in dietlibc-0.31_orig/bin: assert_fail.o
Only in dietlibc-0.31_orig/bin: atan.o
Only in dietlibc-0.31_orig/bin: atan2.o
Only in dietlibc-0.31_orig/bin: atexit.o
Only in dietlibc-0.31_orig/bin: atof.o
Only in dietlibc-0.31_orig/bin: atoi.o
Only in dietlibc-0.31_orig/bin: atol.o
Only in dietlibc-0.31_orig/bin: atoll.o
Only in dietlibc-0.31_orig/bin: auth_none.o
Only in dietlibc-0.31_orig/bin: auth_unix.o
Only in dietlibc-0.31_orig/bin: authunix_prot.o
Only in dietlibc-0.31_orig/bin: basename.o
Only in dietlibc-0.31_orig/bin: bcopy.o
Only in dietlibc-0.31_orig/bin: bdflush.o
Only in dietlibc-0.31_orig/bin: bessel.o
Only in dietlibc-0.31_orig/bin: bind.o
Only in dietlibc-0.31_orig/bin: bindresvport.o
Only in dietlibc-0.31_orig/bin: bindtextdomain.o
Only in dietlibc-0.31_orig/bin: binshstr.o
Only in dietlibc-0.31_orig/bin: brk.o
Only in dietlibc-0.31_orig/bin: bsearch.o
Only in dietlibc-0.31_orig/bin: btowc.o
Only in dietlibc-0.31_orig/bin: bzero.o
Only in dietlibc-0.31_orig/bin: capget.o
Only in dietlibc-0.31_orig/bin: capset.o
Only in dietlibc-0.31_orig/bin: ceil.o
Only in dietlibc-0.31_orig/bin: cfgetospeed.o
Only in dietlibc-0.31_orig/bin: cfmakeraw.o
Only in dietlibc-0.31_orig/bin: cfsetispeed.o
Only in dietlibc-0.31_orig/bin: cfsetospeed.o
Only in dietlibc-0.31_orig/bin: chdir.o
Only in dietlibc-0.31_orig/bin: chmod.o
Only in dietlibc-0.31_orig/bin: chown.o
Only in dietlibc-0.31_orig/bin: chown32.o
Only in dietlibc-0.31_orig/bin: chroot.o
Only in dietlibc-0.31_orig/bin: clearenv.o
Only in dietlibc-0.31_orig/bin: clearerr.o
Only in dietlibc-0.31_orig/bin: clnt_generic.o
Only in dietlibc-0.31_orig/bin: clnt_perror.o
Only in dietlibc-0.31_orig/bin: clnt_raw.o
Only in dietlibc-0.31_orig/bin: clnt_simple.o
Only in dietlibc-0.31_orig/bin: clnt_tcp.o
Only in dietlibc-0.31_orig/bin: clnt_udp.o
Only in dietlibc-0.31_orig/bin: clock.o
Only in dietlibc-0.31_orig/bin: clock_getres.o
Only in dietlibc-0.31_orig/bin: clock_gettime.o
Only in dietlibc-0.31_orig/bin: clock_nanosleep.o
Only in dietlibc-0.31_orig/bin: clock_settime.o
Only in dietlibc-0.31_orig/bin: clone.o
Only in dietlibc-0.31_orig/bin: close.o
Only in dietlibc-0.31_orig/bin: closedir.o
Only in dietlibc-0.31_orig/bin: confstr.o
Only in dietlibc-0.31_orig/bin: connect.o
Only in dietlibc-0.31_orig/bin: cos.o
Only in dietlibc-0.31_orig/bin: cosh.o
Only in dietlibc-0.31_orig/bin: creat.o
Only in dietlibc-0.31_orig/bin: creat64.o
Only in dietlibc-0.31_orig/bin: create_module.o
Only in dietlibc-0.31_orig/bin: crypt.o
Only in dietlibc-0.31_orig/bin: ctermid.o
Only in dietlibc-0.31_orig/bin: ctime.o
Only in dietlibc-0.31_orig/bin: ctime_r.o
Only in dietlibc-0.31_orig/bin: cuserid.o
Only in dietlibc-0.31_orig/bin: daemon.o
Only in dietlibc-0.31_orig/bin: dcgettext.o
Only in dietlibc-0.31_orig/bin: delete_module.o
Only in dietlibc-0.31_orig/bin: dgettext.o
Binary files dietlibc-0.31_orig/bin/dietlibc.a and dietlibc-0.31/bin/dietlibc.a differ
Only in dietlibc-0.31_orig/bin: difftime.o
Only in dietlibc-0.31_orig/bin: dirfd.o
Only in dietlibc-0.31_orig/bin: dirname.o
Only in dietlibc-0.31_orig/bin: div.o
Only in dietlibc-0.31_orig/bin: dl_iterate_phdr.o
Only in dietlibc-0.31_orig/bin: dl_support.o
Only in dietlibc-0.31_orig/bin: dn_expand.o
Only in dietlibc-0.31_orig/bin: dn_skipname.o
Only in dietlibc-0.31_orig/bin: dngettext.o
Only in dietlibc-0.31_orig/bin: dnscruft.o
Only in dietlibc-0.31_orig/bin: dnscruft2.o
Only in dietlibc-0.31_orig/bin: dnscruft3.o
Only in dietlibc-0.31_orig/bin: dnscruft4.o
Only in dietlibc-0.31_orig/bin: dup.o
Only in dietlibc-0.31_orig/bin: dup2.o
Only in dietlibc-0.31_orig/bin: dyn_start.o
Only in dietlibc-0.31_orig/bin: dyn_stop.o
Only in dietlibc-0.31_orig/bin: endmntent.o
Only in dietlibc-0.31_orig/bin: environ.o
Only in dietlibc-0.31_orig/bin: epoll_create.o
Only in dietlibc-0.31_orig/bin: epoll_ctl.o
Only in dietlibc-0.31_orig/bin: epoll_wait.o
Only in dietlibc-0.31_orig/bin: erf.o
Only in dietlibc-0.31_orig/bin: err.o
Only in dietlibc-0.31_orig/bin: errlist.o
Only in dietlibc-0.31_orig/bin: errlistu.o
Only in dietlibc-0.31_orig/bin: errno.o
Only in dietlibc-0.31_orig/bin: errno_location.o
Only in dietlibc-0.31_orig/bin: errx.o
Only in dietlibc-0.31_orig/bin: exec_lib.o
Only in dietlibc-0.31_orig/bin: execl.o
Only in dietlibc-0.31_orig/bin: execle.o
Only in dietlibc-0.31_orig/bin: execlp.o
Only in dietlibc-0.31_orig/bin: execv.o
Only in dietlibc-0.31_orig/bin: execve.o
Only in dietlibc-0.31_orig/bin: execvp.o
Only in dietlibc-0.31_orig/bin: exp.o
Only in dietlibc-0.31_orig/bin: fabs.o
Only in dietlibc-0.31_orig/bin: facilitynames.o
Only in dietlibc-0.31_orig/bin: fadvise64.o
Only in dietlibc-0.31_orig/bin: fadvise64_64.o
Only in dietlibc-0.31_orig/bin: fchdir.o
Only in dietlibc-0.31_orig/bin: fchmod.o
Only in dietlibc-0.31_orig/bin: fchown.o
Only in dietlibc-0.31_orig/bin: fchown32.o
Only in dietlibc-0.31_orig/bin: fclose.o
Only in dietlibc-0.31_orig/bin: fcntl.o
Only in dietlibc-0.31_orig/bin: fcntl64.o
Only in dietlibc-0.31_orig/bin: fdatasync.o
Only in dietlibc-0.31_orig/bin: fdglue.o
Only in dietlibc-0.31_orig/bin: fdglue2.o
Only in dietlibc-0.31_orig/bin: fdopen.o
Only in dietlibc-0.31_orig/bin: fdprintf.o
Only in dietlibc-0.31_orig/bin: feof.o
Only in dietlibc-0.31_orig/bin: ferror.o
Only in dietlibc-0.31_orig/bin: fflush.o
Only in dietlibc-0.31_orig/bin: ffs.o
Only in dietlibc-0.31_orig/bin: ffsl.o
Only in dietlibc-0.31_orig/bin: ffsll.o
Only in dietlibc-0.31_orig/bin: fgetc_unlocked.o
Only in dietlibc-0.31_orig/bin: fgetpos.o
Only in dietlibc-0.31_orig/bin: fgetpwent.o
Only in dietlibc-0.31_orig/bin: fgetpwent_r.o
Only in dietlibc-0.31_orig/bin: fgets.o
Only in dietlibc-0.31_orig/bin: fgetxattr.o
Only in dietlibc-0.31_orig/bin: fileno.o
Only in dietlibc-0.31_orig/bin: flistxattr.o
Only in dietlibc-0.31_orig/bin: flock.o
Only in dietlibc-0.31_orig/bin: floor.o
Only in dietlibc-0.31_orig/bin: fmod.o
Only in dietlibc-0.31_orig/bin: fnmatch.o
Only in dietlibc-0.31_orig/bin: fopen.o
Only in dietlibc-0.31_orig/bin: fork.o
Only in dietlibc-0.31_orig/bin: fprintf.o
Only in dietlibc-0.31_orig/bin: fputc_unlocked.o
Only in dietlibc-0.31_orig/bin: fputs.o
Only in dietlibc-0.31_orig/bin: fread.o
Only in dietlibc-0.31_orig/bin: freeaddrinfo.o
Only in dietlibc-0.31_orig/bin: fremovexattr.o
Only in dietlibc-0.31_orig/bin: freopen.o
Only in dietlibc-0.31_orig/bin: fscanf.o
Only in dietlibc-0.31_orig/bin: fseek.o
Only in dietlibc-0.31_orig/bin: fseeko.o
Only in dietlibc-0.31_orig/bin: fseeko64.o
Only in dietlibc-0.31_orig/bin: fsetpos.o
Only in dietlibc-0.31_orig/bin: fsetxattr.o
Only in dietlibc-0.31_orig/bin: fstat.o
Only in dietlibc-0.31_orig/bin: fstat64.o
Only in dietlibc-0.31_orig/bin: fstatfs.o
Only in dietlibc-0.31_orig/bin: fstatfs64.o
Only in dietlibc-0.31_orig/bin: fstatvfs.o
Only in dietlibc-0.31_orig/bin: fsync.o
Only in dietlibc-0.31_orig/bin: ftell.o
Only in dietlibc-0.31_orig/bin: ftello.o
Only in dietlibc-0.31_orig/bin: ftello64.o
Only in dietlibc-0.31_orig/bin: ftime.o
Only in dietlibc-0.31_orig/bin: ftok.o
Only in dietlibc-0.31_orig/bin: ftruncate.o
Only in dietlibc-0.31_orig/bin: ftruncate64.o
Only in dietlibc-0.31_orig/bin: ftw.o
Only in dietlibc-0.31_orig/bin: ftw64.o
Only in dietlibc-0.31_orig/bin: futex.o
Only in dietlibc-0.31_orig/bin: fwrite.o
Only in dietlibc-0.31_orig/bin: gai_strerror.o
Only in dietlibc-0.31_orig/bin: gamma.o
Only in dietlibc-0.31_orig/bin: get_mempolicy.o
Only in dietlibc-0.31_orig/bin: get_myaddress.o
Only in dietlibc-0.31_orig/bin: get_thread_area.o
Only in dietlibc-0.31_orig/bin: getaddrinfo.o
Only in dietlibc-0.31_orig/bin: getcwd.o
Only in dietlibc-0.31_orig/bin: getdelim.o
Only in dietlibc-0.31_orig/bin: getdents.o
Only in dietlibc-0.31_orig/bin: getdents64.o
Only in dietlibc-0.31_orig/bin: getdomainname.o
Only in dietlibc-0.31_orig/bin: getdtablesize.o
Only in dietlibc-0.31_orig/bin: getegid.o
Only in dietlibc-0.31_orig/bin: getegid32.o
Only in dietlibc-0.31_orig/bin: getenv.o
Only in dietlibc-0.31_orig/bin: geteuid.o
Only in dietlibc-0.31_orig/bin: geteuid32.o
Only in dietlibc-0.31_orig/bin: getgid.o
Only in dietlibc-0.31_orig/bin: getgid32.o
Only in dietlibc-0.31_orig/bin: getgrent.o
Only in dietlibc-0.31_orig/bin: getgrent_r.o
Only in dietlibc-0.31_orig/bin: getgrgid.o
Only in dietlibc-0.31_orig/bin: getgrgid_r.o
Only in dietlibc-0.31_orig/bin: getgrnam.o
Only in dietlibc-0.31_orig/bin: getgrnam_r.o
Only in dietlibc-0.31_orig/bin: getgroups.o
Only in dietlibc-0.31_orig/bin: getgroups32.o
Only in dietlibc-0.31_orig/bin: gethostbyaddr.o
Only in dietlibc-0.31_orig/bin: gethostbyaddr2_r.o
Only in dietlibc-0.31_orig/bin: gethostbyaddr_r.o
Only in dietlibc-0.31_orig/bin: gethostbyname.o
Only in dietlibc-0.31_orig/bin: gethostbyname2.o
Only in dietlibc-0.31_orig/bin: gethostbyname2_r.o
Only in dietlibc-0.31_orig/bin: gethostbyname_r.o
Only in dietlibc-0.31_orig/bin: gethostent.o
Only in dietlibc-0.31_orig/bin: gethostent_nor.o
Only in dietlibc-0.31_orig/bin: gethostname.o
Only in dietlibc-0.31_orig/bin: getitimer.o
Only in dietlibc-0.31_orig/bin: getline.o
Only in dietlibc-0.31_orig/bin: getlogin.o
Only in dietlibc-0.31_orig/bin: getmntent.o
Only in dietlibc-0.31_orig/bin: getnameinfo.o
Only in dietlibc-0.31_orig/bin: getopt.o
Only in dietlibc-0.31_orig/bin: getopt_data.o
Only in dietlibc-0.31_orig/bin: getopt_long.o
Only in dietlibc-0.31_orig/bin: getopt_long_only.o
Only in dietlibc-0.31_orig/bin: getpagesize.o
Only in dietlibc-0.31_orig/bin: getpass.o
Only in dietlibc-0.31_orig/bin: getpeername.o
Only in dietlibc-0.31_orig/bin: getpgid.o
Only in dietlibc-0.31_orig/bin: getpgrp.o
Only in dietlibc-0.31_orig/bin: getpid.o
Only in dietlibc-0.31_orig/bin: getppid.o
Only in dietlibc-0.31_orig/bin: getpriority.o
Only in dietlibc-0.31_orig/bin: getprotobyname.o
Only in dietlibc-0.31_orig/bin: getprotobyname_r.o
Only in dietlibc-0.31_orig/bin: getprotobynumber.o
Only in dietlibc-0.31_orig/bin: getprotobynumber_r.o
Only in dietlibc-0.31_orig/bin: getprotoent.o
Only in dietlibc-0.31_orig/bin: getprotoent_r.o
Only in dietlibc-0.31_orig/bin: getpwent.o
Only in dietlibc-0.31_orig/bin: getpwent_r.o
Only in dietlibc-0.31_orig/bin: getpwnam.o
Only in dietlibc-0.31_orig/bin: getpwnam_r.o
Only in dietlibc-0.31_orig/bin: getpwuid.o
Only in dietlibc-0.31_orig/bin: getpwuid_r.o
Only in dietlibc-0.31_orig/bin: getresgid.o
Only in dietlibc-0.31_orig/bin: getresgid32.o
Only in dietlibc-0.31_orig/bin: getresuid.o
Only in dietlibc-0.31_orig/bin: getrlimit.o
Only in dietlibc-0.31_orig/bin: getrpcent.o
Only in dietlibc-0.31_orig/bin: getrpcport.o
Only in dietlibc-0.31_orig/bin: getrusage.o
Only in dietlibc-0.31_orig/bin: getservbyname.o
Only in dietlibc-0.31_orig/bin: getservbyname_r.o
Only in dietlibc-0.31_orig/bin: getservbyport.o
Only in dietlibc-0.31_orig/bin: getservbyport_r.o
Only in dietlibc-0.31_orig/bin: getservent.o
Only in dietlibc-0.31_orig/bin: getservent_r.o
Only in dietlibc-0.31_orig/bin: getsid.o
Only in dietlibc-0.31_orig/bin: getsockname.o
Only in dietlibc-0.31_orig/bin: getsockopt.o
Only in dietlibc-0.31_orig/bin: getspent.o
Only in dietlibc-0.31_orig/bin: getspent_r.o
Only in dietlibc-0.31_orig/bin: getspnam.o
Only in dietlibc-0.31_orig/bin: getspnam_r.o
Only in dietlibc-0.31_orig/bin: gettext.o
Only in dietlibc-0.31_orig/bin: gettid.o
Only in dietlibc-0.31_orig/bin: gettimeofday.o
Only in dietlibc-0.31_orig/bin: getuid.o
Only in dietlibc-0.31_orig/bin: getuid32.o
Only in dietlibc-0.31_orig/bin: getusershell.o
Only in dietlibc-0.31_orig/bin: getutmp.o
Only in dietlibc-0.31_orig/bin: getxattr.o
Only in dietlibc-0.31_orig/bin: glob.o
Only in dietlibc-0.31_orig/bin: gmtime.o
Only in dietlibc-0.31_orig/bin: gmtime_r.o
Only in dietlibc-0.31_orig/bin: grantpt.o
Only in dietlibc-0.31_orig/bin: grbuf.o
Only in dietlibc-0.31_orig/bin: h_errno.o
Only in dietlibc-0.31_orig/bin: hasmntopt.o
Only in dietlibc-0.31_orig/bin: herrno_location.o
Only in dietlibc-0.31_orig/bin: herror.o
Only in dietlibc-0.31_orig/bin: hstrerror.o
Only in dietlibc-0.31_orig/bin: htonl.o
Only in dietlibc-0.31_orig/bin: htons.o
Only in dietlibc-0.31_orig/bin: hypot.o
Only in dietlibc-0.31_orig/bin: iconv.o
Only in dietlibc-0.31_orig/bin: iconv_close.o
Only in dietlibc-0.31_orig/bin: iconv_open.o
Only in dietlibc-0.31_orig/bin: if_indextoname.o
Only in dietlibc-0.31_orig/bin: if_nameindex.o
Only in dietlibc-0.31_orig/bin: if_nametoindex.o
Only in dietlibc-0.31_orig/bin: inet_addr.o
Only in dietlibc-0.31_orig/bin: inet_aton.o
Only in dietlibc-0.31_orig/bin: inet_ntoa.o
Only in dietlibc-0.31_orig/bin: inet_ntoa_r.o
Only in dietlibc-0.31_orig/bin: inet_ntop.o
Only in dietlibc-0.31_orig/bin: inet_pton.o
Only in dietlibc-0.31_orig/bin: init_module.o
Only in dietlibc-0.31_orig/bin: initgroups.o
Only in dietlibc-0.31_orig/bin: inotify_add_watch.o
Only in dietlibc-0.31_orig/bin: inotify_init.o
Only in dietlibc-0.31_orig/bin: inotify_rm_watch.o
Only in dietlibc-0.31_orig/bin: io_cancel.o
Only in dietlibc-0.31_orig/bin: io_destroy.o
Only in dietlibc-0.31_orig/bin: io_getevents.o
Only in dietlibc-0.31_orig/bin: io_setup.o
Only in dietlibc-0.31_orig/bin: io_submit.o
Only in dietlibc-0.31_orig/bin: ioctl.o
Only in dietlibc-0.31_orig/bin: ioperm.o
Only in dietlibc-0.31_orig/bin: iopl.o
Only in dietlibc-0.31_orig/bin: ipc.o
Only in dietlibc-0.31_orig/bin: ipow.o
Only in dietlibc-0.31_orig/bin: ipv6constants.o
Only in dietlibc-0.31_orig/bin: isalnum.o
Only in dietlibc-0.31_orig/bin: isalpha.o
Only in dietlibc-0.31_orig/bin: isascii.o
Only in dietlibc-0.31_orig/bin: isatty.o
Only in dietlibc-0.31_orig/bin: isblank.o
Only in dietlibc-0.31_orig/bin: iscntrl.o
Only in dietlibc-0.31_orig/bin: isdigit.o
Only in dietlibc-0.31_orig/bin: isgraph.o
Only in dietlibc-0.31_orig/bin: isleap.o
Only in dietlibc-0.31_orig/bin: islower.o
Only in dietlibc-0.31_orig/bin: isprint.o
Only in dietlibc-0.31_orig/bin: ispunct.o
Only in dietlibc-0.31_orig/bin: isspace.o
Only in dietlibc-0.31_orig/bin: isupper.o
Only in dietlibc-0.31_orig/bin: iswalnum.o
Only in dietlibc-0.31_orig/bin: iswalpha.o
Only in dietlibc-0.31_orig/bin: iswblank.o
Only in dietlibc-0.31_orig/bin: iswcntrl.o
Only in dietlibc-0.31_orig/bin: iswctype.o
Only in dietlibc-0.31_orig/bin: iswdigit.o
Only in dietlibc-0.31_orig/bin: iswgraph.o
Only in dietlibc-0.31_orig/bin: iswlower.o
Only in dietlibc-0.31_orig/bin: iswprint.o
Only in dietlibc-0.31_orig/bin: iswpunct.o
Only in dietlibc-0.31_orig/bin: iswspace.o
Only in dietlibc-0.31_orig/bin: iswupper.o
Only in dietlibc-0.31_orig/bin: iswxdigit.o
Only in dietlibc-0.31_orig/bin: isxdigit.o
Only in dietlibc-0.31_orig/bin: keyctl.o
Only in dietlibc-0.31_orig/bin: kill.o
Only in dietlibc-0.31_orig/bin: killpg.o
Only in dietlibc-0.31_orig/bin: labs.o
Only in dietlibc-0.31_orig/bin: latin1-iscntrl.o
Only in dietlibc-0.31_orig/bin: latin1-isgraph.o
Only in dietlibc-0.31_orig/bin: latin1-islower.o
Only in dietlibc-0.31_orig/bin: latin1-isprint.o
Only in dietlibc-0.31_orig/bin: latin1-isupper.o
Only in dietlibc-0.31_orig/bin: lc_ctype.o
Only in dietlibc-0.31_orig/bin: lchown.o
Only in dietlibc-0.31_orig/bin: lchown32.o
Only in dietlibc-0.31_orig/bin: ldexp.o
Only in dietlibc-0.31_orig/bin: ldiv.o
Only in dietlibc-0.31_orig/bin: lgetxattr.o
Binary files dietlibc-0.31_orig/bin/libcompat.a and dietlibc-0.31/bin/libcompat.a differ
Binary files dietlibc-0.31_orig/bin/libcrypt.a and dietlibc-0.31/bin/libcrypt.a differ
Binary files dietlibc-0.31_orig/bin/libgmon.a and dietlibc-0.31/bin/libgmon.a differ
Binary files dietlibc-0.31_orig/bin/liblatin1.a and dietlibc-0.31/bin/liblatin1.a differ
Binary files dietlibc-0.31_orig/bin/libm.a and dietlibc-0.31/bin/libm.a differ
Only in dietlibc-0.31_orig/bin: libm2.o
Binary files dietlibc-0.31_orig/bin/libpthread.a and dietlibc-0.31/bin/libpthread.a differ
Binary files dietlibc-0.31_orig/bin/librpc.a and dietlibc-0.31/bin/librpc.a differ
Only in dietlibc-0.31_orig/bin: link.o
Only in dietlibc-0.31_orig/bin: listen.o
Only in dietlibc-0.31_orig/bin: listxattr.o
Only in dietlibc-0.31_orig/bin: llabs.o
Only in dietlibc-0.31_orig/bin: lldiv.o
Only in dietlibc-0.31_orig/bin: llistxattr.o
Only in dietlibc-0.31_orig/bin: localeconv.o
Only in dietlibc-0.31_orig/bin: localtime.o
Only in dietlibc-0.31_orig/bin: localtime_r.o
Only in dietlibc-0.31_orig/bin: lockf.o
Only in dietlibc-0.31_orig/bin: log.o
Only in dietlibc-0.31_orig/bin: log10.o
Only in dietlibc-0.31_orig/bin: log1p.o
Only in dietlibc-0.31_orig/bin: logging.o
Only in dietlibc-0.31_orig/bin: longjmp.o
Only in dietlibc-0.31_orig/bin: lremovexattr.o
Only in dietlibc-0.31_orig/bin: lseek.o
Only in dietlibc-0.31_orig/bin: lseek64.o
Only in dietlibc-0.31_orig/bin: lsetxattr.o
Only in dietlibc-0.31_orig/bin: lstat.o
Only in dietlibc-0.31_orig/bin: lstat64.o
Only in dietlibc-0.31_orig/bin: madvise.o
Only in dietlibc-0.31_orig/bin: mbind.o
Only in dietlibc-0.31_orig/bin: mblen.o
Only in dietlibc-0.31_orig/bin: mbrlen.o
Only in dietlibc-0.31_orig/bin: mbrtowc.o
Only in dietlibc-0.31_orig/bin: mbsinit.o
Only in dietlibc-0.31_orig/bin: mbsrtowcs.o
Only in dietlibc-0.31_orig/bin: mbstowcs.o
Only in dietlibc-0.31_orig/bin: mbtowc.o
Only in dietlibc-0.31_orig/bin: mcount.o
Only in dietlibc-0.31_orig/bin: md5.o
Only in dietlibc-0.31_orig/bin: md5asm.o
Only in dietlibc-0.31_orig/bin: md5crypt.o
Only in dietlibc-0.31_orig/bin: memccpy.o
Only in dietlibc-0.31_orig/bin: memchr.o
Only in dietlibc-0.31_orig/bin: memcmp.o
Only in dietlibc-0.31_orig/bin: memcpy.o
Only in dietlibc-0.31_orig/bin: memmem.o
Only in dietlibc-0.31_orig/bin: memmove.o
Only in dietlibc-0.31_orig/bin: memrchr.o
Only in dietlibc-0.31_orig/bin: memset.o
Only in dietlibc-0.31_orig/bin: mincore.o
Only in dietlibc-0.31_orig/bin: mkdir.o
Only in dietlibc-0.31_orig/bin: mkdtemp.o
Only in dietlibc-0.31_orig/bin: mkfifo.o
Only in dietlibc-0.31_orig/bin: mknod.o
Only in dietlibc-0.31_orig/bin: mkstemp.o
Only in dietlibc-0.31_orig/bin: mktemp.o
Only in dietlibc-0.31_orig/bin: mktime.o
Only in dietlibc-0.31_orig/bin: mlock.o
Only in dietlibc-0.31_orig/bin: mlockall.o
Only in dietlibc-0.31_orig/bin: mmap.o
Only in dietlibc-0.31_orig/bin: mmap2.o
Only in dietlibc-0.31_orig/bin: mmap64.o
Only in dietlibc-0.31_orig/bin: modf.o
Only in dietlibc-0.31_orig/bin: monitor.o
Only in dietlibc-0.31_orig/bin: mount.o
Only in dietlibc-0.31_orig/bin: mprotect.o
Only in dietlibc-0.31_orig/bin: mq_getattr.o
Only in dietlibc-0.31_orig/bin: mq_getsetattr.o
Only in dietlibc-0.31_orig/bin: mq_notify.o
Only in dietlibc-0.31_orig/bin: mq_open.o
Only in dietlibc-0.31_orig/bin: mq_receive.o
Only in dietlibc-0.31_orig/bin: mq_send.o
Only in dietlibc-0.31_orig/bin: mq_timedreceive.o
Only in dietlibc-0.31_orig/bin: mq_timedsend.o
Only in dietlibc-0.31_orig/bin: mq_unlink.o
Only in dietlibc-0.31_orig/bin: mremap.o
Only in dietlibc-0.31_orig/bin: msgctl.o
Only in dietlibc-0.31_orig/bin: msgget.o
Only in dietlibc-0.31_orig/bin: msgrcv.o
Only in dietlibc-0.31_orig/bin: msgsnd.o
Only in dietlibc-0.31_orig/bin: msync.o
Only in dietlibc-0.31_orig/bin: munlock.o
Only in dietlibc-0.31_orig/bin: munlockall.o
Only in dietlibc-0.31_orig/bin: munmap.o
Only in dietlibc-0.31_orig/bin: n_sigaction.o
Only in dietlibc-0.31_orig/bin: n_sigpending.o
Only in dietlibc-0.31_orig/bin: n_sigprocmask.o
Only in dietlibc-0.31_orig/bin: n_sigsuspend.o
Only in dietlibc-0.31_orig/bin: nanosleep.o
Only in dietlibc-0.31_orig/bin: netent.o
Only in dietlibc-0.31_orig/bin: ngettext.o
Only in dietlibc-0.31_orig/bin: nice.o
Only in dietlibc-0.31_orig/bin: nl_langinfo.o
Only in dietlibc-0.31_orig/bin: open.o
Only in dietlibc-0.31_orig/bin: open64.o
Only in dietlibc-0.31_orig/bin: opendir.o
Only in dietlibc-0.31_orig/bin: openpty.o
Only in dietlibc-0.31_orig/bin: pause.o
Only in dietlibc-0.31_orig/bin: pclose.o
Only in dietlibc-0.31_orig/bin: perror.o
Only in dietlibc-0.31_orig/bin: personality.o
Only in dietlibc-0.31_orig/bin: pipe.o
Only in dietlibc-0.31_orig/bin: pivot_root.o
Only in dietlibc-0.31_orig/bin: pmap_clnt.o
Only in dietlibc-0.31_orig/bin: pmap_getmaps.o
Only in dietlibc-0.31_orig/bin: pmap_getport.o
Only in dietlibc-0.31_orig/bin: pmap_prot.o
Only in dietlibc-0.31_orig/bin: pmap_prot2.o
Only in dietlibc-0.31_orig/bin: pmap_rmt.o
Only in dietlibc-0.31_orig/bin: poll.o
Only in dietlibc-0.31_orig/bin: poly.o
Only in dietlibc-0.31_orig/bin: popen.o
Only in dietlibc-0.31_orig/bin: pow.o
Only in dietlibc-0.31_orig/bin: prctl.o
Only in dietlibc-0.31_orig/bin: pread.o
Only in dietlibc-0.31_orig/bin: printf.o
Only in dietlibc-0.31_orig/bin: prioritynames.o
Only in dietlibc-0.31_orig/bin: profil.o
Only in dietlibc-0.31_orig/bin: protoent_buf.o
Only in dietlibc-0.31_orig/bin: pselect.o
Only in dietlibc-0.31_orig/bin: pstart.o
Only in dietlibc-0.31_orig/bin: pthread_atfork.o
Only in dietlibc-0.31_orig/bin: pthread_attr_getdetachstate.o
Only in dietlibc-0.31_orig/bin: pthread_attr_getinheritsched.o
Only in dietlibc-0.31_orig/bin: pthread_attr_getschedparam.o
Only in dietlibc-0.31_orig/bin: pthread_attr_getschedpolicy.o
Only in dietlibc-0.31_orig/bin: pthread_attr_getscope.o
Only in dietlibc-0.31_orig/bin: pthread_attr_getstackaddr.o
Only in dietlibc-0.31_orig/bin: pthread_attr_getstacksize.o
Only in dietlibc-0.31_orig/bin: pthread_attr_setdetachstate.o
Only in dietlibc-0.31_orig/bin: pthread_attr_setinheritsched.o
Only in dietlibc-0.31_orig/bin: pthread_attr_setschedparam.o
Only in dietlibc-0.31_orig/bin: pthread_attr_setschedpolicy.o
Only in dietlibc-0.31_orig/bin: pthread_attr_setscope.o
Only in dietlibc-0.31_orig/bin: pthread_attr_setstackaddr.o
Only in dietlibc-0.31_orig/bin: pthread_attr_setstacksize.o
Only in dietlibc-0.31_orig/bin: pthread_cleanup.o
Only in dietlibc-0.31_orig/bin: pthread_cond_broadcast.o
Only in dietlibc-0.31_orig/bin: pthread_cond_destroy.o
Only in dietlibc-0.31_orig/bin: pthread_cond_init.o
Only in dietlibc-0.31_orig/bin: pthread_cond_signal.o
Only in dietlibc-0.31_orig/bin: pthread_cond_timedwait.o
Only in dietlibc-0.31_orig/bin: pthread_cond_wait.o
Only in dietlibc-0.31_orig/bin: pthread_condattr_getshared.o
Only in dietlibc-0.31_orig/bin: pthread_condattr_setshared.o
Only in dietlibc-0.31_orig/bin: pthread_create.o
Only in dietlibc-0.31_orig/bin: pthread_detach.o
Only in dietlibc-0.31_orig/bin: pthread_dns_cruft.o
Only in dietlibc-0.31_orig/bin: pthread_equal.o
Only in dietlibc-0.31_orig/bin: pthread_errno.o
Only in dietlibc-0.31_orig/bin: pthread_fdglue2.o
Only in dietlibc-0.31_orig/bin: pthread_fflush.o
Only in dietlibc-0.31_orig/bin: pthread_fgetc.o
Only in dietlibc-0.31_orig/bin: pthread_flockfile.o
Only in dietlibc-0.31_orig/bin: pthread_fputc.o
Only in dietlibc-0.31_orig/bin: pthread_ftrylockfile.o
Only in dietlibc-0.31_orig/bin: pthread_funlockfile.o
Only in dietlibc-0.31_orig/bin: pthread_internal.o
Only in dietlibc-0.31_orig/bin: pthread_join.o
Only in dietlibc-0.31_orig/bin: pthread_key.o
Only in dietlibc-0.31_orig/bin: pthread_kill.o
Only in dietlibc-0.31_orig/bin: pthread_mutex_destroy.o
Only in dietlibc-0.31_orig/bin: pthread_mutex_init.o
Only in dietlibc-0.31_orig/bin: pthread_mutex_lock.o
Only in dietlibc-0.31_orig/bin: pthread_mutex_trylock.o
Only in dietlibc-0.31_orig/bin: pthread_mutex_unlock.o
Only in dietlibc-0.31_orig/bin: pthread_mutexattr_getkind_np.o
Only in dietlibc-0.31_orig/bin: pthread_mutexattr_init.o
Only in dietlibc-0.31_orig/bin: pthread_mutexattr_setkind_np.o
Only in dietlibc-0.31_orig/bin: pthread_self.o
Only in dietlibc-0.31_orig/bin: pthread_semaphore_destroy.o
Only in dietlibc-0.31_orig/bin: pthread_semaphore_getvalue.o
Only in dietlibc-0.31_orig/bin: pthread_semaphore_init.o
Only in dietlibc-0.31_orig/bin: pthread_semaphore_misc.o
Only in dietlibc-0.31_orig/bin: pthread_semaphore_post.o
Only in dietlibc-0.31_orig/bin: pthread_semaphore_trywait.o
Only in dietlibc-0.31_orig/bin: pthread_semaphore_wait.o
Only in dietlibc-0.31_orig/bin: pthread_setcancelstate.o
Only in dietlibc-0.31_orig/bin: pthread_setschedparam.o
Only in dietlibc-0.31_orig/bin: pthread_sigaction.o
Only in dietlibc-0.31_orig/bin: pthread_sigmask.o
Only in dietlibc-0.31_orig/bin: pthread_spinlock.o
Only in dietlibc-0.31_orig/bin: pthread_sys_alloc.o
Only in dietlibc-0.31_orig/bin: pthread_sys_close.o
Only in dietlibc-0.31_orig/bin: pthread_sys_create.o
Only in dietlibc-0.31_orig/bin: pthread_sys_fcntl.o
Only in dietlibc-0.31_orig/bin: pthread_sys_fdatasync.o
Only in dietlibc-0.31_orig/bin: pthread_sys_fsync.o
Only in dietlibc-0.31_orig/bin: pthread_sys_logging.o
Only in dietlibc-0.31_orig/bin: pthread_sys_msync.o
Only in dietlibc-0.31_orig/bin: pthread_sys_nanosleep.o
Only in dietlibc-0.31_orig/bin: pthread_sys_open.o
Only in dietlibc-0.31_orig/bin: pthread_sys_pause.o
Only in dietlibc-0.31_orig/bin: pthread_sys_read.o
Only in dietlibc-0.31_orig/bin: pthread_sys_sigsuspend.o
Only in dietlibc-0.31_orig/bin: pthread_sys_sleep.o
Only in dietlibc-0.31_orig/bin: pthread_sys_tcdrain.o
Only in dietlibc-0.31_orig/bin: pthread_sys_waitpid.o
Only in dietlibc-0.31_orig/bin: pthread_sys_write.o
Only in dietlibc-0.31_orig/bin: ptrace.o
Only in dietlibc-0.31_orig/bin: ptsname.o
Only in dietlibc-0.31_orig/bin: putchar.o
Only in dietlibc-0.31_orig/bin: putenv.o
Only in dietlibc-0.31_orig/bin: putpwent.o
Only in dietlibc-0.31_orig/bin: puts.o
Only in dietlibc-0.31_orig/bin: pwbuf.o
Only in dietlibc-0.31_orig/bin: pwrite.o
Only in dietlibc-0.31_orig/bin: qsort.o
Only in dietlibc-0.31_orig/bin: query_module.o
Only in dietlibc-0.31_orig/bin: quotactl.o
Only in dietlibc-0.31_orig/bin: raise.o
Only in dietlibc-0.31_orig/bin: rand.o
Only in dietlibc-0.31_orig/bin: rand48.o
Only in dietlibc-0.31_orig/bin: rand_r.o
Only in dietlibc-0.31_orig/bin: rcmd.o
Only in dietlibc-0.31_orig/bin: re_bsd.o
Only in dietlibc-0.31_orig/bin: read.o
Only in dietlibc-0.31_orig/bin: readahead.o
Only in dietlibc-0.31_orig/bin: readdir.o
Only in dietlibc-0.31_orig/bin: readdir64.o
Only in dietlibc-0.31_orig/bin: readlink.o
Only in dietlibc-0.31_orig/bin: readv.o
Only in dietlibc-0.31_orig/bin: realpath.o
Only in dietlibc-0.31_orig/bin: reboot.o
Only in dietlibc-0.31_orig/bin: recv.o
Only in dietlibc-0.31_orig/bin: recvfrom.o
Only in dietlibc-0.31_orig/bin: recvmsg.o
Only in dietlibc-0.31_orig/bin: remap_file_pages.o
Only in dietlibc-0.31_orig/bin: remove.o
Only in dietlibc-0.31_orig/bin: removexattr.o
Only in dietlibc-0.31_orig/bin: rename.o
Only in dietlibc-0.31_orig/bin: request_key.o
Only in dietlibc-0.31_orig/bin: res_close.o
Only in dietlibc-0.31_orig/bin: res_init.o
Only in dietlibc-0.31_orig/bin: res_mkquery.o
Only in dietlibc-0.31_orig/bin: res_query.o
Only in dietlibc-0.31_orig/bin: res_search.o
Only in dietlibc-0.31_orig/bin: rewind.o
Only in dietlibc-0.31_orig/bin: rewinddir.o
Only in dietlibc-0.31_orig/bin: rint.o
Only in dietlibc-0.31_orig/bin: rmdir.o
Only in dietlibc-0.31_orig/bin: rpc_callmsg.o
Only in dietlibc-0.31_orig/bin: rpc_commondata.o
Only in dietlibc-0.31_orig/bin: rpc_dtablesize.o
Only in dietlibc-0.31_orig/bin: rpc_prot.o
Only in dietlibc-0.31_orig/bin: rt_sigaction.o
Only in dietlibc-0.31_orig/bin: rt_sigpending.o
Only in dietlibc-0.31_orig/bin: rt_sigprocmask.o
Only in dietlibc-0.31_orig/bin: rt_sigqueueinfo.o
Only in dietlibc-0.31_orig/bin: rt_sigreturn.o
Only in dietlibc-0.31_orig/bin: rt_sigsuspend.o
Only in dietlibc-0.31_orig/bin: rt_sigtimedwait.o
Only in dietlibc-0.31_orig/bin: rx.o
Only in dietlibc-0.31_orig/bin: sbrk.o
Only in dietlibc-0.31_orig/bin: scan_ulong.o
Only in dietlibc-0.31_orig/bin: scandir.o
Only in dietlibc-0.31_orig/bin: scandir64.o
Only in dietlibc-0.31_orig/bin: scanf.o
Only in dietlibc-0.31_orig/bin: sched_get_priority_max.o
Only in dietlibc-0.31_orig/bin: sched_get_priority_min.o
Only in dietlibc-0.31_orig/bin: sched_getparam.o
Only in dietlibc-0.31_orig/bin: sched_getscheduler.o
Only in dietlibc-0.31_orig/bin: sched_rr_get_interval.o
Only in dietlibc-0.31_orig/bin: sched_setparam.o
Only in dietlibc-0.31_orig/bin: sched_setscheduler.o
Only in dietlibc-0.31_orig/bin: sched_yield.o
Only in dietlibc-0.31_orig/bin: seekdir.o
Only in dietlibc-0.31_orig/bin: select.o
Only in dietlibc-0.31_orig/bin: semctl.o
Only in dietlibc-0.31_orig/bin: semget.o
Only in dietlibc-0.31_orig/bin: semop.o
Only in dietlibc-0.31_orig/bin: send.o
Only in dietlibc-0.31_orig/bin: sendfile.o
Only in dietlibc-0.31_orig/bin: sendfile64.o
Only in dietlibc-0.31_orig/bin: sendmsg.o
Only in dietlibc-0.31_orig/bin: sendto.o
Only in dietlibc-0.31_orig/bin: servent_buf.o
Only in dietlibc-0.31_orig/bin: set_mempolicy.o
Only in dietlibc-0.31_orig/bin: set_thread_area.o
Only in dietlibc-0.31_orig/bin: set_tid_address.o
Only in dietlibc-0.31_orig/bin: setdomainname.o
Only in dietlibc-0.31_orig/bin: setegid.o
Only in dietlibc-0.31_orig/bin: setenv.o
Only in dietlibc-0.31_orig/bin: seteuid.o
Only in dietlibc-0.31_orig/bin: setfsgid.o
Only in dietlibc-0.31_orig/bin: setfsgid32.o
Only in dietlibc-0.31_orig/bin: setfsuid.o
Only in dietlibc-0.31_orig/bin: setfsuid32.o
Only in dietlibc-0.31_orig/bin: setgid.o
Only in dietlibc-0.31_orig/bin: setgid32.o
Only in dietlibc-0.31_orig/bin: setgroups.o
Only in dietlibc-0.31_orig/bin: sethostname.o
Only in dietlibc-0.31_orig/bin: setitimer.o
Only in dietlibc-0.31_orig/bin: setjmp.o
Only in dietlibc-0.31_orig/bin: setlinebuf.o
Only in dietlibc-0.31_orig/bin: setlocale.o
Only in dietlibc-0.31_orig/bin: setmntent.o
Only in dietlibc-0.31_orig/bin: setpgid.o
Only in dietlibc-0.31_orig/bin: setpgrp.o
Only in dietlibc-0.31_orig/bin: setpriority.o
Only in dietlibc-0.31_orig/bin: setregid.o
Only in dietlibc-0.31_orig/bin: setregid32.o
Only in dietlibc-0.31_orig/bin: setresgid.o
Only in dietlibc-0.31_orig/bin: setresgid32.o
Only in dietlibc-0.31_orig/bin: setresuid.o
Only in dietlibc-0.31_orig/bin: setreuid.o
Only in dietlibc-0.31_orig/bin: setreuid32.o
Only in dietlibc-0.31_orig/bin: setrlimit.o
Only in dietlibc-0.31_orig/bin: setsid.o
Only in dietlibc-0.31_orig/bin: setsockopt.o
Only in dietlibc-0.31_orig/bin: settimeofday.o
Only in dietlibc-0.31_orig/bin: setuid.o
Only in dietlibc-0.31_orig/bin: setuid32.o
Only in dietlibc-0.31_orig/bin: setvbuf.o
Only in dietlibc-0.31_orig/bin: setxattr.o
Only in dietlibc-0.31_orig/bin: shmat.o
Only in dietlibc-0.31_orig/bin: shmctl.o
Only in dietlibc-0.31_orig/bin: shmdt.o
Only in dietlibc-0.31_orig/bin: shmget.o
Only in dietlibc-0.31_orig/bin: shutdown.o
Only in dietlibc-0.31_orig/bin: sigaction.o
Only in dietlibc-0.31_orig/bin: sigaddset.o
Only in dietlibc-0.31_orig/bin: sigaltstack.o
Only in dietlibc-0.31_orig/bin: sigandset.o
Only in dietlibc-0.31_orig/bin: sigdelset.o
Only in dietlibc-0.31_orig/bin: sigemptyset.o
Only in dietlibc-0.31_orig/bin: sigfillset.o
Only in dietlibc-0.31_orig/bin: siginterrupt.o
Only in dietlibc-0.31_orig/bin: sigisemptyset.o
Only in dietlibc-0.31_orig/bin: sigismember.o
Only in dietlibc-0.31_orig/bin: sigjmp.o
Only in dietlibc-0.31_orig/bin: signal.o
Only in dietlibc-0.31_orig/bin: signalfd.o
Only in dietlibc-0.31_orig/bin: sigorset.o
Only in dietlibc-0.31_orig/bin: sigpending.o
Only in dietlibc-0.31_orig/bin: sigprocmask.o
Only in dietlibc-0.31_orig/bin: sigqueueinfo.o
Only in dietlibc-0.31_orig/bin: sigsuspend.o
Only in dietlibc-0.31_orig/bin: sigtimedwait.o
Only in dietlibc-0.31_orig/bin: sigwait.o
Only in dietlibc-0.31_orig/bin: sin.o
Only in dietlibc-0.31_orig/bin: sincos.o
Only in dietlibc-0.31_orig/bin: sinh.o
Only in dietlibc-0.31_orig/bin: sleep.o
Only in dietlibc-0.31_orig/bin: snprintf.o
Only in dietlibc-0.31_orig/bin: socket.o
Only in dietlibc-0.31_orig/bin: socketcall.o
Only in dietlibc-0.31_orig/bin: socketpair.o
Only in dietlibc-0.31_orig/bin: spbuf.o
Only in dietlibc-0.31_orig/bin: splice.o
Only in dietlibc-0.31_orig/bin: sprintf.o
Only in dietlibc-0.31_orig/bin: sqrt.o
Only in dietlibc-0.31_orig/bin: sscanf.o
Only in dietlibc-0.31_orig/bin: stack_smash_handler.o
Only in dietlibc-0.31_orig/bin: stack_smash_handler2.o
Only in dietlibc-0.31_orig/bin: stackgap.o
Only in dietlibc-0.31_orig/bin: start.o
Only in dietlibc-0.31_orig/bin: stat.o
Only in dietlibc-0.31_orig/bin: stat64.o
Only in dietlibc-0.31_orig/bin: statfs.o
Only in dietlibc-0.31_orig/bin: statfs64.o
Only in dietlibc-0.31_orig/bin: statfs64_cvt.o
Only in dietlibc-0.31_orig/bin: statvfs.o
Only in dietlibc-0.31_orig/bin: statvfs_cvt.o
Only in dietlibc-0.31_orig/bin: stderr.o
Only in dietlibc-0.31_orig/bin: stdin.o
Only in dietlibc-0.31_orig/bin: stdout.o
Only in dietlibc-0.31_orig/bin: stime.o
Only in dietlibc-0.31_orig/bin: stpcpy.o
Only in dietlibc-0.31_orig/bin: strcasecmp.o
Only in dietlibc-0.31_orig/bin: strcat.o
Only in dietlibc-0.31_orig/bin: strchr.o
Only in dietlibc-0.31_orig/bin: strcmp.o
Only in dietlibc-0.31_orig/bin: strcpy.o
Only in dietlibc-0.31_orig/bin: strcspn.o
Only in dietlibc-0.31_orig/bin: strdup.o
Only in dietlibc-0.31_orig/bin: strerror.o
Only in dietlibc-0.31_orig/bin: strerror_r.o
Only in dietlibc-0.31_orig/bin: strftime.o
Only in dietlibc-0.31_orig/bin: strlcat.o
Only in dietlibc-0.31_orig/bin: strlcpy.o
Only in dietlibc-0.31_orig/bin: strlen.o
Only in dietlibc-0.31_orig/bin: strncasecmp.o
Only in dietlibc-0.31_orig/bin: strncat.o
Only in dietlibc-0.31_orig/bin: strncmp.o
Only in dietlibc-0.31_orig/bin: strncpy.o
Only in dietlibc-0.31_orig/bin: strndup.o
Only in dietlibc-0.31_orig/bin: strnlen.o
Only in dietlibc-0.31_orig/bin: strpbrk.o
Only in dietlibc-0.31_orig/bin: strptime.o
Only in dietlibc-0.31_orig/bin: strrchr.o
Only in dietlibc-0.31_orig/bin: strsep.o
Only in dietlibc-0.31_orig/bin: strsignal.o
Only in dietlibc-0.31_orig/bin: strspn.o
Only in dietlibc-0.31_orig/bin: strstr.o
Only in dietlibc-0.31_orig/bin: strtod.o
Only in dietlibc-0.31_orig/bin: strtof.o
Only in dietlibc-0.31_orig/bin: strtok.o
Only in dietlibc-0.31_orig/bin: strtok_r.o
Only in dietlibc-0.31_orig/bin: strtol.o
Only in dietlibc-0.31_orig/bin: strtold.o
Only in dietlibc-0.31_orig/bin: strtoll.o
Only in dietlibc-0.31_orig/bin: strtoul.o
Only in dietlibc-0.31_orig/bin: strtoull.o
Only in dietlibc-0.31_orig/bin: strxfrm.o
Only in dietlibc-0.31_orig/bin: svc.o
Only in dietlibc-0.31_orig/bin: svc_auth.o
Only in dietlibc-0.31_orig/bin: svc_auth_unix.o
Only in dietlibc-0.31_orig/bin: svc_raw.o
Only in dietlibc-0.31_orig/bin: svc_run.o
Only in dietlibc-0.31_orig/bin: svc_simple.o
Only in dietlibc-0.31_orig/bin: svc_tcp.o
Only in dietlibc-0.31_orig/bin: svc_udp.o
Only in dietlibc-0.31_orig/bin: swab.o
Only in dietlibc-0.31_orig/bin: swapoff.o
Only in dietlibc-0.31_orig/bin: swapon.o
Only in dietlibc-0.31_orig/bin: symlink.o
Only in dietlibc-0.31_orig/bin: sync.o
Only in dietlibc-0.31_orig/bin: sys_siglist.o
Only in dietlibc-0.31_orig/bin: syscall.o
Only in dietlibc-0.31_orig/bin: sysconf.o
Only in dietlibc-0.31_orig/bin: sysconf_cpus.o
Only in dietlibc-0.31_orig/bin: sysctl.o
Only in dietlibc-0.31_orig/bin: sysfs.o
Only in dietlibc-0.31_orig/bin: sysinfo.o
Only in dietlibc-0.31_orig/bin: syslog.o
Only in dietlibc-0.31_orig/bin: system.o
Only in dietlibc-0.31_orig/bin: tan.o
Only in dietlibc-0.31_orig/bin: tanh.o
Only in dietlibc-0.31_orig/bin: tcdrain.o
Only in dietlibc-0.31_orig/bin: tcflow.o
Only in dietlibc-0.31_orig/bin: tcflush.o
Only in dietlibc-0.31_orig/bin: tcgetattr.o
Only in dietlibc-0.31_orig/bin: tcgetpgrp.o
Only in dietlibc-0.31_orig/bin: tcgetsid.o
Only in dietlibc-0.31_orig/bin: tcsendbreak.o
Only in dietlibc-0.31_orig/bin: tcsetattr.o
Only in dietlibc-0.31_orig/bin: tcsetpgrp.o
Only in dietlibc-0.31_orig/bin: tee.o
Only in dietlibc-0.31_orig/bin: telldir.o
Only in dietlibc-0.31_orig/bin: tempnam.o
Only in dietlibc-0.31_orig/bin: textdomain.o
Only in dietlibc-0.31_orig/bin: tgkill.o
Only in dietlibc-0.31_orig/bin: time.o
Only in dietlibc-0.31_orig/bin: time_table_spd.o
Only in dietlibc-0.31_orig/bin: timegm.o
Only in dietlibc-0.31_orig/bin: timer_create.o
Only in dietlibc-0.31_orig/bin: timer_delete.o
Only in dietlibc-0.31_orig/bin: timer_getoverrun.o
Only in dietlibc-0.31_orig/bin: timer_gettime.o
Only in dietlibc-0.31_orig/bin: timer_settime.o
Only in dietlibc-0.31_orig/bin: timerfd.o
Only in dietlibc-0.31_orig/bin: timerfd_create.o
Only in dietlibc-0.31_orig/bin: times.o
Only in dietlibc-0.31_orig/bin: timezone.o
Only in dietlibc-0.31_orig/bin: tkill.o
Only in dietlibc-0.31_orig/bin: tmpfile.o
Only in dietlibc-0.31_orig/bin: tmpnam.o
Only in dietlibc-0.31_orig/bin: toascii.o
Only in dietlibc-0.31_orig/bin: tolower.o
Only in dietlibc-0.31_orig/bin: toupper.o
Only in dietlibc-0.31_orig/bin: towlower.o
Only in dietlibc-0.31_orig/bin: towupper.o
Only in dietlibc-0.31_orig/bin: truncate.o
Only in dietlibc-0.31_orig/bin: truncate64.o
Only in dietlibc-0.31_orig/bin: ttyname.o
Only in dietlibc-0.31_orig/bin: tzfile.o
Only in dietlibc-0.31_orig/bin: umask.o
Only in dietlibc-0.31_orig/bin: umount.o
Only in dietlibc-0.31_orig/bin: umount2.o
Only in dietlibc-0.31_orig/bin: uname.o
Only in dietlibc-0.31_orig/bin: ungetc.o
Only in dietlibc-0.31_orig/bin: unified.o
Only in dietlibc-0.31_orig/bin: unlink.o
Only in dietlibc-0.31_orig/bin: unlockpt.o
Only in dietlibc-0.31_orig/bin: unsetenv.o
Only in dietlibc-0.31_orig/bin: usleep.o
Only in dietlibc-0.31_orig/bin: ustat.o
Only in dietlibc-0.31_orig/bin: utent.o
Only in dietlibc-0.31_orig/bin: utime.o
Only in dietlibc-0.31_orig/bin: utimes.o
Only in dietlibc-0.31_orig/bin: utmpxname.o
Only in dietlibc-0.31_orig/bin: utxent.o
Only in dietlibc-0.31_orig/bin: vasprintf.o
Only in dietlibc-0.31_orig/bin: verr.o
Only in dietlibc-0.31_orig/bin: verrx.o
Only in dietlibc-0.31_orig/bin: vfdprintf.o
Only in dietlibc-0.31_orig/bin: vfork.o
Only in dietlibc-0.31_orig/bin: vfprintf.o
Only in dietlibc-0.31_orig/bin: vfscanf.o
Only in dietlibc-0.31_orig/bin: vhangup.o
Only in dietlibc-0.31_orig/bin: vmsplice.o
Only in dietlibc-0.31_orig/bin: vprintf.o
Only in dietlibc-0.31_orig/bin: vscanf.o
Only in dietlibc-0.31_orig/bin: vserver.o
Only in dietlibc-0.31_orig/bin: vsnprintf.o
Only in dietlibc-0.31_orig/bin: vsprintf.o
Only in dietlibc-0.31_orig/bin: vsscanf.o
Only in dietlibc-0.31_orig/bin: vwarn.o
Only in dietlibc-0.31_orig/bin: vwarnx.o
Only in dietlibc-0.31_orig/bin: wait.o
Only in dietlibc-0.31_orig/bin: wait3.o
Only in dietlibc-0.31_orig/bin: wait4.o
Only in dietlibc-0.31_orig/bin: waitpid.o
Only in dietlibc-0.31_orig/bin: warn.o
Only in dietlibc-0.31_orig/bin: warnx.o
Only in dietlibc-0.31_orig/bin: wcrtomb.o
Only in dietlibc-0.31_orig/bin: wcscat.o
Only in dietlibc-0.31_orig/bin: wcschr.o
Only in dietlibc-0.31_orig/bin: wcscmp.o
Only in dietlibc-0.31_orig/bin: wcscpy.o
Only in dietlibc-0.31_orig/bin: wcslen.o
Only in dietlibc-0.31_orig/bin: wcsncat.o
Only in dietlibc-0.31_orig/bin: wcsncpy.o
Only in dietlibc-0.31_orig/bin: wcsrchr.o
Only in dietlibc-0.31_orig/bin: wcsstr.o
Only in dietlibc-0.31_orig/bin: wctomb.o
Only in dietlibc-0.31_orig/bin: wctype.o
Only in dietlibc-0.31_orig/bin: wcwidth.o
Only in dietlibc-0.31_orig/bin: wmemcmp.o
Only in dietlibc-0.31_orig/bin: wmemcpy.o
Only in dietlibc-0.31_orig/bin: wmemset.o
Only in dietlibc-0.31_orig/bin: write.o
Only in dietlibc-0.31_orig/bin: write12.o
Only in dietlibc-0.31_orig/bin: writev.o
Only in dietlibc-0.31_orig/bin: wtent.o
Only in dietlibc-0.31_orig/bin: wtxent.o
Only in dietlibc-0.31_orig/bin: xdr.o
Only in dietlibc-0.31_orig/bin: xdr_array.o
Only in dietlibc-0.31_orig/bin: xdr_float.o
Only in dietlibc-0.31_orig/bin: xdr_mem.o
Only in dietlibc-0.31_orig/bin: xdr_rec.o
Only in dietlibc-0.31_orig/bin: xdr_reference.o
Only in dietlibc-0.31_orig/bin: xdr_stdio.o
diff -ru dietlibc-0.31_orig/i386/dyn_syscalls.S dietlibc-0.31/i386/dyn_syscalls.S
--- dietlibc-0.31_orig/i386/dyn_syscalls.S	2008-05-26 12:56:07.000000000 +0200
+++ dietlibc-0.31/i386/dyn_syscalls.S	2008-05-26 13:24:01.000000000 +0200
@@ -139,6 +139,7 @@
 #include "../syscalls.s/__pread.S"
 #include "../syscalls.s/__pwrite.S"
 #include "../syscalls.s/__reboot.S"
+#include "../syscalls.s/__signalfd.S"
 #include "../syscalls.s/access.S"
 #include "../syscalls.s/adjtimex.S"
 #include "../syscalls.s/brk.S"
@@ -281,6 +282,8 @@
 #include "../syscalls.s/capset.S"
 #include "../syscalls.s/fcntl64.S"
 
+#include "../syscalls.s/timerfd.S"
+
 #include "../syscalls.s/timer_create.S"
 #include "../syscalls.s/timer_settime.S"
 #include "../syscalls.s/timer_gettime.S"
Only in dietlibc-0.31: la.s
diff -ru dietlibc-0.31_orig/lib/stack_smash_handler2.c dietlibc-0.31/lib/stack_smash_handler2.c
--- dietlibc-0.31_orig/lib/stack_smash_handler2.c	2008-05-26 12:56:07.000000000 +0200
+++ dietlibc-0.31/lib/stack_smash_handler2.c	2008-05-26 13:28:50.000000000 +0200
@@ -10,3 +10,8 @@
   __write2("smashed stack detected, program terminated.\n");
   _exit(127);
 }
+
+void __stack_chk_fail_local(void) {
+  __attribute__((weak,alias("__stack_chk_fail")));
+}
+
Binary files dietlibc-0.31_orig/pic-i386/__dtostr.o and dietlibc-0.31/pic-i386/__dtostr.o differ
Binary files dietlibc-0.31_orig/pic-i386/__end_parse.o and dietlibc-0.31/pic-i386/__end_parse.o differ
Binary files dietlibc-0.31_orig/pic-i386/__fcntl64.o and dietlibc-0.31/pic-i386/__fcntl64.o differ
Binary files dietlibc-0.31_orig/pic-i386/__finite.o and dietlibc-0.31/pic-i386/__finite.o differ
Binary files dietlibc-0.31_orig/pic-i386/__fprepare_parse.o and dietlibc-0.31/pic-i386/__fprepare_parse.o differ
Binary files dietlibc-0.31_orig/pic-i386/__fstat64.o and dietlibc-0.31/pic-i386/__fstat64.o differ
Binary files dietlibc-0.31_orig/pic-i386/__fstatfs64.o and dietlibc-0.31/pic-i386/__fstatfs64.o differ
Binary files dietlibc-0.31_orig/pic-i386/__ftruncate64.o and dietlibc-0.31/pic-i386/__ftruncate64.o differ
Binary files dietlibc-0.31_orig/pic-i386/__getcwd.o and dietlibc-0.31/pic-i386/__getcwd.o differ
Only in dietlibc-0.31_orig/pic-i386: __half.o
Binary files dietlibc-0.31_orig/pic-i386/__isinf.o and dietlibc-0.31/pic-i386/__isinf.o differ
Binary files dietlibc-0.31_orig/pic-i386/__lltostr.o and dietlibc-0.31/pic-i386/__lltostr.o differ
Binary files dietlibc-0.31_orig/pic-i386/__lstat64.o and dietlibc-0.31/pic-i386/__lstat64.o differ
Binary files dietlibc-0.31_orig/pic-i386/__ltostr.o and dietlibc-0.31/pic-i386/__ltostr.o differ
Binary files dietlibc-0.31_orig/pic-i386/__parse.o and dietlibc-0.31/pic-i386/__parse.o differ
Binary files dietlibc-0.31_orig/pic-i386/__parse_nws.o and dietlibc-0.31/pic-i386/__parse_nws.o differ
Binary files dietlibc-0.31_orig/pic-i386/__parse_ws.o and dietlibc-0.31/pic-i386/__parse_ws.o differ
Binary files dietlibc-0.31_orig/pic-i386/__prepare_parse.o and dietlibc-0.31/pic-i386/__prepare_parse.o differ
Binary files dietlibc-0.31_orig/pic-i386/__ptrace.o and dietlibc-0.31/pic-i386/__ptrace.o differ
Binary files dietlibc-0.31_orig/pic-i386/__restore_rt.o and dietlibc-0.31/pic-i386/__restore_rt.o differ
Binary files dietlibc-0.31_orig/pic-i386/__sendfile64.o and dietlibc-0.31/pic-i386/__sendfile64.o differ
Binary files dietlibc-0.31_orig/pic-i386/__stat64.o and dietlibc-0.31/pic-i386/__stat64.o differ
Binary files dietlibc-0.31_orig/pic-i386/__statfs64.o and dietlibc-0.31/pic-i386/__statfs64.o differ
Binary files dietlibc-0.31_orig/pic-i386/__sysctl.o and dietlibc-0.31/pic-i386/__sysctl.o differ
Only in dietlibc-0.31_orig/pic-i386: __testandset.o
Binary files dietlibc-0.31_orig/pic-i386/__truncate64.o and dietlibc-0.31/pic-i386/__truncate64.o differ
Binary files dietlibc-0.31_orig/pic-i386/__utmp_block_signals.o and dietlibc-0.31/pic-i386/__utmp_block_signals.o differ
Binary files dietlibc-0.31_orig/pic-i386/__utmp_io.o and dietlibc-0.31/pic-i386/__utmp_io.o differ
Binary files dietlibc-0.31_orig/pic-i386/__v_printf.o and dietlibc-0.31/pic-i386/__v_printf.o differ
Binary files dietlibc-0.31_orig/pic-i386/__v_scanf.o and dietlibc-0.31/pic-i386/__v_scanf.o differ
Binary files dietlibc-0.31_orig/pic-i386/_brk.o and dietlibc-0.31/pic-i386/_brk.o differ
Binary files dietlibc-0.31_orig/pic-i386/abort.o and dietlibc-0.31/pic-i386/abort.o differ
Only in dietlibc-0.31_orig/pic-i386: acos.o
Binary files dietlibc-0.31_orig/pic-i386/addmntent.o and dietlibc-0.31/pic-i386/addmntent.o differ
Binary files dietlibc-0.31_orig/pic-i386/adjtime.o and dietlibc-0.31/pic-i386/adjtime.o differ
Binary files dietlibc-0.31_orig/pic-i386/alloc.o and dietlibc-0.31/pic-i386/alloc.o differ
Binary files dietlibc-0.31_orig/pic-i386/alphasort.o and dietlibc-0.31/pic-i386/alphasort.o differ
Binary files dietlibc-0.31_orig/pic-i386/alphasort64.o and dietlibc-0.31/pic-i386/alphasort64.o differ
Binary files dietlibc-0.31_orig/pic-i386/asctime.o and dietlibc-0.31/pic-i386/asctime.o differ
Binary files dietlibc-0.31_orig/pic-i386/asctime_r.o and dietlibc-0.31/pic-i386/asctime_r.o differ
Only in dietlibc-0.31_orig/pic-i386: asin.o
Binary files dietlibc-0.31_orig/pic-i386/asprintf.o and dietlibc-0.31/pic-i386/asprintf.o differ
Binary files dietlibc-0.31_orig/pic-i386/assert_fail.o and dietlibc-0.31/pic-i386/assert_fail.o differ
Only in dietlibc-0.31_orig/pic-i386: atan.o
Only in dietlibc-0.31_orig/pic-i386: atan2.o
Binary files dietlibc-0.31_orig/pic-i386/atexit.o and dietlibc-0.31/pic-i386/atexit.o differ
Binary files dietlibc-0.31_orig/pic-i386/atof.o and dietlibc-0.31/pic-i386/atof.o differ
Binary files dietlibc-0.31_orig/pic-i386/atol.o and dietlibc-0.31/pic-i386/atol.o differ
Binary files dietlibc-0.31_orig/pic-i386/atoll.o and dietlibc-0.31/pic-i386/atoll.o differ
Binary files dietlibc-0.31_orig/pic-i386/basename.o and dietlibc-0.31/pic-i386/basename.o differ
Binary files dietlibc-0.31_orig/pic-i386/bcopy.o and dietlibc-0.31/pic-i386/bcopy.o differ
Only in dietlibc-0.31_orig/pic-i386: bessel.o
Binary files dietlibc-0.31_orig/pic-i386/bindtextdomain.o and dietlibc-0.31/pic-i386/bindtextdomain.o differ
Binary files dietlibc-0.31_orig/pic-i386/bsearch.o and dietlibc-0.31/pic-i386/bsearch.o differ
Only in dietlibc-0.31_orig/pic-i386: ceil.o
Binary files dietlibc-0.31_orig/pic-i386/cfsetispeed.o and dietlibc-0.31/pic-i386/cfsetispeed.o differ
Binary files dietlibc-0.31_orig/pic-i386/cfsetospeed.o and dietlibc-0.31/pic-i386/cfsetospeed.o differ
Only in dietlibc-0.31_orig/pic-i386: clearenv.o
Binary files dietlibc-0.31_orig/pic-i386/clock.o and dietlibc-0.31/pic-i386/clock.o differ
Binary files dietlibc-0.31_orig/pic-i386/closedir.o and dietlibc-0.31/pic-i386/closedir.o differ
Binary files dietlibc-0.31_orig/pic-i386/confstr.o and dietlibc-0.31/pic-i386/confstr.o differ
Only in dietlibc-0.31_orig/pic-i386: cos.o
Only in dietlibc-0.31_orig/pic-i386: cosh.o
Binary files dietlibc-0.31_orig/pic-i386/creat.o and dietlibc-0.31/pic-i386/creat.o differ
Binary files dietlibc-0.31_orig/pic-i386/creat64.o and dietlibc-0.31/pic-i386/creat64.o differ
Binary files dietlibc-0.31_orig/pic-i386/crypt.o and dietlibc-0.31/pic-i386/crypt.o differ
Only in dietlibc-0.31_orig/pic-i386: ctermid.o
Binary files dietlibc-0.31_orig/pic-i386/ctime.o and dietlibc-0.31/pic-i386/ctime.o differ
Binary files dietlibc-0.31_orig/pic-i386/ctime_r.o and dietlibc-0.31/pic-i386/ctime_r.o differ
Only in dietlibc-0.31_orig/pic-i386: cuserid.o
Only in dietlibc-0.31_orig/pic-i386: daemon.o
Binary files dietlibc-0.31_orig/pic-i386/dirname.o and dietlibc-0.31/pic-i386/dirname.o differ
Binary files dietlibc-0.31_orig/pic-i386/dl_iterate_phdr.o and dietlibc-0.31/pic-i386/dl_iterate_phdr.o differ
Binary files dietlibc-0.31_orig/pic-i386/dl_support.o and dietlibc-0.31/pic-i386/dl_support.o differ
Binary files dietlibc-0.31_orig/pic-i386/dn_expand.o and dietlibc-0.31/pic-i386/dn_expand.o differ
Binary files dietlibc-0.31_orig/pic-i386/dnscruft.o and dietlibc-0.31/pic-i386/dnscruft.o differ
Binary files dietlibc-0.31_orig/pic-i386/dnscruft2.o and dietlibc-0.31/pic-i386/dnscruft2.o differ
Binary files dietlibc-0.31_orig/pic-i386/dnscruft4.o and dietlibc-0.31/pic-i386/dnscruft4.o differ
Only in dietlibc-0.31_orig/pic-i386: dstart.o
Only in dietlibc-0.31_orig/pic-i386: dyn_so_start.o
Only in dietlibc-0.31_orig/pic-i386: dyn_start.o
Only in dietlibc-0.31_orig/pic-i386: dyn_stop.o
Binary files dietlibc-0.31_orig/pic-i386/dyn_syscalls.o and dietlibc-0.31/pic-i386/dyn_syscalls.o differ
Binary files dietlibc-0.31_orig/pic-i386/endmntent.o and dietlibc-0.31/pic-i386/endmntent.o differ
Only in dietlibc-0.31_orig/pic-i386: erf.o
Only in dietlibc-0.31_orig/pic-i386: err.o
Only in dietlibc-0.31_orig/pic-i386: errx.o
Binary files dietlibc-0.31_orig/pic-i386/exec_lib.o and dietlibc-0.31/pic-i386/exec_lib.o differ
Binary files dietlibc-0.31_orig/pic-i386/execl.o and dietlibc-0.31/pic-i386/execl.o differ
Binary files dietlibc-0.31_orig/pic-i386/execle.o and dietlibc-0.31/pic-i386/execle.o differ
Binary files dietlibc-0.31_orig/pic-i386/execlp.o and dietlibc-0.31/pic-i386/execlp.o differ
Binary files dietlibc-0.31_orig/pic-i386/execv.o and dietlibc-0.31/pic-i386/execv.o differ
Binary files dietlibc-0.31_orig/pic-i386/execvp.o and dietlibc-0.31/pic-i386/execvp.o differ
Only in dietlibc-0.31_orig/pic-i386: exp.o
Only in dietlibc-0.31_orig/pic-i386: fabs.o
Binary files dietlibc-0.31_orig/pic-i386/fclose.o and dietlibc-0.31/pic-i386/fclose.o differ
Binary files dietlibc-0.31_orig/pic-i386/fdglue2.o and dietlibc-0.31/pic-i386/fdglue2.o differ
Binary files dietlibc-0.31_orig/pic-i386/fdopen.o and dietlibc-0.31/pic-i386/fdopen.o differ
Binary files dietlibc-0.31_orig/pic-i386/fdprintf.o and dietlibc-0.31/pic-i386/fdprintf.o differ
Binary files dietlibc-0.31_orig/pic-i386/fflush.o and dietlibc-0.31/pic-i386/fflush.o differ
Binary files dietlibc-0.31_orig/pic-i386/ffs.o and dietlibc-0.31/pic-i386/ffs.o differ
Only in dietlibc-0.31_orig/pic-i386: ffsl.o
Only in dietlibc-0.31_orig/pic-i386: ffsll.o
Binary files dietlibc-0.31_orig/pic-i386/fgetc_unlocked.o and dietlibc-0.31/pic-i386/fgetc_unlocked.o differ
Binary files dietlibc-0.31_orig/pic-i386/fgetpos.o and dietlibc-0.31/pic-i386/fgetpos.o differ
Binary files dietlibc-0.31_orig/pic-i386/fgetpwent.o and dietlibc-0.31/pic-i386/fgetpwent.o differ
Binary files dietlibc-0.31_orig/pic-i386/fgetpwent_r.o and dietlibc-0.31/pic-i386/fgetpwent_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/fgets.o and dietlibc-0.31/pic-i386/fgets.o differ
Only in dietlibc-0.31_orig/pic-i386: floor.o
Only in dietlibc-0.31_orig/pic-i386: fmod.o
Binary files dietlibc-0.31_orig/pic-i386/fnmatch.o and dietlibc-0.31/pic-i386/fnmatch.o differ
Binary files dietlibc-0.31_orig/pic-i386/fopen.o and dietlibc-0.31/pic-i386/fopen.o differ
Binary files dietlibc-0.31_orig/pic-i386/fprintf.o and dietlibc-0.31/pic-i386/fprintf.o differ
Binary files dietlibc-0.31_orig/pic-i386/fputc_unlocked.o and dietlibc-0.31/pic-i386/fputc_unlocked.o differ
Binary files dietlibc-0.31_orig/pic-i386/fputs.o and dietlibc-0.31/pic-i386/fputs.o differ
Binary files dietlibc-0.31_orig/pic-i386/fread.o and dietlibc-0.31/pic-i386/fread.o differ
Binary files dietlibc-0.31_orig/pic-i386/freeaddrinfo.o and dietlibc-0.31/pic-i386/freeaddrinfo.o differ
Binary files dietlibc-0.31_orig/pic-i386/freopen.o and dietlibc-0.31/pic-i386/freopen.o differ
Binary files dietlibc-0.31_orig/pic-i386/fscanf.o and dietlibc-0.31/pic-i386/fscanf.o differ
Binary files dietlibc-0.31_orig/pic-i386/fseek.o and dietlibc-0.31/pic-i386/fseek.o differ
Binary files dietlibc-0.31_orig/pic-i386/fseeko.o and dietlibc-0.31/pic-i386/fseeko.o differ
Binary files dietlibc-0.31_orig/pic-i386/fseeko64.o and dietlibc-0.31/pic-i386/fseeko64.o differ
Binary files dietlibc-0.31_orig/pic-i386/fsetpos.o and dietlibc-0.31/pic-i386/fsetpos.o differ
Binary files dietlibc-0.31_orig/pic-i386/fstatvfs.o and dietlibc-0.31/pic-i386/fstatvfs.o differ
Binary files dietlibc-0.31_orig/pic-i386/ftell.o and dietlibc-0.31/pic-i386/ftell.o differ
Binary files dietlibc-0.31_orig/pic-i386/ftello.o and dietlibc-0.31/pic-i386/ftello.o differ
Binary files dietlibc-0.31_orig/pic-i386/ftello64.o and dietlibc-0.31/pic-i386/ftello64.o differ
Binary files dietlibc-0.31_orig/pic-i386/ftime.o and dietlibc-0.31/pic-i386/ftime.o differ
Only in dietlibc-0.31_orig/pic-i386: ftok.o
Binary files dietlibc-0.31_orig/pic-i386/ftw.o and dietlibc-0.31/pic-i386/ftw.o differ
Binary files dietlibc-0.31_orig/pic-i386/ftw64.o and dietlibc-0.31/pic-i386/ftw64.o differ
Binary files dietlibc-0.31_orig/pic-i386/fwrite.o and dietlibc-0.31/pic-i386/fwrite.o differ
Only in dietlibc-0.31_orig/pic-i386: gamma.o
Binary files dietlibc-0.31_orig/pic-i386/getaddrinfo.o and dietlibc-0.31/pic-i386/getaddrinfo.o differ
Only in dietlibc-0.31_orig/pic-i386: getdelim.o
Binary files dietlibc-0.31_orig/pic-i386/getdomainname.o and dietlibc-0.31/pic-i386/getdomainname.o differ
Binary files dietlibc-0.31_orig/pic-i386/getenv.o and dietlibc-0.31/pic-i386/getenv.o differ
Binary files dietlibc-0.31_orig/pic-i386/getgrent.o and dietlibc-0.31/pic-i386/getgrent.o differ
Binary files dietlibc-0.31_orig/pic-i386/getgrent_r.o and dietlibc-0.31/pic-i386/getgrent_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getgrgid.o and dietlibc-0.31/pic-i386/getgrgid.o differ
Binary files dietlibc-0.31_orig/pic-i386/getgrgid_r.o and dietlibc-0.31/pic-i386/getgrgid_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getgrnam.o and dietlibc-0.31/pic-i386/getgrnam.o differ
Binary files dietlibc-0.31_orig/pic-i386/getgrnam_r.o and dietlibc-0.31/pic-i386/getgrnam_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/gethostbyaddr.o and dietlibc-0.31/pic-i386/gethostbyaddr.o differ
Binary files dietlibc-0.31_orig/pic-i386/gethostbyaddr_r.o and dietlibc-0.31/pic-i386/gethostbyaddr_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/gethostbyname.o and dietlibc-0.31/pic-i386/gethostbyname.o differ
Binary files dietlibc-0.31_orig/pic-i386/gethostbyname2.o and dietlibc-0.31/pic-i386/gethostbyname2.o differ
Binary files dietlibc-0.31_orig/pic-i386/gethostbyname2_r.o and dietlibc-0.31/pic-i386/gethostbyname2_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/gethostbyname_r.o and dietlibc-0.31/pic-i386/gethostbyname_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/gethostent.o and dietlibc-0.31/pic-i386/gethostent.o differ
Binary files dietlibc-0.31_orig/pic-i386/gethostent_nor.o and dietlibc-0.31/pic-i386/gethostent_nor.o differ
Binary files dietlibc-0.31_orig/pic-i386/gethostname.o and dietlibc-0.31/pic-i386/gethostname.o differ
Only in dietlibc-0.31_orig/pic-i386: getline.o
Binary files dietlibc-0.31_orig/pic-i386/getlogin.o and dietlibc-0.31/pic-i386/getlogin.o differ
Binary files dietlibc-0.31_orig/pic-i386/getmntent.o and dietlibc-0.31/pic-i386/getmntent.o differ
Binary files dietlibc-0.31_orig/pic-i386/getnameinfo.o and dietlibc-0.31/pic-i386/getnameinfo.o differ
Binary files dietlibc-0.31_orig/pic-i386/getopt.o and dietlibc-0.31/pic-i386/getopt.o differ
Binary files dietlibc-0.31_orig/pic-i386/getopt_long.o and dietlibc-0.31/pic-i386/getopt_long.o differ
Binary files dietlibc-0.31_orig/pic-i386/getopt_long_only.o and dietlibc-0.31/pic-i386/getopt_long_only.o differ
Binary files dietlibc-0.31_orig/pic-i386/getpass.o and dietlibc-0.31/pic-i386/getpass.o differ
Binary files dietlibc-0.31_orig/pic-i386/getpgrp.o and dietlibc-0.31/pic-i386/getpgrp.o differ
Binary files dietlibc-0.31_orig/pic-i386/getprotobyname.o and dietlibc-0.31/pic-i386/getprotobyname.o differ
Binary files dietlibc-0.31_orig/pic-i386/getprotobyname_r.o and dietlibc-0.31/pic-i386/getprotobyname_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getprotobynumber.o and dietlibc-0.31/pic-i386/getprotobynumber.o differ
Binary files dietlibc-0.31_orig/pic-i386/getprotobynumber_r.o and dietlibc-0.31/pic-i386/getprotobynumber_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getprotoent.o and dietlibc-0.31/pic-i386/getprotoent.o differ
Binary files dietlibc-0.31_orig/pic-i386/getprotoent_r.o and dietlibc-0.31/pic-i386/getprotoent_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getpwent.o and dietlibc-0.31/pic-i386/getpwent.o differ
Binary files dietlibc-0.31_orig/pic-i386/getpwent_r.o and dietlibc-0.31/pic-i386/getpwent_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getpwnam.o and dietlibc-0.31/pic-i386/getpwnam.o differ
Binary files dietlibc-0.31_orig/pic-i386/getpwnam_r.o and dietlibc-0.31/pic-i386/getpwnam_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getpwuid.o and dietlibc-0.31/pic-i386/getpwuid.o differ
Binary files dietlibc-0.31_orig/pic-i386/getpwuid_r.o and dietlibc-0.31/pic-i386/getpwuid_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getservbyname.o and dietlibc-0.31/pic-i386/getservbyname.o differ
Binary files dietlibc-0.31_orig/pic-i386/getservbyname_r.o and dietlibc-0.31/pic-i386/getservbyname_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getservbyport.o and dietlibc-0.31/pic-i386/getservbyport.o differ
Binary files dietlibc-0.31_orig/pic-i386/getservbyport_r.o and dietlibc-0.31/pic-i386/getservbyport_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getservent.o and dietlibc-0.31/pic-i386/getservent.o differ
Binary files dietlibc-0.31_orig/pic-i386/getservent_r.o and dietlibc-0.31/pic-i386/getservent_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getspent.o and dietlibc-0.31/pic-i386/getspent.o differ
Binary files dietlibc-0.31_orig/pic-i386/getspent_r.o and dietlibc-0.31/pic-i386/getspent_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getspnam.o and dietlibc-0.31/pic-i386/getspnam.o differ
Binary files dietlibc-0.31_orig/pic-i386/getspnam_r.o and dietlibc-0.31/pic-i386/getspnam_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/getusershell.o and dietlibc-0.31/pic-i386/getusershell.o differ
Only in dietlibc-0.31_orig/pic-i386: getutmp.o
Binary files dietlibc-0.31_orig/pic-i386/glob.o and dietlibc-0.31/pic-i386/glob.o differ
Binary files dietlibc-0.31_orig/pic-i386/gmtime.o and dietlibc-0.31/pic-i386/gmtime.o differ
Binary files dietlibc-0.31_orig/pic-i386/gmtime_r.o and dietlibc-0.31/pic-i386/gmtime_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/grantpt.o and dietlibc-0.31/pic-i386/grantpt.o differ
Binary files dietlibc-0.31_orig/pic-i386/hasmntopt.o and dietlibc-0.31/pic-i386/hasmntopt.o differ
Binary files dietlibc-0.31_orig/pic-i386/herror.o and dietlibc-0.31/pic-i386/herror.o differ
Binary files dietlibc-0.31_orig/pic-i386/htonl.o and dietlibc-0.31/pic-i386/htonl.o differ
Binary files dietlibc-0.31_orig/pic-i386/htons.o and dietlibc-0.31/pic-i386/htons.o differ
Only in dietlibc-0.31_orig/pic-i386: hypot.o
Binary files dietlibc-0.31_orig/pic-i386/iconv.o and dietlibc-0.31/pic-i386/iconv.o differ
Binary files dietlibc-0.31_orig/pic-i386/iconv_open.o and dietlibc-0.31/pic-i386/iconv_open.o differ
Binary files dietlibc-0.31_orig/pic-i386/if_indextoname.o and dietlibc-0.31/pic-i386/if_indextoname.o differ
Binary files dietlibc-0.31_orig/pic-i386/if_nameindex.o and dietlibc-0.31/pic-i386/if_nameindex.o differ
Binary files dietlibc-0.31_orig/pic-i386/if_nametoindex.o and dietlibc-0.31/pic-i386/if_nametoindex.o differ
Binary files dietlibc-0.31_orig/pic-i386/inet_addr.o and dietlibc-0.31/pic-i386/inet_addr.o differ
Binary files dietlibc-0.31_orig/pic-i386/inet_aton.o and dietlibc-0.31/pic-i386/inet_aton.o differ
Binary files dietlibc-0.31_orig/pic-i386/inet_ntoa.o and dietlibc-0.31/pic-i386/inet_ntoa.o differ
Binary files dietlibc-0.31_orig/pic-i386/inet_ntoa_r.o and dietlibc-0.31/pic-i386/inet_ntoa_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/inet_ntop.o and dietlibc-0.31/pic-i386/inet_ntop.o differ
Binary files dietlibc-0.31_orig/pic-i386/inet_pton.o and dietlibc-0.31/pic-i386/inet_pton.o differ
Binary files dietlibc-0.31_orig/pic-i386/initgroups.o and dietlibc-0.31/pic-i386/initgroups.o differ
Only in dietlibc-0.31_orig/pic-i386: ipow.o
Binary files dietlibc-0.31_orig/pic-i386/isatty.o and dietlibc-0.31/pic-i386/isatty.o differ
Binary files dietlibc-0.31_orig/pic-i386/isleap.o and dietlibc-0.31/pic-i386/isleap.o differ
Binary files dietlibc-0.31_orig/pic-i386/ispunct.o and dietlibc-0.31/pic-i386/ispunct.o differ
Binary files dietlibc-0.31_orig/pic-i386/iswalnum.o and dietlibc-0.31/pic-i386/iswalnum.o differ
Binary files dietlibc-0.31_orig/pic-i386/iswalpha.o and dietlibc-0.31/pic-i386/iswalpha.o differ
Binary files dietlibc-0.31_orig/pic-i386/iswctype.o and dietlibc-0.31/pic-i386/iswctype.o differ
Binary files dietlibc-0.31_orig/pic-i386/iswpunct.o and dietlibc-0.31/pic-i386/iswpunct.o differ
Binary files dietlibc-0.31_orig/pic-i386/killpg.o and dietlibc-0.31/pic-i386/killpg.o differ
Only in dietlibc-0.31_orig/pic-i386: ldexp.o
Binary files dietlibc-0.31_orig/pic-i386/libc.so and dietlibc-0.31/pic-i386/libc.so differ
Only in dietlibc-0.31_orig/pic-i386: libm2.o
Binary files dietlibc-0.31_orig/pic-i386/lldiv.o and dietlibc-0.31/pic-i386/lldiv.o differ
Binary files dietlibc-0.31_orig/pic-i386/localtime.o and dietlibc-0.31/pic-i386/localtime.o differ
Binary files dietlibc-0.31_orig/pic-i386/localtime_r.o and dietlibc-0.31/pic-i386/localtime_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/lockf.o and dietlibc-0.31/pic-i386/lockf.o differ
Only in dietlibc-0.31_orig/pic-i386: log.o
Only in dietlibc-0.31_orig/pic-i386: log10.o
Only in dietlibc-0.31_orig/pic-i386: log1p.o
Binary files dietlibc-0.31_orig/pic-i386/logging.o and dietlibc-0.31/pic-i386/logging.o differ
Binary files dietlibc-0.31_orig/pic-i386/longjmp.o and dietlibc-0.31/pic-i386/longjmp.o differ
Binary files dietlibc-0.31_orig/pic-i386/lseek64.o and dietlibc-0.31/pic-i386/lseek64.o differ
Binary files dietlibc-0.31_orig/pic-i386/mblen.o and dietlibc-0.31/pic-i386/mblen.o differ
Binary files dietlibc-0.31_orig/pic-i386/mbsrtowcs.o and dietlibc-0.31/pic-i386/mbsrtowcs.o differ
Binary files dietlibc-0.31_orig/pic-i386/mbstowcs.o and dietlibc-0.31/pic-i386/mbstowcs.o differ
Binary files dietlibc-0.31_orig/pic-i386/mbtowc.o and dietlibc-0.31/pic-i386/mbtowc.o differ
Binary files dietlibc-0.31_orig/pic-i386/md5.o and dietlibc-0.31/pic-i386/md5.o differ
Binary files dietlibc-0.31_orig/pic-i386/md5asm.o and dietlibc-0.31/pic-i386/md5asm.o differ
Binary files dietlibc-0.31_orig/pic-i386/md5crypt.o and dietlibc-0.31/pic-i386/md5crypt.o differ
Binary files dietlibc-0.31_orig/pic-i386/memccpy.o and dietlibc-0.31/pic-i386/memccpy.o differ
Binary files dietlibc-0.31_orig/pic-i386/memchr.o and dietlibc-0.31/pic-i386/memchr.o differ
Binary files dietlibc-0.31_orig/pic-i386/memcmp.o and dietlibc-0.31/pic-i386/memcmp.o differ
Binary files dietlibc-0.31_orig/pic-i386/memcpy.o and dietlibc-0.31/pic-i386/memcpy.o differ
Binary files dietlibc-0.31_orig/pic-i386/memmem.o and dietlibc-0.31/pic-i386/memmem.o differ
Binary files dietlibc-0.31_orig/pic-i386/memset.o and dietlibc-0.31/pic-i386/memset.o differ
Binary files dietlibc-0.31_orig/pic-i386/mkdtemp.o and dietlibc-0.31/pic-i386/mkdtemp.o differ
Binary files dietlibc-0.31_orig/pic-i386/mkfifo.o and dietlibc-0.31/pic-i386/mkfifo.o differ
Binary files dietlibc-0.31_orig/pic-i386/mkstemp.o and dietlibc-0.31/pic-i386/mkstemp.o differ
Binary files dietlibc-0.31_orig/pic-i386/mktemp.o and dietlibc-0.31/pic-i386/mktemp.o differ
Binary files dietlibc-0.31_orig/pic-i386/mktime.o and dietlibc-0.31/pic-i386/mktime.o differ
Binary files dietlibc-0.31_orig/pic-i386/mmap64.o and dietlibc-0.31/pic-i386/mmap64.o differ
Only in dietlibc-0.31_orig/pic-i386: modf.o
Binary files dietlibc-0.31_orig/pic-i386/mq_getattr.o and dietlibc-0.31/pic-i386/mq_getattr.o differ
Binary files dietlibc-0.31_orig/pic-i386/mq_receive.o and dietlibc-0.31/pic-i386/mq_receive.o differ
Binary files dietlibc-0.31_orig/pic-i386/mq_send.o and dietlibc-0.31/pic-i386/mq_send.o differ
Binary files dietlibc-0.31_orig/pic-i386/msgctl.o and dietlibc-0.31/pic-i386/msgctl.o differ
Binary files dietlibc-0.31_orig/pic-i386/msgget.o and dietlibc-0.31/pic-i386/msgget.o differ
Binary files dietlibc-0.31_orig/pic-i386/msgrcv.o and dietlibc-0.31/pic-i386/msgrcv.o differ
Binary files dietlibc-0.31_orig/pic-i386/msgsnd.o and dietlibc-0.31/pic-i386/msgsnd.o differ
Binary files dietlibc-0.31_orig/pic-i386/netent.o and dietlibc-0.31/pic-i386/netent.o differ
Binary files dietlibc-0.31_orig/pic-i386/nl_langinfo.o and dietlibc-0.31/pic-i386/nl_langinfo.o differ
Binary files dietlibc-0.31_orig/pic-i386/open64.o and dietlibc-0.31/pic-i386/open64.o differ
Binary files dietlibc-0.31_orig/pic-i386/opendir.o and dietlibc-0.31/pic-i386/opendir.o differ
Binary files dietlibc-0.31_orig/pic-i386/openpty.o and dietlibc-0.31/pic-i386/openpty.o differ
Binary files dietlibc-0.31_orig/pic-i386/pclose.o and dietlibc-0.31/pic-i386/pclose.o differ
Binary files dietlibc-0.31_orig/pic-i386/perror.o and dietlibc-0.31/pic-i386/perror.o differ
Only in dietlibc-0.31_orig/pic-i386: poly.o
Binary files dietlibc-0.31_orig/pic-i386/popen.o and dietlibc-0.31/pic-i386/popen.o differ
Only in dietlibc-0.31_orig/pic-i386: pow.o
Binary files dietlibc-0.31_orig/pic-i386/pread.o and dietlibc-0.31/pic-i386/pread.o differ
Binary files dietlibc-0.31_orig/pic-i386/printf.o and dietlibc-0.31/pic-i386/printf.o differ
Binary files dietlibc-0.31_orig/pic-i386/pselect.o and dietlibc-0.31/pic-i386/pselect.o differ
Only in dietlibc-0.31_orig/pic-i386: pthread_atfork.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_getdetachstate.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_getinheritsched.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_getschedparam.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_getschedpolicy.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_getscope.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_getstackaddr.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_getstacksize.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_setdetachstate.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_setinheritsched.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_setschedparam.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_setschedpolicy.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_setscope.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_setstackaddr.o
Only in dietlibc-0.31_orig/pic-i386: pthread_attr_setstacksize.o
Only in dietlibc-0.31_orig/pic-i386: pthread_cleanup.o
Only in dietlibc-0.31_orig/pic-i386: pthread_cond_broadcast.o
Only in dietlibc-0.31_orig/pic-i386: pthread_cond_destroy.o
Only in dietlibc-0.31_orig/pic-i386: pthread_cond_init.o
Only in dietlibc-0.31_orig/pic-i386: pthread_cond_signal.o
Only in dietlibc-0.31_orig/pic-i386: pthread_cond_timedwait.o
Only in dietlibc-0.31_orig/pic-i386: pthread_cond_wait.o
Only in dietlibc-0.31_orig/pic-i386: pthread_condattr_getshared.o
Only in dietlibc-0.31_orig/pic-i386: pthread_condattr_setshared.o
Only in dietlibc-0.31_orig/pic-i386: pthread_create.o
Only in dietlibc-0.31_orig/pic-i386: pthread_detach.o
Only in dietlibc-0.31_orig/pic-i386: pthread_dns_cruft.o
Only in dietlibc-0.31_orig/pic-i386: pthread_equal.o
Only in dietlibc-0.31_orig/pic-i386: pthread_errno.o
Only in dietlibc-0.31_orig/pic-i386: pthread_fdglue2.o
Only in dietlibc-0.31_orig/pic-i386: pthread_fflush.o
Only in dietlibc-0.31_orig/pic-i386: pthread_fgetc.o
Only in dietlibc-0.31_orig/pic-i386: pthread_flockfile.o
Only in dietlibc-0.31_orig/pic-i386: pthread_fputc.o
Only in dietlibc-0.31_orig/pic-i386: pthread_ftrylockfile.o
Only in dietlibc-0.31_orig/pic-i386: pthread_funlockfile.o
Only in dietlibc-0.31_orig/pic-i386: pthread_internal.o
Only in dietlibc-0.31_orig/pic-i386: pthread_join.o
Only in dietlibc-0.31_orig/pic-i386: pthread_key.o
Only in dietlibc-0.31_orig/pic-i386: pthread_kill.o
Only in dietlibc-0.31_orig/pic-i386: pthread_mutex_destroy.o
Only in dietlibc-0.31_orig/pic-i386: pthread_mutex_init.o
Only in dietlibc-0.31_orig/pic-i386: pthread_mutex_lock.o
Only in dietlibc-0.31_orig/pic-i386: pthread_mutex_trylock.o
Only in dietlibc-0.31_orig/pic-i386: pthread_mutex_unlock.o
Only in dietlibc-0.31_orig/pic-i386: pthread_mutexattr_getkind_np.o
Only in dietlibc-0.31_orig/pic-i386: pthread_mutexattr_init.o
Only in dietlibc-0.31_orig/pic-i386: pthread_mutexattr_setkind_np.o
Only in dietlibc-0.31_orig/pic-i386: pthread_self.o
Only in dietlibc-0.31_orig/pic-i386: pthread_semaphore_destroy.o
Only in dietlibc-0.31_orig/pic-i386: pthread_semaphore_getvalue.o
Only in dietlibc-0.31_orig/pic-i386: pthread_semaphore_init.o
Only in dietlibc-0.31_orig/pic-i386: pthread_semaphore_misc.o
Only in dietlibc-0.31_orig/pic-i386: pthread_semaphore_post.o
Only in dietlibc-0.31_orig/pic-i386: pthread_semaphore_trywait.o
Only in dietlibc-0.31_orig/pic-i386: pthread_semaphore_wait.o
Only in dietlibc-0.31_orig/pic-i386: pthread_setcancelstate.o
Only in dietlibc-0.31_orig/pic-i386: pthread_setschedparam.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sigaction.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sigmask.o
Only in dietlibc-0.31_orig/pic-i386: pthread_spinlock.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_alloc.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_close.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_create.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_fcntl.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_fdatasync.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_fsync.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_logging.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_msync.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_nanosleep.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_open.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_pause.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_read.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_sigsuspend.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_sleep.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_tcdrain.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_waitpid.o
Only in dietlibc-0.31_orig/pic-i386: pthread_sys_write.o
Binary files dietlibc-0.31_orig/pic-i386/ptsname.o and dietlibc-0.31/pic-i386/ptsname.o differ
Binary files dietlibc-0.31_orig/pic-i386/putchar.o and dietlibc-0.31/pic-i386/putchar.o differ
Binary files dietlibc-0.31_orig/pic-i386/putenv.o and dietlibc-0.31/pic-i386/putenv.o differ
Binary files dietlibc-0.31_orig/pic-i386/putpwent.o and dietlibc-0.31/pic-i386/putpwent.o differ
Binary files dietlibc-0.31_orig/pic-i386/puts.o and dietlibc-0.31/pic-i386/puts.o differ
Binary files dietlibc-0.31_orig/pic-i386/pwrite.o and dietlibc-0.31/pic-i386/pwrite.o differ
Binary files dietlibc-0.31_orig/pic-i386/qsort.o and dietlibc-0.31/pic-i386/qsort.o differ
Binary files dietlibc-0.31_orig/pic-i386/raise.o and dietlibc-0.31/pic-i386/raise.o differ
Binary files dietlibc-0.31_orig/pic-i386/rand.o and dietlibc-0.31/pic-i386/rand.o differ
Binary files dietlibc-0.31_orig/pic-i386/rand48.o and dietlibc-0.31/pic-i386/rand48.o differ
Only in dietlibc-0.31_orig/pic-i386: rcmd.o
Only in dietlibc-0.31_orig/pic-i386: re_bsd.o
Binary files dietlibc-0.31_orig/pic-i386/readdir.o and dietlibc-0.31/pic-i386/readdir.o differ
Binary files dietlibc-0.31_orig/pic-i386/readdir64.o and dietlibc-0.31/pic-i386/readdir64.o differ
Binary files dietlibc-0.31_orig/pic-i386/realpath.o and dietlibc-0.31/pic-i386/realpath.o differ
Binary files dietlibc-0.31_orig/pic-i386/reboot.o and dietlibc-0.31/pic-i386/reboot.o differ
Binary files dietlibc-0.31_orig/pic-i386/remove.o and dietlibc-0.31/pic-i386/remove.o differ
Binary files dietlibc-0.31_orig/pic-i386/res_close.o and dietlibc-0.31/pic-i386/res_close.o differ
Binary files dietlibc-0.31_orig/pic-i386/res_init.o and dietlibc-0.31/pic-i386/res_init.o differ
Binary files dietlibc-0.31_orig/pic-i386/res_mkquery.o and dietlibc-0.31/pic-i386/res_mkquery.o differ
Binary files dietlibc-0.31_orig/pic-i386/res_query.o and dietlibc-0.31/pic-i386/res_query.o differ
Binary files dietlibc-0.31_orig/pic-i386/res_search.o and dietlibc-0.31/pic-i386/res_search.o differ
Binary files dietlibc-0.31_orig/pic-i386/rewind.o and dietlibc-0.31/pic-i386/rewind.o differ
Binary files dietlibc-0.31_orig/pic-i386/rewinddir.o and dietlibc-0.31/pic-i386/rewinddir.o differ
Only in dietlibc-0.31_orig/pic-i386: rint.o
Binary files dietlibc-0.31_orig/pic-i386/rx.o and dietlibc-0.31/pic-i386/rx.o differ
Binary files dietlibc-0.31_orig/pic-i386/sbrk.o and dietlibc-0.31/pic-i386/sbrk.o differ
Binary files dietlibc-0.31_orig/pic-i386/scandir.o and dietlibc-0.31/pic-i386/scandir.o differ
Binary files dietlibc-0.31_orig/pic-i386/scandir64.o and dietlibc-0.31/pic-i386/scandir64.o differ
Binary files dietlibc-0.31_orig/pic-i386/scanf.o and dietlibc-0.31/pic-i386/scanf.o differ
Binary files dietlibc-0.31_orig/pic-i386/seekdir.o and dietlibc-0.31/pic-i386/seekdir.o differ
Binary files dietlibc-0.31_orig/pic-i386/semctl.o and dietlibc-0.31/pic-i386/semctl.o differ
Binary files dietlibc-0.31_orig/pic-i386/semget.o and dietlibc-0.31/pic-i386/semget.o differ
Binary files dietlibc-0.31_orig/pic-i386/semop.o and dietlibc-0.31/pic-i386/semop.o differ
Binary files dietlibc-0.31_orig/pic-i386/setegid.o and dietlibc-0.31/pic-i386/setegid.o differ
Binary files dietlibc-0.31_orig/pic-i386/setenv.o and dietlibc-0.31/pic-i386/setenv.o differ
Binary files dietlibc-0.31_orig/pic-i386/seteuid.o and dietlibc-0.31/pic-i386/seteuid.o differ
Binary files dietlibc-0.31_orig/pic-i386/setlinebuf.o and dietlibc-0.31/pic-i386/setlinebuf.o differ
Binary files dietlibc-0.31_orig/pic-i386/setlocale.o and dietlibc-0.31/pic-i386/setlocale.o differ
Binary files dietlibc-0.31_orig/pic-i386/setmntent.o and dietlibc-0.31/pic-i386/setmntent.o differ
Binary files dietlibc-0.31_orig/pic-i386/setpgrp.o and dietlibc-0.31/pic-i386/setpgrp.o differ
Binary files dietlibc-0.31_orig/pic-i386/setvbuf.o and dietlibc-0.31/pic-i386/setvbuf.o differ
Binary files dietlibc-0.31_orig/pic-i386/shmat.o and dietlibc-0.31/pic-i386/shmat.o differ
Binary files dietlibc-0.31_orig/pic-i386/shmctl.o and dietlibc-0.31/pic-i386/shmctl.o differ
Binary files dietlibc-0.31_orig/pic-i386/shmdt.o and dietlibc-0.31/pic-i386/shmdt.o differ
Binary files dietlibc-0.31_orig/pic-i386/shmget.o and dietlibc-0.31/pic-i386/shmget.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigaction.o and dietlibc-0.31/pic-i386/sigaction.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigaddset.o and dietlibc-0.31/pic-i386/sigaddset.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigdelset.o and dietlibc-0.31/pic-i386/sigdelset.o differ
Binary files dietlibc-0.31_orig/pic-i386/siginterrupt.o and dietlibc-0.31/pic-i386/siginterrupt.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigismember.o and dietlibc-0.31/pic-i386/sigismember.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigjmp.o and dietlibc-0.31/pic-i386/sigjmp.o differ
Binary files dietlibc-0.31_orig/pic-i386/signal.o and dietlibc-0.31/pic-i386/signal.o differ
Binary files dietlibc-0.31_orig/pic-i386/signalfd.o and dietlibc-0.31/pic-i386/signalfd.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigpending.o and dietlibc-0.31/pic-i386/sigpending.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigprocmask.o and dietlibc-0.31/pic-i386/sigprocmask.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigqueueinfo.o and dietlibc-0.31/pic-i386/sigqueueinfo.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigsuspend.o and dietlibc-0.31/pic-i386/sigsuspend.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigtimedwait.o and dietlibc-0.31/pic-i386/sigtimedwait.o differ
Binary files dietlibc-0.31_orig/pic-i386/sigwait.o and dietlibc-0.31/pic-i386/sigwait.o differ
Only in dietlibc-0.31_orig/pic-i386: sin.o
Only in dietlibc-0.31_orig/pic-i386: sincos.o
Only in dietlibc-0.31_orig/pic-i386: sinh.o
Binary files dietlibc-0.31_orig/pic-i386/sleep.o and dietlibc-0.31/pic-i386/sleep.o differ
Binary files dietlibc-0.31_orig/pic-i386/snprintf.o and dietlibc-0.31/pic-i386/snprintf.o differ
Binary files dietlibc-0.31_orig/pic-i386/sprintf.o and dietlibc-0.31/pic-i386/sprintf.o differ
Only in dietlibc-0.31_orig/pic-i386: sqrt.o
Binary files dietlibc-0.31_orig/pic-i386/sscanf.o and dietlibc-0.31/pic-i386/sscanf.o differ
Binary files dietlibc-0.31_orig/pic-i386/stack_smash_handler.o and dietlibc-0.31/pic-i386/stack_smash_handler.o differ
Binary files dietlibc-0.31_orig/pic-i386/stack_smash_handler2.o and dietlibc-0.31/pic-i386/stack_smash_handler2.o differ
Binary files dietlibc-0.31_orig/pic-i386/stackgap.o and dietlibc-0.31/pic-i386/stackgap.o differ
Binary files dietlibc-0.31_orig/pic-i386/statvfs.o and dietlibc-0.31/pic-i386/statvfs.o differ
Binary files dietlibc-0.31_orig/pic-i386/stderr.o and dietlibc-0.31/pic-i386/stderr.o differ
Binary files dietlibc-0.31_orig/pic-i386/stdin.o and dietlibc-0.31/pic-i386/stdin.o differ
Binary files dietlibc-0.31_orig/pic-i386/stdout.o and dietlibc-0.31/pic-i386/stdout.o differ
Only in dietlibc-0.31_orig/pic-i386: stpcpy.o
Binary files dietlibc-0.31_orig/pic-i386/strcasecmp.o and dietlibc-0.31/pic-i386/strcasecmp.o differ
Binary files dietlibc-0.31_orig/pic-i386/strcat.o and dietlibc-0.31/pic-i386/strcat.o differ
Binary files dietlibc-0.31_orig/pic-i386/strchr.o and dietlibc-0.31/pic-i386/strchr.o differ
Binary files dietlibc-0.31_orig/pic-i386/strcmp.o and dietlibc-0.31/pic-i386/strcmp.o differ
Binary files dietlibc-0.31_orig/pic-i386/strcpy.o and dietlibc-0.31/pic-i386/strcpy.o differ
Binary files dietlibc-0.31_orig/pic-i386/strdup.o and dietlibc-0.31/pic-i386/strdup.o differ
Binary files dietlibc-0.31_orig/pic-i386/strerror_r.o and dietlibc-0.31/pic-i386/strerror_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/strftime.o and dietlibc-0.31/pic-i386/strftime.o differ
Binary files dietlibc-0.31_orig/pic-i386/strlen.o and dietlibc-0.31/pic-i386/strlen.o differ
Binary files dietlibc-0.31_orig/pic-i386/strncmp.o and dietlibc-0.31/pic-i386/strncmp.o differ
Binary files dietlibc-0.31_orig/pic-i386/strncpy.o and dietlibc-0.31/pic-i386/strncpy.o differ
Binary files dietlibc-0.31_orig/pic-i386/strndup.o and dietlibc-0.31/pic-i386/strndup.o differ
Only in dietlibc-0.31_orig/pic-i386: strnlen.o
Binary files dietlibc-0.31_orig/pic-i386/strptime.o and dietlibc-0.31/pic-i386/strptime.o differ
Binary files dietlibc-0.31_orig/pic-i386/strrchr.o and dietlibc-0.31/pic-i386/strrchr.o differ
Binary files dietlibc-0.31_orig/pic-i386/strstr.o and dietlibc-0.31/pic-i386/strstr.o differ
Binary files dietlibc-0.31_orig/pic-i386/strtod.o and dietlibc-0.31/pic-i386/strtod.o differ
Binary files dietlibc-0.31_orig/pic-i386/strtof.o and dietlibc-0.31/pic-i386/strtof.o differ
Binary files dietlibc-0.31_orig/pic-i386/strtok.o and dietlibc-0.31/pic-i386/strtok.o differ
Binary files dietlibc-0.31_orig/pic-i386/strtok_r.o and dietlibc-0.31/pic-i386/strtok_r.o differ
Binary files dietlibc-0.31_orig/pic-i386/strtol.o and dietlibc-0.31/pic-i386/strtol.o differ
Binary files dietlibc-0.31_orig/pic-i386/strtold.o and dietlibc-0.31/pic-i386/strtold.o differ
Binary files dietlibc-0.31_orig/pic-i386/strtoll.o and dietlibc-0.31/pic-i386/strtoll.o differ
Binary files dietlibc-0.31_orig/pic-i386/strtoul.o and dietlibc-0.31/pic-i386/strtoul.o differ
Binary files dietlibc-0.31_orig/pic-i386/strtoull.o and dietlibc-0.31/pic-i386/strtoull.o differ
Binary files dietlibc-0.31_orig/pic-i386/strxfrm.o and dietlibc-0.31/pic-i386/strxfrm.o differ
Only in dietlibc-0.31_orig/pic-i386: syscall.o
Binary files dietlibc-0.31_orig/pic-i386/sysconf.o and dietlibc-0.31/pic-i386/sysconf.o differ
Binary files dietlibc-0.31_orig/pic-i386/sysconf_cpus.o and dietlibc-0.31/pic-i386/sysconf_cpus.o differ
Binary files dietlibc-0.31_orig/pic-i386/system.o and dietlibc-0.31/pic-i386/system.o differ
Only in dietlibc-0.31_orig/pic-i386: tan.o
Only in dietlibc-0.31_orig/pic-i386: tanh.o
Binary files dietlibc-0.31_orig/pic-i386/tcdrain.o and dietlibc-0.31/pic-i386/tcdrain.o differ
Binary files dietlibc-0.31_orig/pic-i386/tcflow.o and dietlibc-0.31/pic-i386/tcflow.o differ
Binary files dietlibc-0.31_orig/pic-i386/tcflush.o and dietlibc-0.31/pic-i386/tcflush.o differ
Binary files dietlibc-0.31_orig/pic-i386/tcgetattr.o and dietlibc-0.31/pic-i386/tcgetattr.o differ
Binary files dietlibc-0.31_orig/pic-i386/tcgetpgrp.o and dietlibc-0.31/pic-i386/tcgetpgrp.o differ
Binary files dietlibc-0.31_orig/pic-i386/tcgetsid.o and dietlibc-0.31/pic-i386/tcgetsid.o differ
Binary files dietlibc-0.31_orig/pic-i386/tcsendbreak.o and dietlibc-0.31/pic-i386/tcsendbreak.o differ
Binary files dietlibc-0.31_orig/pic-i386/tcsetattr.o and dietlibc-0.31/pic-i386/tcsetattr.o differ
Binary files dietlibc-0.31_orig/pic-i386/tcsetpgrp.o and dietlibc-0.31/pic-i386/tcsetpgrp.o differ
Binary files dietlibc-0.31_orig/pic-i386/telldir.o and dietlibc-0.31/pic-i386/telldir.o differ
Binary files dietlibc-0.31_orig/pic-i386/tempnam.o and dietlibc-0.31/pic-i386/tempnam.o differ
Binary files dietlibc-0.31_orig/pic-i386/textdomain.o and dietlibc-0.31/pic-i386/textdomain.o differ
Binary files dietlibc-0.31_orig/pic-i386/timegm.o and dietlibc-0.31/pic-i386/timegm.o differ
Binary files dietlibc-0.31_orig/pic-i386/timerfd_create.o and dietlibc-0.31/pic-i386/timerfd_create.o differ
Binary files dietlibc-0.31_orig/pic-i386/tmpfile.o and dietlibc-0.31/pic-i386/tmpfile.o differ
Binary files dietlibc-0.31_orig/pic-i386/tmpnam.o and dietlibc-0.31/pic-i386/tmpnam.o differ
Only in dietlibc-0.31_orig/pic-i386: toascii.o
Binary files dietlibc-0.31_orig/pic-i386/ttyname.o and dietlibc-0.31/pic-i386/ttyname.o differ
Binary files dietlibc-0.31_orig/pic-i386/tzfile.o and dietlibc-0.31/pic-i386/tzfile.o differ
Binary files dietlibc-0.31_orig/pic-i386/unlockpt.o and dietlibc-0.31/pic-i386/unlockpt.o differ
Binary files dietlibc-0.31_orig/pic-i386/unsetenv.o and dietlibc-0.31/pic-i386/unsetenv.o differ
Binary files dietlibc-0.31_orig/pic-i386/usleep.o and dietlibc-0.31/pic-i386/usleep.o differ
Binary files dietlibc-0.31_orig/pic-i386/utent.o and dietlibc-0.31/pic-i386/utent.o differ
Only in dietlibc-0.31_orig/pic-i386: utimes.o
Only in dietlibc-0.31_orig/pic-i386: utmpxname.o
Binary files dietlibc-0.31_orig/pic-i386/utxent.o and dietlibc-0.31/pic-i386/utxent.o differ
Binary files dietlibc-0.31_orig/pic-i386/vasprintf.o and dietlibc-0.31/pic-i386/vasprintf.o differ
Only in dietlibc-0.31_orig/pic-i386: verr.o
Only in dietlibc-0.31_orig/pic-i386: verrx.o
Binary files dietlibc-0.31_orig/pic-i386/vfdprintf.o and dietlibc-0.31/pic-i386/vfdprintf.o differ
Binary files dietlibc-0.31_orig/pic-i386/vfork.o and dietlibc-0.31/pic-i386/vfork.o differ
Binary files dietlibc-0.31_orig/pic-i386/vfprintf.o and dietlibc-0.31/pic-i386/vfprintf.o differ
Binary files dietlibc-0.31_orig/pic-i386/vfscanf.o and dietlibc-0.31/pic-i386/vfscanf.o differ
Binary files dietlibc-0.31_orig/pic-i386/vprintf.o and dietlibc-0.31/pic-i386/vprintf.o differ
Binary files dietlibc-0.31_orig/pic-i386/vscanf.o and dietlibc-0.31/pic-i386/vscanf.o differ
Binary files dietlibc-0.31_orig/pic-i386/vsnprintf.o and dietlibc-0.31/pic-i386/vsnprintf.o differ
Binary files dietlibc-0.31_orig/pic-i386/vsprintf.o and dietlibc-0.31/pic-i386/vsprintf.o differ
Binary files dietlibc-0.31_orig/pic-i386/vsscanf.o and dietlibc-0.31/pic-i386/vsscanf.o differ
Only in dietlibc-0.31_orig/pic-i386: vwarn.o
Only in dietlibc-0.31_orig/pic-i386: vwarnx.o
Binary files dietlibc-0.31_orig/pic-i386/wait.o and dietlibc-0.31/pic-i386/wait.o differ
Binary files dietlibc-0.31_orig/pic-i386/wait3.o and dietlibc-0.31/pic-i386/wait3.o differ
Only in dietlibc-0.31_orig/pic-i386: warn.o
Only in dietlibc-0.31_orig/pic-i386: warnx.o
Binary files dietlibc-0.31_orig/pic-i386/wctomb.o and dietlibc-0.31/pic-i386/wctomb.o differ
Binary files dietlibc-0.31_orig/pic-i386/wctype.o and dietlibc-0.31/pic-i386/wctype.o differ
Binary files dietlibc-0.31_orig/pic-i386/write12.o and dietlibc-0.31/pic-i386/write12.o differ
Binary files dietlibc-0.31_orig/pic-i386/wtent.o and dietlibc-0.31/pic-i386/wtent.o differ
Only in dietlibc-0.31_orig/pic-i386: wtxent.o
diff -ru dietlibc-0.31_orig/syscalls.s/timerfd.S dietlibc-0.31/syscalls.s/timerfd.S
--- dietlibc-0.31_orig/syscalls.s/timerfd.S	2008-05-26 12:56:07.000000000 +0200
+++ dietlibc-0.31/syscalls.s/timerfd.S	2008-05-26 13:23:13.000000000 +0200
@@ -1,5 +1,6 @@
 #include "syscalls.h"
 
 #ifdef __NR_timerfd
+#warning __timerfd call
 syscall(timerfd,__timerfd)
 #endif
diff -ru dietlibc-0.31_orig/x86_64/dyn_syscalls.S dietlibc-0.31/x86_64/dyn_syscalls.S
--- dietlibc-0.31_orig/x86_64/dyn_syscalls.S	2008-05-26 12:56:07.000000000 +0200
+++ dietlibc-0.31/x86_64/dyn_syscalls.S	2008-05-26 13:14:21.000000000 +0200
@@ -44,6 +44,7 @@
 #include "../syscalls.s/__pread.S"
 #include "../syscalls.s/__pwrite.S"
 #include "../syscalls.s/__reboot.S"
+#include "../syscalls.s/__signalfd.S"
 #include "../syscalls.s/_llseek.S"
 #include "../syscalls.s/access.S"
 #include "../syscalls.s/adjtimex.S"
@@ -208,12 +209,13 @@
 #include "../syscalls.s/sysinfo.S"
 #include "../syscalls.s/syslog.S"
 #include "../syscalls.s/time.S"
+#include "../syscalls.s/timerfd.S"
 #include "../syscalls.s/timer_create.S"
 #include "../syscalls.s/timer_delete.S"
 #include "../syscalls.s/timer_getoverrun.S"
 #include "../syscalls.s/timer_gettime.S"
 #include "../syscalls.s/timer_settime.S"
-#include "../syscalls.s/times.S"
+#include "../syscalls.s/timer_settime.S"
 #include "../syscalls.s/truncate.S"
 #include "../syscalls.s/truncate64.S"
 #include "../syscalls.s/umask.S"
