args: -r -u -- cvs-importer-0.3_orig cvs-importer-0.3
diff -r -u -- cvs-importer-0.3_orig/setup.py cvs-importer-0.3/setup.py
--- cvs-importer-0.3_orig/setup.py	2008-10-28 17:36:25.000000000 +0100
+++ cvs-importer-0.3/setup.py	2008-10-28 17:36:37.000000000 +0100
@@ -37,6 +37,6 @@
 	cmdclass = {'install_scripts':my_install_scripts, 'install_data':my_install_data},
 	scripts = ["cvs-importer"],
 	#distutils does not handle manpages (yet?), so fudge it :)
-	data_files = [("man/man1", ["cvs-importer.1"])],
+	data_files = [("share/man/man1", ["cvs-importer.1"])],
 	)
 
