diff -ru gmp-4.2.2_orig/gmp-impl.h gmp-4.2.2/gmp-impl.h
--- gmp-4.2.2_orig/gmp-impl.h	2008-05-30 11:54:01.000000000 +0200
+++ gmp-4.2.2/gmp-impl.h	2008-05-30 11:55:17.000000000 +0200
@@ -108,10 +108,10 @@
 
 #if HAVE_INTTYPES_H      /* for uint_least32_t */
 # include <inttypes.h>
-#else
-# if HAVE_STDINT_H
-#  include <stdint.h>
-# endif
+#endif
+
+#if HAVE_STDINT_H
+# include <stdint.h>
 #endif
 
 #ifdef __cplusplus
