diff -wru jabberd-1.4.3.1_orig/jabberd/mio_ssl.c jabberd-1.4.3.1/jabberd/mio_ssl.c
--- jabberd-1.4.3.1_orig/jabberd/mio_ssl.c	2008-08-08 15:45:50.000000000 +0200
+++ jabberd-1.4.3.1/jabberd/mio_ssl.c	2008-08-08 15:46:39.000000000 +0200
@@ -63,6 +63,7 @@
 
     log_debug(ZONE, "Handling configuration using: %s", xmlnode2str(x));
     /* Generic SSL Inits */
+    SSL_library_init();
 	OpenSSL_add_all_algorithms();    
     SSL_load_error_strings();
 
