Index: libchaos/include/libchaos/Makefile.in
===================================================================
--- libchaos/include/libchaos/Makefile.in	(revision 21)
+++ libchaos/include/libchaos/Makefile.in	(revision 22)
@@ -68,10 +68,9 @@
 
 # install config
 # ---------------------------------------------------------------------------
-include_DATA = $(HEADERS)
+include_DATA = $(HEADERS) $(CONFIG_H)
 include_PREFIX = $(PREFIX)
 
 # include global make targets
 # ---------------------------------------------------------------------------
 include $(top_builddir)/build.mk
-#@WIN32_TRUE@include ../../build.mk
