diff -ru ifile-1.3.8_orig/argp/argp-fs-xinl.c ifile-1.3.8/argp/argp-fs-xinl.c
--- ifile-1.3.8_orig/argp/argp-fs-xinl.c	2008-08-22 19:43:11.000000000 +0200
+++ ifile-1.3.8/argp/argp-fs-xinl.c	2008-08-22 19:43:44.000000000 +0200
@@ -23,8 +23,8 @@
 #endif
 
 #define ARGP_FS_EI
-#undef __OPTIMIZE__
-#define __OPTIMIZE__
+/*#undef __OPTIMIZE__
+#define __OPTIMIZE__*/
 #include "argp-fmtstream.h"
 
 /* Add weak aliases.  */
diff -ru ifile-1.3.8_orig/argp/argp-xinl.c ifile-1.3.8/argp/argp-xinl.c
--- ifile-1.3.8_orig/argp/argp-xinl.c	2008-08-22 19:43:11.000000000 +0200
+++ ifile-1.3.8/argp/argp-xinl.c	2008-08-22 19:43:58.000000000 +0200
@@ -23,8 +23,8 @@
 #endif
 
 #define ARGP_EI
-#undef __OPTIMIZE__
-#define __OPTIMIZE__
+/*#undef __OPTIMIZE__
+#define __OPTIMIZE__*/
 #include "argp.h"
 
 /* Add weak aliases.  */
