args: -r -u -- libbonobo-2.22.0_orig libbonobo-2.22.0
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/activation-context-corba.Po libbonobo-2.22.0/activation-server/.deps/activation-context-corba.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/activation-context-query-lexer.Po libbonobo-2.22.0/activation-server/.deps/activation-context-query-lexer.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/activation-context-query-parser.Po libbonobo-2.22.0/activation-server/.deps/activation-context-query-parser.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/activation-context-query.Po libbonobo-2.22.0/activation-server/.deps/activation-context-query.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/activation-server-corba-extensions.Po libbonobo-2.22.0/activation-server/.deps/activation-server-corba-extensions.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/activation-server-main.Po libbonobo-2.22.0/activation-server/.deps/activation-server-main.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/client.Po libbonobo-2.22.0/activation-server/.deps/client.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/object-directory-activate.Po libbonobo-2.22.0/activation-server/.deps/object-directory-activate.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/object-directory-config-file.Po libbonobo-2.22.0/activation-server/.deps/object-directory-config-file.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/object-directory-corba.Po libbonobo-2.22.0/activation-server/.deps/object-directory-corba.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/object-directory-load.Po libbonobo-2.22.0/activation-server/.deps/object-directory-load.Po
diff -r -u -- libbonobo-2.22.0_orig/activation-server/.deps/test-performance.Po libbonobo-2.22.0/activation-server/.deps/test-performance.Po
diff -r -u -- libbonobo-2.22.0_orig/bonobo/.deps/Bonobo-common.Plo libbonobo-2.22.0/bonobo/.deps/Bonobo-common.Plo
diff -r -u -- libbonobo-2.22.0_orig/bonobo/.deps/Bonobo-skels.Plo libbonobo-2.22.0/bonobo/.deps/Bonobo-skels.Plo
diff -r -u -- libbonobo-2.22.0_orig/bonobo/.deps/Bonobo-stubs.Plo libbonobo-2.22.0/bonobo/.deps/Bonobo-stubs.Plo
diff -r -u -- libbonobo-2.22.0_orig/bonobo/.deps/bonobo-app-client.Plo libbonobo-2.22.0/bonobo/.deps/bonobo-app-client.Plo
diff -r -u -- libbonobo-2.22.0_orig/bonobo/.deps/bonobo-application.Plo libbonobo-2.22.0/bonobo/.deps/bonobo-application.Plo
diff -r -u -- libbonobo-2.22.0_orig/bonobo/.deps/bonobo-debug.Plo libbonobo-2.22.0/bonobo/.deps/bonobo-debug.Plo
diff -r -u -- libbonobo-2.22.0_orig/bonobo/.deps/bonobo-running-context.Plo libbonobo-2.22.0/bonobo/.deps/bonobo-running-context.Plo
diff -r -u -- libbonobo-2.22.0_orig/bonobo/.deps/bonobo-types.Plo libbonobo-2.22.0/bonobo/.deps/bonobo-types.Plo
diff -r -u -- libbonobo-2.22.0_orig/bonobo/bonobo-running-context.c libbonobo-2.22.0/bonobo/bonobo-running-context.c
--- libbonobo-2.22.0_orig/bonobo/bonobo-running-context.c	2008-11-21 20:14:02.000000000 +0100
+++ libbonobo-2.22.0/bonobo/bonobo-running-context.c	2008-11-21 20:14:36.000000000 +0100
@@ -9,6 +9,7 @@
  */
 #include <config.h>
 #include <stdio.h>
+#include <glib/gtypes.h>
 #include <glib/gmain.h>
 #include <glib-object.h>
 
diff -r -u -- libbonobo-2.22.0_orig/monikers/.deps/bonobo-moniker-cache.Plo libbonobo-2.22.0/monikers/.deps/bonobo-moniker-cache.Plo
diff -r -u -- libbonobo-2.22.0_orig/monikers/.deps/bonobo-moniker-extender-stream.Plo libbonobo-2.22.0/monikers/.deps/bonobo-moniker-extender-stream.Plo
diff -r -u -- libbonobo-2.22.0_orig/monikers/.deps/bonobo-moniker-ior.Plo libbonobo-2.22.0/monikers/.deps/bonobo-moniker-ior.Plo
diff -r -u -- libbonobo-2.22.0_orig/monikers/.deps/bonobo-moniker-item.Plo libbonobo-2.22.0/monikers/.deps/bonobo-moniker-item.Plo
diff -r -u -- libbonobo-2.22.0_orig/monikers/.deps/bonobo-moniker-new.Plo libbonobo-2.22.0/monikers/.deps/bonobo-moniker-new.Plo
diff -r -u -- libbonobo-2.22.0_orig/monikers/.deps/bonobo-moniker-oaf.Plo libbonobo-2.22.0/monikers/.deps/bonobo-moniker-oaf.Plo
diff -r -u -- libbonobo-2.22.0_orig/monikers/.deps/bonobo-moniker-query.Plo libbonobo-2.22.0/monikers/.deps/bonobo-moniker-query.Plo
diff -r -u -- libbonobo-2.22.0_orig/monikers/.deps/bonobo-moniker-std.Plo libbonobo-2.22.0/monikers/.deps/bonobo-moniker-std.Plo
diff -r -u -- libbonobo-2.22.0_orig/monikers/.deps/bonobo-stream-cache.Plo libbonobo-2.22.0/monikers/.deps/bonobo-stream-cache.Plo
diff -r -u -- libbonobo-2.22.0_orig/samples/echo/.deps/Bonobo_Sample_Echo-common.Po libbonobo-2.22.0/samples/echo/.deps/Bonobo_Sample_Echo-common.Po
diff -r -u -- libbonobo-2.22.0_orig/samples/echo/.deps/Bonobo_Sample_Echo-skels.Po libbonobo-2.22.0/samples/echo/.deps/Bonobo_Sample_Echo-skels.Po
diff -r -u -- libbonobo-2.22.0_orig/samples/echo/.deps/Bonobo_Sample_Echo-stubs.Po libbonobo-2.22.0/samples/echo/.deps/Bonobo_Sample_Echo-stubs.Po
diff -r -u -- libbonobo-2.22.0_orig/samples/echo/.deps/echo-client.Po libbonobo-2.22.0/samples/echo/.deps/echo-client.Po
diff -r -u -- libbonobo-2.22.0_orig/samples/echo/.deps/echo.Po libbonobo-2.22.0/samples/echo/.deps/echo.Po
diff -r -u -- libbonobo-2.22.0_orig/samples/echo/.deps/main.Po libbonobo-2.22.0/samples/echo/.deps/main.Po
diff -r -u -- libbonobo-2.22.0_orig/samples/echo/Bonobo_Sample_Echo.server libbonobo-2.22.0/samples/echo/Bonobo_Sample_Echo.server
--- libbonobo-2.22.0_orig/samples/echo/Bonobo_Sample_Echo.server	2008-11-21 20:14:02.000000000 +0100
+++ libbonobo-2.22.0/samples/echo/Bonobo_Sample_Echo.server	2008-11-21 20:15:10.000000000 +0100
@@ -1,7 +1,7 @@
 <oaf_info>
 
 <oaf_server iid="OAFIID:Bonobo_Sample_Echo_Factory" type="exe" 
-	location="/opt/gnome2/lib/bonobo-2.0/samples/bonobo-echo-2">
+	location="/usr/lib/bonobo-2.0/samples/bonobo-echo-2">
 	<oaf_attribute name="repo_ids" type="stringv">
 		<item value="IDL:GNOME/ObjectFactory:1.0"/>
 	</oaf_attribute>
diff -r -u -- libbonobo-2.22.0_orig/tests/test-activation/.deps/bonobo-activation-run-query.Po libbonobo-2.22.0/tests/test-activation/.deps/bonobo-activation-run-query.Po
diff -r -u -- libbonobo-2.22.0_orig/tests/test-activation/.deps/bonobo-activation-testplugin.Plo libbonobo-2.22.0/tests/test-activation/.deps/bonobo-activation-testplugin.Plo
diff -r -u -- libbonobo-2.22.0_orig/tests/test-activation/.deps/empty-common.Po libbonobo-2.22.0/tests/test-activation/.deps/empty-common.Po
diff -r -u -- libbonobo-2.22.0_orig/tests/test-activation/.deps/empty-stubs.Po libbonobo-2.22.0/tests/test-activation/.deps/empty-stubs.Po
diff -r -u -- libbonobo-2.22.0_orig/tests/test-activation/.deps/plugin-common.Plo libbonobo-2.22.0/tests/test-activation/.deps/plugin-common.Plo
diff -r -u -- libbonobo-2.22.0_orig/tests/test-activation/.deps/plugin-skels.Plo libbonobo-2.22.0/tests/test-activation/.deps/plugin-skels.Plo
diff -r -u -- libbonobo-2.22.0_orig/tests/test-activation/.deps/plugin-stubs.Plo libbonobo-2.22.0/tests/test-activation/.deps/plugin-stubs.Plo
diff -r -u -- libbonobo-2.22.0_orig/utils/.deps/bonobo-activation-sysconf.Po libbonobo-2.22.0/utils/.deps/bonobo-activation-sysconf.Po
