diff -ru libomlet-0.98_orig/src/libomlet/Makefile.am libomlet-0.98/src/libomlet/Makefile.am
--- libomlet-0.98_orig/src/libomlet/Makefile.am	2008-07-12 21:58:14.000000000 +0200
+++ libomlet-0.98/src/libomlet/Makefile.am	2008-07-12 21:59:06.000000000 +0200
@@ -1,4 +1,4 @@
 INCLUDES = -I$(top_srcdir)/src/include
 lib_LTLIBRARIES = libomlet.la
 libomlet_la_SOURCES = adaboost.cc dataset.cc example.cc weakrule.cc
-libomlet_la_LDFLAGS = -version-info 0:9
+libomlet_la_LDFLAGS = -rpath $(libdir) -no-undefined -version-info 0:9
