diff -ru metafs-0.1.5_orig/plugins/.deps/libtaglib_la-taglib.Plo metafs-0.1.5/plugins/.deps/libtaglib_la-taglib.Plo
diff -ru metafs-0.1.5_orig/plugins/memfs/.deps/dir.Plo metafs-0.1.5/plugins/memfs/.deps/dir.Plo
diff -ru metafs-0.1.5_orig/plugins/memfs/.deps/file.Plo metafs-0.1.5/plugins/memfs/.deps/file.Plo
diff -ru metafs-0.1.5_orig/plugins/memfs/.deps/inode.Plo metafs-0.1.5/plugins/memfs/.deps/inode.Plo
diff -ru metafs-0.1.5_orig/plugins/memfs/.deps/plugin.Plo metafs-0.1.5/plugins/memfs/.deps/plugin.Plo
diff -ru metafs-0.1.5_orig/plugins/memfs/.deps/symlink.Plo metafs-0.1.5/plugins/memfs/.deps/symlink.Plo
diff -ru metafs-0.1.5_orig/plugins/memfs/inode.cc metafs-0.1.5/plugins/memfs/inode.cc
--- metafs-0.1.5_orig/plugins/memfs/inode.cc	2008-06-13 15:56:21.000000000 +0200
+++ metafs-0.1.5/plugins/memfs/inode.cc	2008-06-13 15:57:38.000000000 +0200
@@ -3,6 +3,10 @@
 #include <attr/xattr.h>
 #include <time.h>
 
+#ifndef ENOATTR
+#define ENOATTR ENOENT
+#endif
+
 using namespace memfs;
 using metafs::SysError;
 
diff -ru metafs-0.1.5_orig/plugins/metameta/.deps/xattr.Plo metafs-0.1.5/plugins/metameta/.deps/xattr.Plo
diff -ru metafs-0.1.5_orig/plugins/metameta/xattr.cc metafs-0.1.5/plugins/metameta/xattr.cc
--- metafs-0.1.5_orig/plugins/metameta/xattr.cc	2008-06-13 15:56:21.000000000 +0200
+++ metafs-0.1.5/plugins/metameta/xattr.cc	2008-06-13 15:56:45.000000000 +0200
@@ -2,6 +2,10 @@
 
 #include <attr/xattr.h>
 
+#ifndef ENOATTR
+#define ENOATTR ENOENT
+#endif
+
 using namespace metameta;
 
 ////////// XAttrInode {
diff -ru metafs-0.1.5_orig/plugins/mime/.deps/lex.yy.Plo metafs-0.1.5/plugins/mime/.deps/lex.yy.Plo
diff -ru metafs-0.1.5_orig/plugins/mime/.deps/plugin.Plo metafs-0.1.5/plugins/mime/.deps/plugin.Plo
diff -ru metafs-0.1.5_orig/plugins/realfs/.deps/dir.Plo metafs-0.1.5/plugins/realfs/.deps/dir.Plo
diff -ru metafs-0.1.5_orig/plugins/realfs/.deps/file.Plo metafs-0.1.5/plugins/realfs/.deps/file.Plo
diff -ru metafs-0.1.5_orig/plugins/realfs/.deps/inode.Plo metafs-0.1.5/plugins/realfs/.deps/inode.Plo
diff -ru metafs-0.1.5_orig/plugins/realfs/.deps/plugin.Plo metafs-0.1.5/plugins/realfs/.deps/plugin.Plo
diff -ru metafs-0.1.5_orig/plugins/realfs/.deps/symlink.Plo metafs-0.1.5/plugins/realfs/.deps/symlink.Plo
diff -ru metafs-0.1.5_orig/plugins/realfs/.deps/xattr_db.Plo metafs-0.1.5/plugins/realfs/.deps/xattr_db.Plo
diff -ru metafs-0.1.5_orig/plugins/realfs/xattr_db.cc metafs-0.1.5/plugins/realfs/xattr_db.cc
--- metafs-0.1.5_orig/plugins/realfs/xattr_db.cc	2008-06-13 15:56:21.000000000 +0200
+++ metafs-0.1.5/plugins/realfs/xattr_db.cc	2008-06-13 15:57:14.000000000 +0200
@@ -2,6 +2,10 @@
 #include <sys/stat.h>
 #include <attr/xattr.h>
 
+#ifndef ENOATTR
+#define ENOATTR ENOENT
+#endif
+
 using namespace realfs;
 using metafs::SysError;
 
