diff -r -u gscanbus-0.7.1_orig/gscanbus.c gscanbus-0.7.1/gscanbus.c
--- gscanbus-0.7.1_orig/gscanbus.c	2007-12-20 08:46:11.000000000 +0100
+++ gscanbus-0.7.1/gscanbus.c	2007-12-20 08:46:37.000000000 +0100
@@ -109,14 +109,14 @@
 
 static GdkPixmap *pixmap = NULL;
 
-const char not_compatible[] = "\
-This libraw1394 does not work with your version of Linux. You need a different
-version that matches your kernel (see kernel help text for the raw1394 option to
-find out which is the correct version).\n";
-
-const char not_loaded[] = "\
-This probably means that you don't have raw1394 support in the kernel or that
-you haven't loaded the raw1394 module.\n";
+const char not_compatible[] = "\n" \
+"This libraw1394 does not work with your version of Linux. You need a different\n" \
+"version that matches your kernel (see kernel help text for the raw1394 option to\n" \
+"find out which is the correct version).\n";
+
+const char not_loaded[] = "\n" \
+"This probably means that you don't have raw1394 support in the kernel or that\n" \
+"you haven't loaded the raw1394 module.\n";
 
 /*---------------------------------------------------------------------------
  * Drawing routines
