diff -r -u lcc-4.1_orig/etc/linux.c lcc-4.1/etc/linux.c
--- lcc-4.1_orig/etc/linux.c	2007-09-21 05:06:00.000000000 +0200
+++ lcc-4.1/etc/linux.c	2007-09-21 05:20:35.000000000 +0200
@@ -4,47 +4,55 @@
 
 static char rcsid[] = "$Id: linux.c,v 1.5 1998/09/16 20:41:09 drh Exp $";
 
+#ifndef PREFIX
+#define PREFIX "/usr/"
+#endif
+
 #ifndef LCCDIR
-#define LCCDIR "/usr/local/lib/lcc/"
+#define LCCDIR PREFIX "lib/lcc/x86/linux/4.1/"
+#endif
+
+#ifndef DYNAMIC_LINKER
+#define DYNAMIC_LINKER "/lib/ld-linux.so.2"
 #endif
 
 char *suffixes[] = { ".c", ".i", ".s", ".o", ".out", 0 };
 char inputs[256] = "";
-char *cpp[] = { LCCDIR "gcc/cpp",
+char *cpp[] = { LCCDIR "cpp",
 	"-U__GNUC__", "-D_POSIX_SOURCE", "-D__STDC__=1", "-D__STRICT_ANSI__",
 	"-Dunix", "-Di386", "-Dlinux",
 	"-D__unix__", "-D__i386__", "-D__linux__", "-D__signed__=signed",
 	"$1", "$2", "$3", 0 };
-char *include[] = {"-I" LCCDIR "include", "-I" LCCDIR "gcc/include", "-I/usr/include", 0 };
+char *include[] = {"-I" LCCDIR "include", "-I" PREFIX "include", 0 };
 char *com[] = {LCCDIR "rcc", "-target=x86/linux", "$1", "$2", "$3", 0 };
-char *as[] = { "/usr/bin/as", "-o", "$3", "$1", "$2", 0 };
+char *as[] = { PREFIX "bin/as", "-o", "$3", "$1", "$2", 0 };
 char *ld[] = {
-	/*  0 */ "/usr/bin/ld", "-m", "elf_i386", "-dynamic-linker",
-	/*  4 */ "/lib/ld-linux.so.2", "-o", "$3",
-	/*  7 */ "/usr/lib/crt1.o", "/usr/lib/crti.o",
-	/*  9 */ LCCDIR "/gcc/crtbegin.o", 
+	/*  0 */ PREFIX "bin/ld", "-m", "elf_i386", "-dynamic-linker",
+	/*  4 */ DYNAMIC_LINKER, "-o", "$3",
+	/*  7 */ PREFIX "lib/crt1.o", PREFIX "lib/crti.o",
+	/*  9 */ LCCDIR "crtbegin.o", 
                  "$1", "$2",
 	/* 12 */ "-L" LCCDIR,
 	/* 13 */ "-llcc",
-	/* 14 */ "-L" LCCDIR "/gcc", "-lgcc", "-lc", "-lm",
+	/* 14 */ "-L" LCCDIR, "-lgcc", "-lc", "-lm",
 	/* 18 */ "",
-	/* 19 */ LCCDIR "/gcc/crtend.o", "/usr/lib/crtn.o",
+	/* 19 */ LCCDIR "crtend.o", PREFIX "lib/crtn.o",
 	0 };
 
 extern char *concat(char *, char *);
 
 int option(char *arg) {
   	if (strncmp(arg, "-lccdir=", 8) == 0) {
-		cpp[0] = concat(&arg[8], "/gcc/cpp");
+		cpp[0] = concat(&arg[8], "/cpp");
 		include[0] = concat("-I", concat(&arg[8], "/include"));
-		include[1] = concat("-I", concat(&arg[8], "/gcc/include"));
-		ld[9]  = concat(&arg[8], "/gcc/crtbegin.o");
+/*		include[1] = concat("-I", concat(&arg[8], "/gcc/include"));*/
+		ld[9]  = concat(&arg[8], "/crtbegin.o");
 		ld[12] = concat("-L", &arg[8]);
-		ld[14] = concat("-L", concat(&arg[8], "/gcc"));
-		ld[19] = concat(&arg[8], "/gcc/crtend.o");
+/*		ld[14] = concat("-L", concat(&arg[8], "/gcc"));*/
+		ld[19] = concat(&arg[8], "/crtend.o");
 		com[0] = concat(&arg[8], "/rcc");
 	} else if (strcmp(arg, "-p") == 0 || strcmp(arg, "-pg") == 0) {
-		ld[7] = "/usr/lib/gcrt1.o";
+		ld[7] = PREFIX "lib/gcrt1.o";
 		ld[18] = "-lgmon";
 	} else if (strcmp(arg, "-b") == 0) 
 		;
