diff -ru libfries-0.96_orig/src/libfries/Makefile.am libfries-0.96/src/libfries/Makefile.am
--- libfries-0.96_orig/src/libfries/Makefile.am	2008-07-12 22:01:45.000000000 +0200
+++ libfries-0.96/src/libfries/Makefile.am	2008-07-12 22:02:05.000000000 +0200
@@ -2,4 +2,4 @@
 EXTRA_DIST = features.l features.y
 lib_LTLIBRARIES = libfries.la
 libfries_la_SOURCES = util.cc language.cc features_l.cc features_y.cc fex.cc RGF.cc sensor.cc
-libfries_la_LDFLAGS = -version-info 0:9
+libfries_la_LDFLAGS = -rpath $(libdir) -no-undefined -version-info 0:9
