diff -ru libfries-0.97_orig/src/include/regexp-pcre++.h libfries-0.97/src/include/regexp-pcre++.h
--- libfries-0.97_orig/src/include/regexp-pcre++.h	2008-11-14 13:34:53.000000000 +0000
+++ libfries-0.97/src/include/regexp-pcre++.h	2008-11-14 13:34:59.000000000 +0000
@@ -30,6 +30,7 @@
 #define _REGEX_H
 
 #include <string>
+#include <cstring>
 
 #ifndef _PCRE_H
 #include "pcre.h"
diff -ru libfries-0.97_orig/src/libfries/RGF.cc libfries-0.97/src/libfries/RGF.cc
--- libfries-0.97_orig/src/libfries/RGF.cc	2008-11-14 13:34:53.000000000 +0000
+++ libfries-0.97/src/libfries/RGF.cc	2008-11-14 13:44:14.000000000 +0000
@@ -48,6 +48,7 @@
 #include <iostream>
 #include <string>
 #include <vector>
+#include <algorithm>
 
 using namespace std;
 
