diff -ru -ru pdftk-1.12_orig/pdftk/Makefile.Base pdftk-1.12/pdftk/Makefile.Base
--- pdftk-1.12_orig/pdftk/Makefile.Base	2006-06-06 16:52:51.000000000 +0200
+++ pdftk-1.12/pdftk/Makefile.Base	2006-06-06 16:54:11.000000000 +0200
@@ -37,8 +37,7 @@
 libgcj_local_libs = \
 $(java_libs_root)/java_local/security/security.a \
 $(java_libs_root)/gnu_local/java/security/provider/provider.a \
-$(java_libs_root)/gnu_local/java/security/security.a \
-$(java_libs_root)/gnu/gcj/convert/convert.a
+$(java_libs_root)/gnu_local/java/security/security.a 
 
 # this must already be set according to your platform Makefile;
 # we're just appending to it, here
