--- rxvt-2.7.10_orig/autoconf/configure.in	2008-08-09 07:54:59.000000000 +0200
+++ rxvt-2.7.10/autoconf/configure.in	2008-08-09 07:55:41.000000000 +0200
@@ -346,12 +346,12 @@
     CFLAGS="$CCOPTS"
 fi
 
-AC_PATH_PROG(MV, mv, mv)
-AC_PATH_PROG(RM, rm, rm)
-AC_PATH_PROG(CP, cp, cp)
-AC_PATH_PROG(LN, ln, ln)
-AC_PATH_PROG(SED, sed, sed)
-AC_PATH_PROG(ECHO, echo, echo)
+dnl AC_PATH_PROG(MV, mv, mv)
+dnl AC_PATH_PROG(RM, rm, rm)
+dnl AC_PATH_PROG(CP, cp, cp)
+dnl AC_PATH_PROG(LN, ln, ln)
+dnl AC_PATH_PROG(SED, sed, sed)
+dnl AC_PATH_PROG(ECHO, echo, echo)
 AC_PATH_PROG(CMP, cmp, cmp)
 AC_PATH_PROG(TBL, tbl)
 
--- rxvt-2.7.10_orig/autoconf/Make.common.in	2008-08-09 08:02:58.000000000 +0200
+++ rxvt-2.7.10/autoconf/Make.common.in	2008-08-09 08:04:15.000000000 +0200
@@ -35,14 +35,14 @@
 # Tools & program stuff
 CC = @CC@
 CPP = @CPP@
-MV = @MV@
-RM = @RM@
-RMF = @RM@ -f
-CP = @CP@
-LN = @LN@
-SED = @SED@
+MV = mv
+RM = rm -f
+RMF = rm -f
+CP = cp
+LN = ln
+SED = sed
 AWK = @AWK@
-ECHO = @ECHO@
+ECHO = echo
 CMP = @CMP@
 TBL = @TBL@
 INSTALL = @INSTALL@
