diff -ru rxvt-2.7.10_orig/autoconf/configure.in rxvt-2.7.10/autoconf/configure.in
--- rxvt-2.7.10_orig/autoconf/configure.in	2007-10-26 11:02:21.000000000 +0200
+++ rxvt-2.7.10/autoconf/configure.in	2007-10-26 12:23:31.000000000 +0200
@@ -1218,7 +1224,6 @@
 
 AC_OUTPUT(autoconf/Make.common \
 Makefile \
-doc/Makefile \
 src/Makefile \
 src/graphics/Makefile \
 src/test/Makefile \
diff -ru rxvt-2.7.10_orig/src/feature.h rxvt-2.7.10/src/feature.h
--- rxvt-2.7.10_orig/src/feature.h	2007-10-26 11:02:21.000000000 +0200
+++ rxvt-2.7.10/src/feature.h	2007-10-26 12:26:27.000000000 +0200
@ -427,7 +427,7 @@
 /*
  * Default number of lines in the scrollback buffer
  */
-#define SAVELINES	64
+#define SAVELINES	256
 
 /*
  * Provide termcap/terminfo bw support
@@ -441,6 +441,6 @@
  * sets (NFONT_LIST) have been matched up
  */
 #define MAX_NFONTS	7
-#define FONT0_IDX	2
+#define FONT0_IDX	0
 
 #endif
diff -ru rxvt-2.7.10_orig/src/init.c rxvt-2.7.10/src/init.c
--- rxvt-2.7.10_orig/src/init.c	2007-10-26 11:02:21.000000000 +0200
+++ rxvt-2.7.10/src/init.c	2007-10-26 12:36:00.000000000 +0200
@@ -640,7 +640,7 @@
 			 rs[Rs_scrollBar_thickness]);
 #endif
 
-#if 0	/* #ifndef NO_BOLDFONT */
+#ifndef NO_BOLDFONT 
     if (rs[Rs_font] == NULL && rs[Rs_boldFont] != NULL) {
 	rs[Rs_font] = rs[Rs_boldFont];
 	rs[Rs_boldFont] = NULL;
@@ -1096,7 +1099,7 @@
 /* grab colors before netscape does */
     rxvt_Get_Colours(r);
 
-    rxvt_change_font(r, 1, NULL);
+    rxvt_change_font(r, 1, A_12);
     rxvt_window_calc(r, 0, 0);
     r->h->old_width = r->szHint.width;
     r->h->old_height = r->szHint.height;
diff -ru rxvt-2.7.10_orig/src/defaultfont.h rxvt-2.7.10/src/defaultfont.h
--- rxvt-2.7.10_orig/src/defaultfont.h	2008-06-08 03:27:16.000000000 +0200
+++ rxvt-2.7.10/src/defaultfont.h	2008-06-08 03:31:16.000000000 +0200
@@ -90,6 +90,7 @@
  * ASCII font definitions (only used in this file)
  */
 
+#define A_10 "-misc-fixed-medium-r-semicondensed--10-90-75-75-c-50-iso8859-1"
 #define A_12 "-misc-fixed-medium-r-semicondensed--12-110-75-75-c-60-iso8859-1"
 #define A_14 "-misc-fixed-medium-r-normal--14-130-75-75-c-70-iso8859-1"
 #define A_16 "-etl-fixed-medium-r-normal--16-160-72-72-c-80-iso8859-1"
diff -ru rxvt-2.7.10_orig/src/init.c rxvt-2.7.10/src/init.c
--- rxvt-2.7.10_orig/src/init.c	2008-06-08 03:27:16.000000000 +0200
+++ rxvt-2.7.10/src/init.c	2008-06-08 03:31:40.000000000 +0200
@@ -34,6 +34,7 @@
 #include "../config.h"		/* NECESSARY */
 #include "rxvt.h"		/* NECESSARY */
 #include "init.h"
+#include "defaultfont.h"
 
 #include <signal.h>
 
@@ -1099,7 +1100,7 @@
 /* grab colors before netscape does */
     rxvt_Get_Colours(r);
 
-    rxvt_change_font(r, 1, A_12);
+    rxvt_change_font(r, 1, "6x13");
     rxvt_window_calc(r, 0, 0);
     r->h->old_width = r->szHint.width;
     r->h->old_height = r->szHint.height;
