diff -ru ming-0.2a_orig/mingpp.h ming-0.2a/mingpp.h
--- ming-0.2a_orig/mingpp.h	2008-03-24 00:22:04.000000000 +0000
+++ ming-0.2a/mingpp.h	2008-03-24 00:22:21.000000000 +0000
@@ -627,7 +627,7 @@
   void setColor(unsigned char r, unsigned char g, unsigned char b, int a=0xff)
     { SWFText_setColor(this->text, r, g, b, a); }
 
-  void addString(const char *string, float *advance=NULL)
+  void addString(const char *string, int *advance=NULL)
     { SWFText_addString(this->text, string, advance); }
 
   void setSpacing(float spacing)
