diff -r -u -N -- reflex_orig/inc/Reflex/internal/Makefile.am reflex/inc/Reflex/internal/Makefile.am
--- reflex_orig/inc/Reflex/internal/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ reflex/inc/Reflex/internal/Makefile.am	2007-06-13 12:06:23.000000000 +0200
@@ -0,0 +1 @@
+EXTRA_DIST = InternalTools.h MemberBase.h MemberTemplateImpl.h MemberTemplateName.h OwnedMember.h OwnedMemberTemplate.h OwnedPropertyList.h PropertyListImpl.h ScopeBase.h ScopeName.h TypeBase.h TypeName.h TypeTemplateImpl.h TypeTemplateName.h
