--- pstoedit-3.45_orig/config/pstoedit.m4	2008-03-24 14:12:57.000000000 +0000
+++ pstoedit-3.45/config/pstoedit.m4	2008-03-24 14:13:30.000000000 +0000
@@ -5,6 +5,9 @@
 dnl Test to see if libpstoedit is installed, and define PSTOEDIT_CFLAGS, LIBS
 dnl
 
+AC_DEFUN([AM_PATH_PSTOEDIT],
+[
+
 AC_ARG_WITH(pstoedit-prefix,[  --with-pstoedit-prefix=PFX   Prefix where Pstoedit is installed (optional)],
             pstoedit_prefix="$withval", pstoedit_prefix="")
 AC_ARG_WITH(pstoedit-exec-prefix,[  --with-pstoedit-exec-prefix=PFX Exec prefix where Pstoedit is installed (optional)],
@@ -155,3 +158,6 @@
   AC_SUBST(PSTOEDIT_LIBS)
   rm -f conf.pstoedittest
 ])
+
+
+])
