args: -r -u -- libtorrent-0.12.2_orig libtorrent-0.12.2
diff -r -u -- libtorrent-0.12.2_orig/rak/address_info.h libtorrent-0.12.2/rak/address_info.h
--- libtorrent-0.12.2_orig/rak/address_info.h	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/rak/address_info.h	2009-03-01 08:11:15.000000000 +0100
@@ -90,7 +90,7 @@
   hints.set_family(pfamily);
   hints.set_socket_type(stype);
   
-  return ::getaddrinfo(node, NULL, hints.c_addrinfo(), reinterpret_cast<addrinfo**>(ai));
+  return ::getaddrinfo(node, 0, hints.c_addrinfo(), reinterpret_cast<addrinfo**>(ai));
 }
 
 }
diff -r -u -- libtorrent-0.12.2_orig/rak/functional.h libtorrent-0.12.2/rak/functional.h
--- libtorrent-0.12.2_orig/rak/functional.h	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/rak/functional.h	2009-03-01 08:11:15.000000000 +0100
@@ -480,7 +480,7 @@
   typedef Ret result_type;
   typedef Ret (Object::*Function)();
 
-  mem_fun0() : m_object(NULL) {}
+  mem_fun0() : m_object(0) {}
   mem_fun0(Object* o, Function f) : m_object(o), m_function(f) {}
 
   bool is_valid() const { return m_object; }
@@ -498,7 +498,7 @@
   typedef Ret result_type;
   typedef Ret (Object::*Function)() const;
 
-  const_mem_fun0() : m_object(NULL) {}
+  const_mem_fun0() : m_object(0) {}
   const_mem_fun0(const Object* o, Function f) : m_object(o), m_function(f) {}
 
   bool is_valid() const { return m_object; }
@@ -516,7 +516,7 @@
   typedef Ret result_type;
   typedef Ret (Object::*Function)(Arg1);
 
-  mem_fun1() : m_object(NULL) {}
+  mem_fun1() : m_object(0) {}
   mem_fun1(Object* o, Function f) : m_object(o), m_function(f) {}
 
   bool is_valid() const { return m_object; }
@@ -534,7 +534,7 @@
   typedef Ret result_type;
   typedef Ret (Object::*Function)(Arg1) const;
 
-  const_mem_fun1() : m_object(NULL) {}
+  const_mem_fun1() : m_object(0) {}
   const_mem_fun1(const Object* o, Function f) : m_object(o), m_function(f) {}
 
   bool is_valid() const { return m_object; }
@@ -553,7 +553,7 @@
   typedef Ret (Object::*Function)(Arg1, Arg2);
   typedef Object object_type;
 
-  mem_fun2() : m_object(NULL) {}
+  mem_fun2() : m_object(0) {}
   mem_fun2(Object* o, Function f) : m_object(o), m_function(f) {}
 
   bool         is_valid() const { return m_object; }
@@ -574,7 +574,7 @@
   typedef Ret result_type;
   typedef Ret (Object::*Function)(Arg1, Arg2, Arg3);
 
-  mem_fun3() : m_object(NULL) {}
+  mem_fun3() : m_object(0) {}
   mem_fun3(Object* o, Function f) : m_object(o), m_function(f) {}
 
   bool is_valid() const { return m_object; }
diff -r -u -- libtorrent-0.12.2_orig/rak/functional_fun.h libtorrent-0.12.2/rak/functional_fun.h
--- libtorrent-0.12.2_orig/rak/functional_fun.h	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/rak/functional_fun.h	2009-03-01 08:11:15.000000000 +0100
@@ -94,7 +94,7 @@
   typedef Result                 result_type;
   typedef function_base0<Result> base_type;
 
-  bool                is_valid() const     { return m_base.get() != NULL; }
+  bool                is_valid() const     { return m_base.get() != 0; }
 
   void                set(base_type* base) { m_base = std::auto_ptr<base_type>(base); }
   base_type*          release()            { return m_base.release(); }
@@ -111,7 +111,7 @@
   typedef Result                       result_type;
   typedef function_base1<Result, Arg1> base_type;
 
-  bool                is_valid() const     { return m_base.get() != NULL; }
+  bool                is_valid() const     { return m_base.get() != 0; }
 
   void                set(base_type* base) { m_base = std::auto_ptr<base_type>(base); }
   base_type*          release()            { return m_base.release(); }
@@ -128,7 +128,7 @@
   typedef Result                             result_type;
   typedef function_base2<Result, Arg1, Arg2> base_type;
 
-  bool                is_valid() const     { return m_base.get() != NULL; }
+  bool                is_valid() const     { return m_base.get() != 0; }
 
   void                set(base_type* base) { m_base = std::auto_ptr<base_type>(base); }
   base_type*          release()            { return m_base.release(); }
@@ -145,7 +145,7 @@
   typedef Result                                   result_type;
   typedef function_base3<Result, Arg1, Arg2, Arg3> base_type;
 
-  bool                is_valid() const     { return m_base.get() != NULL; }
+  bool                is_valid() const     { return m_base.get() != 0; }
 
   void                set(base_type* base) { m_base = std::auto_ptr<base_type>(base); }
   base_type*          release()            { return m_base.release(); }
diff -r -u -- libtorrent-0.12.2_orig/rak/partial_queue.h libtorrent-0.12.2/rak/partial_queue.h
--- libtorrent-0.12.2_orig/rak/partial_queue.h	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/rak/partial_queue.h	2009-03-01 08:11:15.000000000 +0100
@@ -58,13 +58,13 @@
 
   static const size_type num_layers = 8;
 
-  partial_queue() : m_data(NULL), m_maxLayerSize(0) {}
+  partial_queue() : m_data(0), m_maxLayerSize(0) {}
   ~partial_queue() { disable(); }
 
   bool                is_full() const                         { return m_ceiling == 0; }
   bool                is_layer_full(size_type l) const        { return m_layers[l].second >= m_maxLayerSize; }
 
-  bool                is_enabled() const                      { return m_data != NULL; }
+  bool                is_enabled() const                      { return m_data != 0; }
 
   // Add check to see if we can add more. Also make it possible to
   // check how full we are in the lower parts so the caller knows when
@@ -124,14 +124,14 @@
 inline void
 partial_queue::disable() {
   delete [] m_data;
-  m_data = NULL;
+  m_data = 0;
 
   m_maxLayerSize = 0;
 }
 
 inline void
 partial_queue::clear() {
-  if (m_data == NULL)
+  if (m_data == 0)
     return;
 
   m_index = 0;
diff -r -u -- libtorrent-0.12.2_orig/rak/path.h libtorrent-0.12.2/rak/path.h
--- libtorrent-0.12.2_orig/rak/path.h	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/rak/path.h	2009-03-01 08:11:15.000000000 +0100
@@ -52,7 +52,7 @@
 
   char* home = std::getenv("HOME");
 
-  if (home == NULL)
+  if (home == 0)
     return path;
   
   return home + path.substr(1);
@@ -88,7 +88,7 @@
   if (*src == '~') {
     char* home = std::getenv("HOME");
 
-    if (home == NULL)
+    if (home == 0)
       return first;
 
     first += strlcpy(first, home, std::distance(first, last));
diff -r -u -- libtorrent-0.12.2_orig/rak/priority_queue.h libtorrent-0.12.2/rak/priority_queue.h
--- libtorrent-0.12.2_orig/rak/priority_queue.h	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/rak/priority_queue.h	2009-03-01 08:11:15.000000000 +0100
@@ -117,7 +117,7 @@
 public:
   typedef Queue container_type;
 
-  queue_pop_iterator() : m_queue(NULL) {}
+  queue_pop_iterator() : m_queue(0) {}
   queue_pop_iterator(Queue* q, Compare c) : m_queue(q), m_compare(c) {}
 
   queue_pop_iterator& operator ++ ()                     { m_queue->pop(); return *this; }
diff -r -u -- libtorrent-0.12.2_orig/rak/priority_queue_default.h libtorrent-0.12.2/rak/priority_queue_default.h
--- libtorrent-0.12.2_orig/rak/priority_queue_default.h	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/rak/priority_queue_default.h	2009-03-01 08:11:15.000000000 +0100
@@ -53,7 +53,7 @@
       throw std::logic_error("priority_item::~priority_item() called on a queued item.");
 
     m_time = timer();
-    m_slot.set(NULL);
+    m_slot.set(0);
   }
 
   bool                is_valid() const                        { return m_slot.is_valid(); }
diff -r -u -- libtorrent-0.12.2_orig/src/.deps/globals.Plo libtorrent-0.12.2/src/.deps/globals.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/.deps/manager.Plo libtorrent-0.12.2/src/.deps/manager.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/.deps/resource_manager.Plo libtorrent-0.12.2/src/.deps/resource_manager.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/data/.deps/chunk.Plo libtorrent-0.12.2/src/data/.deps/chunk.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/data/.deps/chunk_list.Plo libtorrent-0.12.2/src/data/.deps/chunk_list.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/data/.deps/chunk_part.Plo libtorrent-0.12.2/src/data/.deps/chunk_part.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/data/.deps/hash_chunk.Plo libtorrent-0.12.2/src/data/.deps/hash_chunk.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/data/.deps/hash_queue.Plo libtorrent-0.12.2/src/data/.deps/hash_queue.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/data/.deps/hash_queue_node.Plo libtorrent-0.12.2/src/data/.deps/hash_queue_node.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/data/.deps/hash_torrent.Plo libtorrent-0.12.2/src/data/.deps/hash_torrent.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/data/.deps/memory_chunk.Plo libtorrent-0.12.2/src/data/.deps/memory_chunk.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/data/.deps/socket_file.Plo libtorrent-0.12.2/src/data/.deps/socket_file.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/data/chunk.cc libtorrent-0.12.2/src/data/chunk.cc
--- libtorrent-0.12.2_orig/src/data/chunk.cc	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/src/data/chunk.cc	2009-03-01 08:11:15.000000000 +0100
@@ -38,6 +38,7 @@
 
 #include <algorithm>
 #include <functional>
+#include <cstring>
 
 #include "torrent/exceptions.h"
 
diff -r -u -- libtorrent-0.12.2_orig/src/data/chunk_list.h libtorrent-0.12.2/src/data/chunk_list.h
--- libtorrent-0.12.2_orig/src/data/chunk_list.h	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/src/data/chunk_list.h	2009-03-01 08:11:15.000000000 +0100
@@ -37,6 +37,7 @@
 #ifndef LIBTORRENT_DATA_CHUNK_LIST_H
 #define LIBTORRENT_DATA_CHUNK_LIST_H
 
+#include <string>
 #include <vector>
 #include <rak/error_number.h>
 #include <rak/functional.h>
diff -r -u -- libtorrent-0.12.2_orig/src/dht/.deps/dht_bucket.Plo libtorrent-0.12.2/src/dht/.deps/dht_bucket.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/dht/.deps/dht_node.Plo libtorrent-0.12.2/src/dht/.deps/dht_node.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/dht/.deps/dht_router.Plo libtorrent-0.12.2/src/dht/.deps/dht_router.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/dht/.deps/dht_server.Plo libtorrent-0.12.2/src/dht/.deps/dht_server.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/dht/.deps/dht_tracker.Plo libtorrent-0.12.2/src/dht/.deps/dht_tracker.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/dht/.deps/dht_transaction.Plo libtorrent-0.12.2/src/dht/.deps/dht_transaction.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/download/.deps/available_list.Plo libtorrent-0.12.2/src/download/.deps/available_list.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/download/.deps/choke_manager.Plo libtorrent-0.12.2/src/download/.deps/choke_manager.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/download/.deps/chunk_selector.Plo libtorrent-0.12.2/src/download/.deps/chunk_selector.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/download/.deps/chunk_statistics.Plo libtorrent-0.12.2/src/download/.deps/chunk_statistics.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/download/.deps/delegator.Plo libtorrent-0.12.2/src/download/.deps/delegator.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/download/.deps/download_constructor.Plo libtorrent-0.12.2/src/download/.deps/download_constructor.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/download/.deps/download_main.Plo libtorrent-0.12.2/src/download/.deps/download_main.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/download/.deps/download_manager.Plo libtorrent-0.12.2/src/download/.deps/download_manager.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/download/.deps/download_wrapper.Plo libtorrent-0.12.2/src/download/.deps/download_wrapper.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/net/.deps/address_list.Plo libtorrent-0.12.2/src/net/.deps/address_list.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/net/.deps/listen.Plo libtorrent-0.12.2/src/net/.deps/listen.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/net/.deps/socket_base.Plo libtorrent-0.12.2/src/net/.deps/socket_base.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/net/.deps/socket_datagram.Plo libtorrent-0.12.2/src/net/.deps/socket_datagram.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/net/.deps/socket_fd.Plo libtorrent-0.12.2/src/net/.deps/socket_fd.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/net/.deps/socket_set.Plo libtorrent-0.12.2/src/net/.deps/socket_set.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/net/.deps/socket_stream.Plo libtorrent-0.12.2/src/net/.deps/socket_stream.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/net/.deps/throttle_list.Plo libtorrent-0.12.2/src/net/.deps/throttle_list.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/net/address_list.cc libtorrent-0.12.2/src/net/address_list.cc
--- libtorrent-0.12.2_orig/src/net/address_list.cc	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/src/net/address_list.cc	2009-03-01 08:11:15.000000000 +0100
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <algorithm>
+
 #include <rak/functional.h>
 
 #include "download/download_info.h"  // for SocketAddressCompact
diff -r -u -- libtorrent-0.12.2_orig/src/protocol/.deps/extensions.Plo libtorrent-0.12.2/src/protocol/.deps/extensions.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/protocol/.deps/handshake.Plo libtorrent-0.12.2/src/protocol/.deps/handshake.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/protocol/.deps/handshake_encryption.Plo libtorrent-0.12.2/src/protocol/.deps/handshake_encryption.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/protocol/.deps/handshake_manager.Plo libtorrent-0.12.2/src/protocol/.deps/handshake_manager.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/protocol/.deps/initial_seed.Plo libtorrent-0.12.2/src/protocol/.deps/initial_seed.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/protocol/.deps/peer_connection_base.Plo libtorrent-0.12.2/src/protocol/.deps/peer_connection_base.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/protocol/.deps/peer_connection_leech.Plo libtorrent-0.12.2/src/protocol/.deps/peer_connection_leech.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/protocol/.deps/peer_factory.Plo libtorrent-0.12.2/src/protocol/.deps/peer_factory.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/protocol/.deps/request_list.Plo libtorrent-0.12.2/src/protocol/.deps/request_list.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/bitfield.Plo libtorrent-0.12.2/src/torrent/.deps/bitfield.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/chunk_manager.Plo libtorrent-0.12.2/src/torrent/.deps/chunk_manager.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/connection_manager.Plo libtorrent-0.12.2/src/torrent/.deps/connection_manager.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/dht_manager.Plo libtorrent-0.12.2/src/torrent/.deps/dht_manager.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/download.Plo libtorrent-0.12.2/src/torrent/.deps/download.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/error.Plo libtorrent-0.12.2/src/torrent/.deps/error.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/exceptions.Plo libtorrent-0.12.2/src/torrent/.deps/exceptions.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/http.Plo libtorrent-0.12.2/src/torrent/.deps/http.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/object.Plo libtorrent-0.12.2/src/torrent/.deps/object.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/object_stream.Plo libtorrent-0.12.2/src/torrent/.deps/object_stream.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/path.Plo libtorrent-0.12.2/src/torrent/.deps/path.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/poll_epoll.Plo libtorrent-0.12.2/src/torrent/.deps/poll_epoll.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/poll_kqueue.Plo libtorrent-0.12.2/src/torrent/.deps/poll_kqueue.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/poll_select.Plo libtorrent-0.12.2/src/torrent/.deps/poll_select.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/rate.Plo libtorrent-0.12.2/src/torrent/.deps/rate.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/resume.Plo libtorrent-0.12.2/src/torrent/.deps/resume.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/throttle.Plo libtorrent-0.12.2/src/torrent/.deps/throttle.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/torrent.Plo libtorrent-0.12.2/src/torrent/.deps/torrent.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/tracker.Plo libtorrent-0.12.2/src/torrent/.deps/tracker.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/.deps/tracker_list.Plo libtorrent-0.12.2/src/torrent/.deps/tracker_list.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/data/.deps/file_list_iterator.Plo libtorrent-0.12.2/src/torrent/data/.deps/file_list_iterator.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/data/.deps/file_manager.Plo libtorrent-0.12.2/src/torrent/data/.deps/file_manager.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/data/.deps/file_utils.Plo libtorrent-0.12.2/src/torrent/data/.deps/file_utils.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/data/.deps/transfer_list.Plo libtorrent-0.12.2/src/torrent/data/.deps/transfer_list.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/data/file_list_iterator.h libtorrent-0.12.2/src/torrent/data/file_list_iterator.h
--- libtorrent-0.12.2_orig/src/torrent/data/file_list_iterator.h	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/src/torrent/data/file_list_iterator.h	2009-03-01 08:11:15.000000000 +0100
@@ -40,6 +40,8 @@
 #include <torrent/common.h>
 #include <torrent/data/file_list.h>
 
+#include <cmath>
+
 namespace torrent {
 
 class File;
@@ -61,7 +63,7 @@
   bool                is_entering() const;
   bool                is_leaving() const  { return m_depth < 0; }
 
-  uint32_t            depth() const       { return std::abs(m_depth); }
+  uint32_t            depth() const       { return std::abs((float)m_depth); }
 
   iterator            base() const        { return m_position; }
   reference           file() const        { return *m_position; }
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/exceptions.cc libtorrent-0.12.2/src/torrent/exceptions.cc
--- libtorrent-0.12.2_orig/src/torrent/exceptions.cc	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/src/torrent/exceptions.cc	2009-03-01 08:11:15.000000000 +0100
@@ -37,6 +37,7 @@
 #include "config.h"
 
 #include <cerrno>
+#include <cstring>
 
 #include "exceptions.h"
 
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/peer/.deps/client_info.Plo libtorrent-0.12.2/src/torrent/peer/.deps/client_info.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/peer/.deps/client_list.Plo libtorrent-0.12.2/src/torrent/peer/.deps/client_list.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/peer/.deps/connection_list.Plo libtorrent-0.12.2/src/torrent/peer/.deps/connection_list.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/peer/.deps/peer.Plo libtorrent-0.12.2/src/torrent/peer/.deps/peer.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/peer/.deps/peer_info.Plo libtorrent-0.12.2/src/torrent/peer/.deps/peer_info.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/peer/.deps/peer_list.Plo libtorrent-0.12.2/src/torrent/peer/.deps/peer_list.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/torrent/tracker_list.h libtorrent-0.12.2/src/torrent/tracker_list.h
--- libtorrent-0.12.2_orig/src/torrent/tracker_list.h	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/src/torrent/tracker_list.h	2009-03-01 08:24:20.000000000 +0100
@@ -40,6 +40,7 @@
 #include <algorithm>
 #include <vector>
 #include <torrent/common.h>
+#include <string>
 
 namespace torrent {
 
diff -r -u -- libtorrent-0.12.2_orig/src/tracker/.deps/tracker_dht.Plo libtorrent-0.12.2/src/tracker/.deps/tracker_dht.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/tracker/.deps/tracker_http.Plo libtorrent-0.12.2/src/tracker/.deps/tracker_http.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/tracker/.deps/tracker_manager.Plo libtorrent-0.12.2/src/tracker/.deps/tracker_manager.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/tracker/.deps/tracker_udp.Plo libtorrent-0.12.2/src/tracker/.deps/tracker_udp.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/utils/.deps/diffie_hellman.Plo libtorrent-0.12.2/src/utils/.deps/diffie_hellman.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/utils/.deps/sha_fast.Plo libtorrent-0.12.2/src/utils/.deps/sha_fast.Plo
diff -r -u -- libtorrent-0.12.2_orig/src/utils/diffie_hellman.cc libtorrent-0.12.2/src/utils/diffie_hellman.cc
--- libtorrent-0.12.2_orig/src/utils/diffie_hellman.cc	2009-03-01 08:10:49.000000000 +0100
+++ libtorrent-0.12.2/src/utils/diffie_hellman.cc	2009-03-01 08:11:15.000000000 +0100
@@ -37,6 +37,7 @@
 #include "config.h"
 
 #include <string>
+#include <cstring>
 
 #ifdef USE_OPENSSL
 #include <openssl/bn.h>
