--- binutils-2.18.50_orig/bfd/elf.c	2008-03-10 14:53:34.000000000 +0100
+++ binutils-2.18.50/bfd/elf.c	2008-03-10 14:53:48.000000000 +0100
@@ -6383,9 +6383,8 @@
 	  if (type == STT_OBJECT)
 	    sym.st_info = ELF_ST_INFO (STB_GLOBAL, STT_COMMON);
 	  else
-#else
-	    sym.st_info = ELF_ST_INFO (STB_GLOBAL, type);
 #endif
+	    sym.st_info = ELF_ST_INFO (STB_GLOBAL, type);
 	}
       else if (bfd_is_und_section (syms[idx]->section))
 	sym.st_info = ELF_ST_INFO (((flags & BSF_WEAK)
