Only in libishtarservices-1.0/ishtar: libishtarservices.a
diff -ru libishtarservices-1.0_orig/ishtar/services.cpp libishtarservices-1.0/ishtar/services.cpp
--- libishtarservices-1.0_orig/ishtar/services.cpp	2009-02-25 23:12:58.000000000 +0100
+++ libishtarservices-1.0/ishtar/services.cpp	2009-02-25 23:13:09.000000000 +0100
@@ -22,6 +22,7 @@
 #include "services.h"
 #include <sstream>
 #include <list>
+#include <cstring>
 
 /*!	\file services.cpp
 	\brief Implementation of Ishtar services classes.
Only in libishtarservices-1.0/ishtar: services.o
