--- gettext-0.17_orig/gettext-runtime/intl/dcigettext.c	2009-04-20 09:52:56.000000000 +0200
+++ gettext-0.17/gettext-runtime/intl/dcigettext.c	2009-04-20 10:04:23.000000000 +0200
@@ -553,7 +553,7 @@
 #  ifdef _LIBC
   localename = __current_locale_name (category);
 #  else
-#   if HAVE_NL_LOCALE_NAME
+#   if HAVE_NL_LOCALE_NAME && defined(NL_LOCALE_NAME)
   /* NL_LOCALE_NAME is public glibc API introduced in glibc-2.4.  */
   localename = nl_langinfo (NL_LOCALE_NAME (category));
 #   else
