diff -r -u CSL_orig/CSL/IO/FileIO.cpp CSL/CSL/IO/FileIO.cpp
--- CSL_orig/CSL/IO/FileIO.cpp	2007-07-31 21:54:49.000000000 +0200
+++ CSL/CSL/IO/FileIO.cpp	2007-07-31 22:12:50.000000000 +0200
@@ -4,6 +4,7 @@
 //
 
 #include "FileIO.h"
+#include <cstring>
 
 using namespace csl;
 
diff -r -u CSL_orig/CSL/IO/NullIO.h CSL/CSL/IO/NullIO.h
--- CSL_orig/CSL/IO/NullIO.h	2007-07-31 21:54:49.000000000 +0200
+++ CSL/CSL/IO/NullIO.h	2007-07-31 22:13:51.000000000 +0200
@@ -39,5 +39,5 @@
 
 }	// end of namespace
 
-#endif CSL_NULLIO_H
+#endif //ndef CSL_NULLIO_H
 
diff -r -u CSL_orig/CSL/IO/PAIO.cpp CSL/CSL/IO/PAIO.cpp
--- CSL_orig/CSL/IO/PAIO.cpp	2007-07-31 21:54:49.000000000 +0200
+++ CSL/CSL/IO/PAIO.cpp	2007-07-31 22:09:52.000000000 +0200
@@ -6,6 +6,8 @@
 #include "PAIO.h"
 #include "Interleaver.h"
 #include <iostream>
+#include <cstring>
+#include <cstdlib>
 
 using namespace csl;
 
diff -r -u CSL_orig/CSL/Processors/Convolver.cpp CSL/CSL/Processors/Convolver.cpp
--- CSL_orig/CSL/Processors/Convolver.cpp	2007-07-31 21:54:49.000000000 +0200
+++ CSL/CSL/Processors/Convolver.cpp	2007-07-31 22:05:36.000000000 +0200
@@ -4,6 +4,7 @@
 /// 
 
 #include "Convolver.h"
+#include <cstring>
 
 using namespace csl;
 
diff -r -u CSL_orig/CSL/Processors/FIR.cpp CSL/CSL/Processors/FIR.cpp
--- CSL_orig/CSL/Processors/FIR.cpp	2007-07-31 21:54:49.000000000 +0200
+++ CSL/CSL/Processors/FIR.cpp	2007-07-31 22:03:13.000000000 +0200
@@ -6,6 +6,7 @@
 #include "FIR.h"
 #include <stdlib.h>
 #include <math.h>
+#include <string.h>
 
 using namespace csl;
 
diff -r -u CSL_orig/CSL/Sources/Spectral.cpp CSL/CSL/Sources/Spectral.cpp
--- CSL_orig/CSL/Sources/Spectral.cpp	2007-07-31 21:54:49.000000000 +0200
+++ CSL/CSL/Sources/Spectral.cpp	2007-07-31 22:01:29.000000000 +0200
@@ -7,6 +7,7 @@
 #include "Window.h"
 #include <stdlib.h>
 #include <math.h>
+#include <string.h>
 
 using namespace csl;
 
