diff -ru libtholoura-0.1.0_orig/plugins/Makefile libtholoura-0.1.0/plugins/Makefile
--- libtholoura-0.1.0_orig/plugins/Makefile	2009-02-28 11:42:07.000000000 +0100
+++ libtholoura-0.1.0/plugins/Makefile	2009-02-28 11:43:46.000000000 +0100
@@ -9,11 +9,11 @@
 endif
 
 all:
-	for i in `ls -l | grep ^d | ${AWK} '{print $$8}'` ; do \
+	for i in `find * -type d -maxdepth 0` ; do \
 		(cd $$i ; if [ -f Makefile ] ; then make all ; fi) ;\
 	done
 
 distclean:
-	for i in `ls -l | grep ^d | ${AWK} '{print $$8}'` ; do \
+	for i in `find * -type d -maxdepth 0` ; do \
 		(cd $$i ; if [ -f Makefile ] ; then make distclean ; fi) ;\
 	done
Only in libtholoura-0.1.0/plugins/chess: chess.o
Only in libtholoura-0.1.0/plugins/chess: chess.so
Only in libtholoura-0.1.0/plugins/diagonal: diagonal.o
Only in libtholoura-0.1.0/plugins/diagonal: diagonal.so
Only in libtholoura-0.1.0/plugins/dither: dither.o
Only in libtholoura-0.1.0/plugins/dither: dither.so
Only in libtholoura-0.1.0/plugins/dizzy: dizzy.o
Only in libtholoura-0.1.0/plugins/dizzy: dizzy.so
Only in libtholoura-0.1.0/plugins/grid: grid.o
Only in libtholoura-0.1.0/plugins/grid: grid.so
Only in libtholoura-0.1.0/plugins/lines: lines.o
Only in libtholoura-0.1.0/plugins/lines: lines.so
Only in libtholoura-0.1.0/plugins/noise: noise.o
Only in libtholoura-0.1.0/plugins/noise: noise.so
Only in libtholoura-0.1.0/plugins/plasma: plasma.o
Only in libtholoura-0.1.0/plugins/plasma: plasma.so
Only in libtholoura-0.1.0/plugins/rwalk: rwalk.o
Only in libtholoura-0.1.0/plugins/rwalk: rwalk.so
Only in libtholoura-0.1.0/plugins/sines: sines.o
Only in libtholoura-0.1.0/plugins/sines: sines.so
Only in libtholoura-0.1.0/plugins/stir: stir.o
Only in libtholoura-0.1.0/plugins/stir: stir.so
Only in libtholoura-0.1.0/plugins/stripes: stripes.o
Only in libtholoura-0.1.0/plugins/stripes: stripes.so
Only in libtholoura-0.1.0/plugins/wipeout: wipeout.o
Only in libtholoura-0.1.0/plugins/wipeout: wipeout.so
Only in libtholoura-0.1.0/src: export.o
Only in libtholoura-0.1.0/src: library.o
Only in libtholoura-0.1.0/src: libtholoura.so
Only in libtholoura-0.1.0/src: plugins.o
Only in libtholoura-0.1.0/src: text.o
Only in libtholoura-0.1.0/test: test
Only in libtholoura-0.1.0/test: test.o
