diff -wru stk-4.3.1_orig/include/ADSR.h stk-4.3.1/include/ADSR.h
--- stk-4.3.1_orig/include/ADSR.h	2007-12-06 01:58:33.000000000 +0100
+++ stk-4.3.1/include/ADSR.h	2008-01-11 12:13:53.609375000 +0100
@@ -20,7 +20,7 @@
 
 #include "Envelope.h"
 
-class ADSR : public Envelope
+class STK_EXPORT ADSR : public Envelope
 {
  public:
 
diff -wru stk-4.3.1_orig/include/Asymp.h stk-4.3.1/include/Asymp.h
--- stk-4.3.1_orig/include/Asymp.h	2007-12-06 01:58:33.000000000 +0100
+++ stk-4.3.1/include/Asymp.h	2008-01-11 12:13:53.609375000 +0100
@@ -30,7 +30,7 @@
 
 const StkFloat TARGET_THRESHOLD = 0.000001;
 
-class Asymp : public Envelope
+class STK_EXPORT Asymp : public Envelope
 {
  public:
 
diff -wru stk-4.3.1_orig/include/BandedWG.h stk-4.3.1/include/BandedWG.h
--- stk-4.3.1_orig/include/BandedWG.h	2007-12-06 01:58:33.000000000 +0100
+++ stk-4.3.1/include/BandedWG.h	2008-01-11 12:13:53.609375000 +0100
@@ -40,7 +40,7 @@
 #include "ADSR.h"
 #include "BiQuad.h"
 
-class BandedWG : public Instrmnt
+class STK_EXPORT BandedWG : public Instrmnt
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/BeeThree.h stk-4.3.1/include/BeeThree.h
--- stk-4.3.1_orig/include/BeeThree.h	2007-12-06 01:58:33.000000000 +0100
+++ stk-4.3.1/include/BeeThree.h	2008-01-11 12:13:53.625000000 +0100
@@ -37,7 +37,7 @@
 
 #include "FM.h"
 
-class BeeThree : public FM
+class STK_EXPORT BeeThree : public FM
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/BiQuad.h stk-4.3.1/include/BiQuad.h
--- stk-4.3.1_orig/include/BiQuad.h	2007-12-06 01:58:33.000000000 +0100
+++ stk-4.3.1/include/BiQuad.h	2008-01-11 12:13:53.625000000 +0100
@@ -17,7 +17,7 @@
 
 #include "Filter.h"
 
-class BiQuad : protected Filter
+class STK_EXPORT BiQuad : protected Filter
 {
 public:
 
diff -wru stk-4.3.1_orig/include/Blit.h stk-4.3.1/include/Blit.h
--- stk-4.3.1_orig/include/Blit.h	2007-12-06 01:58:33.000000000 +0100
+++ stk-4.3.1/include/Blit.h	2008-01-11 12:13:53.625000000 +0100
@@ -26,7 +26,7 @@
 
 #include "Generator.h"
 
-class Blit: public Generator
+class STK_EXPORT Blit: public Generator
 {
  public:
   //! Default constructor that initializes BLIT frequency to 220 Hz.
diff -wru stk-4.3.1_orig/include/BlitSaw.h stk-4.3.1/include/BlitSaw.h
--- stk-4.3.1_orig/include/BlitSaw.h	2007-12-06 01:58:33.000000000 +0100
+++ stk-4.3.1/include/BlitSaw.h	2008-01-11 12:13:53.625000000 +0100
@@ -24,7 +24,7 @@
 
 #include "Generator.h"
 
-class BlitSaw: public Generator
+class STK_EXPORT BlitSaw: public Generator
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/BlitSquare.h stk-4.3.1/include/BlitSquare.h
--- stk-4.3.1_orig/include/BlitSquare.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/BlitSquare.h	2008-01-11 12:13:53.640625000 +0100
@@ -35,7 +35,7 @@
 
 #include "Generator.h"
 
-class BlitSquare: public Generator
+class STK_EXPORT BlitSquare: public Generator
 {
  public:
   //! Default constructor that initializes BLIT frequency to 220 Hz.
diff -wru stk-4.3.1_orig/include/BlowBotl.h stk-4.3.1/include/BlowBotl.h
--- stk-4.3.1_orig/include/BlowBotl.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/BlowBotl.h	2008-01-11 12:13:53.640625000 +0100
@@ -27,7 +27,7 @@
 #include "ADSR.h"
 #include "SineWave.h"
 
-class BlowBotl : public Instrmnt
+class STK_EXPORT BlowBotl : public Instrmnt
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/BlowHole.h stk-4.3.1/include/BlowHole.h
--- stk-4.3.1_orig/include/BlowHole.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/BlowHole.h	2008-01-11 12:13:53.640625000 +0100
@@ -45,7 +45,7 @@
 #include "Noise.h"
 #include "SineWave.h"
 
-class BlowHole : public Instrmnt
+class STK_EXPORT BlowHole : public Instrmnt
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/BowTable.h stk-4.3.1/include/BowTable.h
--- stk-4.3.1_orig/include/BowTable.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/BowTable.h	2008-01-11 12:13:53.640625000 +0100
@@ -14,7 +14,7 @@
 
 #include "Function.h"
 
-class BowTable : public Function
+class STK_EXPORT BowTable : public Function
 {
 public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/Bowed.h stk-4.3.1/include/Bowed.h
--- stk-4.3.1_orig/include/Bowed.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Bowed.h	2008-01-11 12:13:53.656250000 +0100
@@ -32,7 +32,7 @@
 #include "SineWave.h"
 #include "ADSR.h"
 
-class Bowed : public Instrmnt
+class STK_EXPORT Bowed : public Instrmnt
 {
  public:
   //! Class constructor, taking the lowest desired playing frequency.
diff -wru stk-4.3.1_orig/include/Brass.h stk-4.3.1/include/Brass.h
--- stk-4.3.1_orig/include/Brass.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Brass.h	2008-01-11 12:13:53.656250000 +0100
@@ -30,7 +30,7 @@
 #include "ADSR.h"
 #include "SineWave.h"
 
-class Brass: public Instrmnt
+class STK_EXPORT Brass: public Instrmnt
 {
  public:
   //! Class constructor, taking the lowest desired playing frequency.
diff -wru stk-4.3.1_orig/include/Chorus.h stk-4.3.1/include/Chorus.h
--- stk-4.3.1_orig/include/Chorus.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Chorus.h	2008-01-11 12:13:53.656250000 +0100
@@ -15,7 +15,7 @@
 #include "DelayL.h" 
 #include "SineWave.h" 
 
-class Chorus : public Effect
+class STK_EXPORT Chorus : public Effect
 {
  public:
   //! Class constructor, taking the median desired delay length.
diff -wru stk-4.3.1_orig/include/Clarinet.h stk-4.3.1/include/Clarinet.h
--- stk-4.3.1_orig/include/Clarinet.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Clarinet.h	2008-01-11 12:13:53.656250000 +0100
@@ -33,7 +33,7 @@
 #include "Noise.h"
 #include "SineWave.h"
 
-class Clarinet : public Instrmnt
+class STK_EXPORT Clarinet : public Instrmnt
 {
  public:
   //! Class constructor, taking the lowest desired playing frequency.
diff -wru stk-4.3.1_orig/include/Delay.h stk-4.3.1/include/Delay.h
--- stk-4.3.1_orig/include/Delay.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Delay.h	2008-01-11 12:13:53.656250000 +0100
@@ -23,7 +23,7 @@
 
 #include "Filter.h"
 
-class Delay : protected Filter
+class STK_EXPORT Delay : protected Filter
 {
 public:
 
diff -wru stk-4.3.1_orig/include/DelayA.h stk-4.3.1/include/DelayA.h
--- stk-4.3.1_orig/include/DelayA.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/DelayA.h	2008-01-11 12:13:53.671875000 +0100
@@ -24,7 +24,7 @@
 
 #include "Delay.h"
 
-class DelayA : public Delay
+class STK_EXPORT DelayA : public Delay
 {
 public:
 
diff -wru stk-4.3.1_orig/include/DelayL.h stk-4.3.1/include/DelayL.h
--- stk-4.3.1_orig/include/DelayL.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/DelayL.h	2008-01-11 12:13:53.671875000 +0100
@@ -27,7 +27,7 @@
 
 #include "Delay.h"
 
-class DelayL : public Delay
+class STK_EXPORT DelayL : public Delay
 {
 public:
 
diff -wru stk-4.3.1_orig/include/Drummer.h stk-4.3.1/include/Drummer.h
--- stk-4.3.1_orig/include/Drummer.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Drummer.h	2008-01-11 12:13:53.671875000 +0100
@@ -25,7 +25,7 @@
 const int DRUM_NUMWAVES = 11;
 const int DRUM_POLYPHONY = 4;
 
-class Drummer : public Instrmnt
+class STK_EXPORT Drummer : public Instrmnt
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Echo.h stk-4.3.1/include/Echo.h
--- stk-4.3.1_orig/include/Echo.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Echo.h	2008-01-11 12:13:53.671875000 +0100
@@ -14,7 +14,7 @@
 #include "Effect.h" 
 #include "Delay.h" 
 
-class Echo : public Effect
+class STK_EXPORT Echo : public Effect
 {
  public:
   //! Class constructor, taking the longest desired delay length (one second default value).
diff -wru stk-4.3.1_orig/include/Effect.h stk-4.3.1/include/Effect.h
--- stk-4.3.1_orig/include/Effect.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Effect.h	2008-01-11 12:13:53.671875000 +0100
@@ -14,7 +14,7 @@
 #ifndef STK_EFFECT_H
 #define STK_EFFECT_H
 
-class Effect : public Stk
+class STK_EXPORT Effect : public Stk
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Envelope.h stk-4.3.1/include/Envelope.h
--- stk-4.3.1_orig/include/Envelope.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Envelope.h	2008-01-11 12:13:53.687500000 +0100
@@ -18,7 +18,7 @@
 
 #include "Generator.h"
 
-class Envelope : public Generator
+class STK_EXPORT Envelope : public Generator
 {
  public:
 
diff -wru stk-4.3.1_orig/include/FM.h stk-4.3.1/include/FM.h
--- stk-4.3.1_orig/include/FM.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/FM.h	2008-01-11 12:13:53.687500000 +0100
@@ -32,7 +32,7 @@
 #include "SineWave.h"
 #include "TwoZero.h"
 
-class FM : public Instrmnt
+class STK_EXPORT FM : public Instrmnt
 {
  public:
   //! Class constructor, taking the number of wave/envelope operators to control.
diff -wru stk-4.3.1_orig/include/FMVoices.h stk-4.3.1/include/FMVoices.h
--- stk-4.3.1_orig/include/FMVoices.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/FMVoices.h	2008-01-11 12:13:53.687500000 +0100
@@ -35,7 +35,7 @@
 
 #include "FM.h"
 
-class FMVoices : public FM
+class STK_EXPORT FMVoices : public FM
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/FileRead.h stk-4.3.1/include/FileRead.h
--- stk-4.3.1_orig/include/FileRead.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/FileRead.h	2008-01-11 12:13:53.687500000 +0100
@@ -34,7 +34,7 @@
 
 #include "Stk.h"
 
-class FileRead : public Stk
+class STK_EXPORT FileRead : public Stk
 {
 public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/FileWrite.h stk-4.3.1/include/FileWrite.h
--- stk-4.3.1_orig/include/FileWrite.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/FileWrite.h	2008-01-11 12:13:53.703125000 +0100
@@ -26,7 +26,7 @@
 
 #include "Stk.h"
 
-class FileWrite : public Stk
+class STK_EXPORT FileWrite : public Stk
 {
  public:
 
diff -wru stk-4.3.1_orig/include/FileWvIn.h stk-4.3.1/include/FileWvIn.h
--- stk-4.3.1_orig/include/FileWvIn.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/FileWvIn.h	2008-01-11 12:13:53.703125000 +0100
@@ -37,7 +37,7 @@
 #include "WvIn.h"
 #include "FileRead.h"
 
-class FileWvIn : public WvIn
+class STK_EXPORT FileWvIn : public WvIn
 {
 public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/FileWvOut.h stk-4.3.1/include/FileWvOut.h
--- stk-4.3.1_orig/include/FileWvOut.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/FileWvOut.h	2008-01-11 12:13:53.703125000 +0100
@@ -27,7 +27,7 @@
 #include "WvOut.h"
 #include "FileWrite.h"
 
-class FileWvOut : public WvOut
+class STK_EXPORT FileWvOut : public WvOut
 {
  public:
 
diff -wru stk-4.3.1_orig/include/Filter.h stk-4.3.1/include/Filter.h
--- stk-4.3.1_orig/include/Filter.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Filter.h	2008-01-11 12:13:53.718750000 +0100
@@ -33,7 +33,7 @@
 #include "Stk.h"
 #include <vector>
 
-class Filter : public Stk
+class STK_EXPORT Filter : public Stk
 {
 public:
   //! Default constructor creates a zero-order pass-through "filter".
diff -wru stk-4.3.1_orig/include/Flute.h stk-4.3.1/include/Flute.h
--- stk-4.3.1_orig/include/Flute.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Flute.h	2008-01-11 12:13:53.718750000 +0100
@@ -34,7 +34,7 @@
 #include "ADSR.h"
 #include "SineWave.h"
 
-class Flute : public Instrmnt
+class STK_EXPORT Flute : public Instrmnt
 {
  public:
   //! Class constructor, taking the lowest desired playing frequency.
diff -wru stk-4.3.1_orig/include/FormSwep.h stk-4.3.1/include/FormSwep.h
--- stk-4.3.1_orig/include/FormSwep.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/FormSwep.h	2008-01-11 12:13:53.718750000 +0100
@@ -17,7 +17,7 @@
 
 #include "BiQuad.h"
 
-class FormSwep : public BiQuad
+class STK_EXPORT FormSwep : public BiQuad
 {
  public:
 
diff -wru stk-4.3.1_orig/include/Function.h stk-4.3.1/include/Function.h
--- stk-4.3.1_orig/include/Function.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Function.h	2008-01-11 12:13:53.718750000 +0100
@@ -15,7 +15,7 @@
 #ifndef STK_FUNCTION_H
 #define STK_FUNCTION_H
 
-class Function : public Stk
+class STK_EXPORT Function : public Stk
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Generator.h stk-4.3.1/include/Generator.h
--- stk-4.3.1_orig/include/Generator.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Generator.h	2008-01-11 12:13:53.718750000 +0100
@@ -14,7 +14,7 @@
 
 #include "Stk.h"
 
-class Generator : public Stk
+class STK_EXPORT Generator : public Stk
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Granulate.h stk-4.3.1/include/Granulate.h
--- stk-4.3.1_orig/include/Granulate.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Granulate.h	2008-01-11 12:13:53.734375000 +0100
@@ -23,7 +23,7 @@
 #include "Envelope.h"
 #include "Noise.h"
 
-class Granulate: public Generator
+class STK_EXPORT Granulate: public Generator
 {
  public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/HevyMetl.h stk-4.3.1/include/HevyMetl.h
--- stk-4.3.1_orig/include/HevyMetl.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/HevyMetl.h	2008-01-11 12:13:53.734375000 +0100
@@ -33,7 +33,7 @@
 
 #include "FM.h"
 
-class HevyMetl : public FM
+class STK_EXPORT HevyMetl : public FM
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/InetWvIn.h stk-4.3.1/include/InetWvIn.h
--- stk-4.3.1_orig/include/InetWvIn.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/InetWvIn.h	2008-01-11 12:13:53.734375000 +0100
@@ -38,7 +38,7 @@
   void *object;
 } ThreadInfo;
 
-class InetWvIn : public WvIn
+class STK_EXPORT InetWvIn : public WvIn
 {
 public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/InetWvOut.h stk-4.3.1/include/InetWvOut.h
--- stk-4.3.1_orig/include/InetWvOut.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/InetWvOut.h	2008-01-11 12:13:53.734375000 +0100
@@ -26,7 +26,7 @@
 #include "WvOut.h"
 #include "Socket.h"
 
-class InetWvOut : public WvOut
+class STK_EXPORT InetWvOut : public WvOut
 {
  public:
   //! Default constructor ... the socket is not instantiated.
diff -wru stk-4.3.1_orig/include/Instrmnt.h stk-4.3.1/include/Instrmnt.h
--- stk-4.3.1_orig/include/Instrmnt.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Instrmnt.h	2008-01-11 12:13:53.750000000 +0100
@@ -14,7 +14,7 @@
 
 #include "Stk.h"
 
-class Instrmnt : public Stk
+class STK_EXPORT Instrmnt : public Stk
 {
  public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/JCRev.h stk-4.3.1/include/JCRev.h
--- stk-4.3.1_orig/include/JCRev.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/JCRev.h	2008-01-11 12:13:53.750000000 +0100
@@ -20,7 +20,7 @@
 #include "Effect.h"
 #include "Delay.h" 
 
-class JCRev : public Effect
+class STK_EXPORT JCRev : public Effect
 {
  public:
   //! Class constructor taking a T60 decay time argument (one second default value).
diff -wru stk-4.3.1_orig/include/JetTable.h stk-4.3.1/include/JetTable.h
--- stk-4.3.1_orig/include/JetTable.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/JetTable.h	2008-01-11 12:13:53.750000000 +0100
@@ -18,7 +18,7 @@
 
 #include "Function.h"
 
-class JetTable : public Function
+class STK_EXPORT JetTable : public Function
 {
 public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/Mandolin.h stk-4.3.1/include/Mandolin.h
--- stk-4.3.1_orig/include/Mandolin.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Mandolin.h	2008-01-11 12:13:53.750000000 +0100
@@ -33,7 +33,7 @@
 #include "PluckTwo.h"
 #include "FileWvIn.h"
 
-class Mandolin : public PluckTwo
+class STK_EXPORT Mandolin : public PluckTwo
 {
  public:
   //! Class constructor, taking the lowest desired playing frequency.
diff -wru stk-4.3.1_orig/include/Mesh2D.h stk-4.3.1/include/Mesh2D.h
--- stk-4.3.1_orig/include/Mesh2D.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Mesh2D.h	2008-01-11 12:13:53.765625000 +0100
@@ -33,7 +33,7 @@
 const short NXMAX = 12;
 const short NYMAX = 12;
 
-class Mesh2D : public Instrmnt
+class STK_EXPORT Mesh2D : public Instrmnt
 {
  public:
   //! Class constructor, taking the x and y dimensions in samples.
diff -wru stk-4.3.1_orig/include/Messager.h stk-4.3.1/include/Messager.h
--- stk-4.3.1_orig/include/Messager.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Messager.h	2008-01-11 12:13:53.765625000 +0100
@@ -54,7 +54,7 @@
 
 #endif // __STK_REALTIME__
 
-class Messager : public Stk
+class STK_EXPORT Messager : public Stk
 {
  public:
 
diff -wru stk-4.3.1_orig/include/MidiFileIn.h stk-4.3.1/include/MidiFileIn.h
--- stk-4.3.1_orig/include/MidiFileIn.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/MidiFileIn.h	2008-01-11 12:13:53.765625000 +0100
@@ -24,7 +24,7 @@
 #include <fstream>
 #include <sstream>
 
-class MidiFileIn : public Stk
+class STK_EXPORT MidiFileIn : public Stk
 {
  public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/Modal.h stk-4.3.1/include/Modal.h
--- stk-4.3.1_orig/include/Modal.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Modal.h	2008-01-11 12:13:53.765625000 +0100
@@ -21,7 +21,7 @@
 #include "BiQuad.h"
 #include "OnePole.h"
 
-class Modal : public Instrmnt
+class STK_EXPORT Modal : public Instrmnt
 {
 public:
   //! Class constructor, taking the desired number of modes to create.
diff -wru stk-4.3.1_orig/include/ModalBar.h stk-4.3.1/include/ModalBar.h
--- stk-4.3.1_orig/include/ModalBar.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/ModalBar.h	2008-01-11 12:13:53.781250000 +0100
@@ -33,7 +33,7 @@
 
 #include "Modal.h"
 
-class ModalBar : public Modal
+class STK_EXPORT ModalBar : public Modal
 {
 public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Modulate.h stk-4.3.1/include/Modulate.h
--- stk-4.3.1_orig/include/Modulate.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Modulate.h	2008-01-11 12:13:53.781250000 +0100
@@ -18,7 +18,7 @@
 #include "SubNoise.h"
 #include "OnePole.h"
 
-class Modulate : public Generator
+class STK_EXPORT Modulate : public Generator
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Moog.h stk-4.3.1/include/Moog.h
--- stk-4.3.1_orig/include/Moog.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Moog.h	2008-01-11 12:13:53.781250000 +0100
@@ -24,7 +24,7 @@
 #include "Sampler.h"
 #include "FormSwep.h"
 
-class Moog : public Sampler
+class STK_EXPORT Moog : public Sampler
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Mutex.h stk-4.3.1/include/Mutex.h
--- stk-4.3.1_orig/include/Mutex.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Mutex.h	2008-01-11 12:13:53.781250000 +0100
@@ -31,7 +31,7 @@
 
 #endif
 
-class Mutex : public Stk
+class STK_EXPORT Mutex : public Stk
 {
  public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/NRev.h stk-4.3.1/include/NRev.h
--- stk-4.3.1_orig/include/NRev.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/NRev.h	2008-01-11 12:13:53.781250000 +0100
@@ -22,7 +22,7 @@
 #include "Effect.h" 
 #include "Delay.h" 
 
-class NRev : public Effect
+class STK_EXPORT NRev : public Effect
 {
  public:
   //! Class constructor taking a T60 decay time argument (one second default value).
diff -wru stk-4.3.1_orig/include/Noise.h stk-4.3.1/include/Noise.h
--- stk-4.3.1_orig/include/Noise.h	2007-12-06 01:58:32.000000000 +0100
+++ stk-4.3.1/include/Noise.h	2008-01-11 12:13:53.796875000 +0100
@@ -15,7 +15,7 @@
 
 #include "Generator.h"
 
-class Noise : public Generator
+class STK_EXPORT Noise : public Generator
 {
 public:
 
diff -wru stk-4.3.1_orig/include/OnePole.h stk-4.3.1/include/OnePole.h
--- stk-4.3.1_orig/include/OnePole.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/OnePole.h	2008-01-11 12:13:53.796875000 +0100
@@ -17,7 +17,7 @@
 
 #include "Filter.h"
 
-class OnePole : protected Filter
+class STK_EXPORT OnePole : protected Filter
 {
 public:
 
diff -wru stk-4.3.1_orig/include/OneZero.h stk-4.3.1/include/OneZero.h
--- stk-4.3.1_orig/include/OneZero.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/OneZero.h	2008-01-11 12:13:53.796875000 +0100
@@ -17,7 +17,7 @@
 
 #include "Filter.h"
 
-class OneZero : protected Filter
+class STK_EXPORT OneZero : protected Filter
 {
  public:
 
diff -wru stk-4.3.1_orig/include/PRCRev.h stk-4.3.1/include/PRCRev.h
--- stk-4.3.1_orig/include/PRCRev.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/PRCRev.h	2008-01-11 12:13:53.796875000 +0100
@@ -20,7 +20,7 @@
 #include "Effect.h" 
 #include "Delay.h" 
 
-class PRCRev : public Effect
+class STK_EXPORT PRCRev : public Effect
 {
 public:
   //! Class constructor taking a T60 decay time argument (one second default value).
diff -wru stk-4.3.1_orig/include/PercFlut.h stk-4.3.1/include/PercFlut.h
--- stk-4.3.1_orig/include/PercFlut.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/PercFlut.h	2008-01-11 12:13:53.796875000 +0100
@@ -31,7 +31,7 @@
 
 #include "FM.h"
 
-class PercFlut : public FM
+class STK_EXPORT PercFlut : public FM
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Phonemes.h stk-4.3.1/include/Phonemes.h
--- stk-4.3.1_orig/include/Phonemes.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Phonemes.h	2008-01-11 12:13:53.812500000 +0100
@@ -15,7 +15,7 @@
 
 #include "Stk.h"
 
-class Phonemes : public Stk
+class STK_EXPORT Phonemes : public Stk
 {
 public:
 
diff -wru stk-4.3.1_orig/include/PitShift.h stk-4.3.1/include/PitShift.h
--- stk-4.3.1_orig/include/PitShift.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/PitShift.h	2008-01-11 12:13:53.812500000 +0100
@@ -15,7 +15,7 @@
 #include "Effect.h" 
 #include "DelayL.h" 
 
-class PitShift : public Effect
+class STK_EXPORT PitShift : public Effect
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/PluckTwo.h stk-4.3.1/include/PluckTwo.h
--- stk-4.3.1_orig/include/PluckTwo.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/PluckTwo.h	2008-01-11 12:13:53.812500000 +0100
@@ -26,7 +26,7 @@
 #include "DelayA.h"
 #include "OneZero.h"
 
-class PluckTwo : public Instrmnt
+class STK_EXPORT PluckTwo : public Instrmnt
 {
  public:
   //! Class constructor, taking the lowest desired playing frequency.
diff -wru stk-4.3.1_orig/include/Plucked.h stk-4.3.1/include/Plucked.h
--- stk-4.3.1_orig/include/Plucked.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Plucked.h	2008-01-11 12:13:53.812500000 +0100
@@ -26,7 +26,7 @@
 #include "OnePole.h"
 #include "Noise.h"
 
-class Plucked : public Instrmnt
+class STK_EXPORT Plucked : public Instrmnt
 {
  public:
   //! Class constructor, taking the lowest desired playing frequency.
diff -wru stk-4.3.1_orig/include/PoleZero.h stk-4.3.1/include/PoleZero.h
--- stk-4.3.1_orig/include/PoleZero.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/PoleZero.h	2008-01-11 12:13:53.812500000 +0100
@@ -17,7 +17,7 @@
 
 #include "Filter.h"
 
-class PoleZero : protected Filter
+class STK_EXPORT PoleZero : protected Filter
 {
  public:
 
diff -wru stk-4.3.1_orig/include/ReedTable.h stk-4.3.1/include/ReedTable.h
--- stk-4.3.1_orig/include/ReedTable.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/ReedTable.h	2008-01-11 12:13:53.828125000 +0100
@@ -22,7 +22,7 @@
 
 #include "Function.h"
 
-class ReedTable : public Function
+class STK_EXPORT ReedTable : public Function
 {
 public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/Resonate.h stk-4.3.1/include/Resonate.h
--- stk-4.3.1_orig/include/Resonate.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Resonate.h	2008-01-11 12:13:53.828125000 +0100
@@ -25,7 +25,7 @@
 #include "BiQuad.h"
 #include "Noise.h"
 
-class Resonate : public Instrmnt
+class STK_EXPORT Resonate : public Instrmnt
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Rhodey.h stk-4.3.1/include/Rhodey.h
--- stk-4.3.1_orig/include/Rhodey.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Rhodey.h	2008-01-11 12:13:53.828125000 +0100
@@ -35,7 +35,7 @@
 
 #include "FM.h"
 
-class Rhodey : public FM
+class STK_EXPORT Rhodey : public FM
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/RtAudio.h stk-4.3.1/include/RtAudio.h
--- stk-4.3.1_orig/include/RtAudio.h	2007-12-07 09:04:31.000000000 +0100
+++ stk-4.3.1/include/RtAudio.h	2008-01-11 12:13:53.843750000 +0100
@@ -188,9 +188,9 @@
 //
 // **************************************************************** //
 
-class RtApi;
+class STK_EXPORT RtApi;
 
-class RtAudio
+class STK_EXPORT RtAudio
 {
  public:
 
@@ -530,7 +530,7 @@
 
 #include <sstream>
 
-class RtApi
+class STK_EXPORT RtApi
 {
 public:
 
@@ -700,7 +700,7 @@
 
 #include <CoreAudio/AudioHardware.h>
 
-class RtApiCore: public RtApi
+class STK_EXPORT RtApiCore: public RtApi
 {
 public:
 
@@ -738,7 +738,7 @@
 
 #if defined(__UNIX_JACK__)
 
-class RtApiJack: public RtApi
+class STK_EXPORT RtApiJack: public RtApi
 {
 public:
 
@@ -771,7 +771,7 @@
 
 #if defined(__WINDOWS_ASIO__)
 
-class RtApiAsio: public RtApi
+class STK_EXPORT RtApiAsio: public RtApi
 {
 public:
 
@@ -807,7 +807,7 @@
 
 #if defined(__WINDOWS_DS__)
 
-class RtApiDs: public RtApi
+class STK_EXPORT RtApiDs: public RtApi
 {
 public:
 
@@ -845,7 +845,7 @@
 
 #if defined(__LINUX_ALSA__)
 
-class RtApiAlsa: public RtApi
+class STK_EXPORT RtApiAlsa: public RtApi
 {
 public:
 
@@ -877,7 +877,7 @@
 
 #if defined(__LINUX_OSS__)
 
-class RtApiOss: public RtApi
+class STK_EXPORT RtApiOss: public RtApi
 {
 public:
 
@@ -909,7 +909,7 @@
 
 #if defined(__RTAUDIO_DUMMY__)
 
-class RtApiDummy: public RtApi
+class STK_EXPORT RtApiDummy: public RtApi
 {
 public:
 
diff -wru stk-4.3.1_orig/include/RtError.h stk-4.3.1/include/RtError.h
--- stk-4.3.1_orig/include/RtError.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/RtError.h	2008-01-11 12:19:22.093750000 +0100
@@ -12,11 +12,13 @@
 #ifndef RTERROR_H
 #define RTERROR_H
 
+#include "Stk.h"
+
 #include <exception>
 #include <iostream>
 #include <string>
 
-class RtError : public std::exception
+class STK_EXPORT RtError : public std::exception
 {
  public:
   //! Defined RtError types.
diff -wru stk-4.3.1_orig/include/RtMidi.h stk-4.3.1/include/RtMidi.h
--- stk-4.3.1_orig/include/RtMidi.h	2007-12-07 07:25:40.000000000 +0100
+++ stk-4.3.1/include/RtMidi.h	2008-01-11 12:13:53.859375000 +0100
@@ -43,7 +43,7 @@
 #include "RtError.h"
 #include <string>
 
-class RtMidi
+class STK_EXPORT RtMidi
 {
  public:
 
@@ -98,7 +98,7 @@
 #include <vector>
 #include <queue>
 
-class RtMidiIn : public RtMidi
+class STK_EXPORT RtMidiIn : public RtMidi
 {
  public:
 
@@ -236,7 +236,7 @@
 */
 /**********************************************************************/
 
-class RtMidiOut : public RtMidi
+class STK_EXPORT RtMidiOut : public RtMidi
 {
  public:
 
diff -wru stk-4.3.1_orig/include/RtWvIn.h stk-4.3.1/include/RtWvIn.h
--- stk-4.3.1_orig/include/RtWvIn.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/RtWvIn.h	2008-01-11 12:13:53.859375000 +0100
@@ -24,7 +24,7 @@
 #include "WvIn.h"
 #include "RtAudio.h"
 
-class RtWvIn : public WvIn
+class STK_EXPORT RtWvIn : public WvIn
 {
 public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/RtWvOut.h stk-4.3.1/include/RtWvOut.h
--- stk-4.3.1_orig/include/RtWvOut.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/RtWvOut.h	2008-01-11 12:13:53.859375000 +0100
@@ -24,7 +24,7 @@
 #include "WvOut.h"
 #include "RtAudio.h"
 
-class RtWvOut : public WvOut
+class STK_EXPORT RtWvOut : public WvOut
 {
  public:
 
diff -wru stk-4.3.1_orig/include/Sampler.h stk-4.3.1/include/Sampler.h
--- stk-4.3.1_orig/include/Sampler.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Sampler.h	2008-01-11 12:13:53.875000000 +0100
@@ -17,7 +17,7 @@
 #include "WaveLoop.h"
 #include "OnePole.h"
 
-class Sampler : public Instrmnt
+class STK_EXPORT Sampler : public Instrmnt
 {
  public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/Saxofony.h stk-4.3.1/include/Saxofony.h
--- stk-4.3.1_orig/include/Saxofony.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Saxofony.h	2008-01-11 12:13:53.875000000 +0100
@@ -46,7 +46,7 @@
 #include "Noise.h"
 #include "SineWave.h"
 
-class Saxofony : public Instrmnt
+class STK_EXPORT Saxofony : public Instrmnt
 {
  public:
   //! Class constructor, taking the lowest desired playing frequency.
diff -wru stk-4.3.1_orig/include/Shakers.h stk-4.3.1/include/Shakers.h
--- stk-4.3.1_orig/include/Shakers.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Shakers.h	2008-01-11 12:13:53.875000000 +0100
@@ -60,7 +60,7 @@
 const int MAX_FREQS = 8;
 const int NUM_INSTR = 24;
 
-class Shakers : public Instrmnt
+class STK_EXPORT Shakers : public Instrmnt
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Simple.h stk-4.3.1/include/Simple.h
--- stk-4.3.1_orig/include/Simple.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Simple.h	2008-01-11 12:13:53.875000000 +0100
@@ -27,7 +27,7 @@
 #include "BiQuad.h"
 #include "Noise.h"
 
-class Simple : public Instrmnt
+class STK_EXPORT Simple : public Instrmnt
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/SineWave.h stk-4.3.1/include/SineWave.h
--- stk-4.3.1_orig/include/SineWave.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/SineWave.h	2008-01-11 12:13:53.890625000 +0100
@@ -20,7 +20,7 @@
 
 #include "Generator.h"
 
-class SineWave : public Generator
+class STK_EXPORT SineWave : public Generator
 {
 public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/SingWave.h stk-4.3.1/include/SingWave.h
--- stk-4.3.1_orig/include/SingWave.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/SingWave.h	2008-01-11 12:13:53.890625000 +0100
@@ -20,7 +20,7 @@
 #include "Modulate.h"
 #include "Envelope.h"
 
-class SingWave : public Generator
+class STK_EXPORT SingWave : public Generator
 {
  public:
   //! Class constructor taking filename argument.
diff -wru stk-4.3.1_orig/include/Sitar.h stk-4.3.1/include/Sitar.h
--- stk-4.3.1_orig/include/Sitar.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Sitar.h	2008-01-11 12:13:53.890625000 +0100
@@ -26,7 +26,7 @@
 #include "Noise.h"
 #include "ADSR.h"
 
-class Sitar : public Instrmnt
+class STK_EXPORT Sitar : public Instrmnt
 {
  public:
   //! Class constructor, taking the lowest desired playing frequency.
diff -wru stk-4.3.1_orig/include/Skini.h stk-4.3.1/include/Skini.h
--- stk-4.3.1_orig/include/Skini.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Skini.h	2008-01-11 12:13:53.890625000 +0100
@@ -33,7 +33,7 @@
 #include <string>
 #include <fstream>
 
-class Skini : public Stk
+class STK_EXPORT Skini : public Stk
 {
  public:
 
diff -wru stk-4.3.1_orig/include/Socket.h stk-4.3.1/include/Socket.h
--- stk-4.3.1_orig/include/Socket.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Socket.h	2008-01-11 12:13:53.890625000 +0100
@@ -33,7 +33,7 @@
 
 #endif
 
-class Socket : public Stk
+class STK_EXPORT Socket : public Stk
 {
  public:
 
diff -wru stk-4.3.1_orig/include/Sphere.h stk-4.3.1/include/Sphere.h
--- stk-4.3.1_orig/include/Sphere.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Sphere.h	2008-01-11 12:13:53.906250000 +0100
@@ -15,7 +15,7 @@
 #include "Stk.h"
 #include "Vector3D.h"
 
-class Sphere : public Stk
+class STK_EXPORT Sphere : public Stk
 {
 public:
   //! Constructor taking an initial radius value.
diff -wru stk-4.3.1_orig/include/StifKarp.h stk-4.3.1/include/StifKarp.h
--- stk-4.3.1_orig/include/StifKarp.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/StifKarp.h	2008-01-11 12:13:53.906250000 +0100
@@ -31,7 +31,7 @@
 #include "Noise.h"
 #include "BiQuad.h"
 
-class StifKarp : public Instrmnt
+class STK_EXPORT StifKarp : public Instrmnt
 {
  public:
   //! Class constructor, taking the lowest desired playing frequency.
diff -wru stk-4.3.1_orig/include/Stk.h stk-4.3.1/include/Stk.h
--- stk-4.3.1_orig/include/Stk.h	2007-12-06 01:58:33.000000000 +0100
+++ stk-4.3.1/include/Stk.h	2008-01-11 12:58:27.828125000 +0100
@@ -58,6 +58,24 @@
 #include <sstream>
 #include <vector>
 
+#ifdef WIN32
+# ifdef STK_DLL
+#  ifdef STK_BUILD
+#   define STK_EXPORT __declspec(dllexport)
+#  else
+#   define STK_EXPORT __declspec(dllimport)
+#  endif
+# endif
+#endif
+
+#ifdef _DEBUG
+#define _STK_DEBUG_ 1
+#endif
+
+#ifndef STK_EXPORT
+# define STK_EXPORT
+#endif
+
 // Most data in STK is passed and calculated with the
 // following user-definable floating-point type.  You
 // can change this to "float" if you prefer or perhaps
@@ -70,7 +88,7 @@
   be sub-classes to take care of more specific error conditions ... or
   not.
 */
-class StkError
+class STK_EXPORT StkError
 {
 public:
   enum Type {
@@ -117,7 +135,7 @@
 };
 
 
-class Stk
+class STK_EXPORT Stk
 {
 public:
 
@@ -239,7 +257,7 @@
 */
 /***************************************************/
 
-class StkFrames
+class STK_EXPORT StkFrames
 {
 public:
 
diff -wru stk-4.3.1_orig/include/SubNoise.h stk-4.3.1/include/SubNoise.h
--- stk-4.3.1_orig/include/SubNoise.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/SubNoise.h	2008-01-11 12:13:53.921875000 +0100
@@ -15,7 +15,7 @@
 
 #include "Noise.h"
 
-class SubNoise : public Noise
+class STK_EXPORT SubNoise : public Noise
 {
  public:
 
diff -wru stk-4.3.1_orig/include/TcpClient.h stk-4.3.1/include/TcpClient.h
--- stk-4.3.1_orig/include/TcpClient.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/TcpClient.h	2008-01-11 12:13:53.921875000 +0100
@@ -28,7 +28,7 @@
 
 #include "Socket.h"
 
-class TcpClient : public Socket
+class STK_EXPORT TcpClient : public Socket
 {
  public:
   //! Default class constructor creates a socket client connection to the specified host and port.
diff -wru stk-4.3.1_orig/include/TcpServer.h stk-4.3.1/include/TcpServer.h
--- stk-4.3.1_orig/include/TcpServer.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/TcpServer.h	2008-01-11 12:13:53.921875000 +0100
@@ -28,7 +28,7 @@
 
 #include "Socket.h"
 
-class TcpServer : public Socket
+class STK_EXPORT TcpServer : public Socket
 {
  public:
   //! Default constructor creates a local socket server on port 2006 (or the specified port number).
diff -wru stk-4.3.1_orig/include/Thread.h stk-4.3.1/include/Thread.h
--- stk-4.3.1_orig/include/Thread.h	2007-12-06 01:58:31.000000000 +0100
+++ stk-4.3.1/include/Thread.h	2008-01-11 12:13:53.921875000 +0100
@@ -44,7 +44,7 @@
 
 #endif
 
-class Thread : public Stk
+class STK_EXPORT Thread : public Stk
 {
  public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/TubeBell.h stk-4.3.1/include/TubeBell.h
--- stk-4.3.1_orig/include/TubeBell.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/TubeBell.h	2008-01-11 12:13:53.937500000 +0100
@@ -35,7 +35,7 @@
 
 #include "FM.h"
 
-class TubeBell : public FM
+class STK_EXPORT TubeBell : public FM
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/TwoPole.h stk-4.3.1/include/TwoPole.h
--- stk-4.3.1_orig/include/TwoPole.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/TwoPole.h	2008-01-11 12:13:53.937500000 +0100
@@ -17,7 +17,7 @@
 
 #include "Filter.h"
 
-class TwoPole : protected Filter
+class STK_EXPORT TwoPole : protected Filter
 {
  public:
 
diff -wru stk-4.3.1_orig/include/TwoZero.h stk-4.3.1/include/TwoZero.h
--- stk-4.3.1_orig/include/TwoZero.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/TwoZero.h	2008-01-11 12:13:53.937500000 +0100
@@ -17,7 +17,7 @@
 
 #include "Filter.h"
 
-class TwoZero : protected Filter
+class STK_EXPORT TwoZero : protected Filter
 {
  public:
   //! Default constructor creates a second-order pass-through filter.
diff -wru stk-4.3.1_orig/include/UdpSocket.h stk-4.3.1/include/UdpSocket.h
--- stk-4.3.1_orig/include/UdpSocket.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/UdpSocket.h	2008-01-11 12:13:53.937500000 +0100
@@ -26,7 +26,7 @@
 
 #include "Socket.h"
 
-class UdpSocket : public Socket
+class STK_EXPORT UdpSocket : public Socket
 {
  public:
   //! Default constructor creates a local UDP socket on port 2006 (or the specified port number).
diff -wru stk-4.3.1_orig/include/Vector3D.h stk-4.3.1/include/Vector3D.h
--- stk-4.3.1_orig/include/Vector3D.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/Vector3D.h	2008-01-11 12:13:53.953125000 +0100
@@ -13,7 +13,7 @@
 
 #include "Stk.h"
 
-class Vector3D : public Stk
+class STK_EXPORT Vector3D : public Stk
 {
 
 public:
diff -wru stk-4.3.1_orig/include/VoicForm.h stk-4.3.1/include/VoicForm.h
--- stk-4.3.1_orig/include/VoicForm.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/VoicForm.h	2008-01-11 12:13:53.953125000 +0100
@@ -36,7 +36,7 @@
 #include "OnePole.h"
 #include "OneZero.h"
 
-class VoicForm : public Instrmnt
+class STK_EXPORT VoicForm : public Instrmnt
 {
   public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Voicer.h stk-4.3.1/include/Voicer.h
--- stk-4.3.1_orig/include/Voicer.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/Voicer.h	2008-01-11 12:13:53.953125000 +0100
@@ -35,7 +35,7 @@
 #include "Instrmnt.h"
 #include <vector>
 
-class Voicer : public Stk
+class STK_EXPORT Voicer : public Stk
 {
 public:
   //! Class constructor taking an optional note decay time (in seconds).
diff -wru stk-4.3.1_orig/include/WaveLoop.h stk-4.3.1/include/WaveLoop.h
--- stk-4.3.1_orig/include/WaveLoop.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/WaveLoop.h	2008-01-11 12:13:53.953125000 +0100
@@ -21,7 +21,7 @@
 
 #include "FileWvIn.h"
 
-class WaveLoop : public FileWvIn
+class STK_EXPORT WaveLoop : public FileWvIn
 {
 public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/Whistle.h stk-4.3.1/include/Whistle.h
--- stk-4.3.1_orig/include/Whistle.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/Whistle.h	2008-01-11 12:13:53.968750000 +0100
@@ -27,7 +27,7 @@
 #include "OnePole.h"
 #include "Envelope.h"
 
-class Whistle : public Instrmnt
+class STK_EXPORT Whistle : public Instrmnt
 {
 public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/Wurley.h stk-4.3.1/include/Wurley.h
--- stk-4.3.1_orig/include/Wurley.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/Wurley.h	2008-01-11 12:13:53.968750000 +0100
@@ -35,7 +35,7 @@
 
 #include "FM.h"
 
-class Wurley : public FM
+class STK_EXPORT Wurley : public FM
 {
  public:
   //! Class constructor.
diff -wru stk-4.3.1_orig/include/WvIn.h stk-4.3.1/include/WvIn.h
--- stk-4.3.1_orig/include/WvIn.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/WvIn.h	2008-01-11 12:13:53.968750000 +0100
@@ -23,7 +23,7 @@
 #include "Stk.h"
 #include <vector>
 
-class WvIn : public Stk
+class STK_EXPORT WvIn : public Stk
 {
 public:
   //! Default constructor.
diff -wru stk-4.3.1_orig/include/WvOut.h stk-4.3.1/include/WvOut.h
--- stk-4.3.1_orig/include/WvOut.h	2007-12-06 01:58:30.000000000 +0100
+++ stk-4.3.1/include/WvOut.h	2008-01-11 12:13:53.968750000 +0100
@@ -26,7 +26,7 @@
 #include "Stk.h"
 #include <vector>
 
-class WvOut : public Stk
+class STK_EXPORT WvOut : public Stk
 {
  public:
 
