diff -ru stk-4.3.1_orig/include/Stk.h stk-4.3.1/include/Stk.h
--- stk-4.3.1_orig/include/Stk.h	2008-02-13 13:55:35.000000000 +0100
+++ stk-4.3.1/include/Stk.h	2008-02-13 13:56:13.000000000 +0100
@@ -399,10 +399,10 @@
 const StkFloat TWO_PI       = 2 * PI;
 const StkFloat ONE_OVER_128 = 0.0078125;
 
-#if defined(__WINDOWS_DS__) || defined(__WINDOWS_ASIO__) || defined(__WINDOWS_MM__)
+#if defined(WIN32) || defined(__WINDOWS_DS__) || defined(__WINDOWS_ASIO__) || defined(__WINDOWS_MM__)
   #define __OS_WINDOWS__
   #define __STK_REALTIME__
-#elif defined(__LINUX_OSS__) || defined(__LINUX_ALSA__) || defined(__UNIX_JACK__)
+#elif defined(__linux__) || defined(__LINUX_OSS__) || defined(__LINUX_ALSA__) || defined(__UNIX_JACK__)
   #define __OS_LINUX__
   #define __STK_REALTIME__
 #elif defined(__IRIX_AL__)
