diff -ru screen-4.0.3_orig/config.h.in screen-4.0.3/config.h.in
--- screen-4.0.3_orig/config.h.in	2009-03-01 19:01:38.000000000 +0100
+++ screen-4.0.3/config.h.in	2009-03-01 19:03:39.000000000 +0100
@@ -563,7 +563,7 @@
  * If you are on a SYS V machine that restricts filename length to 14 
  * characters, you may need to enforce that by setting NAME_MAX to 14
  */
-#undef NAME_MAX		/* KEEP_UNDEF_HERE override system value */
+//#undef NAME_MAX		/* KEEP_UNDEF_HERE override system value */
 #undef NAME_MAX
 
 /*
