diff -ru httpd-2.0.63_orig/configure httpd-2.0.63/configure
--- httpd-2.0.63_orig/configure	2008-11-20 07:53:07.000000000 +0000
+++ httpd-2.0.63/configure	2008-11-20 07:54:52.000000000 +0000
@@ -5241,7 +5241,7 @@
       ;;
   *)
       if test "x$LTFLAGS" = "x"; then
-          LTFLAGS='--silent'
+          LTFLAGS='--silent --tag=CC'
       fi
       my_libtool=`$apr_config --apr-libtool`
       LIBTOOL="$my_libtool \$(LTFLAGS)"
diff -ru httpd-2.0.63_orig/configure.in httpd-2.0.63/configure.in
--- httpd-2.0.63_orig/configure.in	2008-11-20 07:53:07.000000000 +0000
+++ httpd-2.0.63/configure.in	2008-11-20 07:54:22.000000000 +0000
@@ -166,7 +166,7 @@
       ;;
   *)
       if test "x$LTFLAGS" = "x"; then
-          LTFLAGS='--silent'
+          LTFLAGS='--silent --tag=CC'
       fi
       my_libtool=`$apr_config --apr-libtool`
       LIBTOOL="$my_libtool \$(LTFLAGS)"
