diff -wru pitchtune-0.0.4_orig/Makefile.am pitchtune-0.0.4/Makefile.am
--- pitchtune-0.0.4_orig/Makefile.am	2008-11-11 18:59:52.000000000 +0100
+++ pitchtune-0.0.4/Makefile.am	2008-11-11 19:00:59.000000000 +0100
@@ -8,13 +8,6 @@
 docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
 doc_DATA =   AUTHORS ChangeLog README REQUIRED TODO
 
-CLEANFILES = AUTHORS.gz ChangeLog.gz README.gz TODO.gz
-
-
-%.gz: %
-	gzip -c -9 $< > $<.gz
-
-
 rpm:
 	$(MAKE) dist
 	rpm -ta --clean $(PACKAGE)-$(VERSION).tar.gz
