args: -r -u -N -- pitchtune-0.0.4_orig pitchtune-0.0.4
diff -r -u -N -- pitchtune-0.0.4_orig/Makefile pitchtune-0.0.4/Makefile
diff -r -u -N -- pitchtune-0.0.4_orig/Makefile.am pitchtune-0.0.4/Makefile.am
--- pitchtune-0.0.4_orig/Makefile.am	2008-11-12 20:38:45.000000000 +0100
+++ pitchtune-0.0.4/Makefile.am	2008-11-12 20:41:54.000000000 +0100
@@ -1,6 +1,8 @@
 ## Process this file with automake to create Makefile.in
 # pitchtune timestamp update 1
 
+AUTOMAKE_AMFLAGS = --foreign --add-missing
+
 SUBDIRS = debian doc src
 
 EXTRA_DIST = pitchtune.spec
diff -r -u -N -- pitchtune-0.0.4_orig/Makefile.in pitchtune-0.0.4/Makefile.in
diff -r -u -N -- pitchtune-0.0.4_orig/NEWS pitchtune-0.0.4/NEWS
--- pitchtune-0.0.4_orig/NEWS	1970-01-01 01:00:00.000000000 +0100
+++ pitchtune-0.0.4/NEWS	2008-11-12 20:40:59.000000000 +0100
@@ -0,0 +1 @@
+
diff -r -u -N -- pitchtune-0.0.4_orig/autogen.sh pitchtune-0.0.4/autogen.sh
--- pitchtune-0.0.4_orig/autogen.sh	2008-11-12 20:38:45.000000000 +0100
+++ pitchtune-0.0.4/autogen.sh	2008-11-12 20:42:00.000000000 +0100
@@ -1,159 +1,9 @@
 #!/bin/sh
-# Run this to generate all the initial makefiles, etc.
+#
+# 20081112
 
-srcdir=`dirname $0`
-test -z "$srcdir" && srcdir=.
 
-DIE=0
-
-if [ -n "$GNOME2_DIR" ]; then
-	ACLOCAL_FLAGS="-I $GNOME2_DIR/share/aclocal $ACLOCAL_FLAGS"
-	LD_LIBRARY_PATH="$GNOME2_DIR/lib:$LD_LIBRARY_PATH"
-	PATH="$GNOME2_DIR/bin:$PATH"
-	export PATH
-	export LD_LIBRARY_PATH
-fi
-
-(test -f $srcdir/configure.in) || {
-    echo -n "**Error**: Directory "\`$srcdir\'" does not look like the"
-    echo " top-level package directory"
-    exit 1
-}
-
-(autoconf --version) < /dev/null > /dev/null 2>&1 || {
-  echo
-  echo "**Error**: You must have \`autoconf' installed."
-  echo "Download the appropriate package for your distribution,"
-  echo "or get the source tarball at ftp://ftp.gnu.org/pub/gnu/"
-  DIE=1
-}
-
-(grep "^AC_PROG_INTLTOOL" $srcdir/configure.in >/dev/null) && {
-  (intltoolize --version) < /dev/null > /dev/null 2>&1 || {
-    echo 
-    echo "**Error**: You must have \`intltool' installed."
-    echo "You can get it from:"
-    echo "  ftp://ftp.gnome.org/pub/GNOME/"
-    DIE=1
-  }
-}
-
-(grep "^AM_PROG_XML_I18N_TOOLS" $srcdir/configure.in >/dev/null) && {
-  (xml-i18n-toolize --version) < /dev/null > /dev/null 2>&1 || {
-    echo
-    echo "**Error**: You must have \`xml-i18n-toolize' installed."
-    echo "You can get it from:"
-    echo "  ftp://ftp.gnome.org/pub/GNOME/"
-    DIE=1
-  }
-}
-
-(grep "^AM_PROG_LIBTOOL" $srcdir/configure.in >/dev/null) && {
-  (libtool --version) < /dev/null > /dev/null 2>&1 || {
-    echo
-    echo "**Error**: You must have \`libtool' installed."
-    echo "You can get it from: ftp://ftp.gnu.org/pub/gnu/"
-    DIE=1
-  }
-}
-
-(grep "^AM_GLIB_GNU_GETTEXT" $srcdir/configure.in >/dev/null) && {
-  (grep "sed.*POTFILES" $srcdir/configure.in) > /dev/null || \
-  (glib-gettextize --version) < /dev/null > /dev/null 2>&1 || {
-    echo
-    echo "**Error**: You must have \`glib' installed."
-    echo "You can get it from: ftp://ftp.gtk.org/pub/gtk"
-    DIE=1
-  }
-}
-
-(automake --version) < /dev/null > /dev/null 2>&1 || {
-  echo
-  echo "**Error**: You must have \`automake' installed."
-  echo "You can get it from: ftp://ftp.gnu.org/pub/gnu/"
-  DIE=1
-  NO_AUTOMAKE=yes
-}
-
-
-# if no automake, don't bother testing for aclocal
-test -n "$NO_AUTOMAKE" || (aclocal --version) < /dev/null > /dev/null 2>&1 || {
-  echo
-  echo "**Error**: Missing \`aclocal'.  The version of \`automake'"
-  echo "installed doesn't appear recent enough."
-  echo "You can get automake from ftp://ftp.gnu.org/pub/gnu/"
-  DIE=1
-}
-
-if test "$DIE" -eq 1; then
-  exit 1
-fi
-
-if test -z "$*"; then
-  echo "**Warning**: I am going to run \`configure' with no arguments."
-  echo "If you wish to pass any to it, please specify them on the"
-  echo \`$0\'" command line."
-  echo
-fi
-
-case $CC in
-xlc )
-  am_opt=--include-deps;;
-esac
-
-for coin in `find $srcdir -path $srcdir/CVS -prune -o -name configure.in -print`
-do 
-  dr=`dirname $coin`
-  if test -f $dr/NO-AUTO-GEN; then
-    echo skipping $dr -- flagged as no auto-gen
-  else
-    echo processing $dr
-    ( cd $dr
-
-      aclocalinclude="$ACLOCAL_FLAGS"
-
-      if grep "^AM_GLIB_GNU_GETTEXT" configure.in >/dev/null; then
-	echo "Creating $dr/aclocal.m4 ..."
-	test -r $dr/aclocal.m4 || touch $dr/aclocal.m4
-	echo "Running glib-gettextize...  Ignore non-fatal messages."
-	echo "no" | glib-gettextize --force --copy
-	echo "Making $dr/aclocal.m4 writable ..."
-	test -r $dr/aclocal.m4 && chmod u+w $dr/aclocal.m4
-      fi
-      if grep "^AC_PROG_INTLTOOL" configure.in >/dev/null; then
-        echo "Running intltoolize..."
-	intltoolize --copy --force --automake
-      fi
-      if grep "^AM_PROG_XML_I18N_TOOLS" configure.in >/dev/null; then
-        echo "Running xml-i18n-toolize..."
-	xml-i18n-toolize --copy --force --automake
-      fi
-      if grep "^AM_PROG_LIBTOOL" configure.in >/dev/null; then
-	if test -z "$NO_LIBTOOLIZE" ; then 
-	  echo "Running libtoolize..."
-	  libtoolize --force --copy
-	fi
-      fi
-      echo "Running aclocal $aclocalinclude ..."
-      aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
-	echo "Running autoheader..."
-	autoheader
-      fi
-      echo "Running automake --gnu $am_opt ..."
-      automake --add-missing --gnu $am_opt
-      echo "Running autoconf ..."
-      autoconf
-    )
-  fi
-done
-
-conf_flags="--enable-maintainer-mode"
-
-if test x$NOCONFIGURE = x; then
-  echo Running $srcdir/configure $conf_flags "$@" ...
-  $srcdir/configure $conf_flags "$@" \
-  && echo Now type \`make\' to compile. || exit 1
-else
-  echo Skipping configure process.
-fi
+aclocal --force
+automake --force --copy --add-missing --foreign --add-missing
+aclocal --force
+autoconf --force
diff -r -u -N -- pitchtune-0.0.4_orig/config.log pitchtune-0.0.4/config.log
diff -r -u -N -- pitchtune-0.0.4_orig/config.status pitchtune-0.0.4/config.status
diff -r -u -N -- pitchtune-0.0.4_orig/debian/Makefile pitchtune-0.0.4/debian/Makefile
diff -r -u -N -- pitchtune-0.0.4_orig/debian/Makefile.in pitchtune-0.0.4/debian/Makefile.in
diff -r -u -N -- pitchtune-0.0.4_orig/doc/Makefile pitchtune-0.0.4/doc/Makefile
diff -r -u -N -- pitchtune-0.0.4_orig/doc/Makefile.in pitchtune-0.0.4/doc/Makefile.in
diff -r -u -N -- pitchtune-0.0.4_orig/install-sh pitchtune-0.0.4/install-sh
diff -r -u -N -- pitchtune-0.0.4_orig/missing pitchtune-0.0.4/missing
diff -r -u -N -- pitchtune-0.0.4_orig/mkinstalldirs pitchtune-0.0.4/mkinstalldirs
diff -r -u -N -- pitchtune-0.0.4_orig/src/.deps/control.Po pitchtune-0.0.4/src/.deps/control.Po
diff -r -u -N -- pitchtune-0.0.4_orig/src/.deps/glade_callbacks.Po pitchtune-0.0.4/src/.deps/glade_callbacks.Po
diff -r -u -N -- pitchtune-0.0.4_orig/src/.deps/glade_interface.Po pitchtune-0.0.4/src/.deps/glade_interface.Po
diff -r -u -N -- pitchtune-0.0.4_orig/src/.deps/glade_support.Po pitchtune-0.0.4/src/.deps/glade_support.Po
diff -r -u -N -- pitchtune-0.0.4_orig/src/.deps/main.Po pitchtune-0.0.4/src/.deps/main.Po
diff -r -u -N -- pitchtune-0.0.4_orig/src/.deps/sel_guitar.Po pitchtune-0.0.4/src/.deps/sel_guitar.Po
diff -r -u -N -- pitchtune-0.0.4_orig/src/.deps/sound.Po pitchtune-0.0.4/src/.deps/sound.Po
diff -r -u -N -- pitchtune-0.0.4_orig/src/.deps/tuner.Po pitchtune-0.0.4/src/.deps/tuner.Po
diff -r -u -N -- pitchtune-0.0.4_orig/src/.deps/ui.Po pitchtune-0.0.4/src/.deps/ui.Po
diff -r -u -N -- pitchtune-0.0.4_orig/src/Makefile pitchtune-0.0.4/src/Makefile
diff -r -u -N -- pitchtune-0.0.4_orig/src/Makefile.am pitchtune-0.0.4/src/Makefile.am
--- pitchtune-0.0.4_orig/src/Makefile.am	2008-11-12 20:38:45.000000000 +0100
+++ pitchtune-0.0.4/src/Makefile.am	2008-11-12 20:41:16.000000000 +0100
@@ -1,5 +1,7 @@
 ## Process this file with automake to create Makefile.in
 
+AUTOMAKE_AMFLAGS = --foreign --add-missing
+
 EXTRA_DIST = pitchtune.glade draw.c
 
 bin_PROGRAMS = pitchtune
diff -r -u -N -- pitchtune-0.0.4_orig/src/Makefile.in pitchtune-0.0.4/src/Makefile.in
