--- spidermonkey-0.0.1a_orig/spidermonkey.pyx	2008-02-23 03:22:40.000000000 +0100
+++ spidermonkey-0.0.1a/spidermonkey.pyx	2008-02-23 03:25:18.000000000 +0100
@@ -170,7 +170,7 @@
         JSXDRObjectOp xdrObject
         JSHasInstanceOp hasInstance
         JSMarkOp mark
-        jsword spare
+        #jsword spare
 
     cdef JSBool JS_PropertyStub(JSContext *cx, JSObject *obj, jsval id, jsval *vp)
     cdef JSBool JS_EnumerateStub(JSContext *cx, JSObject *obj)
@@ -712,7 +712,7 @@
         self.jsc.xdrObject = NULL
         self.jsc.hasInstance = NULL
         self.jsc.mark = NULL
-        self.jsc.spare = 0
+        #self.jsc.spare = 0
 
         if bind_constructor:
             if JS_InitClass(
