diff -ru wire-swig-0.20_orig/WireWrapper.cc wire-swig-0.20/WireWrapper.cc
--- wire-swig-0.20_orig/WireWrapper.cc	2008-04-03 05:51:46.000000000 +0000
+++ wire-swig-0.20/WireWrapper.cc	2008-04-03 06:40:27.000000000 +0000
@@ -93,7 +93,7 @@
     };
     char * retrieve_text_by_docid(unsigned long docid) {
       static char t[MAX_DOC_LEN];
-      off_t size;
+      off64_t size;
       storage_read( storage, docid, t, &size );
       return t;
     }
@@ -185,7 +185,7 @@
     char* fetch_byid(unsigned long docid) {
       //      char *t = (char *) malloc(sizeof(char) * MAX_DOC_LEN);
       static char t[MAX_DOC_LEN];
-      off_t size;
+      off64_t size;
       //      assert(t != NULL);
       storage_read( storage, docid, t, &size );
       return t;
