args: -r -u -- libirc-svn20080704_orig libirc-svn20080704
diff -r -u -- libirc-svn20080704_orig/Makefile.am libirc-svn20080704/Makefile.am
--- libirc-svn20080704_orig/Makefile.am	2008-07-04 03:52:41.000000000 +0200
+++ libirc-svn20080704/Makefile.am	2008-07-04 03:55:12.000000000 +0200
@@ -56,52 +56,52 @@
 # to emphasize the achievement.
 #
 all-local:
-	@@ECHO@ "Done."
-	@@ECHO@
-	@@ECHO@ "libIRC @LIBIRC_VERSION@, Build @CONFIG_DATE@"
-	@@ECHO@
+	@@lt_ECHO@ "Done."
+	@@lt_ECHO@
+	@@lt_ECHO@ "libIRC @LIBIRC_VERSION@, Build @CONFIG_DATE@"
+	@@lt_ECHO@
 	@if test "x$(MAKECMDGOALS)" = "xall-am" ; then \
-	  @ECHO@ @ECHO_N@ "Elapsed compilation time: " ;\
+	  @lt_ECHO@ @ECHO_N@ "Elapsed compilation time: " ;\
 	  sh $(top_srcdir)/misc/elapsed.sh `cat $(top_builddir)/include/libirc_version.h | grep Compilation` ;\
-	  @ECHO@ @ECHO_N@ "Elapsed time since configuration: " ;\
+	  @lt_ECHO@ @ECHO_N@ "Elapsed time since configuration: " ;\
 	  sh $(top_srcdir)/misc/elapsed.sh @CONFIG_TIME@ ;\
-	  @ECHO@ "---" ;\
-	  @ECHO@ "Run 'make examples' to build the libIRC framework examples" ;\
-	  @ECHO@ "Run 'make install' to begin installation into $(prefix)" ;\
+	  @lt_ECHO@ "---" ;\
+	  @lt_ECHO@ "Run 'make examples' to build the libIRC framework examples" ;\
+	  @lt_ECHO@ "Run 'make install' to begin installation into $(prefix)" ;\
 	elif test "x$(MAKECMDGOALS)" = "xinstall-am" ; then \
-	  @ECHO@ @ECHO_N@ "Elapsed installation time: " ;\
+	  @lt_ECHO@ @ECHO_N@ "Elapsed installation time: " ;\
 	  sh $(top_srcdir)/misc/elapsed.sh `cat $(top_builddir)/include/libirc_version.h | grep Compilation` ;\
-	  @ECHO@ @ECHO_N@ "Elapsed time since configuration: " ;\
+	  @lt_ECHO@ @ECHO_N@ "Elapsed time since configuration: " ;\
 	  sh $(top_srcdir)/misc/elapsed.sh @CONFIG_TIME@ ;\
-	  @ECHO@ "---" ;\
-	  @ECHO@ "Run 'make examples' to build the libIRC framework examples" ;\
-	  @ECHO@ ;\
+	  @lt_ECHO@ "---" ;\
+	  @lt_ECHO@ "Run 'make examples' to build the libIRC framework examples" ;\
+	  @lt_ECHO@ ;\
 	  line1="  libIRC @LIBIRC_VERSION@ is now installed into $(prefix)" ;\
 	  line2="  Be sure to add $(prefix)/bin to your PATH" ;\
-	  length="`@ECHO@ $(prefix)@LIBIRC_VERSION@ | wc | awk '{print $$3}'`" ;\
+	  length="`@lt_ECHO@ $(prefix)@LIBIRC_VERSION@ | wc | awk '{print $$3}'`" ;\
 	  length="`expr $$length + 34`" ;\
 	  len=$$length ;\
 	  while test $$len -gt 0  ; do \
-	    @ECHO@ -n "***" ;\
+	    @lt_ECHO@ -n "***" ;\
 	    len="`expr $$len - 3`" ;\
           done ;\
-	  @ECHO@ ;\
-	  @ECHO@ "$$line1" ;\
-	  @ECHO@ "$$line2" ;\
+	  @lt_ECHO@ ;\
+	  @lt_ECHO@ "$$line1" ;\
+	  @lt_ECHO@ "$$line2" ;\
 	  while [ $$length -gt 0 ] ; do \
-	    @ECHO@ @ECHO_N@ "***" ;\
+	    @lt_ECHO@ @ECHO_N@ "***" ;\
 	    length="`expr $$length - 3`" ;\
           done ;\
-	  @ECHO@ ;\
+	  @lt_ECHO@ ;\
 	elif test "x$(MAKECMDGOALS)" = "x" ; then \
-	  @ECHO@ @ECHO_N@ "Elapsed time: " ;\
+	  @lt_ECHO@ @ECHO_N@ "Elapsed time: " ;\
 	  sh $(top_srcdir)/misc/elapsed.sh `cat $(top_builddir)/include/libirc_version.h | grep Compilation` ;\
-	  @ECHO@ @ECHO_N@ "Elapsed time since configuration: " ;\
+	  @lt_ECHO@ @ECHO_N@ "Elapsed time since configuration: " ;\
 	  sh $(top_srcdir)/misc/elapsed.sh @CONFIG_TIME@ ;\
-	  @ECHO@ "---" ;\
-	  @ECHO@ "Run 'make test' to test the libIRC framework" ;\
+	  @lt_ECHO@ "---" ;\
+	  @lt_ECHO@ "Run 'make test' to test the libIRC framework" ;\
 	fi
-	@@ECHO@
+	@@lt_ECHO@
 
 DISTCLEANFILES = \
 	$(CONFIG_CACHE) \
diff -r -u -- libirc-svn20080704_orig/Makefile.in libirc-svn20080704/Makefile.in
diff -r -u -- libirc-svn20080704_orig/autom4te.cache/output.0 libirc-svn20080704/autom4te.cache/output.0
diff -r -u -- libirc-svn20080704_orig/autom4te.cache/output.1 libirc-svn20080704/autom4te.cache/output.1
diff -r -u -- libirc-svn20080704_orig/autom4te.cache/output.2 libirc-svn20080704/autom4te.cache/output.2
diff -r -u -- libirc-svn20080704_orig/autom4te.cache/requests libirc-svn20080704/autom4te.cache/requests
diff -r -u -- libirc-svn20080704_orig/autom4te.cache/traces.0 libirc-svn20080704/autom4te.cache/traces.0
diff -r -u -- libirc-svn20080704_orig/autom4te.cache/traces.1 libirc-svn20080704/autom4te.cache/traces.1
diff -r -u -- libirc-svn20080704_orig/autom4te.cache/traces.2 libirc-svn20080704/autom4te.cache/traces.2
diff -r -u -- libirc-svn20080704_orig/botlib/Makefile.in libirc-svn20080704/botlib/Makefile.in
diff -r -u -- libirc-svn20080704_orig/botlib/inc/Makefile.in libirc-svn20080704/botlib/inc/Makefile.in
diff -r -u -- libirc-svn20080704_orig/configure libirc-svn20080704/configure
diff -r -u -- libirc-svn20080704_orig/configure.ac libirc-svn20080704/configure.ac
--- libirc-svn20080704_orig/configure.ac	2008-07-04 03:52:41.000000000 +0200
+++ libirc-svn20080704/configure.ac	2008-07-04 03:53:04.000000000 +0200
@@ -294,6 +294,7 @@
 BC_CONFIGURE_STAGE([programs], [2 of 9])
 
 AC_PROG_CC
+AC_PROG_CXX
 AM_PROG_CC_STDC
 AC_PROG_CPP
 AC_PROG_INSTALL
@@ -305,7 +305,7 @@
 AC_PATH_PROG(SH, sh)
 AC_PATH_PROG(MV, mv)
 AC_PATH_PROG(CP, cp)
-AC_PATH_PROG(RM, rm)
+dnl AC_PATH_PROG(RM, rm)
 
 dnl figure out which compiler is seems like we are using
 dnl detected values include: gcc, mipspro, sunpro
