--- wxWindows-2.4.2_orig/wxwin.m4	2008-04-01 17:52:06.000000000 +0000
+++ wxWindows-2.4.2/wxwin.m4	2008-04-01 17:51:39.000000000 +0000
@@ -33,7 +33,7 @@
 dnl command line options
 dnl ---------------------------------------------------------------------------
 
-AC_DEFUN(AM_OPTIONS_WXCONFIG,
+AC_DEFUN([AM_OPTIONS_WXCONFIG],
 [
    AC_ARG_WITH(wx-prefix, [  --with-wx-prefix=PREFIX   Prefix where wxWindows is installed (optional)],
                wx_config_prefix="$withval", wx_config_prefix="")
@@ -56,7 +56,7 @@
 dnl
 dnl Get the cflags and libraries from the wx-config script
 dnl
-AC_DEFUN(AM_PATH_WXCONFIG,
+AC_DEFUN([AM_PATH_WXCONFIG],
 [
   dnl do we have wx-config name: it can be wx-config or wxd-config or ...
   if test x${WX_CONFIG_NAME+set} != xset ; then
