diff -ru ffmpeg-export-2009-02-07_orig/libavcodec/Makefile ffmpeg-export-2009-02-07/libavcodec/Makefile
--- ffmpeg-export-2009-02-07_orig/libavcodec/Makefile	2009-02-07 22:10:07.000000000 +0100
+++ ffmpeg-export-2009-02-07/libavcodec/Makefile	2009-02-07 23:02:43.000000000 +0100
@@ -3,7 +3,7 @@
 NAME = avcodec
 FFLIBS = avutil
 
-HEADERS = avcodec.h opt.h vdpau.h
+HEADERS = avcodec.h opt.h vdpau.h imgconvert.h
 
 OBJS = allcodecs.o                                                      \
        audioconvert.o                                                   \
diff -ru ffmpeg_orig/libavcodec/imgconvert.c ffmpeg/libavcodec/imgconvert.c
--- ffmpeg_orig/libavcodec/imgconvert.c	2009-02-26 00:50:10.000000000 +0100
+++ ffmpeg/libavcodec/imgconvert.c	2009-02-26 00:49:05.000000000 +0100
@@ -2101,7 +2101,7 @@
     return 0;
 }
 
-#if !CONFIG_SWSCALE
+#if 1 //!CONFIG_SWSCALE
 static uint8_t y_ccir_to_jpeg[256];
 static uint8_t y_jpeg_to_ccir[256];
 static uint8_t c_ccir_to_jpeg[256];
