diff -r -u sex-1.0_orig/sex.c sex-1.0/sex.c
--- sex-1.0_orig/sex.c	2007-10-16 14:14:53.000000000 +0000
+++ sex-1.0/sex.c	2007-10-16 14:15:23.000000000 +0000
@@ -240,7 +240,7 @@
         {his,            SZ(his)},        {dongadj,        SZ(dongadj)},
         {dong,           SZ(dong)},       {intoher,        SZ(intoher)},
         {twatadj,        SZ(twatadj)},    {twat,           SZ(twat)},
-        {(char **)NULL,  (short)NULL},
+        {0,  0},
 };
 
 #define LLINE   50
