diff -ru -ru cairo-1.2.0_orig/configure cairo-1.2.0/configure
--- cairo-1.2.0_orig/configure	2006-07-21 15:11:47.000000000 +0200
+++ cairo-1.2.0/configure	2006-07-21 15:12:22.000000000 +0200
@@ -23007,7 +23007,7 @@
 
   use_png=no
   # libpng13 is GnuWin32's libpng-1.2.8 :-(
-  for l in libpng12 libpng13 libpng10 ; do
+  for l in libpng libpng13 libpng10 ; do
     if $PKG_CONFIG --exists $l ; then
       png_REQUIRES=$l
       use_png=yes
diff -ru -ru cairo-1.2.0_orig/configure.in cairo-1.2.0/configure.in
--- cairo-1.2.0_orig/configure.in	2006-07-21 15:11:47.000000000 +0200
+++ cairo-1.2.0/configure.in	2006-07-21 15:12:21.000000000 +0200
@@ -273,7 +273,7 @@
 CAIRO_BACKEND_ENABLE(png, PNG, png, PNG_FUNCTIONS, yes, [
   use_png=no
   # libpng13 is GnuWin32's libpng-1.2.8 :-(
-  for l in libpng12 libpng13 libpng10 ; do
+  for l in libpng libpng13 libpng10 ; do
     if $PKG_CONFIG --exists $l ; then
       png_REQUIRES=$l
       use_png=yes
