diff -ru flac-1.2.1_orig/include/share/alloc.h flac-1.2.1/include/share/alloc.h
--- flac-1.2.1_orig/include/share/alloc.h	2008-12-01 04:52:07.000000000 +0100
+++ flac-1.2.1/include/share/alloc.h	2008-12-01 04:52:29.000000000 +0100
@@ -34,14 +34,14 @@
 #include <stdlib.h> /* for size_t, malloc(), etc */
 
 #ifndef SIZE_MAX
-# ifndef SIZE_T_MAX
+# ifndef ULONG_MAX
 #  ifdef _MSC_VER
-#   define SIZE_T_MAX UINT_MAX
+#   define ULONG_MAX UINT_MAX
 #  else
 #   error
 #  endif
 # endif
-# define SIZE_MAX SIZE_T_MAX
+# define SIZE_MAX ULONG_MAX
 #endif
 
 #ifndef FLaC__INLINE
