--- flac-1.2.1_orig/src/libFLAC/Makefile.am	2008-12-03 16:11:08.000000000 +0100
+++ flac-1.2.1/src/libFLAC/Makefile.am	2008-12-03 16:39:46.000000000 +0100
@@ -97,7 +97,7 @@
 endif
 # see 'http://www.gnu.org/software/libtool/manual.html#Libtool-versioning' for numbering convention
 libFLAC_la_LDFLAGS = -no-undefined -version-info 10:0:2 -lm $(LOCAL_EXTRA_LDFLAGS)
-libFLAC_la_LIBADD = -lwsock32
+libFLAC_la_LIBADD = -lwsock32 @OGG_LIBS@
 libFLAC_la_SOURCES = \
 	bitmath.c \
 	bitreader.c \
