--- icecast-2.3.1_orig/src/auth_url.c	2008-03-24 17:31:57.000000000 +0000
+++ icecast-2.3.1/src/auth_url.c	2008-03-24 17:34:43.000000000 +0000
@@ -518,7 +518,7 @@
     curl_easy_setopt (url_info->handle, CURLOPT_WRITEDATA, url_info->handle);
     curl_easy_setopt (url_info->handle, CURLOPT_NOSIGNAL, 1L);
     curl_easy_setopt (url_info->handle, CURLOPT_TIMEOUT, 15L);
-    curl_easy_setopt (url_info->handle, CURLOPT_PASSWDFUNCTION, my_getpass);
+/*    curl_easy_setopt (url_info->handle, CURLOPT_PASSWDFUNCTION, my_getpass);*/
     curl_easy_setopt (url_info->handle, CURLOPT_ERRORBUFFER, &url_info->errormsg[0]);
 
     if (url_info->username && url_info->password)
