diff -ru gitfs-0.04_orig/gitworker.c /usr/src/fuse/gitfs-0.04/gitworker.c
--- gitfs-0.04_orig/gitworker.c	2008-06-13 16:50:55.000000000 +0200
+++ gitfs-0.04/gitworker.c	2008-06-13 17:09:47.000000000 +0200
@@ -118,7 +118,7 @@
 	if (cmd->answer.open.type == GFN_FILE ||
 	    cmd->answer.open.type == GFN_SYMLINK) {
 		char type[20];
-		if (sha1_object_info(&cmd->gptr->sha1[0], type,
+		if (sha1_object_info(&cmd->gptr->sha1[0], //type,
 				     &cmd->answer.open.size) == 0 &&
 		    0 == strcmp(type, "blob")) {
 			cmd->error = 0;
