--- libcomprex-0.3.3_orig/libcomprex.m4.in	2008-04-01 20:30:49.000000000 +0000
+++ libcomprex-0.3.3/libcomprex.m4.in	2008-04-01 20:31:00.000000000 +0000
@@ -5,7 +5,7 @@
 dnl AM_PATH_COMPREX([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for COMPREX, and define COMPREX_CFLAGS and COMPREX_LIBS
 dnl
-AC_DEFUN(AM_PATH_COMPREX,
+AC_DEFUN([AM_PATH_COMPREX],
 [dnl
 dnl Get the cflags and libraries from the comprex-config script
 dnl
