diff -ru libcomprex-0.3.3_orig/comprex-config.in libcomprex-0.3.3/comprex-config.in
--- libcomprex-0.3.3_orig/comprex-config.in	2008-02-22 13:08:11.000000000 +0100
+++ libcomprex-0.3.3/comprex-config.in	2008-02-22 13:08:50.000000000 +0100
@@ -44,10 +44,10 @@
 			;;
 
 		--cflags)
-			if test @includedir@ != /usr/include ; then
+			if test @includedir@ != $prefix/include ; then
 				includes=-I@includedir@
 			fi
-			echo $includes -I/usr/include
+			echo $includes -I$prefix/include/libcomprex
 			;;
 
 		--libs)
