--- file-5.00_orig/magic/Magdir/microsoft	1970-01-01 01:00:00.000000000 +0100
+++ file-5.00/magic/Magdir/microsoft	2009-04-17 13:50:35.000000000 +0200
@@ -0,0 +1,7 @@
+#------------------------------------------------------------------------------
+# microsoft:  file(1) magic for crappy Micro$oft data formats
+#
+
+# WAB - Windows address book format, as exported by Outlook
+0 string	\x9c\xcb\xcb\x8d\x13\x75\xd2\x11	Windows address book
+!:mime  application/ms-wab
