diff -ru -ru file-4.17_orig/magic/magic.mime file-4.17/magic/magic.mime
--- file-4.17_orig/magic/magic.mime	2007-01-12 15:22:12.117760464 +0100
+++ file-4.17/magic/magic.mime	2007-01-12 15:22:24.003953488 +0100
@@ -827,10 +827,10 @@
 #
 
 #0       string          \320\317\021\340\241\261\032\341
-#>48     byte            0x1B            application/excel
+#>48     byte            0x1B            application/msexcel
 
-2080	string	Microsoft\ Excel\ 5.0\ Worksheet	application/excel
-2114	string	Biff5					application/excel
+2080	string	Microsoft\ Excel\ 5.0\ Worksheet	application/msexcel
+2114	string	Biff5					application/msexcel
 
 0       string	\224\246\056		application/msword
 
@@ -838,18 +838,37 @@
 
 0	string	PO^Q`			application/msword
 
-0	string	\320\317\021\340\241\261\032\341
->546	string	bjbj			application/msword
->546	string	jbjb			application/msword
-
 512	string	R\0o\0o\0t\0\ \0E\0n\0t\0r\0y	application/msword
 
 2080	string	Microsoft\ Word\ 6.0\ Document	application/msword
 2080	string	Documento\ Microsoft\ Word\ 6	application/msword
 2112	string	MSWordDoc			application/msword
 
-#0	string	\320\317\021\340\241\261\032\341	application/powerpoint
-0	string	\320\317\021\340\241\261\032\341	application/msword
+2112	string	MSWordDoc	application/msword
+0	belong	0x31be0000	application/msword
+0	string	PO^Q`		application/msword
+
+0	string	\376\067\0\043				application/msoffice
+0	string	\320\317\021\340\241\261\032\341	application/msoffice
+0	string	\333\245-\0\0\0				application/msoffice
+
+2080	string	Microsoft\ Excel\ 5.0\ Worksheet	application/msexcel
+2080	string	Foglio\ di\ lavoro\ Microsoft\ Exce	application/msexcel
+2114	string	Biff5					application/msexcel
+2121	string	Biff5					application/msexcel
+0	string	\x09\x04\x06\x00\x00\x00\x10\x00	application/msexcel
+
+#
+0 belong  0x00001a00  Lotus 1-2-3
+>4  belong  0x00100400  wk3 document data
+>4  belong  0x02100400  wk4 document data
+>4  belong  0x07800100  fm3 or fmb document data
+>4  belong  0x07800000  fm3 or fmb document data
+#
+0 belong  0x00000200  Lotus 1-2-3
+>4  belong  0x06040600  wk1 document data
+>4  belong  0x06800200  fmt document data
+
 
 0       string  #\ PaCkAgE\ DaTaStReAm  application/x-svr4-package
 
