--- file-5.00_orig/magic/Magdir/misctools	2009-04-17 13:51:06.000000000 +0200
+++ file-5.00/magic/Magdir/misctools	2009-04-17 13:53:40.000000000 +0200
@@ -4,6 +4,9 @@
 0	search/1	%%!!			X-Post-It-Note text
 0	string/c	BEGIN:VCALENDAR		vCalendar calendar file
 0	string/c	BEGIN:VCARD		vCard visiting card
+0 string/c  begin:vcard             vCard visiting card (UTF-8)
+0 string/c  \0b\0e\0g\0i\0n\0:\0v\0c\0a\0r\0d               vCard visiting card (UTF-16BE)
+0 string/c  b\0e\0g\0i\0n\0:\0v\0c\0a\0r\0d\0               vCard visiting card (UTF-16LE)
 !:mime	text/x-vcard
 
 # From: Alex Beregszaszi <alex@fsn.hu>
