diff -ru -ru file-4.17_orig/magic/magic2mime file-4.17/magic/magic2mime
--- file-4.17_orig/magic/magic2mime	2007-01-12 14:48:09.767244936 +0100
+++ file-4.17/magic/magic2mime	2007-01-12 14:55:03.878290488 +0100
@@ -31,6 +31,7 @@
 	'^BinHex binary text', 'application/mac-binhex40',
 	'^Zip archive data', 'application/zip',
 	'^Microsoft Word', 'application/msword',
+	'^Microsoft Office', 'application/msoffice',
 	'^PGP key', 'application/pgp-keys',
 	'^PGP encrypted', 'application/pgp-encrypted',
 	'^PGP armored data signature', 'application/pgp-signature',
@@ -48,8 +49,11 @@
     '^FLI', 'video/x-fli',
     '^FLC', 'video/x-flc',
     'AVI data', 'video/x-avi',
-    'WAVE', 'audio/x-wav',
     'VOC', 'audio/x-voc',
+    'vCard visiting card (UTF-8)', 'application/vcard; charset=UTF-8',  
+    'vCard visiting card (UTF-16BE)', 'application/vcard; charset=UTF-16BE',  
+    'vCard visiting card (UTF-16LE)', 'application/vcard; charset=UTF-16LE',
+    'WAVE', 'audio/x-wav',
 );
 
 local($mimetype,$index,$regexp);
