diff -ru websvn-2.1.0_orig/include/setup.php websvn-2.1.0/include/setup.php
--- websvn-2.1.0_orig/include/setup.php	2009-04-20 11:44:30.000000000 +0200
+++ websvn-2.1.0/include/setup.php	2009-04-20 11:47:46.000000000 +0200
@@ -222,6 +222,8 @@
   '.pp'    => 'pascal',
   '.ps'    => 'postscript',
   '.s'     => 'asm',
+  '.asm'   => 'asm',
+  '.inc'   => 'asm',
   '.scheme'  => 'scheme',
   '.scm'    => 'scheme',
   '.scr'    => 'synopsys',
