--- php-4.4.2_orig/configure	2006-01-30 13:50:13.000000000 +0100
+++ php-4.4.2/configure	2006-01-30 13:52:16.000000000 +0100
@@ -17519,14 +17519,14 @@
   c|c_r|pthread*) ;;
   *) 
     if test "$ext_shared" = "yes"; then
-        OPENSSL_SHARED_LIBADD="$OPENSSL_SHARED_LIBADD -lgssapi_krb5" 
+        OPENSSL_SHARED_LIBADD="$OPENSSL_SHARED_LIBADD -lkrb5"
     else
       
   
   case gssapi_krb5 in
   c|c_r|pthread*) ;;
   *) 
-      LIBS="$LIBS -lgssapi_krb5" 
+      LIBS="$LIBS -lkrb5"
    ;;
   esac
 
@@ -45411,14 +45411,14 @@
   c|c_r|pthread*) ;;
   *) 
     if test "$ext_shared" = "yes"; then
-        IMAP_SHARED_LIBADD="$IMAP_SHARED_LIBADD -lgssapi_krb5" 
+        IMAP_SHARED_LIBADD="$IMAP_SHARED_LIBADD -lkrb5"
     else
       
   
   case gssapi_krb5 in
   c|c_r|pthread*) ;;
-  *) 
-      LIBS="$LIBS -lgssapi_krb5" 
+  *)
+      LIBS="$LIBS -lkrb5"
    ;;
   esac
 
