--- gcc-4.1.1_orig/libjava/Makefile.in	2006-06-02 15:26:51.000000000 +0200
+++ gcc-4.1.1/libjava/Makefile.in	2006-06-02 15:28:40.000000000 +0200
@@ -738,7 +738,7 @@
 	$(all_packages_source_files:.list=.lo) \
 	$(bc_objects) \
 	$(propertyo_files) \
-	$(LIBFFI) $(ZLIBS) $(GCLIBS)
+	$(LIBFFI) $(ZLIBS) $(GCLIBS) $(LIBICONV)
 
 libgcj_la_DEPENDENCIES = libgcj-$(gcc_version).jar \
 	$(all_packages_source_files:.list=.lo) \
