diff -ru teem-1.0_orig/ga/GenerationalEvolution.cpp teem-1.0/ga/GenerationalEvolution.cpp
--- teem-1.0_orig/ga/GenerationalEvolution.cpp	2009-02-25 23:43:08.000000000 +0100
+++ teem-1.0/ga/GenerationalEvolution.cpp	2009-02-25 23:43:24.000000000 +0100
@@ -39,6 +39,7 @@
 #include <iomanip>
 #include <limits>
 #include <dirent.h>
+#include <cstring>
 
 /*!	\file GenerationalEvolution.cpp
 	\brief Implementation of algorithms for generations based evolution based
diff -ru teem-1.0_orig/ga/SteadyStateEvolution.cpp teem-1.0/ga/SteadyStateEvolution.cpp
--- teem-1.0_orig/ga/SteadyStateEvolution.cpp	2009-02-25 23:43:08.000000000 +0100
+++ teem-1.0/ga/SteadyStateEvolution.cpp	2009-02-25 23:43:38.000000000 +0100
@@ -35,6 +35,7 @@
 #include <iomanip>
 #include <limits>
 #include <dirent.h>
+#include <cstring>
 
 /*!	\file SteadyStateEvolution.cpp
 	\brief Implementation of algorithms for steady-state (continuous) evolution
diff -ru teem-1.0_orig/enkiviewer/videorec.cpp teem-1.0/enkiviewer/videorec.cpp
--- teem-1.0_orig/enkiviewer/videorec.cpp	2009-02-25 23:46:51.000000000 +0100
+++ teem-1.0/enkiviewer/videorec.cpp	2009-02-25 23:48:50.000000000 +0100
@@ -4,6 +4,8 @@
 #include <ffmpeg/avcodec.h>
 #include <ffmpeg/avformat.h>
 #include <iostream>
+#include <cstring>
+#include <cstdlib>
 
 struct Vid
 {
