Only in cint-5.16.19: Makefile.conf
diff -r -u -- cint-5.16.19_orig/configure cint-5.16.19/configure
--- cint-5.16.19_orig/configure	2007-06-13 11:34:33.000000000 +0200
+++ cint-5.16.19/configure	2007-06-13 11:42:34.000000000 +0200
@@ -9,14 +9,14 @@
 # This is a bash script, which works with any reasonable sh.
 # Solaris sh is not, so we have to start bash, and call ourselves again.
 
-[ "x$1" != "x__have_the_proper_shell@" ] && ( \
-uname -a | grep -i sunos > /dev/null \
-  && (bash $0 __have_the_proper_shell@ "$*") \
-  || (sh $0 __have_the_proper_shell@ "$*") \
-)
-
-[ "x$1" != "x__have_the_proper_shell@" ] && exit
-shift
+#[ "x$1" != "x__have_the_proper_shell@" ] && ( \
+#uname -a | grep -i sunos > /dev/null \
+#  && (bash $0 __have_the_proper_shell@ "$*") \
+#  || (sh $0 __have_the_proper_shell@ "$*") \
+#)
+#
+#[ "x$1" != "x__have_the_proper_shell@" ] && exit
+#shift
 
 # configure settings to build CINT
 
@@ -412,7 +416,7 @@
     CINTEXLIBDIR=$1
 }
 
-function processflag_readlinelib {
+function processflag_readline {
     READLINELIB=$1
     BUILDREADLINE=
 }
