diff -r -u -- cint-5.16.19_orig/Makefile cint-5.16.19/Makefile
--- cint-5.16.19_orig/Makefile	2007-06-13 11:50:52.000000000 +0200
+++ cint-5.16.19/Makefile	2007-06-13 11:53:27.000000000 +0200
@@ -42,7 +42,7 @@
 		unset VS_UNICODE_OUTPUT; \
 		LD_LIBRARY_PATH=$$LD_LIBRARY_PATH:. \
 		DYLD_LIBRARY_PATH=$$DYLD_LIBRARY_PATH:. \
-		./cint$(G__CFG_EXEEXT) $(G__CFG_INCP)inc iosenum.cxx); \
+		./cint$(G__CFG_EXEEXT) $(G__CFG_INCP)inc +P include/iosenum.cxx); \
 	fi)
 	@echo > done
 
