diff -ru glibc-2.7_orig/debug/stack_chk_fail_local.c glibc-2.7/debug/stack_chk_fail_local.c
--- glibc-2.7_orig/debug/stack_chk_fail_local.c	2008-11-12 07:19:05.000000000 +0100
+++ glibc-2.7/debug/stack_chk_fail_local.c	2008-11-13 02:28:33.000000000 +0100
@@ -23,8 +23,10 @@
 /* On some architectures, this helps needless PIC pointer setup
    that would be needed just for the __stack_chk_fail call.  */
 
+#if 0
 void __attribute__ ((noreturn)) attribute_hidden
 __stack_chk_fail_local (void)
 {
   __stack_chk_fail ();
 }
+#endif
