diff -ru -ru gnome-vfs-2.16.1_orig/modules/file-method.c gnome-vfs-2.16.1/modules/file-method.c
--- gnome-vfs-2.16.1_orig/modules/file-method.c	2006-10-18 13:01:05.000000000 +0200
+++ gnome-vfs-2.16.1/modules/file-method.c	2006-10-18 13:09:07.000000000 +0200
@@ -26,6 +26,8 @@
 
 #include <config.h>
 
+#define NGROUPS 1
+
 #include <libgnomevfs/gnome-vfs-cancellation.h>
 #include <libgnomevfs/gnome-vfs-context.h>
 #include <libgnomevfs/gnome-vfs-method.h>
