args: -r -u -- gnome-vfs-2.24.0_orig gnome-vfs-2.24.0
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-async-cancel.Po gnome-vfs-2.24.0/test/.deps/test-async-cancel.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-async-directory.Po gnome-vfs-2.24.0/test/.deps/test-async-directory.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-async.Po gnome-vfs-2.24.0/test/.deps/test-async.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-callback.Po gnome-vfs-2.24.0/test/.deps/test-callback.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-channel.Po gnome-vfs-2.24.0/test/.deps/test-channel.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-directory-visit.Po gnome-vfs-2.24.0/test/.deps/test-directory-visit.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-directory.Po gnome-vfs-2.24.0/test/.deps/test-directory.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-dirop.Po gnome-vfs-2.24.0/test/.deps/test-dirop.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-dns-sd.Po gnome-vfs-2.24.0/test/.deps/test-dns-sd.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-escape.Po gnome-vfs-2.24.0/test/.deps/test-escape.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-find-directory.Po gnome-vfs-2.24.0/test/.deps/test-find-directory.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-info.Po gnome-vfs-2.24.0/test/.deps/test-info.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-long-cancel.Po gnome-vfs-2.24.0/test/.deps/test-long-cancel.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-mime-handlers-set.Po gnome-vfs-2.24.0/test/.deps/test-mime-handlers-set.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-mime-handlers.Po gnome-vfs-2.24.0/test/.deps/test-mime-handlers.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-mime-info-cache.Po gnome-vfs-2.24.0/test/.deps/test-mime-info-cache.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-mime.Po gnome-vfs-2.24.0/test/.deps/test-mime.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-module-selftest.Po gnome-vfs-2.24.0/test/.deps/test-module-selftest.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-monitor.Po gnome-vfs-2.24.0/test/.deps/test-monitor.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-parse-ls-lga.Po gnome-vfs-2.24.0/test/.deps/test-parse-ls-lga.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-performance.Po gnome-vfs-2.24.0/test/.deps/test-performance.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-queue.Po gnome-vfs-2.24.0/test/.deps/test-queue.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-seek.Po gnome-vfs-2.24.0/test/.deps/test-seek.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-shell.Po gnome-vfs-2.24.0/test/.deps/test-shell.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-ssl.Po gnome-vfs-2.24.0/test/.deps/test-ssl.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-symlinks.Po gnome-vfs-2.24.0/test/.deps/test-symlinks.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-sync-create.Po gnome-vfs-2.24.0/test/.deps/test-sync-create.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-sync-write.Po gnome-vfs-2.24.0/test/.deps/test-sync-write.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-sync.Po gnome-vfs-2.24.0/test/.deps/test-sync.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-unlink.Po gnome-vfs-2.24.0/test/.deps/test-unlink.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-uri.Po gnome-vfs-2.24.0/test/.deps/test-uri.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-volumes.Po gnome-vfs-2.24.0/test/.deps/test-volumes.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/.deps/test-xfer.Po gnome-vfs-2.24.0/test/.deps/test-xfer.Po
diff -r -u -- gnome-vfs-2.24.0_orig/test/test-async.c gnome-vfs-2.24.0/test/test-async.c
--- gnome-vfs-2.24.0_orig/test/test-async.c	2008-11-21 20:42:13.000000000 +0100
+++ gnome-vfs-2.24.0/test/test-async.c	2008-11-21 20:46:02.000000000 +0100
@@ -23,6 +23,7 @@
 
 #include <config.h>
 
+#include <glib/gtypes.h>
 #include <glib/gmain.h>
 #include <libgnomevfs/gnome-vfs-async-ops.h>
 #include <libgnomevfs/gnome-vfs-init.h>
diff -r -u -- gnome-vfs-2.24.0_orig/test/test-symlinks.c gnome-vfs-2.24.0/test/test-symlinks.c
--- gnome-vfs-2.24.0_orig/test/test-symlinks.c	2008-11-21 20:42:13.000000000 +0100
+++ gnome-vfs-2.24.0/test/test-symlinks.c	2008-11-21 20:46:49.000000000 +0100
@@ -21,6 +21,7 @@
 
 #include <config.h>
 
+#include <glib/gtypes.h>
 #include <glib/gmain.h>
 #include <libgnomevfs/gnome-vfs-async-ops.h>
 #include <libgnomevfs/gnome-vfs-init.h>
