diff -r -u oscpack_orig/examples/OscDump.cpp oscpack/examples/OscDump.cpp
--- oscpack_orig/examples/OscDump.cpp	2007-08-14 09:02:45.000000000 +0200
+++ oscpack/examples/OscDump.cpp	2007-08-14 09:04:09.000000000 +0200
@@ -36,6 +36,8 @@
 
 
 #include <iostream>
+#include <cstring>
+#include <cstdlib>
 
 #include "osc/OscReceivedElements.h"
 #include "osc/OscPrintReceivedElements.h"
diff -r -u oscpack_orig/osc/OscPrintReceivedElements.cpp oscpack/osc/OscPrintReceivedElements.cpp
--- oscpack_orig/osc/OscPrintReceivedElements.cpp	2007-08-14 09:02:45.000000000 +0200
+++ oscpack/osc/OscPrintReceivedElements.cpp	2007-08-14 09:03:16.000000000 +0200
@@ -32,7 +32,7 @@
 #include <iostream>
 #include <iomanip>
 #include <ctime>
-
+#include <cstring>
 
 namespace osc{
 
diff -r -u oscpack_orig/tests/OscReceiveTest.cpp oscpack/tests/OscReceiveTest.cpp
--- oscpack_orig/tests/OscReceiveTest.cpp	2007-08-14 09:02:45.000000000 +0200
+++ oscpack/tests/OscReceiveTest.cpp	2007-08-14 09:03:47.000000000 +0200
@@ -31,6 +31,7 @@
 
 #include <string.h>
 #include <iostream>
+#include <cstdlib>
 
 #include "osc/OscReceivedElements.h"
 
diff -r -u oscpack_orig/tests/OscSendTests.cpp oscpack/tests/OscSendTests.cpp
--- oscpack_orig/tests/OscSendTests.cpp	2007-08-14 09:02:45.000000000 +0200
+++ oscpack/tests/OscSendTests.cpp	2007-08-14 09:03:28.000000000 +0200
@@ -31,6 +31,7 @@
 
 #include <iostream>
 #include <string.h>
+#include <cstdlib>
 
 #include "osc/OscOutboundPacketStream.h"
 
diff -r -u oscpack_orig/tests/OscUnitTests.cpp oscpack/tests/OscUnitTests.cpp
--- oscpack_orig/tests/OscUnitTests.cpp	2007-08-14 09:02:45.000000000 +0200
+++ oscpack/tests/OscUnitTests.cpp	2007-08-14 09:02:57.000000000 +0200
@@ -31,6 +31,7 @@
 
 #include <iostream>
 #include <iomanip>
+#include <cstring>
 
 #include "osc/OscReceivedElements.h"
 #include "osc/OscPrintReceivedElements.h"
