--- netwatch-1.2.0_orig/Makefile.in	2009-03-01 14:27:45.000000000 +0100
+++ netwatch-1.2.0/Makefile.in	2009-03-01 14:30:09.000000000 +0100
@@ -22,10 +22,10 @@
 	$(CC) -o netwatch $(XCFLAGS) $(OBJECTS) $(XLIBS)
 
 install:	$(EXEC)
-	install --owner=root --group=root --mode=0755 -d $(INSTALLDIR)
-	install --owner=root --group=root --mode=0755 -d $(mandir)/man1
-	install --owner=root --group=root --mode=0700  netwatch $(INSTALLDIR)
-	install --owner=root --group=root --mode=0700  netresolv $(INSTALLDIR)
+	install --mode=0755 -d $(INSTALLDIR)
+	install --mode=0755 -d $(mandir)/man1
+	install --mode=0700  netwatch $(INSTALLDIR)
+	install --mode=0700  netresolv $(INSTALLDIR)
 	bzip2 netwatch.1
 	install -D  netwatch.1.bz2 $(mandir)/man1
 
