diff -ru tntdb-0.9.4_orig/src/Makefile.am tntdb-0.9.4/src/Makefile.am
--- tntdb-0.9.4_orig/src/Makefile.am	2007-10-25 15:27:54.000000000 +0200
+++ tntdb-0.9.4/src/Makefile.am	2007-10-25 15:28:48.000000000 +0200
@@ -22,5 +22,5 @@
 	time.cpp \
 	valueimpl.cpp
 
-libtntdb_la_LDFLAGS = -version-info @sonumber@ $(PTHREAD_LIBS) @SHARED_LIB_FLAG@
+libtntdb_la_LDFLAGS = -version-number @sonumber@ $(PTHREAD_LIBS) @SHARED_LIB_FLAG@
 libtntdb_la_CXXFLAGS = -DDRIVERDIR=\"@driverdir@\" $(PTHREAD_CFLAGS)
diff -ru tntdb-0.9.4_orig/src/Makefile.in tntdb-0.9.4/src/Makefile.in
--- tntdb-0.9.4_orig/src/Makefile.in	2007-10-25 15:27:54.000000000 +0200
+++ tntdb-0.9.4/src/Makefile.in	2007-10-25 15:28:48.000000000 +0200
@@ -230,7 +230,7 @@
 	time.cpp \
 	valueimpl.cpp
 
-libtntdb_la_LDFLAGS = -version-info @sonumber@ $(PTHREAD_LIBS) @SHARED_LIB_FLAG@
+libtntdb_la_LDFLAGS = -version-number @sonumber@ $(PTHREAD_LIBS) @SHARED_LIB_FLAG@
 libtntdb_la_CXXFLAGS = -DDRIVERDIR=\"@driverdir@\" $(PTHREAD_CFLAGS)
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-am
Only in tntdb-0.9.4/src: stamp-h1
