diff -r -u tntdb-0.9.4_orig/configure tntdb-0.9.4/configure
--- tntdb-0.9.4_orig/configure	2007-10-25 12:51:44.000000000 +0000
+++ tntdb-0.9.4/configure	2007-10-25 12:53:11.000000000 +0000
@@ -2429,7 +2429,7 @@
 abi_current=0
 abi_revision=0
 abi_age=0
-sonumber=${abi_current}:${abi_revision}:${abi_age}
+sonumber=`IFS=':.'; set -- $PACKAGE_VERSION; echo "$*"`
 
 
 unset CDPATH
diff -r -u tntdb-0.9.4_orig/configure.in tntdb-0.9.4/configure.in
--- tntdb-0.9.4_orig/configure.in	2007-10-25 12:51:44.000000000 +0000
+++ tntdb-0.9.4/configure.in	2007-10-25 12:52:48.000000000 +0000
@@ -4,7 +4,7 @@
 abi_current=0
 abi_revision=0
 abi_age=0
-sonumber=${abi_current}:${abi_revision}:${abi_age}
+sonumber=`IFS=':.'; set -- $PACKAGE_VERSION; echo "$*"`
 AC_SUBST(sonumber)
 
 unset CDPATH
