diff -ru ncurses-5.7_orig/ncurses/tinfo/comp_parse.c ncurses-5.7/ncurses/tinfo/comp_parse.c
--- ncurses-5.7_orig/ncurses/tinfo/comp_parse.c	2008-12-16 12:52:21.000000000 +0100
+++ ncurses-5.7/ncurses/tinfo/comp_parse.c	2008-12-16 12:55:56.000000000 +0100
@@ -204,13 +204,13 @@
 				   _nc_first_name(qp->tterm.term_names)));
 		    multiples++;
 		}
-		if (matchcount >= 1)
+		/*if (matchcount >= 1)
 		    (void) fprintf(stderr, " %s", _nc_first_name(rp->tterm.term_names));
-	    }
+*/	    }
 	}
-	if (matchcount >= 1)
+	/*if (matchcount >= 1)
 	    (void) putc('\n', stderr);
-    }
+*/    }
     if (multiples > 0)
 	return (FALSE);
 
