--- ncurses-5.7_orig/ncurses/tinfo/comp_parse.c	2008-11-21 15:58:33.000000000 +0100
+++ ncurses-5.7/ncurses/tinfo/comp_parse.c	2008-11-21 16:05:09.000000000 +0100
@@ -200,8 +200,8 @@
 		&& _nc_entry_match(qp->tterm.term_names, rp->tterm.term_names)) {
 		matchcount++;
 		if (matchcount == 1) {
-		    (void) fprintf(stderr, "Name collision between %s",
-				   _nc_first_name(qp->tterm.term_names));
+		    DEBUG(2, ("Name collision between %s",
+				   _nc_first_name(qp->tterm.term_names)));
 		    multiples++;
 		}
 		if (matchcount >= 1)
