diff -ru srcconf-0.1_orig/lkc/.deps/mconf.Po srcconf-0.1/lkc/.deps/mconf.Po
Only in srcconf-0.1/lkc: mconf
diff -ru srcconf-0.1_orig/lkc/mconf.c srcconf-0.1/lkc/mconf.c
--- srcconf-0.1_orig/lkc/mconf.c	2008-12-05 22:03:32.000000000 +0100
+++ srcconf-0.1/lkc/mconf.c	2008-12-05 22:04:07.000000000 +0100
@@ -22,6 +22,8 @@
 #define LKC_DIRECT_LINK
 #include "lkc.h"
 
+struct menu *current_menu;
+
 static char menu_backtitle[128];
 static const char menu_instructions[] =
 	"Arrow keys navigate the menu.  "
@@ -88,7 +90,6 @@
 static int indent = 0;
 static struct termios ios_org;
 static int rows, cols;
-static struct menu *current_menu;
 static int child_count;
 static int do_resize;
 static int single_menu_mode;
Only in srcconf-0.1/lkc: mconf.o
