diff -r -u OpenSceneGraph_orig/src/osg/ApplicationUsage.cpp OpenSceneGraph/src/osg/ApplicationUsage.cpp
--- OpenSceneGraph_orig/src/osg/ApplicationUsage.cpp	2007-02-12 14:39:18.000000000 +0100
+++ OpenSceneGraph/src/osg/ApplicationUsage.cpp	2007-07-31 20:25:05.000000000 +0200
@@ -14,6 +14,7 @@
 #include <osg/ApplicationUsage>
 #include <osg/Math>
 #include <osg/ref_ptr>
+#include <stdlib.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/ArgumentParser.cpp OpenSceneGraph/src/osg/ArgumentParser.cpp
--- OpenSceneGraph_orig/src/osg/ArgumentParser.cpp	2007-02-12 14:39:18.000000000 +0100
+++ OpenSceneGraph/src/osg/ArgumentParser.cpp	2007-07-31 20:25:45.000000000 +0200
@@ -17,6 +17,8 @@
 
 #include <set>
 #include <iostream>
+#include <string.h>
+#include <stdlib.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/BlendColor.cpp OpenSceneGraph/src/osg/BlendColor.cpp
--- OpenSceneGraph_orig/src/osg/BlendColor.cpp	2006-09-18 22:54:48.000000000 +0200
+++ OpenSceneGraph/src/osg/BlendColor.cpp	2007-07-31 20:26:33.000000000 +0200
@@ -15,6 +15,7 @@
 #include <osg/State>
 #include <osg/Notify>
 #include <osg/buffered_value>
+#include <string.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/BlendEquation.cpp OpenSceneGraph/src/osg/BlendEquation.cpp
--- OpenSceneGraph_orig/src/osg/BlendEquation.cpp	2006-07-18 17:21:48.000000000 +0200
+++ OpenSceneGraph/src/osg/BlendEquation.cpp	2007-07-31 20:26:56.000000000 +0200
@@ -15,7 +15,7 @@
 #include <osg/State>
 #include <osg/Notify>
 #include <osg/buffered_value>
-
+#include <string.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/BlendFunc.cpp OpenSceneGraph/src/osg/BlendFunc.cpp
--- OpenSceneGraph_orig/src/osg/BlendFunc.cpp	2006-07-18 17:21:48.000000000 +0200
+++ OpenSceneGraph/src/osg/BlendFunc.cpp	2007-07-31 20:27:23.000000000 +0200
@@ -14,6 +14,7 @@
 #include <osg/GLExtensions>
 #include <osg/State>
 #include <osg/Notify>
+#include <string.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/BufferObject.cpp OpenSceneGraph/src/osg/BufferObject.cpp
--- OpenSceneGraph_orig/src/osg/BufferObject.cpp	2007-05-01 20:03:32.000000000 +0200
+++ OpenSceneGraph/src/osg/BufferObject.cpp	2007-07-31 20:27:53.000000000 +0200
@@ -13,6 +13,7 @@
 #include <stdio.h>
 #include <math.h>
 #include <float.h>
+#include <string.h>
 
 #include <osg/BufferObject>
 #include <osg/Notify>
diff -r -u OpenSceneGraph_orig/src/osg/ClampColor.cpp OpenSceneGraph/src/osg/ClampColor.cpp
--- OpenSceneGraph_orig/src/osg/ClampColor.cpp	2006-09-12 11:26:42.000000000 +0200
+++ OpenSceneGraph/src/osg/ClampColor.cpp	2007-07-31 20:28:39.000000000 +0200
@@ -15,7 +15,7 @@
 #include <osg/State>
 #include <osg/Notify>
 #include <osg/buffered_value>
-
+#include <string.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/CullSettings.cpp OpenSceneGraph/src/osg/CullSettings.cpp
--- OpenSceneGraph_orig/src/osg/CullSettings.cpp	2006-07-18 17:21:48.000000000 +0200
+++ OpenSceneGraph/src/osg/CullSettings.cpp	2007-07-31 20:32:53.000000000 +0200
@@ -14,8 +14,9 @@
 #include <osg/CullSettings>
 #include <osg/ArgumentParser>
 #include <osg/ApplicationUsage>
-
 #include <osg/Notify>
+#include <stdlib.h>
+#include <string.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/DisplaySettings.cpp OpenSceneGraph/src/osg/DisplaySettings.cpp
--- OpenSceneGraph_orig/src/osg/DisplaySettings.cpp	2006-07-26 17:29:26.000000000 +0200
+++ OpenSceneGraph/src/osg/DisplaySettings.cpp	2007-07-31 20:34:10.000000000 +0200
@@ -17,6 +17,7 @@
 #include <osg/ref_ptr>
 
 #include <algorithm>
+#include <string.h>
 
 using namespace osg;
 using namespace std;
diff -r -u OpenSceneGraph_orig/src/osg/Geometry.cpp OpenSceneGraph/src/osg/Geometry.cpp
--- OpenSceneGraph_orig/src/osg/Geometry.cpp	2007-05-01 20:03:32.000000000 +0200
+++ OpenSceneGraph/src/osg/Geometry.cpp	2007-07-31 20:36:16.000000000 +0200
@@ -12,6 +12,7 @@
 */
 #include <osg/Geometry>
 #include <osg/Notify>
+#include <stdlib.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/Image.cpp OpenSceneGraph/src/osg/Image.cpp
--- OpenSceneGraph_orig/src/osg/Image.cpp	2007-06-12 18:55:44.000000000 +0200
+++ OpenSceneGraph/src/osg/Image.cpp	2007-07-31 20:51:47.000000000 +0200
@@ -22,6 +22,9 @@
 #include <osg/Texture2D>
 #include <osg/Texture3D>
 
+#include <string.h>
+#include <stdlib.h>
+
 #include "dxtctool.h"
 
 using namespace osg;
diff -r -u OpenSceneGraph_orig/src/osg/LightModel.cpp OpenSceneGraph/src/osg/LightModel.cpp
--- OpenSceneGraph_orig/src/osg/LightModel.cpp	2006-10-02 15:40:58.000000000 +0200
+++ OpenSceneGraph/src/osg/LightModel.cpp	2007-07-31 20:52:54.000000000 +0200
@@ -13,6 +13,7 @@
 #include <string>
 #include <osg/GL>
 #include <osg/LightModel>
+#include <string.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/Notify.cpp OpenSceneGraph/src/osg/Notify.cpp
--- OpenSceneGraph_orig/src/osg/Notify.cpp	2006-07-18 17:21:48.000000000 +0200
+++ OpenSceneGraph/src/osg/Notify.cpp	2007-07-31 20:54:36.000000000 +0200
@@ -14,6 +14,7 @@
 #include <string>
 #include <iostream>
 #include <fstream>
+#include <stdlib.h>
 
 using namespace std;
 
diff -r -u OpenSceneGraph_orig/src/osg/Point.cpp OpenSceneGraph/src/osg/Point.cpp
--- OpenSceneGraph_orig/src/osg/Point.cpp	2006-11-14 13:29:54.000000000 +0100
+++ OpenSceneGraph/src/osg/Point.cpp	2007-07-31 20:55:20.000000000 +0200
@@ -16,6 +16,7 @@
 #include <osg/State>
 #include <osg/Point>
 #include <osg/Notify>
+#include <string.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/PolygonOffset.cpp OpenSceneGraph/src/osg/PolygonOffset.cpp
--- OpenSceneGraph_orig/src/osg/PolygonOffset.cpp	2006-07-18 17:21:48.000000000 +0200
+++ OpenSceneGraph/src/osg/PolygonOffset.cpp	2007-07-31 20:56:24.000000000 +0200
@@ -13,6 +13,7 @@
 #include <osg/GL>
 #include <osg/PolygonOffset>
 #include <osg/Notify>
+#include <string.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/Program.cpp OpenSceneGraph/src/osg/Program.cpp
--- OpenSceneGraph_orig/src/osg/Program.cpp	2007-01-04 17:49:58.000000000 +0100
+++ OpenSceneGraph/src/osg/Program.cpp	2007-07-31 20:57:43.000000000 +0200
@@ -33,6 +33,8 @@
 #include <OpenThreads/ScopedLock>
 #include <OpenThreads/Mutex>
 
+#include <stdlib.h>
+
 using namespace osg;
 
 
diff -r -u OpenSceneGraph_orig/src/osg/Referenced.cpp OpenSceneGraph/src/osg/Referenced.cpp
--- OpenSceneGraph_orig/src/osg/Referenced.cpp	2007-04-06 17:13:02.000000000 +0200
+++ OpenSceneGraph/src/osg/Referenced.cpp	2007-07-31 21:20:17.000000000 +0200
@@ -23,6 +23,7 @@
 #include <OpenThreads/Mutex>
 
 #include <osg/DeleteHandler>
+#include <stdlib.h>
 
 #ifndef OSG_JAVA_BUILD       
 
diff -r -u OpenSceneGraph_orig/src/osg/TexEnvFilter.cpp OpenSceneGraph/src/osg/TexEnvFilter.cpp
--- OpenSceneGraph_orig/src/osg/TexEnvFilter.cpp	2006-07-18 17:21:48.000000000 +0200
+++ OpenSceneGraph/src/osg/TexEnvFilter.cpp	2007-07-31 21:30:24.000000000 +0200
@@ -13,6 +13,7 @@
 #include <osg/GLExtensions>
 #include <osg/TexEnvFilter>
 #include <osg/State>
+#include <stdlib.h>
 
 using namespace osg;
 
diff -r -u OpenSceneGraph_orig/src/osg/Texture.cpp OpenSceneGraph/src/osg/Texture.cpp
--- OpenSceneGraph_orig/src/osg/Texture.cpp	2007-05-01 20:03:32.000000000 +0200
+++ OpenSceneGraph/src/osg/Texture.cpp	2007-07-31 21:33:08.000000000 +0200
@@ -18,6 +18,8 @@
 #include <osg/GLU>
 #include <osg/Timer>
 #include <osg/ApplicationUsage>
+#include <stdlib.h>  
+#include <string.h>  
 
 #include <OpenThreads/ScopedLock>
 #include <OpenThreads/Mutex>
diff -r -u OpenSceneGraph_orig/src/osg/TextureCubeMap.cpp OpenSceneGraph/src/osg/TextureCubeMap.cpp
--- OpenSceneGraph_orig/src/osg/TextureCubeMap.cpp	2006-08-14 21:42:22.000000000 +0200
+++ OpenSceneGraph/src/osg/TextureCubeMap.cpp	2007-07-31 21:35:09.000000000 +0200
@@ -18,6 +18,7 @@
 #include <osg/Notify>
 
 #include <osg/GLU>
+#include <string.h>
 
 
 using namespace osg;
diff -r -u OpenSceneGraph_orig/src/osg/Uniform.cpp OpenSceneGraph/src/osg/Uniform.cpp
--- OpenSceneGraph_orig/src/osg/Uniform.cpp	2007-02-14 14:18:58.000000000 +0100
+++ OpenSceneGraph/src/osg/Uniform.cpp	2007-07-31 21:37:56.000000000 +0200
@@ -18,6 +18,7 @@
 #include <osg/Uniform>
 #include <osg/Program>
 #include <osg/StateSet>
+#include <string.h>
 
 #include <algorithm>
 
diff -r -u OpenSceneGraph_orig/src/osgDB/Field.cpp OpenSceneGraph/src/osgDB/Field.cpp
--- OpenSceneGraph_orig/src/osgDB/Field.cpp	2006-07-18 17:21:48.000000000 +0200
+++ OpenSceneGraph/src/osgDB/Field.cpp	2007-07-31 22:01:00.000000000 +0200
@@ -11,6 +11,7 @@
  * OpenSceneGraph Public License for more details.
 */
 #include <osgDB/Field>
+#include <string.h>
 
 using namespace osgDB;
 using namespace std;
diff -r -u OpenSceneGraph_orig/src/osgDB/FileNameUtils.cpp OpenSceneGraph/src/osgDB/FileNameUtils.cpp
--- OpenSceneGraph_orig/src/osgDB/FileNameUtils.cpp	2007-02-08 23:31:02.000000000 +0100
+++ OpenSceneGraph/src/osgDB/FileNameUtils.cpp	2007-07-31 22:02:56.000000000 +0200
@@ -11,6 +11,9 @@
  * OpenSceneGraph Public License for more details.
 */
 #include <osgDB/FileNameUtils>
+#include <cstring>
+#include <limits.h>
+#include <stdlib.h>
 
 #ifdef WIN32
     #include <windows.h>
diff -r -u OpenSceneGraph_orig/src/osgSim/OpenFlightOptimizer.cpp OpenSceneGraph/src/osgSim/OpenFlightOptimizer.cpp
--- OpenSceneGraph_orig/src/osgSim/OpenFlightOptimizer.cpp	2007-06-15 18:08:33.000000000 +0200
+++ OpenSceneGraph/src/osgSim/OpenFlightOptimizer.cpp	2007-08-01 01:26:07.000000000 +0200
@@ -14,6 +14,7 @@
 
 #include <vector>
 #include <map>
+#include <cstdlib>
 
 using namespace osgSim;
 
diff -r -u OpenSceneGraph_orig/src/osgText/DefaultFont.cpp OpenSceneGraph/src/osgText/DefaultFont.cpp
--- OpenSceneGraph_orig/src/osgText/DefaultFont.cpp	2007-05-04 14:06:31.000000000 +0200
+++ OpenSceneGraph/src/osgText/DefaultFont.cpp	2007-08-01 00:33:49.000000000 +0200
@@ -17,6 +17,7 @@
 #include "DefaultFont.h"
 
 #include <osg/Notify>
+#include <cstdlib>
 
 using namespace osgText;
 
diff -r -u OpenSceneGraph_orig/src/osgUtil/DisplayRequirementsVisitor.cpp OpenSceneGraph/src/osgUtil/DisplayRequirementsVisitor.cpp
--- OpenSceneGraph_orig/src/osgUtil/DisplayRequirementsVisitor.cpp	2006-07-18 17:21:48.000000000 +0200
+++ OpenSceneGraph/src/osgUtil/DisplayRequirementsVisitor.cpp	2007-07-31 22:45:07.000000000 +0200
@@ -19,6 +19,7 @@
 #include <set>
 
 #include <osgUtil/DisplayRequirementsVisitor>
+#include <cstring>
 
 using namespace osg;
 using namespace osgUtil;
diff -r -u OpenSceneGraph_orig/src/osgUtil/Optimizer.cpp OpenSceneGraph/src/osgUtil/Optimizer.cpp
--- OpenSceneGraph_orig/src/osgUtil/Optimizer.cpp	2007-02-14 14:18:58.000000000 +0100
+++ OpenSceneGraph/src/osgUtil/Optimizer.cpp	2007-07-31 22:56:15.000000000 +0200
@@ -39,6 +39,7 @@
 #include <algorithm>
 #include <numeric>
 #include <sstream>
+#include <cstring>
 
 using namespace osgUtil;
 
diff -r -u OpenSceneGraph_orig/src/osgUtil/RenderBin.cpp OpenSceneGraph/src/osgUtil/RenderBin.cpp
--- OpenSceneGraph_orig/src/osgUtil/RenderBin.cpp	2007-04-25 12:32:28.000000000 +0200
+++ OpenSceneGraph/src/osgUtil/RenderBin.cpp	2007-07-31 23:15:22.000000000 +0200
@@ -19,6 +19,7 @@
 #include <osg/AlphaFunc>
 
 #include <algorithm>
+#include <cstring>
 
 using namespace osg;
 using namespace osgUtil;
