diff -ru spiralmodular-0.2.2_orig/SpiralSound/ChannelHandler.C spiralmodular-0.2.2/SpiralSound/ChannelHandler.C
--- spiralmodular-0.2.2_orig/SpiralSound/ChannelHandler.C	2008-03-24 10:36:41.000000000 +0000
+++ spiralmodular-0.2.2/SpiralSound/ChannelHandler.C	2008-03-24 10:39:39.000000000 +0000
@@ -18,6 +18,8 @@
 
 #include "ChannelHandler.h"
 #include <unistd.h>
+#include <cstdlib>
+#include <cstring>
 
 using namespace std;
 
diff -ru spiralmodular-0.2.2_orig/SpiralSound/Plugins/ControllerPlugin/ControllerPluginGUI.C spiralmodular-0.2.2/SpiralSound/Plugins/ControllerPlugin/ControllerPluginGUI.C
--- spiralmodular-0.2.2_orig/SpiralSound/Plugins/ControllerPlugin/ControllerPluginGUI.C	2008-03-24 10:36:41.000000000 +0000
+++ spiralmodular-0.2.2/SpiralSound/Plugins/ControllerPlugin/ControllerPluginGUI.C	2008-03-24 10:56:09.000000000 +0000
@@ -20,6 +20,7 @@
 #include "ControllerPluginGUI.h"
 #include <FL/fl_draw.h>
 #include <FL/fl_draw.H>
+#include <cstdlib>
 
 using namespace std;
 
diff -ru spiralmodular-0.2.2_orig/SpiralSound/Plugins/ControllerPlugin/ControllerPluginGUI.h spiralmodular-0.2.2/SpiralSound/Plugins/ControllerPlugin/ControllerPluginGUI.h
--- spiralmodular-0.2.2_orig/SpiralSound/Plugins/ControllerPlugin/ControllerPluginGUI.h	2008-03-24 10:36:41.000000000 +0000
+++ spiralmodular-0.2.2/SpiralSound/Plugins/ControllerPlugin/ControllerPluginGUI.h	2008-03-24 10:55:47.000000000 +0000
@@ -44,7 +44,7 @@
       Fl_Color m_GUIColour;
       class CVGUI {
          public:
-            CVGUI::CVGUI(int n, ControllerPluginGUI *p, Fl_Color SelColour);
+            CVGUI(int n, ControllerPluginGUI *p, Fl_Color SelColour);
             Fl_Group *m_SliderGroup;
             Fl_Input *m_Title, *m_Min, *m_Max;
             Fl_Slider *m_Chan;
diff -ru spiralmodular-0.2.2_orig/SpiralSound/Plugins/CounterPlugin/CounterPluginGUI.C spiralmodular-0.2.2/SpiralSound/Plugins/CounterPlugin/CounterPluginGUI.C
--- spiralmodular-0.2.2_orig/SpiralSound/Plugins/CounterPlugin/CounterPluginGUI.C	2008-03-24 10:36:41.000000000 +0000
+++ spiralmodular-0.2.2/SpiralSound/Plugins/CounterPlugin/CounterPluginGUI.C	2008-03-24 10:56:28.000000000 +0000
@@ -20,6 +20,7 @@
 #include <FL/fl_draw.h>
 #include <FL/fl_draw.H>
 #include <stdio.h>
+#include <cstdlib>
 
 using namespace std;
 
diff -ru spiralmodular-0.2.2_orig/SpiralSound/Plugins/LADSPAPlugin/LADSPAInfo.C spiralmodular-0.2.2/SpiralSound/Plugins/LADSPAPlugin/LADSPAInfo.C
--- spiralmodular-0.2.2_orig/SpiralSound/Plugins/LADSPAPlugin/LADSPAInfo.C	2008-03-24 10:36:41.000000000 +0000
+++ spiralmodular-0.2.2/SpiralSound/Plugins/LADSPAPlugin/LADSPAInfo.C	2008-03-24 10:55:19.000000000 +0000
@@ -30,6 +30,7 @@
 #include <stdio.h>
 
 #include <cstdlib>
+#include <cstring>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <dirent.h>
diff -ru spiralmodular-0.2.2_orig/SpiralSound/Plugins/MasherPlugin/MasherPlugin.C spiralmodular-0.2.2/SpiralSound/Plugins/MasherPlugin/MasherPlugin.C
--- spiralmodular-0.2.2_orig/SpiralSound/Plugins/MasherPlugin/MasherPlugin.C	2008-03-24 10:36:41.000000000 +0000
+++ spiralmodular-0.2.2/SpiralSound/Plugins/MasherPlugin/MasherPlugin.C	2008-03-24 10:58:33.000000000 +0000
@@ -20,6 +20,7 @@
 #include "MasherPluginGUI.h"
 #include <FL/Fl_Button.h>
 #include "SpiralIcon.xpm"
+#include <cstdlib>
 
 #define PI 3.141592654
 
diff -ru spiralmodular-0.2.2_orig/SpiralSound/Plugins/NoisePlugin/NoisePlugin.C spiralmodular-0.2.2/SpiralSound/Plugins/NoisePlugin/NoisePlugin.C
--- spiralmodular-0.2.2_orig/SpiralSound/Plugins/NoisePlugin/NoisePlugin.C	2008-03-24 10:36:41.000000000 +0000
+++ spiralmodular-0.2.2/SpiralSound/Plugins/NoisePlugin/NoisePlugin.C	2008-03-24 10:59:28.000000000 +0000
@@ -20,6 +20,7 @@
 #include <FL/Fl_Button.h>
 #include <limits.h>
 #include "SpiralIcon.xpm"
+#include <cstdlib>
 
 using namespace std;
 
diff -ru spiralmodular-0.2.2_orig/SpiralSound/Plugins/OscillatorPlugin/OscillatorPlugin.C spiralmodular-0.2.2/SpiralSound/Plugins/OscillatorPlugin/OscillatorPlugin.C
--- spiralmodular-0.2.2_orig/SpiralSound/Plugins/OscillatorPlugin/OscillatorPlugin.C	2008-03-24 10:36:41.000000000 +0000
+++ spiralmodular-0.2.2/SpiralSound/Plugins/OscillatorPlugin/OscillatorPlugin.C	2008-03-24 10:59:43.000000000 +0000
@@ -20,6 +20,7 @@
 #include <FL/Fl_Button.h>
 #include <limits.h>
 #include "SpiralIcon.xpm"
+#include <cstdlib>
 
 using namespace std;
 
diff -ru spiralmodular-0.2.2_orig/SpiralSound/Sample.h spiralmodular-0.2.2/SpiralSound/Sample.h
--- spiralmodular-0.2.2_orig/SpiralSound/Sample.h	2008-03-24 10:36:41.000000000 +0000
+++ spiralmodular-0.2.2/SpiralSound/Sample.h	2008-03-24 10:37:20.000000000 +0000
@@ -24,6 +24,7 @@
 #include <assert.h>
 #include <limits.h>
 #include <iostream>
+#include <cstring>
 
 //#define DEBUG
 
@@ -68,7 +69,7 @@
 	void CropTo(int NewLength);
 	bool IsEmpty() const { return m_IsEmpty; }
 
-	inline float &Sample::operator[](int i) const
+	inline float &operator[](int i) const
 	{
 		#ifdef DEBUG
 			assert(i>=0 && i<m_Length);
@@ -77,7 +78,7 @@
 	}
 
 	// Linear interpolated
-	inline float Sample::operator[](float i) const
+	inline float operator[](float i) const
 	{
 		int ii=(int)i;
 
@@ -91,7 +92,7 @@
 	}
 
 
-	inline void Sample::Set(int i, float v)
+	inline void Set(int i, float v)
 	{
 		m_IsEmpty=false;
 		#ifdef DEBUG
@@ -100,7 +101,7 @@
 		m_Data[i]=v;
 	}
 
-	inline Sample &Sample::operator=(const Sample &rhs)
+	inline Sample &operator=(const Sample &rhs)
 	{
 		Allocate(rhs.GetLength());
 		memcpy(m_Data,rhs.GetBuffer(),GetLengthInBytes());
