Index: src/clamWidgetsPlugin/MelSpectrumViewPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/MelSpectrumViewPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/MelSpectrumViewPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     MelSpectrumViewPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/MelCepstrumViewPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/MelCepstrumViewPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/MelCepstrumViewPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     MelCepstrumViewPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/PolarChromaPeaksPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/PolarChromaPeaksPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/PolarChromaPeaksPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     PolarChromaPeaksPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/ControlSurfaceWidgetPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/ControlSurfaceWidgetPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/ControlSurfaceWidgetPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     ControlSurfaceWidgetPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/SpectrumViewPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/SpectrumViewPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/SpectrumViewPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     SpectrumViewPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/OscilloscopePlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/OscilloscopePlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/OscilloscopePlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     OscilloscopePlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/QSynthKnobPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/QSynthKnobPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/QSynthKnobPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     QSynthKnobPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/TonnetzPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/TonnetzPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/TonnetzPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     TonnetzPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/KeySpacePlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/KeySpacePlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/KeySpacePlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     KeySpacePlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/ChordRankingPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/ChordRankingPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/ChordRankingPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     ChordRankingPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/PeakViewPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/PeakViewPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/PeakViewPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     PeakViewPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/CLAMWidgets.hxx
===================================================================
--- src/clamWidgetsPlugin/CLAMWidgets.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/CLAMWidgets.hxx	(working copy)
@@ -7,7 +7,7 @@
 	public QDesignerCustomWidgetCollectionInterface
 {
 	Q_OBJECT
-	Q_INTERFACES(QDesignerCustomWidgetCollectionInterface)
+//	Q_INTERFACES(QDesignerCustomWidgetCollectionInterface)
 public:
 	CLAMWidgets(QObject* parent=0);
 	virtual ~CLAMWidgets();
Index: src/clamWidgetsPlugin/PixmapWidgetsPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/PixmapWidgetsPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/PixmapWidgetsPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     PixmapSliderPlugin(QObject *parent = 0);
@@ -55,7 +55,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     ThinPixmapButtonPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/LPModelViewPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/LPModelViewPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/LPModelViewPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     LPModelViewPlugin(QObject *parent = 0);
Index: src/clamWidgetsPlugin/VumeterPlugin.hxx
===================================================================
--- src/clamWidgetsPlugin/VumeterPlugin.hxx	(revision 10580)
+++ src/clamWidgetsPlugin/VumeterPlugin.hxx	(working copy)
@@ -30,7 +30,7 @@
                              public QDesignerCustomWidgetInterface
 {
     Q_OBJECT
-    Q_INTERFACES(QDesignerCustomWidgetInterface)
+//    Q_INTERFACES(QDesignerCustomWidgetInterface)
 
 public:
     VumeterPlugin(QObject *parent = 0);
