diff -ru -ru glib-2.12.0_orig/glib/gmessages.c glib-2.12.0/glib/gmessages.c
--- glib-2.12.0_orig/glib/gmessages.c	2006-05-10 02:44:49.000000000 +0200
+++ glib-2.12.0/glib/gmessages.c	2006-11-03 11:04:14.000000000 +0100
@@ -768,9 +768,9 @@
 #endif
 
   if (log_domain)
-    write_string (fd, "\n");
+    write_string (fd, "");
   else
-    write_string (fd, "\n** ");
+    write_string (fd, "** ");
 
 #ifndef G_OS_WIN32
   write_string (fd, "(process:");
@@ -876,7 +876,7 @@
 
   gstring = g_string_new (NULL);
   if (log_level & ALERT_LEVELS)
-    g_string_append (gstring, "\n");
+    g_string_append (gstring, "");
   if (!log_domain)
     g_string_append (gstring, "** ");
 
