--- fam-2.7.0_orig/lib/Client.c++	2008-12-05 07:24:10.000000000 +0100
+++ fam-2.7.0/lib/Client.c++	2008-12-05 07:42:27.000000000 +0100
@@ -34,7 +34,7 @@
 #include <syslog.h>
 #include <errno.h>
 
-#include <iostream.h>
+#include <iostream>
 
 #include "fam.h"
 #include "Client.h"
--- fam-2.7.0_orig/src/IMon.c++	2008-12-05 07:24:11.000000000 +0100
+++ fam-2.7.0/src/IMon.c++	2008-12-05 07:42:57.000000000 +0100
@@ -40,7 +40,7 @@
 #include "Interest.h"
 #include "Log.h"
 #include "Scheduler.h"
-#include "alloc.h"
+//#include "alloc.h"
 
 int		   IMon::imonfd = -2;
 IMon::EventHandler IMon::ehandler = NULL;
