--- vpopmail-5.4.26_orig/configure.in	2008-11-03 09:59:36.000000000 +0100
+++ vpopmail-5.4.26/configure.in	2008-11-03 10:08:05.000000000 +0100
@@ -448,7 +448,9 @@
         AC_MSG_ERROR(No vpopmail home directory)
 fi
 
+
 vpopmaildir=`cat vpopmail.dir`
+AC_ARG_WITH([vpopmaildir],AC_HELP_STRING([--with-vpopmaildir=PATH]),[vpopmaildir="$withval"])
 AC_SUBST(vpopmaildir)
 AC_DEFINE_UNQUOTED(VPOPMAILDIR,"$vpopmaildir","")
 rm -f vpopmail.dir
