diff -r -u tdb-1.0.6_orig/tdbtool.c tdb-1.0.6/tdbtool.c
--- tdb-1.0.6_orig/tdbtool.c	2007-07-28 16:15:50.000000000 +0200
+++ tdb-1.0.6/tdbtool.c	2007-07-28 16:17:28.000000000 +0200
@@ -169,23 +169,23 @@
 
 static void help(void)
 {
-	printf("
-tdbtool: 
-  create    dbname     : create a database
-  open      dbname     : open an existing database
-  erase                : erase the database
-  dump      dumpname   : dump the database as strings
-  insert    key  data  : insert a record
-  store     key  data  : store a record (replace)
-  show      key        : show a record by key
-  delete    key        : delete a record by key
-  list                 : print the database hash table and freelist
-  free                 : print the database freelist
-  1 | first            : print the first record
-  n | next             : print the next record
-  q | quit             : terminate
-  \\n                   : repeat 'next' command
-");
+	printf("\n"
+"tdbtool:\n"
+"  create    dbname     : create a database\n"
+"  open      dbname     : open an existing database\n"
+"  erase                : erase the database\n"
+"  dump      dumpname   : dump the database as strings\n"
+"  insert    key  data  : insert a record\n"
+"  store     key  data  : store a record (replace)\n"
+"  show      key        : show a record by key\n"
+"  delete    key        : delete a record by key\n"
+"  list                 : print the database hash table and freelist\n"
+"  free                 : print the database freelist\n"
+"  1 | first            : print the first record\n"
+"  n | next             : print the next record\n"
+"  q | quit             : terminate"
+"  \\n                   : repeat 'next' command\n"
+);
 }
 
 static void terror(char *why)
