diff -r -u -- php-front-0.1pre350_orig/configure php-front-0.1pre350/configure
--- php-front-0.1pre350_orig/configure	2007-06-05 11:58:39.000000000 +0200
+++ php-front-0.1pre350/configure	2007-06-05 12:12:03.000000000 +0200
@@ -4476,9 +4476,9 @@
 
 
 
-  { echo "$as_me:$LINENO: checking for sdf2table at $SDF/bin/sdf2table" >&5
-echo $ECHO_N "checking for sdf2table at $SDF/bin/sdf2table... $ECHO_C" >&6; }
-  test -x "$SDF/bin/sdf2table"
+  { echo "$as_me:$LINENO: checking for sdf2table" >&5
+echo $ECHO_N "checking for sdf2table... $ECHO_C" >&6; }
+  test -x "${exec_prefix%/$host_alias}/bin/sdf2table"
   if test $? -eq 0; then
     { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
@@ -4497,7 +4497,7 @@
 
   { echo "$as_me:$LINENO: checking for sglr at $SDF/bin/sglr$EXEEXT" >&5
 echo $ECHO_N "checking for sglr at $SDF/bin/sglr$EXEEXT... $ECHO_C" >&6; }
-  test -x "$SDF/bin/sglr$EXEEXT"
+  test -x "${exec_prefix%/$host_alias}/bin/sglr$EXEEXT"
   if test $? -eq 0; then
     { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
@@ -4515,8 +4515,8 @@
 
 
   { echo "$as_me:$LINENO: checking for implodePT at $SDF/bin/implodePT$EXEEXT" >&5
-echo $ECHO_N "checking for implodePT at $SDF/bin/implodePT$EXEEXT... $ECHO_C" >&6; }
-  test -x "$SDF/bin/implodePT$EXEEXT"
+echo $ECHO_N "checking for implodePT at ${exec_prefix%/$host_alias}/bin/implodePT$EXEEXT... $ECHO_C" >&6; }
+  test -x "${exec_prefix%/$host_alias}/bin/implodePT$EXEEXT"
   if test $? -eq 0; then
     { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
@@ -4698,11 +4698,11 @@
 echo "${ECHO_T}$SDF" >&6; }
   fi
 
-test -f "$SDF/bin/sglr$EXEEXT" ||
-  { { echo "$as_me:$LINENO: error: no such file: $SDF/bin/sglr$EXEEXT
+test -f "${exec_prefix%/$host_alias}/bin/sglr$EXEEXT" ||
+  { { echo "$as_me:$LINENO: error: no such file: ${exec_prefix%/$host_alias}/bin/sglr$EXEEXT
         The configuration of package sdf2-bundle >= 2.4 claims that the package is
         installed at this location. Please check if the package is installed correctly." >&5
-echo "$as_me: error: no such file: $SDF/bin/sglr$EXEEXT
+echo "$as_me: error: no such file: ${exec_prefix%/$host_alias}/bin/sglr$EXEEXT
         The configuration of package sdf2-bundle >= 2.4 claims that the package is
         installed at this location. Please check if the package is installed correctly." >&2;}
    { (exit 1); exit 1; }; }
Only in php-front-0.1pre350_orig/src/grammar: PHP4.def
Only in php-front-0.1pre350_orig/src/grammar: PHP4.rtg
Only in php-front-0.1pre350_orig/src/grammar: PHP4.tbl
Only in php-front-0.1pre350_orig/src/grammar: PHP4Mix.def
Only in php-front-0.1pre350_orig/src/grammar: PHP4Test.def
Only in php-front-0.1pre350_orig/src/grammar: PHP4Test.tbl
Only in php-front-0.1pre350_orig/src/grammar: PHP5.def
Only in php-front-0.1pre350_orig/src/grammar: PHP5.rtg
Only in php-front-0.1pre350_orig/src/grammar: PHP5.tbl
Only in php-front-0.1pre350_orig/src/grammar: PHP5Mix.def
Only in php-front-0.1pre350_orig/src/grammar: PHP5Test.def
Only in php-front-0.1pre350_orig/src/grammar: PHP5Test.tbl
Only in php-front-0.1pre350_orig/src/grammar/languages/php/version4: Mix.sdf
Only in php-front-0.1pre350_orig/src/grammar/languages/php/version5: Mix.sdf
Only in php-front-0.1pre350_orig/src/lib: libphp-front-tables.c
Only in php-front-0.1pre350_orig/src/lib: libphp-front-tables.rtree
Only in php-front-0.1pre350_orig/src/lib: libphp-front.c
Only in php-front-0.1pre350_orig/src/lib: libphp-front.rtree
Only in php-front-0.1pre350_orig/src/lib/php/parse: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/parse: options.rtree
Only in php-front-0.1pre350_orig/src/lib/php/parse: postprocess.rtree
Only in php-front-0.1pre350_orig/src/lib/php/parse: tables.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp: altsyntax.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common: expressions.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common/literals: doubleQuotedStringLiterals.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common/literals: heredocLiterals.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common/literals: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common/literals: numberLiterals.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common/literals: singleQuotedStringLiterals.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common/literals: variableLiterals.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common: operators.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common: statements.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/common: tags.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp: helpers.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp: parenthesize4.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp: parenthesize4.str
Only in php-front-0.1pre350_orig/src/lib/php/pp: parenthesize5.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp: parenthesize5.str
Only in php-front-0.1pre350_orig/src/lib/php/pp/version4: statements.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/version5: expressions.rtree
Only in php-front-0.1pre350_orig/src/lib/php/pp/version5: statements.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/common: class.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/common: environment.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/common: function.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/common: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/inclusion: currentdir.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/inclusion: inclusionid.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/inclusion: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/inclusion: path.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/inclusion: workingdir.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/oo: classes.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/version4: class.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/version4: function.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/version4: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/version4: php4-environment.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/version5: class.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/version5: function.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/version5: interface.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/version5: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/reflect/version5: php5-environment.rtree
Only in php-front-0.1pre350/src/lib/php/signature: version4.rtg
Only in php-front-0.1pre350_orig/src/lib/php/signature: version4.rtree
Only in php-front-0.1pre350/src/lib/php/signature: version5.rtg
Only in php-front-0.1pre350_orig/src/lib/php/signature: version5.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis/common: expressions.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis/common: literals.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis/common: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis/common: operators.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis/common: statements.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis/common: variables.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis: ruleset.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis: special.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis: utilities.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop/analysis: wrappers.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop: output.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop: typejuggling.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy/const-prop: types.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy: files.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy: main.rtree
Only in php-front-0.1pre350_orig/src/lib/php/strategy: simple.rtree
Only in php-front-0.1pre350_orig/src/lib: php-front-tables.rtree
Only in php-front-0.1pre350_orig/src/lib: php-front.rtree
Only in php-front-0.1pre350_orig/src/lib: stratego-lib-hackery.rtree
Only in php-front-0.1pre350_orig/src/tools: eval-php.c
Only in php-front-0.1pre350_orig/src/tools: parse-php.c
Only in php-front-0.1pre350_orig/src/tools: parse-php4.c
Only in php-front-0.1pre350_orig/src/tools: parse-php5.c
Only in php-front-0.1pre350_orig/src/tools: php-reflect.c
Only in php-front-0.1pre350_orig/src/tools: pp-php.c
Only in php-front-0.1pre350_orig/src/tools: pp-php4.c
Only in php-front-0.1pre350_orig/src/tools: pp-php5.c
diff -r -u -- php-front-0.1pre350_orig/configure php-front-0.1pre350/configure
--- php-front-0.1pre350_orig/configure	2007-06-05 12:21:52.000000000 +0200
+++ php-front-0.1pre350/configure	2007-06-05 12:25:22.000000000 +0200
@@ -4487,7 +4487,7 @@
 echo "${ECHO_T}no" >&6; }
     { { echo "$as_me:$LINENO: error: cannot find sdf2table. Use --with-sdf or --with-pgen." >&5
 echo "$as_me: error: cannot find sdf2table. Use --with-sdf or --with-pgen." >&2;}
-   { (exit 1); exit 1; }; }
+   :; }
   fi
 
 
@@ -4506,7 +4506,7 @@
 echo "${ECHO_T}no" >&6; }
     { { echo "$as_me:$LINENO: error: cannot find sglr. Use --with-sdf or --with-sglr." >&5
 echo "$as_me: error: cannot find sglr. Use --with-sdf or --with-sglr." >&2;}
-   { (exit 1); exit 1; }; }
+   :; }
   fi
 
 
@@ -4525,7 +4525,7 @@
 echo "${ECHO_T}no" >&6; }
     { { echo "$as_me:$LINENO: error: cannot find implodePT. Use --with-sdf or --with-pt-support." >&5
 echo "$as_me: error: cannot find implodePT. Use --with-sdf or --with-pt-support." >&2;}
-   { (exit 1); exit 1; }; }
+   :; }
   fi
 
 
@@ -4541,7 +4541,7 @@
 echo "${ECHO_T}no" >&6; }
     { { echo "$as_me:$LINENO: error: cannot find sdf-library. Use --with-sdf." >&5
 echo "$as_me: error: cannot find sdf-library. Use --with-sdf." >&2;}
-   { (exit 1); exit 1; }; }
+   :; }
   fi
 
 
