diff -ru -ru gdbm-1.8.3_orig/Makefile.in gdbm-1.8.3/Makefile.in
--- gdbm-1.8.3_orig/Makefile.in	2006-09-29 09:27:13.000000000 +0200
+++ gdbm-1.8.3/Makefile.in	2006-09-29 09:27:34.000000000 +0200
@@ -155,7 +155,7 @@
 
 libgdbm.la: $(LOBJS) gdbm.h
 	rm -f libgdbm.la
-	$(LIBTOOL) --mode=link $(CC) -o libgdbm.la -rpath $(libdir) \
+	$(LIBTOOL) --mode=link $(CC) -o libgdbm.la -rpath $(libdir) -Wl,-rpath,$(libdir) \
 		-version-info $(SHLIB_VER) $(LOBJS)
 
 libgdbm_compat.la: $(C_LOBJS) gdbm.h
