diff -ru -ru openMotif-2.2.3_orig/tools/wml/Makefile.in openMotif-2.2.3/tools/wml/Makefile.in
--- openMotif-2.2.3_orig/tools/wml/Makefile.in	2006-05-16 10:58:54.000000000 +0200
+++ openMotif-2.2.3/tools/wml/Makefile.in	2006-05-16 11:05:11.000000000 +0200
@@ -258,9 +258,9 @@
 	@rm -f wml$(EXEEXT)
 	$(LINK) $(wml_LDFLAGS) $(wml_OBJECTS) $(wml_LDADD) $(LIBS)
 
-wmluiltok$(EXEEXT): $(wmluiltok_OBJECTS) $(wmluiltok_DEPENDENCIES)
+wmluiltok$(EXEEXT): $(wmluiltok_SOURCES) $(wmluiltok_DEPENDENCIES)
 	@rm -f wmluiltok$(EXEEXT)
-	$(LINK) $(wmluiltok_LDFLAGS) $(wmluiltok_OBJECTS) $(wmluiltok_LDADD) $(LIBS)
+	$(LINK) -DYY_MAIN $(wmluiltok_LDFLAGS) wmluiltok.c $(wmluiltok_LDADD) $(LIBS)
 
 wmldbcreate$(EXEEXT): $(wmldbcreate_OBJECTS) $(wmldbcreate_DEPENDENCIES)
 	@rm -f wmldbcreate$(EXEEXT)
