diff -r -u -- openMotif-2.2.3_orig/Makefile.in openMotif-2.2.3/Makefile.in
--- openMotif-2.2.3_orig/Makefile.in	2007-05-18 18:31:51.000000000 +0200
+++ openMotif-2.2.3/Makefile.in	2007-05-18 18:32:08.000000000 +0200
@@ -101,8 +101,7 @@
 MAINTAINERCLEANFILES = Makefile.in              configure libtool              config.guess              config.sub              config.log              ltconfig ltmain.sh              aclocal.m4              config.h.in              install-sh              missing mkinstalldirs
 
 
-SUBDIRS = bindings bitmaps              config              localized              lib              include              tools              clients              doc              demos
-
+SUBDIRS = bindings bitmaps              config              localized              lib              include              tools              clients              doc
 AUTOMAKE_OPTIONS = 1.4
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
