args: -r -u -- aterm-2.7_orig aterm-2.7
diff -r -u -- aterm-2.7_orig/Makefile aterm-2.7/Makefile
diff -r -u -- aterm-2.7_orig/aterm/.libs/libATerm.la aterm-2.7/aterm/.libs/libATerm.la
diff -r -u -- aterm-2.7_orig/aterm/.libs/libATerm.lai aterm-2.7/aterm/.libs/libATerm.lai
diff -r -u -- aterm-2.7_orig/aterm/Makefile aterm-2.7/aterm/Makefile
diff -r -u -- aterm-2.7_orig/aterm/Makefile.am aterm-2.7/aterm/Makefile.am
--- aterm-2.7_orig/aterm/Makefile.am	2008-07-28 15:38:41.000000000 +0200
+++ aterm-2.7/aterm/Makefile.am	2008-07-28 15:41:42.000000000 +0200
@@ -6,7 +6,7 @@
 
 libATerm_la_SOURCES = $(ALLSRCS)
 libATerm_la_CFLAGS = -DNDEBUG
-libATerm_la_LDFLAGS = -avoid-version -no-undefined
+libATerm_la_LDFLAGS = -rpath $(libdir) -no-undefined -version-number 2:7:0
 
 ALLSRCS = \
   afun.c \
diff -r -u -- aterm-2.7_orig/aterm/Makefile.in aterm-2.7/aterm/Makefile.in
diff -r -u -- aterm-2.7_orig/aterm/libATerm.la aterm-2.7/aterm/libATerm.la
diff -r -u -- aterm-2.7_orig/autom4te.cache/requests aterm-2.7/autom4te.cache/requests
diff -r -u -- aterm-2.7_orig/config.log aterm-2.7/config.log
diff -r -u -- aterm-2.7_orig/config.status aterm-2.7/config.status
diff -r -u -- aterm-2.7_orig/test/Makefile aterm-2.7/test/Makefile
diff -r -u -- aterm-2.7_orig/utils/Makefile aterm-2.7/utils/Makefile
