diff -ru nsound-0.5.0_orig/SConstruct nsound-0.5.0/SConstruct
--- nsound-0.5.0_orig/SConstruct	2008-12-04 22:08:26.000000000 +0100
+++ nsound-0.5.0/SConstruct	2008-12-04 22:09:15.000000000 +0100
@@ -129,8 +129,8 @@
 
 if prefix:
     import os.path
-    bindir = os.path.join(prefix, PACKAGE_RELEASE, "bin")
-    libdir = os.path.join(prefix, PACKAGE_RELEASE, "lib")
+    bindir = os.path.join(prefix, "bin")
+    libdir = os.path.join(prefix, "lib")
 else:
     import os.path
     bindir = os.path.join(Dir(".").abspath, "bin")
