diff -ru clucene-core-0.9.21b_orig/Makefile.am clucene-core-0.9.21b/Makefile.am
--- clucene-core-0.9.21b_orig/Makefile.am	2009-07-23 01:58:04.000000000 +0200
+++ clucene-core-0.9.21b/Makefile.am	2009-07-23 01:59:10.000000000 +0200
@@ -5,7 +5,7 @@
 SUBDIRS = src test
 DIST_SUBDIRS = src test doc win32 .
 
-ACLOCAL_AMFLAGS = -I $(top_srcdir)/m4
+ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = $(top_srcdir)/APACHE.license \
            $(top_srcdir)/LGPL.license \
