diff -ru irssi-0.8.12_orig/src/core/network.c irssi-0.8.12/src/core/network.c
--- irssi-0.8.12_orig/src/core/network.c	2008-02-22 21:41:21.000000000 +0100
+++ irssi-0.8.12/src/core/network.c	2008-02-22 21:43:19.000000000 +0100
@@ -27,6 +27,14 @@
 #  define INADDR_NONE INADDR_BROADCAST
 #endif
 
+/*
+ * constants for getnameinfo
+ */
+#ifndef NI_MAXHOST
+#	define NI_MAXHOST  1025
+#	define NI_MAXSERV    32
+#endif
+
 union sockaddr_union {
 	struct sockaddr sa;
 	struct sockaddr_in sin;
