diff -ru transcode-1.0.4_orig/import/af6_decore.cpp transcode-1.0.4/import/af6_decore.cpp
--- transcode-1.0.4_orig/import/af6_decore.cpp	2007-11-29 23:50:46.000000000 +0100
+++ transcode-1.0.4/import/af6_decore.cpp	2007-11-30 00:02:35.000000000 +0100
@@ -43,6 +43,7 @@
 #include <avifile-0.7/version.h>
 #include <avifile-0.7/renderer.h>
 #include <avifile-0.7/avm_creators.h>
+#include <avifile-0.7/videodecoder.h>
 #elif HAVE_AVIFILE_INCLUDES == 0
 #include <avifile/avm_fourcc.h>
 #include <avifile/avifile.h>
