diff -r -u -- nspr-4.6.3_orig/mozilla/nsprpub/config/rules.mk nspr-4.6.3/mozilla/nsprpub/config/rules.mk
--- nspr-4.6.3_orig/mozilla/nsprpub/config/rules.mk	2007-05-08 10:22:44.000000000 +0200
+++ nspr-4.6.3/mozilla/nsprpub/config/rules.mk	2007-05-08 10:24:36.000000000 +0200
@@ -124,12 +124,12 @@
 
 else
 
-LIBRARY		= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
+LIBRARY		= $(OBJDIR)/lib$(LIBRARY_NAME).$(LIB_SUFFIX)
 ifeq ($(OS_ARCH)$(OS_RELEASE), AIX4.1)
 SHARED_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION)_shr.a
 else
 ifdef MKSHLIB
-SHARED_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
+SHARED_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME).$(DLL_SUFFIX).$(LIBRARY_VERSION)
 endif
 endif
 
