--- arts-1.5.10_orig/artsc/Makefile.am	2009-02-26 00:07:15.000000000 +0100
+++ arts-1.5.10/artsc/Makefile.am	2009-02-26 00:12:34.000000000 +0100
@@ -1,4 +1,4 @@
-AM_CFLAGS = -DARTSC_BACKEND='"$(libdir)/libartscbackend.la"' 
+AM_CFLAGS = -DARTSC_BACKEND='"$(libdir)/libartscbackend.la"' -fno-inline
 AM_CPPFLAGS = -DCOMPILING_ARTSC
 lib_LTLIBRARIES = libartsc.la libartscbackend.la libartsdsp.la libartsdsp_st.la
 FLOWLIBS = $(top_builddir)/flow/libartsflow.la
--- arts-1.5.10_orig/artsc/Makefile.in	2009-02-26 00:07:15.000000000 +0100
+++ arts-1.5.10/artsc/Makefile.in	2009-02-26 00:12:37.000000000 +0100
@@ -370,11 +370,12 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
+top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
-AM_CFLAGS = -DARTSC_BACKEND='"$(libdir)/libartscbackend.la"' 
+AM_CFLAGS = -DARTSC_BACKEND='"$(libdir)/libartscbackend.la"' -fno-inline
 AM_CPPFLAGS = -DCOMPILING_ARTSC
 lib_LTLIBRARIES = libartsc.la libartscbackend.la libartsdsp.la libartsdsp_st.la
 FLOWLIBS = $(top_builddir)/flow/libartsflow.la
@@ -793,7 +794,7 @@
 .NOEXPORT:
 
 #>+ 2
-KDE_DIST=artsc-config.in Makefile.in artsc_export.h.in artsdsp.in Makefile.am stdioemu.c 
+KDE_DIST=artsc_export.h.in artsc-config.in artsdsp stamp-h3 Makefile.in artsdsp.in artsc-config Makefile.am artsdsp.loT stdioemu.c 
 
 #>+ 2
 docs-am:
