diff -r -u libsndfile-1.0.17_orig/tests/utils.c libsndfile-1.0.17/tests/utils.c
--- libsndfile-1.0.17_orig/tests/utils.c	2007-09-30 18:29:55.000000000 +0200
+++ libsndfile-1.0.17/tests/utils.c	2007-09-30 18:30:13.000000000 +0200
@@ -29,6 +29,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdint.h>
 
 #if HAVE_UNISTD_H
 #include <unistd.h>
diff -r -u libsndfile-1.0.17_orig/tests/virtual_io_test.c libsndfile-1.0.17/tests/virtual_io_test.c
--- libsndfile-1.0.17_orig/tests/virtual_io_test.c	2007-09-30 18:29:55.000000000 +0200
+++ libsndfile-1.0.17/tests/virtual_io_test.c	2007-09-30 18:30:56.000000000 +0200
@@ -20,6 +20,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdint.h>
 
 #include <fcntl.h>
 #include <math.h>
