diff -r -u highlight-2.6.5_orig/src/astyle/astyle.h highlight-2.6.5/src/astyle/astyle.h
--- highlight-2.6.5_orig/src/astyle/astyle.h	2007-10-16 07:17:14.000000000 +0000
+++ highlight-2.6.5/src/astyle/astyle.h	2007-10-16 07:18:01.000000000 +0000
@@ -37,6 +37,7 @@
 #include <string>
 #include <vector>
 #include <cctype>
+#include <cstring>
 
 using namespace std;
 
diff -r -u highlight-2.6.5_orig/src/codegenerator.cpp highlight-2.6.5/src/codegenerator.cpp
--- highlight-2.6.5_orig/src/codegenerator.cpp	2007-10-16 07:17:14.000000000 +0000
+++ highlight-2.6.5/src/codegenerator.cpp	2007-10-16 07:20:29.000000000 +0000
@@ -33,6 +33,9 @@
   #include "xterm256generator.h"
 #endif
 
+#include <climits>
+#include <memory>
+
 using namespace std;
 
 namespace highlight {
