--- mysql-5.0.27_orig/configure	2007-01-19 15:12:16.000000000 +0100
+++ mysql-5.0.27/configure	2007-01-19 15:13:12.000000000 +0100
@@ -39055,15 +39055,15 @@
   done
 
 
- if test -z "$OPENSSL_LIB" -o -z "$OPENSSL_INCLUDE" ; then
-   echo "Could not find an installation of OpenSSL"
-   if test -n "$OPENSSL_LIB" ; then
-    if test "$TARGET_LINUX" = "true"; then
-      echo "Looks like you've forgotten to install OpenSSL development RPM"
-    fi
-   fi
-  exit 1
- fi
+# if test -z "$OPENSSL_LIB" -o -z "$OPENSSL_INCLUDE" ; then
+#   echo "Could not find an installation of OpenSSL"
+#   if test -n "$OPENSSL_LIB" ; then
+#    if test "$TARGET_LINUX" = "true"; then
+#      echo "Looks like you've forgotten to install OpenSSL development RPM"
+#    fi
+#   fi
+#  exit 1
+# fi
 
 
     #force VIO use
