diff -ru cgilib-0.5_orig/Makefile cgilib-0.5/Makefile
--- cgilib-0.5_orig/Makefile	2006-02-13 11:37:59.000000000 +0000
+++ cgilib-0.5/Makefile	2006-02-13 11:39:06.000000000 +0000
@@ -14,7 +14,7 @@
 #   along with this program; if not, write to the Free Software
 #   Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111, USA.
 
-CFLAGS = -I. -Wall -O2 -g
+CFLAGS = -I. -Wall -O2 -g -DLibrary=\"cgilib\"
 LDFLAGS = -L.
 LIBS = -lcgi
 
diff -ru cgilib-0.5_orig/cgitest.c cgilib-0.5/cgitest.c
--- cgilib-0.5_orig/cgitest.c	2006-02-13 11:37:59.000000000 +0000
+++ cgilib-0.5/cgitest.c	2006-02-13 11:38:10.000000000 +0000
@@ -23,6 +23,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <cgi.h>
 
 s_cgi *cgi;
