--- exscalibar-1.0.4/src/qtextra/qsocketsession.h.gcc42	2007-03-15 12:49:00.000000000 +0900
+++ exscalibar-1.0.4/src/qtextra/qsocketsession.h	2007-03-15 12:55:04.000000000 +0900
@@ -281,7 +281,7 @@
 	 * 
 	 * @sa safeSendWord()
 	 */
-	template<typename T> const T safeReceiveWord() = 0;
+	template<typename T> const T safeReceiveWord();
 	
 	/**
 	 * Receive a number of words from the connection. The word type may be one
@@ -297,7 +297,7 @@
 	 * 
 	 * @sa safeSendWordArray()
 	 */
-	template<typename T> void safeReceiveWordArray(T *t, const uint32_t size) = 0;
+	template<typename T> void safeReceiveWordArray(T *t, const uint32_t size);
 
 	/**
 	 * Send a string value down the connection.
