diff -ru -ru httpd-2.2.2_orig/include/ap_config_auto.h.in httpd-2.2.2/include/ap_config_auto.h.in
--- httpd-2.2.2_orig/include/ap_config_auto.h.in	2006-06-23 11:02:09.000000000 +0200
+++ httpd-2.2.2/include/ap_config_auto.h.in	2006-06-23 11:18:02.000000000 +0200
@@ -115,9 +115,6 @@
 /* Define to 1 if you have the <pwd.h> header file. */
 #undef HAVE_PWD_H
 
-/* Define to 1 if you have the `setsid' function. */
-#undef HAVE_SETSID
-
 /* Define if SSL is supported using SSL-C */
 #undef HAVE_SSLC
 
diff -ru -ru httpd-2.2.2_orig/srclib/apr/include/arch/unix/apr_private.h.in httpd-2.2.2/srclib/apr/include/arch/unix/apr_private.h.in
--- httpd-2.2.2_orig/srclib/apr/include/arch/unix/apr_private.h.in	2006-06-23 11:02:08.000000000 +0200
+++ httpd-2.2.2/srclib/apr/include/arch/unix/apr_private.h.in	2006-06-23 11:21:30.000000000 +0200
@@ -365,9 +365,6 @@
 /* Define to 1 if you have the `setrlimit' function. */
 #undef HAVE_SETRLIMIT
 
-/* Define to 1 if you have the `setsid' function. */
-#undef HAVE_SETSID
-
 /* Define to 1 if you have the `set_h_errno' function. */
 #undef HAVE_SET_H_ERRNO
 
