--- gromjs-1.7.5_orig/install.sh	2008-12-27 13:00:41.000000000 +0100
+++ gromjs-1.7.5/install.sh	2008-12-27 13:00:46.000000000 +0100
@@ -32,7 +32,7 @@
 
 
 #
-makeprog=gmake
+makeprog=make
 
 
 #
@@ -51,7 +51,7 @@
 spidermonkey_origname="js-1.60"
 spidermonkey_name="spidermonkey-1.60"
 spidermonkey_srcdir=$srcpkgdir/$spidermonkey_name/src/
-spidermonkey_gmakecompilesh=$spidermonkey_srcdir/compile.sh
+spidermonkey_makecompilesh=$spidermonkey_srcdir/compile.sh
 
 
 #
@@ -110,8 +110,8 @@
 
 			cd $initialdir;
 
-			echo "gmake -f Makefile.ref  BUILD_OPT=1" >$spidermonkey_gmakecompilesh ;
-			chmod 755 $spidermonkey_gmakecompilesh ;
+			echo "make -f Makefile.ref  BUILD_OPT=1" >$spidermonkey_makecompilesh ;
+			chmod 755 $spidermonkey_makecompilesh ;
 
 
 			cd $initialdir;
