diff -wru gromjs-1.7.5_orig/_gjs_src/bauk-2.0.14/src/gromconnector/gromjavascript/gromjs_compile.sh gromjs-1.7.5/_gjs_src/bauk-2.0.14/src/gromconnector/gromjavascript/gromjs_compile.sh
--- gromjs-1.7.5_orig/_gjs_src/bauk-2.0.14/src/gromconnector/gromjavascript/gromjs_compile.sh	2008-12-27 13:00:41.000000000 +0100
+++ gromjs-1.7.5/_gjs_src/bauk-2.0.14/src/gromconnector/gromjavascript/gromjs_compile.sh	2008-12-27 13:02:44.000000000 +0100
@@ -326,8 +326,8 @@
 echo ""
 echo "[COMPILING]"
 #kompilacija
-echo  gmake -f $MAKEFILEORIG $GMAKE_JS_OPTIMIZE
-gmake -f $MAKEFILEORIG $GMAKE_JS_OPTIMIZE
+echo  make -f $MAKEFILEORIG $GMAKE_JS_OPTIMIZE
+make -f $MAKEFILEORIG $GMAKE_JS_OPTIMIZE
 echo "Compilation complete"
 echo "DONE."
 
