diff -ru libsvm-2.88_orig/svm-toy/gtk/callbacks.cpp libsvm-2.88/svm-toy/gtk/callbacks.cpp
--- libsvm-2.88_orig/svm-toy/gtk/callbacks.cpp	2009-02-19 19:14:59.000000000 +0100
+++ libsvm-2.88/svm-toy/gtk/callbacks.cpp	2009-02-19 19:15:32.000000000 +0100
@@ -1,6 +1,7 @@
 #include <gtk/gtk.h>
 #include <stdio.h>
 #include <ctype.h>
+#include <cstdlib>
 #include <list>
 #include "callbacks.h"
 #include "interface.h"
