diff -r -u reacTIVision-1.3_orig/ext/libdtouch/fiducialdata.cpp reacTIVision-1.3/ext/libdtouch/fiducialdata.cpp
--- reacTIVision-1.3_orig/ext/libdtouch/fiducialdata.cpp	2007-07-30 06:53:00.000000000 +0200
+++ reacTIVision-1.3/ext/libdtouch/fiducialdata.cpp	2007-07-30 06:53:29.000000000 +0200
@@ -33,6 +33,7 @@
 */
 
 #include "fiducialdata.h"
+#include <string.h>
 
 using std::cout;
 using std::endl;
diff -r -u reacTIVision-1.3_orig/ext/libdtouch/fiducialrecognition.cpp reacTIVision-1.3/ext/libdtouch/fiducialrecognition.cpp
--- reacTIVision-1.3_orig/ext/libdtouch/fiducialrecognition.cpp	2007-07-30 06:53:00.000000000 +0200
+++ reacTIVision-1.3/ext/libdtouch/fiducialrecognition.cpp	2007-07-30 06:53:42.000000000 +0200
@@ -42,6 +42,7 @@
 
 #include <iomanip>
 #include <math.h>
+#include <string.h>
 
 using std::cout;
 using std::cerr;
diff -r -u reacTIVision-1.3_orig/ext/libdtouch/ragbuilder.cpp reacTIVision-1.3/ext/libdtouch/ragbuilder.cpp
--- reacTIVision-1.3_orig/ext/libdtouch/ragbuilder.cpp	2007-07-30 06:53:00.000000000 +0200
+++ reacTIVision-1.3/ext/libdtouch/ragbuilder.cpp	2007-07-30 06:54:01.000000000 +0200
@@ -35,6 +35,7 @@
 
 //
 #include "ragbuilder.h"
+#include <string.h>
 
 using std::cout;
 using std::endl;	
diff -r -u reacTIVision-1.3_orig/ext/libdtouch/regionadjacencygraph.cpp reacTIVision-1.3/ext/libdtouch/regionadjacencygraph.cpp
--- reacTIVision-1.3_orig/ext/libdtouch/regionadjacencygraph.cpp	2007-07-30 06:53:00.000000000 +0200
+++ reacTIVision-1.3/ext/libdtouch/regionadjacencygraph.cpp	2007-07-30 06:54:22.000000000 +0200
@@ -17,6 +17,7 @@
 
 #include "regionadjacencygraph.h"
 #include "listhash.h"
+#include <string.h>
 
 using std::cout;
 using std::endl;
diff -r -u reacTIVision-1.3_orig/ext/oscpack/osc/OscReceivedElements.h reacTIVision-1.3/ext/oscpack/osc/OscReceivedElements.h
--- reacTIVision-1.3_orig/ext/oscpack/osc/OscReceivedElements.h	2007-07-30 06:53:00.000000000 +0200
+++ reacTIVision-1.3/ext/oscpack/osc/OscReceivedElements.h	2007-07-30 06:55:12.000000000 +0200
@@ -32,7 +32,7 @@
 
 #include "OscTypes.h"
 #include "OscException.h"
-
+#include <string.h>
 
 namespace osc{
 
