--- biew-5731_orig/biewlib/sysdep/generic/unix/vio.c	2009-04-17 19:24:47.000000000 +0200
+++ biew-5731/biewlib/sysdep/generic/unix/vio.c	2009-04-17 19:29:16.000000000 +0200
@@ -186,6 +186,9 @@
 
 #endif	/* _CURSES_ */
 
+static unsigned is_unicode=0;
+static void *nls_handle;
+
 #ifdef	_VT100_
 
 #include <errno.h>
@@ -207,9 +210,6 @@
 static int out_fd;
 static int _color[8] = {0,4,2,6,1,5,3,7};
 static char *screen_cp;
-static unsigned is_unicode=0;
-
-static void *nls_handle;
 
 static unsigned char frames_vt100[0x30] =
 "aaaxuuukkuxkjjjkmvwtqnttmlvwtqnvvwwmmllnnjlaaaaa";
