diff -r -u foremost-1.5_orig/foremost.conf foremost-1.5/foremost.conf
--- foremost-1.5_orig/foremost.conf	2007-07-29 13:55:13.000000000 +0200
+++ foremost-1.5/foremost.conf	2007-07-29 14:13:10.000000000 +0200
@@ -237,3 +237,7 @@
 # PINs Password Manager program
 #---------------------------------------------------------------------	
 #      pins  y     8000     \x50\x49\x4e\x53\x20\x34\x2e\x32\x30\x0d
+
+pkgfile y 100000 build() build() ASCII
+diff y 1000000 diff\x20 \x00 FORWARD
+
