diff -ru qmail-1.03_orig/qmail-local.c qmail-1.03/qmail-local.c
--- qmail-1.03_orig/qmail-local.c	2008-11-03 16:38:03.000000000 +0100
+++ qmail-1.03/qmail-local.c	2008-11-03 16:49:24.000000000 +0100
@@ -645,7 +645,7 @@
     {
      cmds.s[j] = 0;
      k = j;
-     while ((k > i) && (cmds.s[k - 1] == ' ') || (cmds.s[k - 1] == '\t'))
+     while ((k > i) && ((cmds.s[k - 1] == ' ') || (cmds.s[k - 1] == '\t')))
        cmds.s[--k] = 0;
      switch(cmds.s[i])
       {
