diff -r -u dmake-4.9_orig/configure dmake-4.9/configure
--- dmake-4.9_orig/configure	2007-08-04 06:35:30.000000000 +0200
+++ dmake-4.9/configure	2007-08-04 06:38:33.000000000 +0200
@@ -4168,7 +4168,9 @@
 echo "${ECHO_T}no" >&6; }
 fi
 
+
 eval dmakerootprefix="$datadir"
+eval dmakerootprefix="$dmakerootprefix"
 # $datadir is ${prefix}/share if not overridden
 test "x$dmakerootprefix" != "x${dmakerootprefix##NONE}" && dmakerootprefix=${ac_default_prefix}/share
 
@@ -8647,8 +8649,6 @@
 ' $ac_file_inputs` in
 *datarootdir*) ac_datarootdir_seen=yes;;
 *@datadir@*|*@docdir@*|*@infodir@*|*@localedir@*|*@mandir@*)
-  { echo "$as_me:$LINENO: WARNING: $ac_file_inputs seems to ignore the --datarootdir setting" >&5
-echo "$as_me: WARNING: $ac_file_inputs seems to ignore the --datarootdir setting" >&2;}
 _ACEOF
 cat >>$CONFIG_STATUS <<_ACEOF
   ac_datarootdir_hack='
diff -r -u dmake-4.9_orig/configure.in dmake-4.9/configure.in
--- dmake-4.9_orig/configure.in	2007-08-04 06:35:30.000000000 +0200
+++ dmake-4.9/configure.in	2007-08-04 06:38:29.000000000 +0200
@@ -157,7 +157,9 @@
 fi
 
 dnl Predefine the DMAKEROOT macro for unix targets.
+
 eval dmakerootprefix="$datadir"
+eval dmakerootprefix="$dmakerootprefix"
 # $datadir is ${prefix}/share if not overridden
 test "x$dmakerootprefix" != "x${dmakerootprefix##NONE}" && dmakerootprefix=${ac_default_prefix}/share
 
