Binary files dmake-4.9_orig/dmake and dmake-4.9/dmake differ
diff -r -u dmake-4.9_orig/sysintf.c dmake-4.9/sysintf.c
--- dmake-4.9_orig/sysintf.c	2007-08-04 07:03:45.000000000 +0200
+++ dmake-4.9/sysintf.c	2007-08-04 07:07:04.000000000 +0200
@@ -601,6 +601,7 @@
 
-   mask = umask(0066);
+//   mask = umask(0066);
    fd = mkstemp( *path );
+   chmod(*path, 0755);
-   umask(mask);
+//   umask(mask);
 
 #elif defined(HAVE_TEMPNAM)
Binary files dmake-4.9_orig/sysintf.o and dmake-4.9/sysintf.o differ
