diff -r -u -- graphviz-2.12_orig/configure graphviz-2.12/configure
--- graphviz-2.12_orig/configure	2007-05-08 12:21:34.000000000 +0200
+++ graphviz-2.12/configure	2007-05-08 12:22:12.000000000 +0200
@@ -25552,7 +25552,7 @@
 
 fi;
 if test "x$enable_php" != "xno"; then
-if test `$SWIG -help 2>&1 | $GREP -c '\-php *- Generate'` = 0; then
+if test `$SWIG -help 2>&1 | $GREP -c '\-php[0-9] *- Generate'` = 0; then
 	PHP=
 	{ echo "$as_me:$LINENO: WARNING: swig does not support php. The PHP packages will not be built" >&5
 echo "$as_me: WARNING: swig does not support php. The PHP packages will not be built" >&2;}
