diff -r -u -- graphviz-2.12_orig/contrib/diffimg/Makefile.in graphviz-2.12/contrib/diffimg/Makefile.in
--- graphviz-2.12_orig/contrib/diffimg/Makefile.in	2007-05-08 12:13:37.000000000 +0200
+++ graphviz-2.12/contrib/diffimg/Makefile.in	2007-05-08 12:14:21.000000000 +0200
@@ -367,9 +367,9 @@
 target_alias = @target_alias@
 AM_CPPFLAGS = @GD_INCLUDES@
 diffimg_SOURCES = diffimg.c
-@WITH_LIBGD_FALSE@diffimg_LDADD = @GD_LIBS@
-@WITH_LIBGD_TRUE@diffimg_LDADD = $(top_builddir)/lib/gd/libgvgd_C.la \
-@WITH_LIBGD_TRUE@	@FC_LIBS@ @FT_LIBS@ @JPEG_LIBS@ @PNG_LIBS@ @Z_LIBS@ @MATH_LIBS@
+diffimg_LDADD = @GD_LIBS@
+#@WITH_LIBGD_TRUE@diffimg_LDADD = $(top_builddir)/lib/gd/libgvgd_C.la \
+#@WITH_LIBGD_TRUE@	@FC_LIBS@ @FT_LIBS@ @JPEG_LIBS@ @PNG_LIBS@ @Z_LIBS@ @MATH_LIBS@
 
 GRAPH = "digraph G { hello -> world }"
 CLEANFILES = *.png
Only in graphviz-2.12/contrib/diffimg: diffimg
