diff -r -u -- spuc_orig/functions/scalfunc.cpp spuc/functions/scalfunc.cpp
--- spuc_orig/functions/scalfunc.cpp	2007-06-01 17:32:53.000000000 +0200
+++ spuc/functions/scalfunc.cpp	2007-06-01 17:33:17.000000000 +0200
@@ -171,7 +171,7 @@
     return Y;
 }
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__MINGW32__)
 double lgamma(double x)
 {
 //    it_error("lgamma() not yet implemented for MS C++");
