diff -ru upload-progress-meter-v4.1_orig/upload_progress_meter/upload_progress_meter.c upload-progress-meter-v4.1/upload_progress_meter/upload_progress_meter.c
--- upload-progress-meter-v4.1_orig/upload_progress_meter/upload_progress_meter.c	2006-02-17 10:32:49.000000000 +0100
+++ upload-progress-meter-v4.1/upload_progress_meter/upload_progress_meter.c	2006-02-17 10:33:32.000000000 +0100
@@ -38,6 +38,7 @@
 static void update_file(UPCdata *X);
 static void * callback_file( void *pointer, int read_bytes, int total_bytes, int what_happened );
 static void * callback_mmcache( void *pointer, int read_bytes, int total_bytes, int what_happened );
+static int check_identifier(char * identifier);
 
 static void file_php_get_info(char *, zval * );
 static void mmcache_php_get_info(char *, zval * );
