diff -r -u -- boost_1_34_0_orig/boost/regex/v4/cpp_regex_traits.hpp boost_1_34_0/boost/regex/v4/cpp_regex_traits.hpp
--- boost_1_34_0_orig/boost/regex/v4/cpp_regex_traits.hpp	2007-05-25 12:38:25.000000000 +0200
+++ boost_1_34_0/boost/regex/v4/cpp_regex_traits.hpp	2007-05-25 12:38:46.000000000 +0200
@@ -20,6 +20,7 @@
 #define BOOST_CPP_REGEX_TRAITS_HPP_INCLUDED
 
 #include <boost/config.hpp>
+#include <limits.h>
 
 #ifndef BOOST_NO_STD_LOCALE
 
