args: -r -u -- gst-plugins-good-0.10.10_orig gst-plugins-good-0.10.10
diff -r -u -- gst-plugins-good-0.10.10_orig/Makefile gst-plugins-good-0.10.10/Makefile
diff -r -u -- gst-plugins-good-0.10.10_orig/Makefile.am gst-plugins-good-0.10.10/Makefile.am
--- gst-plugins-good-0.10.10_orig/Makefile.am	2008-10-27 20:19:09.000000000 +0100
+++ gst-plugins-good-0.10.10/Makefile.am	2008-10-27 20:39:47.000000000 +0100
@@ -8,7 +8,6 @@
 
 ALWAYS_SUBDIRS =		\
 	gst sys ext 		\
-	tests			\
 	docs			\
 	po			\
 	common			\
diff -r -u -- gst-plugins-good-0.10.10_orig/Makefile.in gst-plugins-good-0.10.10/Makefile.in
--- gst-plugins-good-0.10.10_orig/Makefile.in	2008-10-27 20:19:08.000000000 +0100
+++ gst-plugins-good-0.10.10/Makefile.in	2008-10-27 20:40:02.000000000 +0100
@@ -593,7 +593,6 @@
 @USE_GCONFTOOL_TRUE@GCONF_DIR = gconf
 ALWAYS_SUBDIRS = \
 	gst sys ext 		\
-	tests			\
 	docs			\
 	po			\
 	common			\
diff -r -u -- gst-plugins-good-0.10.10_orig/config.log gst-plugins-good-0.10.10/config.log
