diff -ru -ru upx-1.25_orig/src/Makefile.bld upx-1.25/src/Makefile.bld
--- upx-1.25_orig/src/Makefile.bld	2006-09-21 17:12:31.000000000 +0200
+++ upx-1.25/src/Makefile.bld	2006-09-21 17:13:24.000000000 +0200
@@ -153,10 +153,10 @@
 include $(srcdir)/Makedefs.gcc
 override arch := $(shell uname -m | sed -e 's/^i[3456789]86$$/i386/')
 ifeq ($(arch),i386)
-  CCARCH += -march=i386 -mcpu=i686
+  CCARCH += -march=i386 -mtune=i686
 endif
 DEFS += -D_FILE_OFFSET_BITS=64
-LDLIBS += -lmcheck
+##LDLIBS += -lmcheck
 ##CFLAGS_WERROR = -Werror
 ifeq ($(DEBUG),1)
 else
@@ -203,7 +203,7 @@
 DEPMODE := linux-intelc
 CC = icc
 CXX = icpc
-CCARCH += -march=i386 -mcpu=i686 -nolib_inline
+CCARCH += -march=i386 -mtune=i686 -nolib_inline
 DEFS += -D_FILE_OFFSET_BITS=64
 CFLAGS_OUTPUT = -o $@
 LINK_EXE_OUTPUT = -o $@
@@ -252,7 +252,7 @@
 include $(srcdir)/Makedefs.gcc
 CC = x86_64-linux-gcc
 CXX = x86_64-linux-g++
-LDLIBS += -lmcheck
+##LDLIBS += -lmcheck
 endif	# cross-amd64-linux
 
 ifeq ($(target),cross-m68k-linux)
@@ -309,7 +309,7 @@
 include $(srcdir)/Makedefs.gcc
 e = .exe
 CXX = gxx
-CCARCH += -march=i386 -mcpu=i686
+CCARCH += -march=i386 -mtune=i686
 CFLAGS_WERROR = -Werror
 STUBEDIT_EXE = stubedit $@ bufsize=0xfc00
 ifneq ($(strip $(wildcard $(DJDIR)/bin/mfxdjstubify.ex[eE])),)
@@ -337,14 +337,14 @@
 include $(srcdir)/Makedefs.gcc
 ##CXX = g++-2
 e = .exe
-CCARCH += -march=i386 -mcpu=i686
+CCARCH += -march=i386 -mtune=i686
 LDLIBS += -lwinmm
 endif
 
 ifeq ($(target),win32-mingw32)
 include $(srcdir)/Makedefs.gcc
 e = .exe
-CCARCH += -march=i386 -mcpu=i686
+CCARCH += -march=i386 -mtune=i686
 LDLIBS += -lwinmm
 endif
 
@@ -354,7 +354,7 @@
 e = .exe
 CC = i586-mingw32msvc-gcc
 CXX = i586-mingw32msvc-g++
-CCARCH += -march=i386 -mcpu=i686
+CCARCH += -march=i386 -mtune=i686
 LDLIBS += -lwinmm
 endif
 
