diff -r -u binutils-2.17/bfd/Makefile.am libbfd-2.17/bfd/Makefile.am
--- binutils-2.17/bfd/Makefile.am	2006-06-23 20:17:17.000000000 +0200
+++ libbfd-2.17/bfd/Makefile.am	2007-08-04 12:58:25.000000000 +0200
@@ -718,7 +718,7 @@
 libbfd_la_SOURCES = $(BFD32_LIBS_CFILES) $(BFD64_LIBS_CFILES)
 libbfd_la_DEPENDENCIES = $(OFILES) ofiles
 libbfd_la_LIBADD = `cat ofiles` @WIN32LIBADD@
-libbfd_la_LDFLAGS = -release $(VERSION) @WIN32LDFLAGS@
+libbfd_la_LDFLAGS = -version-info $(subst .,:,$(VERSION)) @WIN32LDFLAGS@
 
 # libtool will build .libs/libbfd.a.  We create libbfd.a in the build
 # directory so that we don't have to convert all the programs that use
