diff -ru autoconf-2.61_orig/lib/autoconf/general.m4 autoconf-2.61/lib/autoconf/general.m4
--- autoconf-2.61_orig/lib/autoconf/general.m4	2008-07-12 13:23:53.000000000 +0200
+++ autoconf-2.61/lib/autoconf/general.m4	2008-07-12 13:24:24.000000000 +0200
@@ -1654,7 +1654,7 @@
 # Generate the variables THING, THING_{alias cpu vendor os}.
 m4_define([_AC_CANONICAL_SPLIT],
 [case $ac_cv_$1 in
-*-*-*) ;;
+*-*-* | *-*) ;;
 *) AC_MSG_ERROR([invalid value of canonical $1]);;
 esac
 AC_SUBST([$1], [$ac_cv_$1])dnl
