diff -r -u src_orig/autoconf-2.61/lib/autoconf/status.m4 src/autoconf-2.61/lib/autoconf/status.m4
--- src_orig/autoconf-2.61/lib/autoconf/status.m4	2007-04-21 03:35:18.000000000 +0200
+++ src/autoconf-2.61/lib/autoconf/status.m4	2007-04-21 03:35:40.000000000 +0200
@@ -521,7 +521,7 @@
 ])' $ac_file_inputs` in
 *datarootdir*) ac_datarootdir_seen=yes;;
 *@[]m4_join([@*|*@], _AC_datarootdir_vars)@*)
-  AC_MSG_WARN([$ac_file_inputs seems to ignore the --datarootdir setting])
+dnl  AC_MSG_WARN([$ac_file_inputs seems to ignore the --datarootdir setting])
 _ACEOF
 cat >>$CONFIG_STATUS <<_ACEOF
   ac_datarootdir_hack='
